## **Chapter 26. Applied Microorganisms**

Recommended Post: 【Biology】 [Biology Index](https://jb243.github.io/pages/1457)

---

**1.** [Overview](#1-overview)

**2.** [Bacteria](#2-bacteria)

**3.** [Yeast](#3-yeast)

**4.** [Virus](#4-virus)

**5.** [Viroids and Prions](#5-viroids-and-prions)

**6.** [Comparative Microbiology](#6-comparative-microbiology)

---

<br>

## **1. Overview**

 ⑴ Various Growth Forms

> ① Classification Based on Oxygen Requirements and Resistance

<br>

<img width="580" height="233" alt="image" src="https://github.com/user-attachments/assets/a9dfe69d-84d0-43fc-ae96-752506a468e1" />

**Figure 1. Classification Based on Oxygen Requirements and Resistance**

<br>

>> ○ A. Obligate Aerobe: Can only perform aerobic respiration, cannot perform fermentation or anaerobic respiration.

>> ○ B. Facultative: Microorganisms that can adapt their metabolism to both aerobic and anaerobic conditions.

>>> ○ Example: Yeast

>> ○ C. Aerotolerant: Can remove active oxygen like A, B, and E, but cannot perform aerobic respiration.

>> ○ D. Obligate Anaerobe: Suffers lethal damage from oxygen.

>>> ○ Mostly methanogens and acetogens.

>>> ○ Methanogens and acetogens compete using carbon dioxide and hydrogen, with methanogens having an advantage.

>> ○ E. Microaerophile: Prefers moderate oxygen concentrations.

> ② Classification of Bacteria Based on Nutritional Methods

>> ○ Heterotrophic Bacteria

>>> ○ Lack chlorophyll

>>> ○ Contribute to the material cycle in ecosystems by decomposing the remains and excretions of animals and plants

>>> ○ e.g., Escherichia coli

>> ○ Chemosynthetic Bacteria: Autotrophic Bacteria

>>> ○ Lack chlorophyll

>>> ○ Synthesize organic compounds using chemical energy obtained by oxidizing inorganic substances

>>> ○ e.g., nitrite bacteria

>> ○ Photosynthetic Bacteria: Autotrophic Bacteria

>>> ○ Contain chlorophyll a (green)

>>> ○ Synthesize organic compounds using light energy

>>> ○ e.g., Cyanobacteria

> ③ Auxotrophic Mutant

>> ○ A mutant strain of bacteria, fungi, or cultured cells that has undergone mutation and now requires a specific chemical substance for growth

>> ○ **Example 1:** Arginine-requiring strain – requires arginine for growth

> ④ Diauxic Growth

>> ○ Definition: A growth pattern in which microorganisms, when provided with two types of nutrients, consume one nutrient first, and after it is depleted, begin to consume the second nutrient

 ⑵ Types of Culture Media

> ① Complete Media: Contains all nutrients such as amino acids, peptone, and triton.

>> ○ Example 1: Agar medium

>> ○ Example 2: LB medium: Contains yeast extract and is specific for E. coli.

>>> ○ 1st: Mix 700 ml dH2O, 10 g tryptone, 5 g yeast extract, and 10 g NaCl in a flask.

>>> ○ 2nd: Adjust pH to 7.0.

>>> ○ 3rd: Add agar to achieve a 1.5% solution and adjust the volume to 1 L with dH2O.

>>> ○ 4th: Seal the flask and autoclave.

>>> ○ 5th: After cooling to 50-60°C, add ampicillin stock solution to achieve a concentration of 25 μg/ml.

>>> ○ 6th: Pour into culture plates and let solidify.

>>> ○ 7th: Invert the petri dish with LB medium during E. coli culture to prevent contamination.

> ② Minimal Media: Contains minimal essential nutrients like vitamin B7, inorganic salts, and glucose.

> ③ Supplemented Media: Used for nutritional auxotroph confirmation and transformation.

 ⑶ [Cell Culture](https://jb243.github.io/pages/445)

 ⑷ Counting: Serial Dilution Method (Plaque Assay Method), [Absorbance Method](https://jb243.github.io/pages/1404)

<br>

<br>

## **2. Bacteria**

 ⑴ Overview

> ① Unit: Colony

> ② Size: 0.5 - 5.0 μm

> ③ Motility using flagella

 ⑵ Functions of Cell Wall

> ① Prevents swelling and bursting in hypotonic environments, like plant cells.

> ② Reminder of the highly variable osmotic concentration of bacterial growth environments.

 ⑶ Classification Based on Morphology

> ① Type 1: Spirilla

>> ○ Vibrios (e.g., _Vibrio cholerae_)

>> ○ Spirilla (e.g., _Helicobacter pylori_)

>> ○ Spirochaetes (e.g., _Treponema pallidum_)

> ② Type 2: Cocci

>> ○ Diplococci (e.g., _Streptococcus pneumoniae_)

>> ○ Streptococci (e.g., _Streptococcus pyogenes_)

>> ○ Staphylococci (e.g., _Staphylococcus aureus_)

>> ○ Sarcina (e.g., _Sarcina ventriculi_)

> ③ Type 3: Bacilli

>> ○ Chain of bacilli (e.g., _Bacillus anthracis_)

>> ○ Flagellate rods (e.g., _Salmonella typhi_)

>> ○ Spore-formers (e.g., _Clostridium botulinum_)

 ⑷ Classification Based on Oxygen Requirement

> ① Aerobic

> ② Anaerobic

 ⑸ Classification Based on [Gram Staining](https://jb243.github.io/pages/1487): Crystal violet is used first, followed by safranine O.

> ① 1st: Heat-fix bacteria onto a slide after alcohol lamp treatment 2-3 times.

> ② 2nd: Apply **crystal violet** staining reagent to the sample on the slide, let stand for 20 seconds.

>> ○ All cells appear purple at this stage.

>> ○ Crystal violet, a basic reagent with a (+) charge, strongly binds to teichoic acids in Gram-positive bacteria.

> ③ 3rd: Briefly rinse the stained slide with distilled water in a tube and remove excess water.

> ④ 4th: Apply iodine solution (I2-KI) to the slide with the crystal violet-stained sample, let stand for 1 minute.

>> ○ Iodine forms crystal violet - iodine complex, preventing decolorization.

> ⑤ 5th: Rinse the slide in a tube with 95% ethanol while removing the iodine solution.

>> ○ Gram-positive bacteria retain the purple color.

>> ○ In Gram-positive bacteria, alcohol shrinks the cell wall, trapping I2-crystal violet in the cytoplasm.

> ⑥ 6th: Rinse the slide in a tube with distilled water to stop the decolorization.

> ⑦ 7th: Apply **safranine** to the slide with the sample, let stand for 1 minute.

>> ○ Gram-positive bacteria maintain the purple color.

>> ○ Gram-negative bacteria show a red color due to safranine.

> ⑧ 8th: Rinse with water gently for a few seconds, remove excess water.

⑹ Gram-Positive Bacteria

> ① Structure: Cytoplasm - Cell membrane - Thick peptidoglycan

> ② [Peptidoglycan](http://nate9389.tistory.com/1389): Alternating N-acetylglucosamine (N-AG) and N-acetylmuramic acid (N-AM) with β 1→4 linkage.

> ③ Forms endospores for high-temperature and high-pressure resistance.

> ④ Teichoic acid protrudes from peptidoglycan and is connected by covalent bonds, only observed in Gram-positive bacteria.

>> ○ Teichoic acid carries gluconic acid, giving it a negative charge.

> ⑤ Examples: _Bacillus subtilis_, _Bacillus anthracis_, _Staphylococcus_

⑺ Gram-Negative Bacteria

> ① Structure: Cytoplasm - Cell membrane - Periplasmic space - Thin peptidoglycan - Periplasmic space - LPS, Outer membrane

> ② LPS (lipopolysaccharide): Toxic and an immune system target.

>> ○ LPS contributes to botulinum toxin toxicity.

>> ○ Penicillin cannot pass through LPS.

>> ○ Blood clotting and fever induction.

>> ○ It has a structure composed of lipid A + polysaccharide, and the part that acts as an endotoxin is lipid A.

> ③ Outer membrane: More permeable than the cell membrane due to porins.

> ④ Periplasmic space

>> ○ Separates peptidoglycan layer from the cell and outer membrane.

>> ○ Important for cellular functions like metabolism and transport.

> ⑤ Examples: _Escherichia coli_, _Helicobacter pylori_

⑻ [Bacterial Examples](https://jb243.github.io/pages/1436)

 ⑼ Antibiotics

> ① Beta-lactam antibiotics

>> ○ Inhibit penicillin-binding proteins (PBPs) irreversibly, crucial for bacterial cell wall synthesis.

<br>

<img width="382" height="599" alt="image" src="https://github.com/user-attachments/assets/091f860c-d23c-48ac-b247-ae764869267d" />

**Figure 2. Mechanism of Beta-Lactam Antibiotics**

<br>

>> ○ Initially derived from the fungus Acremonium.

>> ○ Carbapenem

>> ○ Cephalosporin  

>> ○ Monobactam

>> ○ Penicillin

>> ○ Clavulanic acid: Irreversibly deactivates beta-lactamase as a beta-lactam antibiotic.

> ② Other Cell Wall Inhibitors

>> ○ Vancomycin: Binds to (D)-Ala-(D)-Ala of pentapeptide. Almost the only drug used against beta-lactam-resistant bacteria.

>> ○ Bacitracin

> ③ Cell Membrane Damage

>> ○ Polymyxin B: Binds to the outer membrane of Gram-negative bacteria through electrostatic interactions, making it unstable.

> ④ Folic Acid Inhibition: Bacteria do not easily absorb folate.

>> ○ Sulfonamide: Competes with p-aminobenzoic acid. The first antibiotic developed by humans.

>> ○ Trimethoprim

<br>

<img width="338" height="500" alt="image" src="https://github.com/user-attachments/assets/ccdc0a87-263e-4e51-8d19-61da60b7743c" />

**Figure 3. Mechanism of Antifolate Agents**

<br>

> ⑤ DNA Gyrase Inhibition

>> ○ Fluoroquinolone

>> ○ Quinolone

> ⑥ RNA Polymerase Inhibition

>> ○ Rifamycin: Includes rifampin, rifapentine, and rifabutin.

> ⑦ 30S Subunit Inhibition

>> ○ Aminoglycoside

>> ○ Gentamycin

>> ○ Neomycin: Inhibits tRNA-mRNA interactions.

>> ○ Streptomycin: Inhibits initiation of translation.

>> ○ Tetracycline: Inhibits tRNA and ribosome binding.

> ⑧ 50S Subunit Inhibition

>> ○ chloramphenicol (trade name: chloromycetin): Inhibitor of peptide bond formation

>>> ○ A powerful antibiotic derived from streptomycesvenezuelae bacteria

>>> ○ Broad-spectrum effectiveness against various bacteria, particularly effective against typhoid

>> ○ clindamycin

>> ○ erythromycin: Inhibits the movement of mRNA along the ribosome

>> ○ linezolid

>> ○ macrolide

>> ○ streptogramin

> ⑨ Antibiotic Sensitivity: Different bacteria species respond differently to various antibiotics

<br>

<img width="960" height="386" alt="image" src="https://github.com/user-attachments/assets/c0dd7ffa-e5c9-4348-9c6b-76a082dedd15" />

 **Figure 4.** Antibiotic Sensitivity

<br>

> ⑩ Antibiotic Resistance: Bacteria start secreting penicillinase, an inhibitor of penicillin, just five years after prescribing penicillin

<br>

<img width="1046" height="491" alt="image" src="https://github.com/user-attachments/assets/e68b2920-a4ae-49cd-a7d7-c318ba145260" />

 **Figure 5.** Antibiotic Resistance

<br>

<br>

## **3. Yeast**

 ⑴ Prefers asexual reproduction, but undergoes sexual reproduction under unfavorable conditions

 ⑵ Mitochondria are inherited from both parents during sexual reproduction

<br>

<br>

## **4. Virus**

 ⑴ Overview

> ① Unit: Plaque

> ② Burst size: Number of phages produced within a bacterium

> ③ Biological characteristics

>> ○ Composed of nucleic acids and proteins

>> ○ Can exhibit activity within cells, involving self-replication and genetic transmission

>> ○ Frequent mutations for rapid adaptation to the environment

> ④ Non-biological characteristics

>> ○ Can exist as protein crystals

>> ○ Existence of auxiliary metabolic enzymes; however, enzymes related to DNA replication or transcription exist

 ⑵ Structure

> ① Virion: Basic unit of a virus. Particle form when outside the host

> ② Nucleic acid + Capsid (capsomer) + Cell entry enzyme + (Envelope) + (Reverse transcriptase)

>> ○ Envelope exists only in animal viruses

>> ○ Reverse transcriptase exists only in retroviruses (e.g., retroviruses)

> ③ Tail fibers: Present in bacteriophages. Recognize specific receptors on the host cell surface for attachment

 ⑶ Classification

> ① Presence of Envelope: Naked virus, Enveloped virus

>> ○ Naked virus: No envelope. Transported into the nucleus via dynein. Cell release through lysozyme.

>> ○ Enveloped virus: Envelope present. Transported into the nucleus via endosomal system.

> ② Host: Bacteriophages, Plant viruses, Animal viruses

> ③ Classification by Genetic Material: ds DNA, ss DNA, ds RNA, ss RNA

> ④ Classification according to [Receptor Binding Protein](https://jb243.github.io/pages/1389)

>> ○ Hemagglutinin

>>> ○ Acts during cell entry

>>> ○ Binds to sialic acid residues at the carbohydrate terminus on the host cell surface

>>> ○ Sialic acid is also known as N-acetylneuraminic acid

>> ○ Neuraminidase

>>> ○ Acts upon cell release

>>> ○ Hemagglutinin temporarily binds to sialic acid residues on the host cell membrane during animal influenza virus budding

>>> ○ Neuraminidase breaks the temporary (glycosidic) linkage

>> ○ Differentiation of influenza viruses depends on the types of hemagglutinin and neuraminidase

>>> ○ Hemagglutinin exists from H1 to H16

>>> ○ Neuraminidase exists from N1 to N9

>>> ○ Virus classification follows the pattern H#N#

>> ○ Tamiflu™, oseltamivir, and Relenza™

>>> ○ Competitive inhibitors of neuraminidase. Similar to sialic acid

>>> ○ Inhibit the replication of influenza viruses

> ⑤ Classification by Antigenic Variation Mechanism

>> ○ Antigenic drift: Continuous reproduction of a virus results in antigenic variations through mutations

>> ○ Antigenic shift: Two different viruses replicate within the same host, leading to new antigenic variations. Involves recombination of segmented viral genomes

>> ○ Influenza viruses acquire diversity through antigenic shifts

>> ○ Burst size: Number of phages produced within a bacterium

 ⑷ Classification by Genetic Material: ds DNA, ss DNA, dsRNA, ssRNA

> ① dsDNA virus

>> ○ Derived from lipid bilayer of the nuclear membrane

>> ○ Examples: AdV, HSV (Herpes virus), Hepatitis B virus, smallpox virus, VACV

> ② ssDNA virus

>> ○ Derived from lipid bilayer of the nuclear membrane

>> ○ Examples: H1-PV, Parvovirus

> ③ dsRNA virus: (+) is sense, (-) is antisense for RNA synthesis

>> ○ Derived from cell membranes.

>> ○ 1st. Double-stranded RNA separates into (-) ssRNA and (+) ssRNA

>> ○ 2nd. (-) ssRNA uses its enzyme (RNA dependent RNA pol) to create dsRNA

>> ○ 3rd. (+) ssRNA functions as mRNA, translating viral proteins

>> ○ Examples: reovirus, rotavirus

> ④ ssRNA virus: Classified as (-) negative-sense ssRNA virus, (+) positive-sense ssRNA virus, and retroviruses

>> ○ Derived from cell membranes.

> ⑤ (+) positive-sense ssRNA virus: (+) is sense, (-) is antisense for RNA synthesis

>> ○ Positive strand carries fundamental genetic information

>> ○ Functions as mRNA for translation of viral proteins, thus exists with 5' cap and poly A tail

>> ○ 1st. (-) Synthesis of multiple ssRNA strands using its RNA pol

>> ○ 2nd. (-) ssRNA is used by its enzyme (RNA dependent RNA pol) to create (+) ssRNA

>> ○ 3rd. (+) ssRNA is used for genome assembly and protein translation

>> ○ High mutation rate as the genome is used for both translation and replication

>> ○ Examples: Hepatitis C virus, SARS, Coronaviruses, poliovirus, coxsackievirus, NDV

> ⑥ (-) negative-sense ssRNA virus: (+) is sense, (-) is antisense for RNA synthesis

>> ○ Negative strand complements the positive strand. Negative strand is used mainly as a template for replication, not genetic information

>> ○ 1st. Uses its RNA dependent RNA pol to create multiple (+) ssRNA strands

>> ○ 2nd. (+) ssRNA is used by its enzyme (RNA dependent RNA pol) to create multiple (-) ssRNA strands.

>> ○ 3rd. (-) ssRNA is used for genome assembly, and (+) ssRNA is used for protein translation

>> ○ Examples: **Influenza virus**, Ebola virus, MV, NDV, VSV

>> ○ **Memory tip:** Influenza... Unfluenza...

> ⑦ Retroviruses: Contain reverse transcriptase (RT), a type of (+) positive-sense ssRNA virus

>> ○ Reverse transcriptase virus: Contains reverse transcriptase (RT) enzyme

>> ○ Reverse transcriptase viruses are classified into retroviruses and non-retroviruses (e.g., Hepatitis B virus, a dsDNA virus)

>> ○ **HIV virus**

>>> ○ Contains two identical ssRNA strands forming the genome. Note that these strands are not complementary

>>> ○ HIV RNA has a 5'-cap and poly-A tail

>>> ○ Key enzymes in HIV virus: Reverse transcriptase (RT), integrase, protease

>> ○ (Note) Influenza virus genome is RNA and can replicate RNA directly from it

>>> ○ However, in the case of avian influenza, a reverse transcription process is present to synthesize cDNA

<br>

<img width="1073" height="658" alt="image" src="https://github.com/user-attachments/assets/ee6d188f-e4b8-499e-8fc1-a8089366c1ce" />

 **Table 1.** Types of DNA viruses

<br>

<img width="1042" height="637" alt="image" src="https://github.com/user-attachments/assets/49afbfc2-f14f-4372-8a74-073a5416435e" />

 **Table 2.** Types of RNA viruses

<br>

<img width="744" height="689" alt="image" src="https://github.com/user-attachments/assets/e334cfab-3400-4943-8bd3-ac9a62b69c26" />

 **Figure 6.** Types of viruses

<br>

 ⑸ Bacteriophage: Has a life cycle of about 30 minutes

> ① Lytic Cycle: Pathogenic viruses like T1, T2, T4 phages (DNA viruses)

>> ○ 1st. Attachment: Tail attaches to the host cell surface

>> ○ 2nd. Entry: Phage action immediately halts host replication, transcription, and translation

>> ○ 3rd. Replication: Early genes expressed → Induction of late gene transcription → Expression of late genes

>>> ○ Early genes: genes that determine the phage lifecycle and degrade the host DNA by hydrolyzing the host chromosome entirely to utilize it for viral DNA synthesis

>>> ○ Late genes: Capsid genes, lysis genes

>> ○ 4th. Assembly: Virus DNA replication using host enzymes. Produced virus proteins cut and degrade host DNA

>> ○ 5th. Release: Cell membrane lytic enzyme produced by phage leads to release

> ② Lysogenic Cycle: Temperate phages like λ phage (DNA viruses)

>> ○ 1st. Attachment

>> ○ 2nd. Entry: Upon insertion into bacteria, linear DNA becomes circular

>> ○ 3rd. Replication: Unlike the lytic cycle, phage DNA integrates into host DNA

>>> ○ Prophage: Phage DNA integrated into host DNA

>> ○ 4th. Assembly and Release: Original phage DNA remains in the host

>> ○ 5th. Switches to lytic cycle if host replication is hindered

>>> ○ More cI factor results in lysogenic cycle, while more cro factor results in lytic cycle

>>> ○ **Memory tip:** I.. in.. O.. out..

>>> ○ In healthy E. coli, phages first synthesize cro, later influenced by cI protein, leading to lysogenic cycle

> ③ Competitive Expression of cro and cI

 **Figure 7.** Gene Expression in Lytic and Lysogenic Cycles]

>> ○ Gene expression in lytic and lysogenic cycles regulated by two genes (cI, cro) and three promoters (PR, PL, PRM)

>> ○ Gene expression pattern in lytic cycle

>>> ○ 1st. Protein from cro inhibits transcription of cI by binding to OR3 within PRM

>>> ○ 2nd. Resultantly, PL and PR promoters are activated during the determination of lytic growth

>> ○ Gene expression pattern in lysogenic cycle

>>> ○ 1st. λ repressor generated by cI gene

>>> ○ 2nd. λ repressor binds to OR1 within PR and spans OR2 across PRM and PR, activating cI expression

>>> ○ 3rd. cro expression is inhibited instead

>>> ○ 4th. As a result, P<sub>RM</sub> promoter is activated during the determination of lysogenic growth.

>> ○ Induction of Lysogeny

>>> ○ 1st. Lysogeny induction

>>> ○ 1st - 1st. Increased cII synthesis under poor growth conditions

>>> ○ 1st - 2nd. cII protein binds to the top of the PRE promoter

>>> ○ 1st - 3rd. Forming DNA loop between OR1-OL1, OR2-OL2, OR3-OL3 with a repressor in between

>>> ○ 1st - 4th. DNA loop promotes transcription of cI gene

>>> ○ 1st - 5th. As cI expression increases due to competitive gene expression of cro and cI, lysogenic growth is induced

>> ○ 2nd. Establishment of Lysogeny

>>> ○ The cI gene is transcribed from the PRE promoter during the establishment of lysogeny, and from the PRM promoter to maintain lysogeny.

>>> ○ λ repressor binding to OR1 and OR2 also promotes the establishment of lysogeny

>>> ○ PR and cII influence the establishment of lysogeny

>>> ○ PL, which regulates lytic genes, also influences the establishment of lysogeny

> ⑤ Induction of Lysogeny

>> ○ The λ repressor protein is shaped like a dumbbell, with its N-terminal and C-terminal regions located at opposite ends.

>> ○ λ repressor monomers form dimers, and dimers combine to form tetramers

>> ○ UV light induces monomer formation of λ repressor, inhibiting cI expression

>> ○ As cro expression increases relatively due to competitive gene expression of cro and cI, lytic growth is induced

> ⑥ Antitermination: Involves N and Q proteins in lambda phage

> ⑦ Retroregulation: Hairpin structure in _int_ expression is resistant to nucleic acid digestion

 ⑹ Animal Viruses

> ① Penetration Mechanisms of Animal Viruses

>> ○ Exposed Viruses: Following endocytosis, the virion disrupts the endocytic vesicle membrane to release its contents.

>> ○ Enveloped Viruses

>>> ○ First Mechanism: Fusion of the envelope and endosomal membrane after endocytosis

>>> ○ Second Mechanism: Fusion of the host cell plasma membrane and virus envelope

> ② Influenza Virus: ssRNA Virus

>> ○ Composed of a total of 8 RNA segments

>> ○ RNA polymerase directly attached to the capsid

>> ○ [Classification of Influenza](https://jb243.github.io/pages/1389): Classification based on types of hemagglutinin and neuraminidase

>> ○ 1<sup>st</sup>. Spike glycoprotein on the viral envelope binds to cell membrane receptors

>> ○ 2<sup>nd</sup>. Fusion of envelope and endosomal membrane → release of virion into the cytoplasm

>> ○ 3<sup>rd</sup>. RNA-dependent RNA polymerase activity

>> ○ 4<sup>th</sup>. Viral protein synthesis → budding

> ③ HIV: ssRNA Virus

>> ○ Capsid contains reverse transcriptase, protease, integrase, and two identical (literally; not complementary) ssRNAs

>> ○ 1<sup>st</sup>. Glycoprotein gp120 on the envelope binds to CD4 on helper T-cells

>>> ○ Envelope consists of glycoproteins gp120 and gp41

>>> ○ Glycoprotein gp120 recognizes CD4

>>> ○ Glycoprotein gp41 recognizes CCR5

>>> ○ Individuals lacking CCR5 exhibit HIV resistance

>> ○ 2<sup>nd</sup>. Fusion with plasma membrane, penetration into the cell, and capsid removal by enzymes

>>> ○ Capsid P24: Primary marker for diagnosing HIV virus infection

>>> ○ Secondary marker for diagnosing virus infection is antibodies

>> ○ 3<sup>rd</sup>. Action of reverse transcriptase

>>> ○ 3<sup>rd</sup>-1<sup>st</sup>. RNA-dependent DNA polymerase activity: Reverse transcription of mRNA using poly T as a primer. ssRNA → ss cDNA

>>> ○ 3<sup>rd</sup>-2<sup>nd</sup>. RNAaseH activity

>>> ○ 3<sup>rd</sup>-3<sup>rd</sup>. DNA-dependent DNA polymerase activity: Functions like DNA pol Ⅰ. ss cDNA → ds
cDNA

>> ○ 4<sup>th</sup>. Double-stranded DNA (ds cDNA) becomes provirus and remains latent in the nucleus

>> ○ 5<sup>th</sup>. RNA transcription: Formation of capsid and nucleic acid after alternative splicing

>> ○ 6<sup>th</sup>. Virus is enveloped by the cell membrane and is released

 ⑺ Plant Viruses

> ① Horizontal Infection: Virus moves to adjacent cells through plasmodesmata

> ② Vertical Infection: Plant viruses infect generations through reproduction

 ⑻ Major Viral Diseases

> ① HBV (Hepatitis B Virus): Hepatocellular carcinoma, increased expression of IGF-2 (Insulin-like Growth Factor)

> ② SV40 (Simian Virus 40): Simian lymphoma, inhibition of pRb and p53 by SV40 T-antigen

> ③ Herpes Simplex Virus (Causing Chickenpox): Lymphoma, Kaposi's sarcoma, inhibition of antigen presentation on MHC I of tumor cells

> ④ HTLV (Human T-leukemia Virus): T-cell lymphoma, excessive secretion of cytokines due to T-cell infection

> ⑤ HIV Virus

<br>

<br>

## **5\. Viroids and Prions**

 ⑴ Viroids

> ① Single-stranded circular RNA composed of 200 to 300 nucleotides

> ② Replicates in plant cells without encoding proteins

 ⑵ Prions

> ① Overview

>> ○ Prions can be eliminated by heating up to 135°C

>> ○ Etymology: protein + infection

> ② Mechanism: Normal prpc in the brain undergoes structural transformation to prpsc

>> ○ prpc exhibits an α-helical structure

>> ○ prpsc exhibits a β-sheet structure

>>> ○ β-sheet structure can compactly stack, allowing abnormal prion proteins to form crystals

>>> ○ This compact structure is called an amyloid structure

>>> ○ Such compact structures are not easily degraded by enzymes

>> ○ Replication: prpc \+ prpsc → prpsc \+ prpsc

> ③ Diseases

>> ○ Symptoms: Causes transmissible spongiform encephalopathy in animals and humans, may even lead to death

>> ○ Examples: Scrapie (sheep), Bovine Spongiform Encephalopathy (BSE, mad cow disease), Creutzfeldt-Jakob Disease (CJD) in humans

>> ○ Currently, there is no cure for prion diseases

> ④ Detection

>> ○ For years, BSE testing was possible only after the animal's death

>> ○ Initial tests took relatively long time: About a week to three years

>> ○ Recent tests provide results within hours, but it's not always possible

> ⑤ History

>> ○ BSE outbreak in the UK in 1980

>> ○ Mid-2005, 2 confirmed cases and 1 suspected case in the US

 **Table. 3.** BSE and vCJD cases by country

<br>

<br>

## **6\. Comparative Microbiology**

 ⑴ Comparison of Animal Cells, BEVS (Baculovirus Expression Vector System)/Insect Cells, Yeast, and Bacteria

> ① Generally, _Escherichia coli_ (E. coli), _Saccharomyces cerevisiae_ (yeast), Chinese hamster ovary (CHO) cells are used

> ② Growth rate: Bacteria > Yeast > BEVS/Insect Cells > Animal Cells

> ③ Cost: Bacteria < Yeast < BEVS/Insect Cells < Animal Cells

> ④ Production yield: Yeast > Bacteria > BEVS/Insect Cells > Animal Cells

> ⑤ FDA approval priority: Animal Cells > Bacteria > Yeast > BEVS/Insect Cells

 ⑵ Comparison of Viruses, Bacteria, and Fungi

<br>

|                   | Virus                    | Bacteria                        | Fungi                          |
|-------------------|--------------------------|----------------------------------|--------------------------------|
| **Size**          | 0.02–0.3 µm              | 0.3–2 µm                        | 3–10 µm                        |
| **Cell Type**     | Acellular                | Prokaryotic                    | Eukaryotic                     |
| **DNA / RNA**     | Either DNA or RNA        | Both DNA and RNA               | Both DNA and RNA               |
| **Nucleic Acid Replication** | Only after entering host cell | Continuous                   | During G1 and S phases         |
| **Replication**   | Complex process          | Binary fission                  | Mitosis, meiosis               |
| **Organelles**    | Uses host organelles     | Non-membranous organelles       | Membranous organelles      |
| **Cell Membrane** | Enveloped / Non-enveloped| No sterol (exception: *Mycoplasma*) | Ergosterol                 |
| **Cell Wall**     | Absent                   | Peptidoglycan                   | Chitin, Glucan                 |

 **Table.** Comparison of Viruses, Bacteria, and Fungi

<br>

---

_Input**:** September 9, 2017, 19:00_

_Last Modified**:** January 22, 2019, 15:56_

