## **Bioinformatics Notes**

Recommended Article **:** 【Bioinformatics】 [Table of Contents for Bioinformatics Analysis](https://jb243.github.io/pages/836)

---

<br>

## **1\. Bayesian Methodology**

 ⑴ Term Definitions

> ① yng **:** single-cell count. Follows a negative binomial distribution.

> ② sn **:** cell size factor.

> ③ zn **:** cell type indicator.

> ④ μngc **:** modeled average expression. Follows a deterministic function.

> ⑤ ϕngc **:** negative binomial dispersion. Follows a deterministic function.

> ⑥ δgc **:** marker overexpression. Follows a log-normal distribution.

> ⑦ ρgc **:** marker/cell type matrix.

> ⑧ xnp **:** covariates (batch or sample).

> ⑨ βpg **:** covariate coefficients. Follows a Gaussian distribution.

> ⑩ a, b **:** dispersion basis coefficients.

> ⑪ πc **:** prior probability of cell type. Follows a Dirichlet distribution.

<br>

<br>

## **2\. Deep Learning Methods to Prioritize Noncoding Variants**

 ⑴ GWAVA(genome-wide annotation of variants)

 ⑵ DeepSea **:** predicting effects of noncoding variants with a deep learning-based sequence model.

 ⑶ DanQ **:** a hybrid convolutional and recurrent deep neural network for quantifying the function of DNA sequences.

 ⑷ DeepFun **:** predicting regulatory variants using a dense epigenomic mapped CNN model elucidated the molecular basis of trait-tissue associations.

 ⑸ DeepC **:** predicting 3D genome folding at megabase-scale using transfer learning (Nature Methods 17:1118-1124 (2020)).

 ⑹ Akita **:** predicting 3D genome folding from DNA sequences (Nature Methods 17:1111-1117 (2020)).

<br>

<br>

## **3\. gnomAD**

 ⑴ Overview

> gnomAD data is available for download through Google Cloud Public Datasets, the Registry of Open Data on AWS, and Azure Open Datasets. We recommend using Hail and our Hail utilities for working with gnomAD data. In addition to the files listed below, Terra has a demo workspace for working with gnomAD data.

 ⑵ Google Cloud Public Datasets

> Files can be browsed and downloaded using gsutil.

> $ gsutil ls gs://gcp-public-data--gnomad/release/

> gnomAD variants are also available as a BigQuery dataset.

 ⑶ Registry of Open Data on AWS

> Files can be browsed and downloaded using the AWS Command Line Interface.

> $ aws s3 ls s3://gnomad-public-us-east-1/release/

 ⑷ Azure Open Datasets

> Files can be browsed and downloaded using AzCopy or Azure Storage Explorer.

> $ azcopy ls https://datasetgnomad.blob.core.windows.net/dataset/

---

<br>

_Input**:** 2022.02.21 12:51_
