## **Chapter 6-1. Programmed Cell Death**

Recommended Post: 【Biology】 Chapter 6. [Signal Transduction](https://jb243.github.io/pages/1424)

---

**1.** [Cell Death](#1-cell-death)

**2.** [Cell Survival](#2-cell-survival)

---

<br>

## **1\. Cell Death** 

⑴ Overview

> ① [NCCD](https://www.nature.com/articles/s41418-017-0012-4)(Nomenclature Committe on Cell Death) explains cell death as a continuous cycle rather than strictly categorizing it.

<br>

![image](https://github.com/user-attachments/assets/64287a69-7534-490c-a9eb-bcccb4422c9f)

**Figure 1.** Subroutines of Major Cell Death Types

<br>

> ② Previously, apoptosis, autophagy, and necrosis were mainly studied based on morphological features.

> ③ Morphological features <span style="color: red;">■</span>, biochemical features <span style="color: green;">■</span>, other features <span style="color: purple;">■</span>.

> ④ Comparison of normal cells, apoptosis (programmed cell death), and necrosis

<br>

![image](https://github.com/user-attachments/assets/3a4b0ba6-a0a6-4088-bb0d-e6ff07eee811)

**Figure 2.** Comparison of Normal Cells, Apoptosis, and Necrosis

<br>

⑵ **Type 1.** Apoptosis (programmed cell death, type I cell death)

> ① Features

>> ○ <span style="color: red;">■</span> Reduction in cell and nuclear volume

>> ○ <span style="color: red;">■</span> Pyknosis: chromatin condensation

>> ○ <span style="color: red;">■</span> Formation of apoptotic bodies: resemble small vesicles.

>> ○ <span style="color: red;">■</span> Breakdown of the cytoskeleton

>> ○ <span style="color: red;">■</span> Minimal structural changes in mitochondria

>> ○ <span style="color: red;">■</span> Formation of blebs, i.e., membrane deformation

>>> ○ Early apoptosis: Asymmetry of the membrane is disrupted, with events such as phosphatidylserine being exposed to the extracellular environment.

>>> ○ Late apoptosis: Pores form in the membrane, resembling necrosis.

>> ○ <span style="color: green;">■</span> Karyorrhexis: Intracellular DNA undergoes regular fragmentation. DNA fragmentation can be observed through electrophoresis.

>> ○ <span style="color: green;">■</span> Phagocytosis 

>> ○ <span style="color: purple;">■</span> Does not harm surrounding cells.

>> ○ <span style="color: purple;">■</span> Progresses gradually

>> ○ <span style="color: purple;">■</span> ATP is used.

>> ○ <span style="color: purple;">■</span> Signal transduction mechanisms involving caspases

> ② **Signal Transduction 1.** Extrinsic Apoptosis

>> ○ 1<sup>st</sup>. Fas or TNF binds to the receptor.

>> ○ 2<sup>nd</sup>. CASP8 (caspase 8) and adaptor bind to form DISC.

>> ○ 3<sup>rd</sup>. DISC activates caspase 3.

>> ○ 4<sup>th</sup>. CASP3 (caspase 3) induces cell death.

> ③ **Signal Transduction 2.** Intrinsic Apoptosis

>> ○ 1<sup>st</sup>. Internal signals cause Bax and Bak on the mitochondrial inner membrane to form oligomers.

>> ○ 2<sup>nd</sup>. The oligomers form a porin, releasing cytochrome c from the intermembrane space into the cytoplasm.

>> ○ 3<sup>rd</sup>. Cyt c binds to Apaf-1, forming CARD (caspase recruitment domain).

>> ○ 4<sup>th</sup>. CARD induces cell death.

>> ○ MOMP, [CASP3](https://www.nature.com/articles/s41418-017-0012-4#:~:text=MOMPandprecipitatedbyexecutionercaspasesmainly-,CASP3,-.) are additionally involved.

> ④ Key Genes

>> ○ Fas

>> ○ Caspase: Initiation of apoptosis and digestion of intracellular components. Related to the mechanism of programmed cell death triggered by external signals. Promotes cell death.

>> ○ p53: Promotes cell death.

>> ○ Bax: Promotes cell death.

>> ○ Cyt c: Promotes cell death.

>> ○ CAD: Promotes cell death.

>> ○ DISC: Promotes cell death.

>> ○ Bcl-2: Inhibits cell death.

> ⑤ Analysis Methods

>> ○ Annexin-V staining: Staining assay for phosphatidylserine on the outer membrane of cells where [phospholipid asymmetry](https://jb243.github.io/pages/1409#2-structure-1-endomembrane) is disrupted.

>> ○ TUNEL assay (terminal transferase dUTP nick-end labeling): An assay to measure dsDNA degraded by apoptosis.

>> ○ γ-H2AX assay: An assay that indicates whether DNA double-strand binding has been disrupted.

>> ○ Calcein-AM assay: High signals indicate a large number of live cells.

>> ○ PI (propidium assay): High signals indicate a large number of dead cells.

>> ○ Ki-67 assay: Ki-67 is used as a cell proliferation marker.

>> ○ DAPI: Binds to the AT-rich regions of the DNA minor groove. Also used as an apoptosis marker.

>> ○ CD8-PD1: Higher values indicate increased apoptosis.

>> ○ Caspase 3/7: Apoptosis marker.

>> ○ PARP cleavage: Apoptosis marker.

>> ○ 8-OHdG (8-hydroxy-2’-deoxyguanosine): DNA oxidative marker.

>> ○ CC3 (cleaved caspase 3) staining: Apoptosis assay.

>> ○ 7-AAD: A fluorescent dye that binds to DNA. It cannot pass through the intact plasma membrane of living cells.

>> ○ DiOC6: A membrane-permeable fluorescent dye. When applied to cells, it accumulates in mitochondria. If the mitochondrial membrane is damaged, DiOC6 accumulation decreases.

>> ○ Trypan blue: stains dead cells blue.

>> ○ [Cytotoxicity assays](https://jb243.github.io/pages/427): MTT, WST, XTT, CCK-8

> ⑥ Examples

>> ○ Rare in unicellular organisms: However, cases of apoptosis have been found in [yeast colonies](https://www.nature.com/articles/s41418-017-0012-4#:~:text=lifecycle\)in-,colonies,-\(suchasseveral).

>> ○ Commonly observed during development.

>> ○ During the metamorphosis of a tadpole, its tail disappears.

⑶ **Type 2.** Autophagy (autophagy-dependent cell death; ADCD, type II cell death)

> ① Features

>> ○ <span style="color: red;">■</span> Formation of DMA (double-membraned autolysosome)

>> ○ <span style="color: red;">■</span> DMA: macroautophagy, microautophagy, chaperone-mediated autophagy, etc.

>> ○ <span style="color: green;">■</span> Increased lysosomal activity

> ② Signal Transduction

>> ○ mTOR, Beclin-1, p53 signaling pathways

> ③ Key Genes

>> ○ ATG5

>> ○ ATG7

>> ○ LC3

>> ○ Beclin-1

>> ○ DRAM3

>> ○ TFEB

⑷ **Type 3.** Necrosis (type III cell death)

> ① Features

>> ○ <span style="color: red;">■</span> Rupture of the plasma membrane, releasing internal enzymes: Damages surrounding cells.

>> ○ <span style="color: red;">■</span> Swelling of cytoplasm and cellular organelles.

>> ○ <span style="color: red;">■</span> Moderate chromatin condensation

>> ○ <span style="color: green;">■</span> Decrease in ATP levels

>> ○ <span style="color: green;">■</span> Inflammatory response

>> ○ <span style="color: green;">■</span> No phagocytosis

>> ○ <span style="color: green;">■</span> No lysosomal activity

>> ○ <span style="color: purple;">■</span> ATP is not used.

>> ○ <span style="color: purple;">■</span> No caspase-mediated signaling mechanisms.

>> ○ <span style="color: purple;">■</span> Progresses rapidly

> ② Signal Transduction

>> ○ TNFR1-RIP1/RIP3-MLKL signaling pathway

>> ○ PKC-MAPK-AP-1 signaling pathway

>> ○ ROS-related metabolic regulation signaling pathway

> ③ Key Genes

>> ○ RIP1

>> ○ RIP3

⑸ **Type 4.** Ferroptosis

> ① Features

>> ○ <span style="color: red;">■</span> Small mitochondria with increased mitochondrial membrane density.

>> ○ <span style="color: red;">■</span> Decrease or loss of mitochondrial cristae

>> ○ <span style="color: red;">■</span> Destruction of the mitochondrial outer membrane

>> ○ <span style="color: green;">■</span> Iron accumulation

>> ○ <span style="color: green;">■</span> Lipid peroxidation: Especially in the cell membrane

> ② Signal Transduction

>> ○ Xc-/GPX4, MVA, sulfur transfer pathways

>> ○ P62-Keap1-NRF2 pathway

>> ○ p53/SLC7A11, ATG5-ATG7-NCOA4 pathway

>> ○ p53-SAT1-ALOX15 pathway

>> ○ HSPB1-TRF1, FSP1-COQ10-NAD(P)H pathway

> ③ Key Genes

>> ○ GPX4

>> ○ TRF1

>> ○ SLC7A11

>> ○ NRF2

>> ○ NCOA4

>> ○ p53

>> ○ HSPB1

>> ○ ACSL4

>> ○ FSP1

> ④ Ferroptosis Inducers

>> ○ Erastin

>> ○ Piperazine erastin

>> ○ Imidazole ketone erastin

>> ○ Sulfasalazine

>> ○ Sorafenib

>> ○ Glutamate

>> ○ (1S,3R)-RSL3

>> ○ C' dots & amino acid starvation

>> ○ FPEF

>> ○ FeGd-HN@Pt@LF/RGD2

>> ○ ZVI

>> ○ DGU:Fe/Dox

>> ○ FeCo-DOX@MCN

>> ○ MON-p53

> ⑤ Ferroptosis Inhibitors

>> ○ Liproxstatin-1

⑹ Other Types

> ① Accidental cell death (ACD): Sudden external shock

> ② Anoikis: Caused by the loss of [integrins](https://jb243.github.io/pages/1409).

> ③ Autosis: A type of autophagy dependent on Na<sup>+</sup>/K<sup>+</sup>-ATPase.

> ④ Cellular senescence: Characterized by the secretion of SASP (senescence-associated secretory phenotype).

> ⑤ Cell scorch death

> ⑥ Copper-induced cell death

> ⑦ Efferocytosis: Involves phagocytes.

> ⑧ Entotic cell death: Actomyosin-dependent cell-in-cell internalization is involved.

> ⑨ Immunogenic cell death (ICD)

> ⑩ Lysosome-dependent cell death (LDCD): Involves primary LMP, cathepsin, MOMP, caspase, etc.

> ⑪ Mitochondrial permeability transition (MPT)-driven necrosis

> ⑫ Mitotic catastrophe

> ⑬ NETotic cell death

> ⑭ Parthanatos: Involves PARP1 hyperactivation.

> ⑮ Pyroptosis: A type of inflammatory cell death characterized by membrane pores formed by the gasdermin protein family.

<br>

<br>

## **2\. Cell Survival** 

⑴ **Example 1.** Inhibition of Apoptosis in Phagocytes

> ① Phagocytes do not undergo apoptosis upon encountering foreign substances: To perform immune responses.

> ② [Toll-like receptors](https://jb243.github.io/pages/86) (TLR)

>> ○ [TLR-1](https://en.wikipedia.org/wiki/TLR_1): Recognizes multiple triacyl lipopeptides.

>> ○ [TLR-2](https://en.wikipedia.org/wiki/TLR2): Recognizes lipoteichoic acid. Activates innate immunity.

>> ○ [TLR-3](https://en.wikipedia.org/wiki/TLR3): Recognizes dsRNA in viruses.

>> ○ [TLR-4](https://en.wikipedia.org/wiki/TLR4): Recognizes LPS (lipopolysaccharide) in Gram-negative bacteria.

>> ○ [TLR-5](https://en.wikipedia.org/wiki/TLR5): Recognizes flagellin, the subunit of prokaryotic flagella.

>> ○ [TLR-6](https://en.wikipedia.org/wiki/TLR6): Recognizes multiple diacyl lipopeptides.

>> ○ [TLR-7](https://en.wikipedia.org/wiki/TLR7): Recognizes single-stranded RNA.

>> ○ [TLR-8](https://en.wikipedia.org/wiki/TLR8): Recognizes small synthetic compounds and single-stranded RNA.

>> ○ [TLR-9](https://en.wikipedia.org/wiki/TLR9): Recognizes unmethylated CG sequences (C<sup>p</sup> DNA) and oligodeoxynucleotide DNA.

> ③ NF-κB Signal Transduction

<br>

![image](https://github.com/user-attachments/assets/67b9fdff-8440-46df-88c3-57a829e21d5f)

**Figure 3.** NF-κB and Iκ-B Experiment (Western Blot Results)

<br>

>> ○ 1<sup>st</sup>. NF-κB is normally bound to the Iκ-B protein in the cytoplasm, forming a complex.

>>> ○ Iκ-B inhibits NF-κB from translocating to the nucleus.

>>> ○ Protein X refers to Iκ-B.

>> ○ 2<sup>nd</sup>. Lipopolysaccharide (LPS), etc. bind to toll-like receptors (TLR).

>> ○ 3<sup>rd</sup>. TNF-α trimer binds to the receptor.

>> ○ 4<sup>th</sup>. IKK (inhibitor kappa kinase) is activated.

>> ○ 5<sup>th</sup>. Iκ-B is activated and degraded in the cytoplasm.

>>> ○ When treated with a proteasome inhibitor, NF-κB does not separate from Iκ-B, indicating that Iκ-B is degraded under normal conditions.

>> ○ 6<sup>th</sup>. NF-κB separates from Iκ-B and translocates to the nucleus.

>> ○ 7<sup>th</sup>. NF-κB produces proteins involved in the inflammatory response of macrophages.

>>> ○ Macrophages exist as inflammatory M1 type and anti-inflammatory M2 type.

>>> ○ NF-κB promotes the conversion of macrophages to the M1 type.

> ④ **Pathway 1.** MyD88-dependent Pathway

>> ○ Overview: Mechanism through signal transduction. Activates NF-κB. Increases TNF-α and IL-1β.

>> ○ TLR → MyD88 → TRAF6 → NF-κB → NF-κB translocates to the nucleus.

>> ○ TLR → MyD88 → TRAF6 → MAPK → AP-1

> ⑤ **Pathway 2.** TRIF-dependent Pathway

>> ○ Overview: Mechanism through endosomes.

>> ○ TLR → TAM, TRIF → IRF3 → IRF3 translocates to the nucleus → type I IFN.

>> ○ TLR → TAM, TRIF → TRAF6 → NF-κB → NF-κB translocates to the nucleus.

>> ○ TLR → TAM, TRIF → TRAF6 → MAPK → AP-1.

⑵ **Example 2.** Immediate Early Stress Response

> ① Stress → ASK-1 (= MAP3K5) → JNK p38<sup>MAPK</sup> → ATF2, JUN, FOS → Translocates to the nucleus → AP-1 (activator protein 1)

<br>

![image](https://github.com/user-attachments/assets/ece5b135-ac24-4edc-9884-4b8327189250)

**Figure 4.** Immediate Early Stress Response

<br>

⑶ **Example 3.** Heat Shock Response (Unfolded Protein Response)

<br>

![image](https://github.com/user-attachments/assets/d067955a-49dd-4536-9702-6b0e622df0d4)

**Figure 5.** Heat Shock Response

<br>

> ① Stress → Proteotoxic stress (endoplasmic reticulum) → PERK, ATF6, HSF1, XBP1 → Translocates to the nucleus → ATF4, ATF6, HSF1, XBP1.

⑷ **Example 4.** Anti-Oxidant Response

> ① Stress → NFE2L2

<br>

![image](https://github.com/user-attachments/assets/e389bf11-7dd1-4843-8b94-8e05a784c73c)

**Figure 6.** Anti-Oxidant Response

<br>

⑸ **Example 5.** Inflammatory Response

> ① Stress, hypoxia → NF-κB → Translocates to the nucleus → NF-κB.

<br>

![image](https://github.com/user-attachments/assets/3a2c7162-5e55-45ea-890a-5cf46c35712d)

**Figure 7.** Inflammatory Response

<br>

⑹ **Example 6.** Hypoxic Response

> ① Stress, hypoxia → HIF-1α → Translocates to the nucleus → HIF-1α, HIF-1β.

<br>

![image](https://github.com/user-attachments/assets/798f74c0-c43c-4b19-89e4-b13943d65472)

**Figure 8.** Hypoxic Response

<br>

---

_Input: 2019.03.14 21:19_

_Modified: 2022.06.13 13:42_
