## **Chapter 10-2. Epigenomics Sequencing**
 
Recommended Reading: 【Biology】 Chapter 10. [Genome Projects and Sequencing Technologies](https://jb243.github.io/pages/75)

---

**1. Type 1.** [Identifying gene function](#1-type-1-identifying-gene-function)

**2.** **Type 2.** [Identifying transcription regulation](#2-type-2-identifying-transcription-regulation)

**3.** **Type 3.** [Identifying post-translational regulation](#3-type-3-identifying-post-translational-regulation)

**4. Type 4.** [Programmable cell function](#4-type-4-programmable-cell-function)

---

**a.** [Epigenetics Library](https://jb243.github.io/pages/398)

---

<br>

## **1. Type 1.** Identifying gene function

⑴ Perturb-seq

> ① 1<sup>st</sup>. Treat Cas9-expressing cells with various types of gRNA libraries.

>> ○ By using CRISPR-Cas9, various perturbations occur in each cell depending on the type of gRNA.

>> ○ At least 2.5 million cells should remain after filtering.

>> ○ There should be at least 100 cells per perturbation on average.

>> ○ Each cell should have approximately 10,000 UMIs

> ② 2<sup>nd</sup>. Use sequencing technologies that detect both gRNA and mRNA (e.g., scRNA-seq, [MERFISH](https://www.biorxiv.org/content/10.1101/2023.11.30.569494v1.full))

> ③ 3<sup>rd</sup>. Grouping cells based on gRNA expression: Cells are naturally grouped according to perturbation conditions.

> ④ 4<sup>th</sup>. Identify gene function.

>> ○ Premise: Genes with similar functions are likely to exhibit similar expression patterns.

>> ○ **Discoverable Result 1.** Changes in gene expression due to perturbations.

>> ○ **Discoverable Result 2.** Differences in perturbation effects due to genetic variants.

<br>

![image](https://github.com/user-attachments/assets/7e16c345-f9e2-46fa-b722-ac9caa937e3e)

**Figure 1.** General schematic of Perturb-seq including validation experiments

<br>

⑵ In vivo Perturb-seq

> ① 1<sup>st</sup>. Deliver AAV (adeno-associated virus) containing Cre to mice to induce Cas9 expression.

> ② 2<sup>nd</sup>. Deliver lentivirus containing sgRNA.

> ③ 3<sup>rd</sup>. In vivo, the CRISPR-Cas9 system induces various perturbations in each Cas9-expressing cell depending on the gRNA.

> ④ 4<sup>th</sup>. Perform scRNA-seq and MERFISH after approximately 10 days.

<br>

![image](https://github.com/user-attachments/assets/08ff071a-7dba-4c71-9bf2-d95478cdb6f2)

**Figure 2.** In vivo Perturb-seq process

<br>

<br>

## **2. Type 2.** Identifying transcription regulation

⑴ BS-seq (bisulfite sequencing)

> ① Treat with bisulfite to determine methylation patterns.

> ② Enables epigenomic profiling.

> ③ **Application 1.** snmC-seq 

>> ○ Plate-base sequencing.

>> ○ Measures sum of 5-methyl- and 5-hydroxymethyl-cytosines.

>> ○ 1-2 million reads per cell.

⑵ **ChIP-seq** (chromatin immunoprecipitation sequencing)

> ① **Definition:** Enables analysis of DNA regions bound to specific proteins, such as transcription factors.

> ② Combines DNA sequencing with ChIP (chromatin immunoprecipitation) to identify binding sites of DNA-associated proteins.

> ③ 1<sup>st</sup>. Treat with a cross-linking agent (e.g., formaldehyde) to fix proteins (e.g., transcription factors) and DNA.

> ④ 2<sup>nd</sup>. Lyse the cells, leaving only the DNA/protein complexes.

> ⑤ 3<sup>rd</sup>. Use sonication to break DNA into small fragments (200-350 bp); cross-linked DNA remains intact.

> ⑥ 4<sup>th</sup>. Add antibodies attached to magnetic beads; applying a magnetic field allows isolation of specific proteins and their linked DNA.

> ⑦ 5<sup>th</sup>. Reverse cross-linking: Apply heat to the precipitate to separate DNA from proteins.

> ⑧ 6<sup>th</sup>. Perform sequencing to determine the DNA sequence. Proceeds with PCR amplification, fragment size selection, adaptor addition.

> ⑨ 7<sup>th</sup>. Compare the sequencing results to the genome reference to identify binding sites of transcription factors, etc.

<br>

![image](https://github.com/user-attachments/assets/1f66c950-10ce-4dab-8444-21698238f282)

**Figure 3.** ChIP-seq process

<br>

> ⑩ **Application 1.** ChIP-chip: HTS version of ChIP-chip is ChIP-seq.

> ⑪ **Application 2.** ChIP-PET

> ⑫ **Application 3.** ChIA-PET: ChIP-seq identifies only binding locations of specific proteins, while ChIA-PET investigates interactions between bound DNA regions.

> ⑬ **Application 4.** RIP-seq

> ⑭ **Application 5.** CLIP-seq 

> ⑮ **Application 6.** meDIP-seq: DNA methylation or hydroxymethylation.

> ⑯ **Application 7.** ChIP-exo: Uses lambda exonuclease digestion. Difficult experimentally.

> ⑰ **Application 8.** CUT&RUN (Cleavage under Targets and Release using Nuclease)

>> ○ Uses Micrococcal Nuclease (MNase).

> ⑱ **Application 9.** CUT&Tag (Cleavage under Targets and Tagmentation)

>> ○ Overview 

>>> ○ Tn5 transposase is an enzyme that cuts open regions of DNA, generating various fragments such as single nucleosomes, dimers, trimers, and more.

>>> ○ The Tn5 transposase catalyzes DNA insertion by creating sticky ends.

>>> ○ It can directly merge adaptors, allowing immediate sequencing.

>> ○ Procedure

>>> ○ **Step 1.** Bind nuclei to beads to anchor them. Magnetic capture leads to high retention of cells.

>>> ○ **Step 2.** Add primary antibody: Quality of data is highly dependent on quality (specificity & sensitivity) of antibody.

>>> ○ **Step 3.** Add secondary antibody.

>>> ○ **Step 4.** Conjugated Tn5-transposase binds to complex and cuts surrounding areas of open DNA.

>>> ○ **Step 5.** PCR amplification, fragment size selection, sequencing 

>> ○ Types

>>> ○ MuLTI-Tag: Minimizes crossover in multiplexing via direct barcode conjugation.

>>> ○ multi-CUT&Tag: Uses barcoded Tn5/pA-antibody complexes. Identifies colocalization of marks.

>>> ○ [spatial-CUT&Tag](https://www.science.org/doi/10.1126/science.abg7216): Spatially visualizes histone modifications and chromatin states on tissue sections.

<br>

|   | **CUT&Tag** | **CUT&RUN** | **ChIP-Seq** |
| --- | --- | --- | --- |
| Native condition? | Yes | Yes | No |
| Sample input | Nuclei | Cells or nuclei | Sheared chromatin |
| Cell number | 100,000 cells | 500,000 cells | 1-10 million cells |
| Chromatin fragmentation | Tn5-based tagmentation | MNase digestion | Sonication |
| Ideal target | Histone PTM | Histone PTM, Chromatin-associated protein, Remodeler | Histone PTM, Chromatin-associated protein |
| Secondary antibody | Yes | No | No |
| Library preparation | No (direct-to-PCR) | Yes | Yes |
| Integrated library | Possible; uses tagmentation | Impossible | Impossible |
| Sequencing depth | 5-8 million reads | 3-5 million reads | 20-50 million reads |
| Workflow length | < 2 days | < 3 days | ~ 1 week |
| Automation compatibility | High | High | Low |
| Signal-to-noise | High | High | Low |

**Table 1.** CUT&Tag vs. CUT&RUN vs. ChIP-Seq ([ref](https://www.activemotif.com/blog-cut-tag), [ref](https://www.activemotif.com/blog-cut-tag), [ref](https://www.epicypher.com/resources/blog/cut-and-run-vs-cut-and-tag-which-one-is-right-for-you/), [ref](https://www.epicypher.com/resources/blog/cut-and-run-vs-cut-and-tag-which-one-is-right-for-you/))

<br>

⑶ **Hi-C seq** (high throughput chromatin conformation capture sequencing)

> ① Definition: Investigates sequences that are naturally close together on chromosomes.

> ② Examines DNA distances to reveal the 3D folding structure of chromosomes within the nucleus.

<br>

![image](https://github.com/user-attachments/assets/6124aa4e-024e-4bc2-a47c-8e59fdd4ca07)

**Figure 4.** Hi-C seq process

<br>

> ③ **Application 1.** ChIA-PET

>> ○ Difference: Hi-C investigates all naturally occurring DNA-DNA interactions, while ChIA-PET focuses on DNA-DNA interactions mediated by specific proteins.

⑷ DNA ticker tape (prime editing)

> ① 1<sup>st</sup>. Initially, only the first site is activated.

> ② 2<sup>nd</sup>. After the first event, the second site becomes activated.

> ③ 3<sup>rd</sup>. Sequentially records molecular events over time.

⑸ ENGRAM (enhancer-driven genomic recording of transcriptional activity in multiplex)

> ① Uses perRNA linked to synthetic enhancers.

> ② Records the sequence and intensity of signaling.

> ③ Reference: Chen et al., bioRxiv (2021)

⑹ ATAC-seq (Assay for transposase-accessible chromatin with sequencing)

> ① Overview

>> ○ Definition: A sequencing technique that identifies euchromatin regions. Since the chromosomes are in pairs, each signal in ATAC-seq can have values of 0, 1, or 2.

>> ○ Pseudo-expression: Euchromatin regions can be inferred as areas of gene expression.

>> ○ Tn5 transposase is an enzyme that cuts open regions of DNA, generating various fragments such as single nucleosomes, dimers, trimers, and more.

>> ○ The Tn5 transposase catalyzes DNA insertion by creating sticky ends.

>> ○ It can directly merge adaptors, allowing immediate sequencing.

>> ○ The ~10.5 bp periodicity observed in ATAC-seq is related to the fact that one full turn of the DNA helix requires 10 bp.

>>> ○ The fourth alpha-helix in Tn5 acts as the major “recognition helix” and makes several base pair-specific contacts in the major groove of the DNA from positions 7 to 13. ([ref](https://biology.kenyon.edu/BMB/Jmol2007/Tn5_Transposase/index.htm))

>>> ○ Tn5 transposase bends the helical axis, leading to an increase in the roll and tilt angles, and in deviations of the major and minor groove widths and depths, compared with average B-DNA. ([ref](https://raymentlab.biochem.wisc.edu/wp-content/uploads/sites/1624/2022/08/117rayment2000_davies.pdf))

>>> ○ This may explain the 10.5 bp periodicity, which is a little different from 10 bp periodicity in average B-DNA.

<br>

![image](https://github.com/user-attachments/assets/7ff7a5c1-21c2-4f48-a895-de0397e62f8d)

**Figure 5.** ATAC-seq fragment size distribution

<br>

> ② **Type 1.** bulk ATAC-seq

>> ○ **Step 1.** Nuclei isolation

>> ○ **Step 2.** Tn5 transposase treatment: This cuts less condensed open regions in the chromosome and inserts DNA sequence tags.

>> ○ **Step 3.** Amplification & sequencing

>> ○ **Step 4.** Data analysis

> ③ **Type 2.** scATAC-seq

>> ○ Defines cell type-specific CREs to identify regulatory TFs and cell types associated with diseases and traits.

>> ○ Used in interpreting GWAS variants.

>> ○ **Comparison of scRNA-seq and scATAC-seq**

>>> ○ scRNA-seq: x<sub>ij</sub> ∈ ℤ<sub>≥0</sub>

>>> ○ scATAC-seq: x<sub>ij</sub> ∈ {0, 1}, j ≫ i

> ④ **Type 3.** spatial ATAC-seq

> ⑤ **Type 4.** FAIRE-seq: Formaldehyde to crosslink chromatin, and phenol-chloroform to extract sheared DNA. 

> ⑥ **Type 5.** DNaseI-seq: DNase I endonuclease to digest chromatin. Can identify the types of transcription factors interacting with chromatin.  

<br>

![image](https://github.com/user-attachments/assets/b4a21bc8-3d4e-4fb2-ba9d-d4f68f002fcd)

**Figure 6.** Example results of DNAseI-seq

<br>

> ⑦ **Type 6.** MNase-seq: Endo-exonuclease to processively digest DNA until obstruction. 

<br>

![image](https://github.com/user-attachments/assets/3b36846f-21eb-4cbd-80d5-adb8c4d1fc9b)

**Figure 7.** Comparison of ATAC-seq with Various Sequencing Techniques

<br>

⑺ NOMe-seq (nucleosome occupancy and methylome sequencing)

> ① Simultaneously measures DNA methylation and nucleosome occupancy.

> ② Defines nucleosome-depleted regions (NDR) where TFs bind.

> ③ Identifies transcription factors interacting with chromatin.

⑻ MBD-seq  

⑼ Bru-seq(Bromouridine sequenceing) & BruChase-seq  

> ① Newly transcribed nascent RNA is labeled with Bru (bromouridine) and then sequenced.  

> ② Used for studying RNA synthesis, RNA stability, and splicing.

> ③ 1<sup>st</sup>. Label nascent RNA in cells with bromouridine

> ④ 2<sup>nd</sup>. Isolate Bru-RNA

> ⑤ 3<sup>rd</sup>. Prepare cDNA libraries from isolated Bru-RNA and perform deep sequencing

⑽ TT-seq(transient transcriptome sequencing)

> ① 1<sup>st</sup>. Incorporation of 4-thiouridine into nascent RNA

> ② 2<sup>nd</sup>. Isolate labelled RNA

> ③ 3<sup>rd</sup>. Prepare cDNA libraries and perform deep sequencing

⑾ MNIST-seq: RNA methylation on chromatin regulation

⑿ GRO-seq(global run-on sequencing)

> ① 1<sup>st</sup>. Cell harvesting and permeabilization

> ② 2<sup>nd</sup>. Nuclear run-on reaction

> ③ 3<sup>rd</sup>. RNA extraction and BrU-labeled RNA is isolated

> ④ 4<sup>th</sup>. Fragmentation and library preparation

⒀ PRO-seq(precision run-on sequencing)

> ① 1<sup>st</sup>. Cell harvesting and permeabilization

> ② 2<sup>nd</sup>. Nuclear run-on with biotinylated NTPs

> ③ 3<sup>th</sup>. RNA extraction and streptavidin pulldown

> ④ 4<sup>th</sup>. 3' adapter ligation and library prep

<br>

<br>

## **3. Type 3.** Identifying post-translational regulation

⑴ Ribo-seq  

> ① Sequencing of ribosome-protected RNA, indicating active translation.  

> ② scRibo-seq

>> ○ Measures ribosomal occupancy per single codon.

>> ○ 1<sup>st</sup>. FACS and lysis.

>> ○ 2<sup>nd</sup>. Nuclease footprinting: MNase nuclease → inactivation → release of footprints.

>> ○ 3<sup>rd</sup>. Create small-RNA library: End repair → 3' ligation → 5' ligation → cDNA synthesis → indexing PCR.

⑵ STAMP-RBP

> ① Overview

>> ○ Over 500 amino acid types have been identified, and 140 of these noncanonical amino acids (NCAAs) are known to be incorporated into proteins.

>> ○ Canonical RNA motifs: GNRA, UNCG, kinkturn, Shine–Dalgarno, sarcin–ricin, etc.

>> ○ STAMP-RBP: Uses scRNA-seq to identify RNA-binding proteins (RBPs).

> ② Process

>> ○ 1<sup>st</sup>. Attach APOBEC to RBPs.

>> ○ 2<sup>nd</sup>. Enable C-U editing at the site where APOBEC and mRNA bind, replacing cytosine (C) with uracil (U).

>> ○ 3<sup>rd</sup>. RNA-seq

>> ○ 4<sup>th</sup>. Use SAILOR to identify C-U editing sites.

>> ○ Can also identify isoform-specific binding profiles using long-read sequencing.₩

<br>

<br>

## **4. Type 4.** Programmable cell function

⑴ RADARS

> ① 1<sup>st</sup>. When the target transcript is present, it forms a dsRNA structure, allowing ADAR to induce A-to-C editing.

> ② 2<sup>nd</sup>. This editing induces cellular behaviors such as GFP expression and caspase activation.

⑵ [LADL (light-activated dynamic looping)](https://www.nature.com/articles/s41592-019-0436-5)

> ① An example of photo-activatable gene expression.

<br>

---

_Input: 2022.01.10 00:03_

_Modified: 2023.01.28 23:12_
