## **Gene Score Library** 

Recommendation: 【Bioinformatics】 [Bioinformatics Index](https://jb243.github.io/pages/836)

---

**1.** [Overview](#1-overview)

**2.** [BP](#2-bpbiological-process)

**3.** [CC](#3-cccellular-component)

**4.** [MF](#4-mfmolecular-function)

**5.** [Others](#5-others)

---

**a.** [Gene Library](https://jb243.github.io/pages/2212) 

**b.** [Cell Type Marker Gene](https://jb243.github.io/pages/1846)

---

<br>

## **1.** Overview

> ○ Database: GO, MSigDB, HALLMARK, PROGENy, KEGG, Reactome, etc.

> ○ For other information, see [https://www.scinapse.io/](https://www.scinapse.io/), [https://typeset.io/](https://typeset.io/), which curated billions of papers.

<br>

<br>

## **2. BP**(biological process)

> ○ `Fibrosis marker`: COL1A1, COL3A1, COL5A1, SPP1, FN1, POSTN, FGF1, PDGFB, CTGF, FGF7, IL34, CDC42, CXCL12 (SDF1), TGFBR1, TGFBR2, CCL3, CCL4, COL5A2, ADIPOQ

> ○ `Corona virus invasion pathway`: ACE2, TMPRSS2, PCK5, PCSK7, CTSL, CTSB, NRP1

> ○ `Hypoxia related score`: DCN, DDIT4, PRKCA, NDRG1

> ○ `Genes that express high in the G1 phase during the cell cycle`: GLIPR1, ORAI3, OSMR

> ○ `Genes that express high in the G1/S phase during the cell cycle`

>> ○ CEP57, CDCA7L ([ref](https://www.biorxiv.org/content/10.1101/2021.07.19.452956v2.full))

>> ○ TYMS, ATAD2, RRM2, KIAA0101, TK1, PCNA, DNMT1, UBE2T, DUT, DHFR, TUBB, CENPM, RAD51AP1, ZWINT, RNASEH2A, TMEM106C, HMGB2, TUBA1B, HIST1H4C, HELLS, CLSPN, FEN1, NUSAP1, STRA13, CDK1, RFC4, RRM1, DNAJC9, WDR34, STMN1, USP1, CENPU, ASF1B, GMNN, SLBP, FAM111B, CENPK, PHGDH, PSMC3IP, CHAF1A, RFC2, WDR76, CDK4, RPA2, GGCT, MCM7, CDC6, MCM3, EZH2, PSIP1, SPC25, NASP, TMPO, TUBG1, TEX30, PKMYT1, SIVA1, MTHFD2 ([ref](https://www-nature-com.proxy.lib.umich.edu/articles/s41588-023-01357-3#Sec46))

> ○ `Genes that express high in the S phase during the cell cycle`: ABHD10, CCDC14, CDKN2AIP, NT5DC1, SVIP, PTAR1 ([ref](https://www.biorxiv.org/content/10.1101/2021.07.19.452956v2.full))

> ○ `Genes that express high in the G2 phase during the cell cycle`: ANKRD36C, YEATS4, DCTPP1 ([ref](https://www.biorxiv.org/content/10.1101/2021.07.19.452956v2.full))

> ○ `Genes that express high in the G2/M phase during the cell cycle`

>> ○ SMC4, TMPO, LMNB1, HINT3 ([ref](https://www.biorxiv.org/content/10.1101/2021.07.19.452956v2.full))

>> ○ CENPF, CCNB1, TOP2A, TPX2, CDC20, NUSAP1, BIRC5, CDKN3, MKI67, PRC1, CCNB2, HMGB2, PTTG1, ARL6IP1, KPNA2, UBE2S, CKS2, UBE2C, TUBB4B, ANP32E, AURKB, CDK1, MAD2L1, KIF20B, TACC3, KNSTRN, CDCA3, CKS1B, SMC4, RACGAP1, TUBA1C, HMGB3, UBE2T, ASPM, SGOL2, PLK1, GTSE1, CEP55, ECT2, CCNA2, NUF2, CKAP2, TROAP, DEPDC1, AURKA, TUBA1B, PBK, CDC25B, CENPW, CALM2, GPSM2, BUB3, CENPA, SKA2, NEK2, HN1, HMMR, KIF23, KIFC1, TUBB, CDCA8, KIF2C, HMGN2, FAM83D, DLGAP5, CENPE, STMN1, TMPO, DBF4, EMC9, DTYMK, FAM64A, PTMS, NDC80, NCAPD2, FOXM1, H2AFX, CKAP5, SGOL1, RAD21, MZT1, SUN2 ([ref](https://www-nature-com.proxy.lib.umich.edu/articles/s41588-023-01357-3#Sec46))

> ○ `Genes that express high in the M phase during the cell cycle`: HMG20B, HMGB3, HPS4 ([ref](https://www.biorxiv.org/content/10.1101/2021.07.19.452956v2.full))

> ○ `Genes that express high in the G0 phase during the cell cycle`: CDKN1A, CDKN1B, CDKN1C ([ref](https://www.biorxiv.org/content/10.1101/2021.07.19.452956v2.full))

> ○ `Cell cycle related score`: CDC20, CCNB1, CENPF, PTTG1

> ○ `Regulation of cell growth`: AREG, EGR1

> ○ `Vascular remodeling`: VEGFB, VEGFC, ANGPTL4

> ○ `Type 1 immunity`: CCL3, CXCL9, CXCL10, CXCL11, IL18

> ○ `Type 2 immunity`: CCL11, CCL13, CD5L, IL4, IL5, IL13, IL24

> ○ `Immune activation`: CD40LG, CD48, CD80, CD86

> ○ `Metastasis`: PLAU, PLAUR, RELB, AREG

> ○ `Immune checkpoint genes`: HAVCR2, CD47, LGALS9, CD276, LAG3, PDCD1, BTLA, CD274 (PD-L1), PDCD1LG2 (PD-L2), VSIR, MIF, SIGLEC10, CTLA4, TIGIT, CD96, TNFRSF1B

> ○ `Proliferation genes`: MKI67, MCM2, PCNA, STMN1, Mki67, Mpp1, Aurkb, Top2a

> ○ `Immunosuppression`: APOE, GPNMB 

> ○ `E2F targets`: ATAD2, MCM2, BIRC5

> ○ `Neuron differentiation`: STMN2, STMN4, NTUBB3

> ○ `Antigen presentation`: CD74, HLA-DRB1, HLA-DQA2, HLA-C, HLA-DRB5, H2-D1, H2-K1, B2m

> ○ `Antigen processing`: LY75, TAP1

> ○ `Migratory marker`: CCR7, CCL19 

> ○ `Keratinization`: SPRR2D, SPRR2E, SPRR2A, CRCT1, CNFN

> ○ `Cell apoptosis`: FOXO3, MCL1

> ○ [`Malignancy score`](https://onlinelibrary.wiley.com/doi/10.1002/cam4.4547): KRT18, IRX2, NAPSA, SPINK13, KRT7, CAPN8, GPRC5A

> ○ `Immune response`: STAT4, MYB

> ○ [`Glycolysis`](https://jb243.github.io/pages/1410): ENO1, GAPDH, GPI, HK1, PGAM1, PKM

> ○ `PPP(pentose phosphate pathway)`: G6PD, PGD, PGLS, RPE, RPIA, TALDO1, TKT

> ○ `TCA cycle`: ME1 (Cytosol), ME2/3 (Mitochondria), DLST, FH, OGDH, SDHA

> ○ `α-ketoglutarate synthesis`: IDH1 (Cytosol), IDH2 (Mitochondria), GLUD1/2 (Mitochondria)

> ○ `Folate metabolism`: MTHFD1 (Cytosol), MTHFD2 (Mitochondria), DHFR

> ○ `NADH metabolism`: NNT (Mitochondria)

> ○ `NADPH generation`: G6PD, GLUD1, ME1

> ○ `Mitochondrial and folate-dependent NADPH metabolism`: GLUD1/2, IDH2, ME2, MTHFD1, MTHFD2, MTHFR

> ○ `Mitochondrial proteins`: TIMM44, PHB1, SAMM50, PRELID1

> ○ `Amino acid metabolism`: PYCR2 

> ○ `Glycine metabolism`: AMT, GCSH, GLDC

> ○ `Oxidation-reduction reaction`: AKR1A1, GSR, PRDX6 

> ○ `Central carbon metabolism`: ACO2, ALDOC, PGAM2, PKM

> ○ `Malignant cancer cell subtype CLS`: AGR2, ANXA10 

> ○ `Malignant cancer cell subtype BSL`: S100A2, SLC2A1, AREG, KRT5, KRT6A, KRT6C 

> ○ `Malignant cancer cell subtype NRP`: CNTN4, KCNJ16, FGG, NR1H4, SPP1 

> ○ `HCC recurrence risk`: SPON2, ZFP36L2, ZFP36, VIM, HLA-DRB1 ([ref](https://doi.org/10.1038/s41586-025-08668-x))

> ○ `neuroinflammatory response`: Tyrobp, Trem2, Igf1

> ○ `endocytosis activity, amyloid-beta clearance`: Apoe, Cd63, Npc2, Lyz2

> ○ `liver zonation marker`: Cyp2f2, Apoa4, Hal, Cyp2e1, Gulo, Cyp1a2, Oat, Glul

> ○ `liver-disease-related gene score`

<br>

| group | gene | steatosis | cirrhosis |
| --- | --- | --- | --- |
| low lipoprotein output | PTPRD, PNPLA3, TM6SF2 | ↑↑↑↑ | ↑↑↑↑ |
| high lipoprotein input | APOE | ↑↑↑ | ↑↑↑ |
| low lipid burn | FTO | ↑↑ | \- |
| insulin | INSR, PNPLA2, GRB14, SREBF1 | ↑↑↑ | \- |
| absorb | MTTP | ↑\* | \- |
| glucose | GCKR, TRIB1 | ↑↑↑ | \- |
| TG divert | MBOAT7, MARC1, TOR1B, ADH1B, GPAM | ↑↑↑↑ | ↑↑↑ |
| epidemiologic association |  | ↑ | ↑ |

**Table 1.** liver-disease-related gene score

<br>

<br> 

## **3. CC**(cellular component)

> ○ `Complement activation`: C1QA, C1QB, C1QC, CFD, C7

> ○ `Immunoglobulin`: IGKC, IGKV4-1, IGHA1, IGHG2, IGLV9-49, IGHV1-46

> ○ `Extracellular matrix`: FBLN, COL1A2, COL3A1, BGN, COL1A1, COL1A2, SPP1, FN1, TIMP1, COL6A2, LYPD3, SLPI, MT2A, NIME2, IFITM3, DCN, TNC

> ○ `Tissue resident marker`: NR4A1, NR4A2, NLRP3

> ○ `T cell receptor (TCR)`: TRAC, TRBC, TRDC, Cd3e, Cd4, Cd8a

> ○ `MHC molecules`: HLA-DRA, HLA-DPA1, HLA-C, HLA-DRB5, HLA-DQA2

> ○ `MHC Class II`: H2-ab1, H2-aa, H2-eb1, Cd74

> ○ `Melanin production and packaging`: Tyrp1, Dct, Pmel, Mlana

> ○ `Glutamate receptors`: GRIK1, GRIK3, GRIK5

> ○ `Voltage-gated potassium channel`: KCND3, KCNK1, KCNMA1, KCNQ2, KCNT1

> ○ `Voltage-gated calcium channel`: CACNA1A, CACNA1C, CACNA2D1

> ○ `Microtubule organization/trafficking`: DYNC2LI1, KIF3A, CFAPs, IFTs, MCTL1

> ○ `Synaptic vesicle transport and release`: KIF1A, KIB3B, KIF26B, MYO5A, MYO9A, BSN

> ○ `Stem-related genes`: SOX4, CD164, SOX9

> ○ `TLS(tertiary lymphoid structure)` ([ref](https://pubmed.ncbi.nlm.nih.gov/35231421/)): IGHA1, IGHG1, IGHG2, IGHG3, IGHG4, IGHGP, IGHM, IGKC, IGLC1, IGLC2, IGLC3, JCHAIN, CD79A, FCRL5, MZB1, SSR4, XBP1, TRBC2, IL7R, CXCL12, LUM, C1QA, C7, CD52, APOE, PTLP, PTGDS, PIM2, DERL3

> ○ `TLS(tertiary lymphoid structure)` ([ref](https://static-content.springer.com/esm/art%3A10.1038%2Fs41587-023-02019-9/MediaObjects/41587_2023_2019_MOESM1_ESM.pdf)) : CD4, CD8A, CD74, CD79A, IL7R, ITGAE, CD1D, CD3D, CD3E, CD8B, CD19, CD22, CD52, CD79B, CR2, CXCL13, CXCR5, FCER2, MS4A1, PDCD1, PTGDS, TRBC2

> ○ `Mitochondrial score`: mt-Nd1, mt-Nd2, mt-Co1, mt-Co2, mt-Atp8, mt-Atp6, mt-Co3, mt-Nd3, mt-Nd4l, mt-Nd4, mt-Nd5, mt-Nd6, mt-Cytb, MT-ND1, MT-ND2, MT-CO1, MT-CO2, MT-ATP8, MT-ATP6, MT-CO3, MT-ND3, MT-ND4L, MT-ND4, MT-ND5, MT-ND6, MT-CYB

> ○ `Hemoglobin score`: Hbb-bt, Hbb-bs, Hbb-bh2, Hbb-bh1, Hbb-y, Hba-x, Hba-a1, Hbq1b, Hba-a2, Hbq1a, HBB, HBD, HBG1, HBG2, HBE1, HBZ, HBM, HBA2, HBA1, HBQ1

> ○ `Naïveness score`: CCR7, TCF7, LEF1, SELL

> ○ `Matrix glycoprotein`: TNC, THBS1 

<br>

<br>

## **4. MF**(molecular function) 

> ○ `Gene summary related to signal transmission`

>> ○ Nuclear hormone receptor: NHR 

>> ○ Jak-Stat: STAT

>> ○ Wnt: TCF/Lef

>> ○ Hedgehog: Gli 

>> ○ TCF-β/BMP: Smad

>> ○ MAPK: ETS

>> ○ Notch: CSL

> ○ `Interferon response`: IFITM1, NFKBIA, IFI27, IFI30, Cxcl9, Cxcl10, Cd274, IFIT3, MX1, IFI44L, OAS1, CXCL10, ISG20, IFI6, C1R, IFI44, C1S, PARP14, LAP3, PLSCR1, IFI35, TNFSF10, GBP1, IFIT1, SOD2, TAP1, EPSTI1, SCO2, XAF1, APOL2, SERPING1, SAMD9L, IL32, IRF7, RNF213, TAP2, IRF1, ICAM1, OASL, TRIM22, GBP4, ISG15, STAT1, HAPLN3, UBE2L6, PARP9, CXCL9, STAT2, USP18, SP100, OAS3, IFIH1, WARS ([ref](https://www-nature-com.proxy.lib.umich.edu/articles/s41588-023-01357-3#Sec46))

> ○ `Inflammatory cytokine`: IL1A, IL1B, CXCL2, CXCL8, CXCL9, CCL4, CCL18, CXCR4, LYZ, IL6, IFNG, NOS2, PTGS2, TNF, CD47, CD274 (PD-L1), GLI1, NFKB1, CCL3, TNF, IRF5, CSF1R, STAB1, NINJ1, JAK3, LITAF, CCL17, CCL21

> ○ `Anti-inflammtory cytokine`: CD163, MSR1 

> ○ `Neurochemical`: Sst, Pvalb, Vip, Reln

> ○ `DNA repair enzyme`: APE1, OGG1, UNG

> ○ `Human leukocyte antigen (HLA)`: HLA-DQA1, HLA-DQB1

> ○ `NF-κB signaling pathway`: NFKB1, NFKBIA

> ○ `Heat shock and DNA damage related gene`: DNAJB1, HSPH1

> ○ `TGFβ signaling pathway`: TGFB2, TGFB3, BMP6

> ○ `JAK–STAT signaling pathway`: IL1B, IL6, IL10

> ○ `Chemokines`: CCL2, CCL4, CCL8, CCL7, CXCL1, CXCL10

> ○ `T cell activation`: TNFRSF9 (4-1BB), TNFRSF18 (GITR)

> ○ `Androgen receptor (AR) target gene score`: KLK2, KLK3, FKBP5, NKX3-1 

> ○ `mTORC1 activity`: RPS6, MLST8

> ○ `Chemokine`: CCL3, CCL4L2, XCL1, XCL2

> ○ `Stress-linked gene` ([ref](https://www-nature-com.proxy.lib.umich.edu/articles/s41588-023-01357-3#Sec46)): GADD45B, FOS, ATF3, FOSB, IER3, EGR1, JUN, ZFP36, IER2, PPP1R15A, ID2, NR4A1, DUSP2, IRF1, INTS6, KLF4, MYC, NFKBIZ, BRD2, CCNL1, NFKBIA, IER5, MCL1, KLF6, SERTAD1, RSRP1, KCNQ1OT1, BTG2, HEXIM1, PLK2, TOB1, PPP1R10, HES1, CITED2, SOCS3, DNAJB1, RHOB, DUSP1, JUNB, CDKN1A, ZC3H12A, DNAJA1, SGK1, ZFP36L2, IFRD1, CTD-3252C9.4, MYLIP, PMAIP1, CLDN4, JUND, ADM, ID1, ID3, MIDN, GEM, GADD45G, HERPUD1, PHLDA2, ERRFI1, CLK1, RND3, RGS16, NCOA7, CXCL2, CD83, Hsp70, Hsp27 

> ○ `Immune cell-homing signals`: CXCL13, CCL3, CCL4, CCL5

> ○ `Adhesion between antigen-presenting cell and T cell`: ICAM1, LFA-3, LFA-1, CD2

> ○ `Co-stimulation between antigen-presenting cell and T cell`: CD40L, OX40, 4-1BB, GITR, CD27, ICOS/CD278, CD28

> ○ `Checkpoint blockade between antigen-presenting cell and T cell`: PD-1, BTLA, TIGIT, TIM-3, LAG-3, CTLA-4 

> ○ `NFkB signaling pathway`: NFKB2, NFKBIA

> ○ `WNT signaling`: Axin2, Celsr2, Lef1, Nkd2, Prickle2, Sfrp1, Sfrp2, Tcf7, Tcf7l1, Tcf7l2, Vangl2, Wnt2, Wnt4, Dkk1, Dkk2, Dkk3, Fzd1, Fzd2, Fzd3, Fzd4, Fzd5, Fzd7, Fzd8

> ○ `Wnt/BMP signaling`: Bmp2, Bmp5, Bmp7

> ○ `Notch signaling`: Ep300, Fbxw7, Maml1, Notch1, Notch2, Notch3, Notch4, Rbpj, Dll1, Dll4, Jag1, Jag2

> ○ `Tumor suppressor gene`: TP53, DCC, ZNF292, HMGN3, UBE2J1

> ○ `Epithelial senescence-associated gene`: SLPI, SPRR1B, SPRR3, IL1RN, LCN2, C15orf48, SPRR2A, S100A8, S100A9, CXCL17, MALL, SPRR2D, KRT13, PRSS22, RHCG, ANXA1, ERO1A, PDZK1IP1, S100P, CEACAM5, TMPRSS11D, SPRR1A, PVRL4, KRT16, EMP1, RAB11FIP1, LMO7, NDRG2, SPRR2E, SERPINB1, KRT6A, CD24, C19orf33, PSCA, CEACAM6, CLDN4, CLDN7, SDCBP2, MUC4, MXD1, LYPD3, KRT6B, LY6D, CLIC3, GPRC5A, TSPAN1, PRSS8, SRD5A3, CD55, CSTB, LYNX1, APOBEC3A, S100A7, RDH10, MUC1, TMPRSS11E, CDKN1A, PI3, DSC2, NCCRP1, CEACAM1, SAT1, DUSP5, LIPH, CSTA, TMEM40 ([ref](https://www-nature-com.proxy.lib.umich.edu/articles/s41588-023-01357-3#Sec46))

> ○ `Oxphos signature`: SRM, ATP5G1, AKR1C1, NDUFAB1, C1QBP, NME1, APRT, H2AFZ, COX5A, NUDT8, NDUFA12, LDHB, CHCHD10, FAM195A, MRPS12, CYC1, NQO1, APOA1BP, PRDX2, PHB, CSRP2, MRPL12, MRPL4, C20orf27, MRPS34, DNPH1, PSMB3, PDCD5, LSM4, DCTPP1, NDUFS8, HSD17B10, NUDT1, BOLA3, NAA38 ([ref](https://www-nature-com.proxy.lib.umich.edu/articles/s41588-023-01357-3#Sec46))

> ○ `Hypoxia signature`: NDRG1, NDUFA4L2, HILPDA, HK2, VEGFA, ADM, ERO1A, ANKRD37, SLC2A1, EGLN3, ERRFI1, SLC6A8, SPRR1B, BHLHE40, IGFBP3, MKNK2, DDIT4, PFKFB3, KDM3A, RHCG, BNIP3L, BNIP3, EDN1, LDHA, C4orf3, PTHLH, PVRL4, FAM210A, FOSL2, PDK1, CXADR, GPI, CLEC2B, RORA, PGK1, GJB2, MAF, GJB6, KDM7A, FAM162A, P4HA1, MT1X, ENO2, NAMPT, KCMF1, ANGPTL4 ([ref](https://www-nature-com.proxy.lib.umich.edu/articles/s41588-023-01357-3#Sec46))

> ○ `EMT signature`: LAMC2, PLAU, TNFRSF12A, EMP3, S100A2, LAMB3, PTHLH, TGFBI, PHLDA1, NDRG1, LAMA3, ITGA6, ITGA3, IGFBP7, KRT17, CDH3, P4HA2, TPM1, ARPC1B, PDPN, ACTN1, CTSB ([ref](https://www-nature-com.proxy.lib.umich.edu/articles/s41588-023-01357-3#Sec46)), FN1, TWIST1, SNAI1, SNAI2, ZEB1, TGFB1, TGFB2, CTNNB1

>> ○ `secreted factors affecting EMT`: HGF, WNT3A, WNT5A, FGF2, IL6, CXCL8, FGF1, TGFB1, TGFB2

>> ○ `EMT upstream regulator`: JAK1, AKT2, SMO, CTNNB1, SMAD2, NFKB2

> ○ `Aldehyde dehydrogenase genes`: ALDH1L1, ALDH1L2, ALDH5A1

> ○ `Genes related to the mevalonate pathway`: HMGCR, FDFT1, DHCR24, SQLE, LSS

> ○ `Genes related to bile metabolism`: BAAT,AKR1D1, HSD3B7, CYP3A5, CYP27A1, CYP7A1, CYP7B1, CYP8B1

> ○ `rRNA transcription`: POLR1G, POLR1E

> ○ `Pre-rRNA processing`: NOL12, ISG20L2

> ○ `Nucleolar ribosome maturation`: WDR75, PDCD11, URB1, URB2, DDX27, NOC3L 

> ○ `Cytoplasmic ribosomal maturation`: RSL24D1 

> ○ `Ribosomal protein`: RPL37A, RPL4

> ○ `Integrin`: ITGB1, ITGAV, ITGA5, ITGA7

<br>

<bR>

## **5. Others**

> ○ `Sequencing low quality signature`: MT-CO3, MT-ATP6, MT-ND2, MT-ND5, MT-ND3, MT-ND4, MT-ND1, MT-CO1, MT-ND4L, MT-CO2, MT-CYB, MTRNR2L12, MT-ND6, COL7A1, SYNE2, MTRNR2L8, PLCG2, RNF213, DST, VMP1, POLR2J3, MDM4, TRIM56, MT-ATP8, NEAT1, MALAT1, LINC00969, FAT1, FAT2, MACF1, SMG1, DYNC1H1, ASH1L, PLXNB2, SLC25A37, LAMA5, DSP, HERC2, BIRC6, USP34, UBR5, HSPG2, TUG1, PARP14, MYH9, SUGP2, EIF4G1, DICER1, ABCA13, PLEC, GABPB1-AS1, KMT2C, AGRN, PRRC2B, ATRX, ARGLU1, DDX17, GOLGB1, DLG1, N4BP2L2 ([ref](https://www-nature-com.proxy.lib.umich.edu/articles/s41588-023-01357-3#Sec46))

---

_Input: 2022.06.21 01:02_
