## **RNA Sequencing Quality Control (QC)**

Recommended Article: 【Bioinformatics】 [Table of Contents for Bioinformatics Analysis](https://jb243.github.io/pages/836)

---

**1.** [Experimental QC](#1-experimental-qc-quality-control)

**2.** [Data QC](#2-data-qc-quality-control)

**3.** [Trouble-shooting](#3-trouble-shooting)

---

**a.** [Genome Projects and Sequencing Techniques](https://jb243.github.io/pages/75)

**b.** [Transcriptome Analysis Pipeline](https://jb243.github.io/pages/2050)

---

<br>

## **1\. Experimental QC** (sample level quality control)

⑴ Definition: Measure of tissue quality

⑵ RNA Sequencing Process

> ① **Step 1.** RNA Purification: Treat with DNase to remove DNA.  

> ② **Step 2.** Poly(A) Selection: Enrich polyadenylated RNA.  

> ③ **Step 3.** Fragmentation: Shear RNA into library-insert sizes of 200–400 nt.  

> ④ **Step 4.** cDNA Synthesis: Convert RNA into complementary DNA (cDNA).  

> ⑤ **Step 5.** cDNA Processing: Ligate adaptors, amplify, and add barcodes.  

> ⑥ **Step 6.** Sequencing: Sequence one or both ends of the fragments, typically 50, 100, or 150 nt per read.  

> ⑦ **Step 7.** Read Mapping: Align sequencing reads to the genome.  

⑶ **Type 1.** RIN (RNA Integrity Number)

> ① Background Knowledge: mRNA comprises less than 3% of total RNA. rRNA makes up more than 80% (mainly 28S [2 kb] and 18S [5 kb] in eukaryotic cells).

> ② Measured by the Agilent 2100 Bioanalyzer.

> ③ RIN Algorithm: It uses features such as the ratio of 18S to total RNA, the ratio of 28S to total RNA, and the 18S normalized height.

> ④ RIN = 10: Intact RNA.

> ⑤ RIN = 1: Completely degraded RNA.

> ⑥ RIN > 7: Generally considered suitable quality level for RNA-seq.

⑷ **Type 2.** DV200: For FFPE tissues, measures the percentage of fragments around 200 nt since RNA is fragmented in FFPE tissues

⑸ **Type 3.** Cq value

⑹ **Type 4.** Nucleic Acid Purity Quantification via Absorbance Ratio

> ① 260 nm / 280 nm Ratio

>> ○ Pure DNA: ~1.8

>> ○ Pure RNA: ~2.0

>> ○ A low 260 nm / 280 nm ratio suggests the presence of proteins or phenol, which absorb at 280 nm.

> ② 260 nm / 230 nm Ratio

>> ○ Pure DNA/RNA molecules: ~2-2.2

>> ○ A low 260 nm / 230 nm ratio indicates the presence of other contaminants.

⑺ **Type 5.** Nucleic Acid Weight Quantification  

> ① Extract RNA using miRNeasy Mini Kit (QIAGEN) or similar methods.  

> ② RNA weight criterion: At least 250 ng.  

⑻ **Type 6.** RNA Quality Score (RQS)  

⑼ **Type 7.** ChIP-seq Experimental QC

> ① ChIP grade monoclonal antibody: Pre-tested with ChIP-seq. 20-30% of the commercially produced antibodies tested were unsatisfactory for ChIP-seq.

> ② qPCR: Best to test positive control region(s) using the ChIP sample. Hope to detect >10-12x fold enrichment over IgG (non-specific antibody).

> ③ Biotinylated transcription factor: Permits factor pull-down on streptavidin. Independent of antibodies.

> ④ CUT&RUN, CUT&Tag, and ChIP-exo are methods to improve the resolution and signal-to-noise ratio of peaks in ChIP-seq.

⑽ **Type 8.** ATAC-seq Experimental QC

> ① Tn5 Concentration: Higher Tn5 concentration relative to DNA concentration increases ATAC-seq signal intensity at promoters and enhancers while reducing fragment size.

> ② Sequencing Lane Cluster Density: Shifts fragment length distribution and TSS enrichment.

⑾ **Type 9.** Spatial Transcriptomics Experimental QC ([ref](https://www.preprints.org/manuscript/202504.1450/v1))

<br>

![image](https://github.com/user-attachments/assets/e5051886-5a3f-40e7-bf4b-93d3a28656d8)

**Table. 1.** Spatial Transcriptomics Experimental QC

<br>

> ① Fresh-frozen tissue has higher RNA integrity than FFPE, but poorer tissue morphology quality.

> ② For FFPE samples, DV200 is a more appropriate metric.

> ③ Image-based spatial transcriptomics platforms like Xenium and CosMx can better tolerate RNA degradation compared to spot-based ST methods.

<br>

<br>

## **2\. Data QC** (sequence level quality control)

⑴ Definition: Evaluation of data quality and improvement as necessary

⑵ **Type 1.** Data QC Metrics: Used for external validity confirmation by comparing with manuals or other datasets

> ① QC Metrics

>> ○ In [dUTP method](https://jb243.github.io/pages/75), "_1.fastq" represents the first strand (anti-sense), and "_2.fastq" represents the second strand (sense; original RNA sequence)

>> ○ **base quality**

>> ○ **mapping rate**

>>> ○ mappability filter 

>>>> ○ **Type 1.** Uniqueness: How unique each sequence is starting at a particular base and of a particular length

>>>> ○ **Type 2.** Alignability: How uniquely k-mer sequences align to a region of the genome (up to 2 mismatches allowed)

>>> ○ mappability score : S = 1 / # of matches found in genome 

>>> ○ Long reads can resolve mapping issue among highly similar regions: Some regions of the genome are troublesome regardless of read length.

>> ○ **Non-coding RNA ratio**

>>> ○ High non-coding RNA ratio indicates lower RNA quality.

>> ○ **GC content**

>>> ○ High GC content: Indicates potential rRNA contamination. In this case, filter out 5S, 18S, 28S rRNA

>>> ○ Low GC content: Indicates potential issues with reverse transcription

>>> ○ Related to CpG island.

>> ○ **Read duplicate**

>>> ○ PCR duplicate: A duplicate that is merely a replication of the same nucleic acid molecule during PCR. High duplication indicates lower RNA quality.

>>> ○ If the sequences are exactly the same, they are considered PCR duplicates. If they are just similar, they are considered biological duplicates.

>>> ○ In paired-end experiments, duplicates occur at the paired-end level.

>>> ○ Generally, DNA-seq involves removing duplicates, while RNA-seq does not: In RNA-seq, the same sequence may repeatedly appear not only due to technical duplicates but also due to high expressing transcripts or short genes. Removing these biological duplicates can reduce the dynamic range of the data or decrease statistical power.

>>> ○ The likelihood of increased duplication rate may rise with the number of cycles during the PCR process: By checking the correlation between the duplicate rate and the number of PCR cycles, one can distinguish between technical and biological duplicates.

>>> ○ Tools for removing duplicates: Samtools, Picard, Trimmomatic, Trim Galore!, fastp 

>> ○ **Unique molecule**(UMI)

>>> ○ RNA quality is significantly lower if the proportion of unique molecules is below 10%

>> ○ **Sequencing depth**

>>> ○ # of reads

>>> ○ In cases of alternative splicing or allele-specific expression: >50 million reads are recommended

>>> ○ DEG analysis of a ribo-depleted library: Approximately 50-60 million total reads are recommended

>>> ○ A ribo-depleted library is recommended to have twice the sequencing depth compared to a poly-A selected library because the ribo-depleted library can capture a wider variety of RNA (e.g., tRNA, rRNA, immature RNA) that the poly-A selected library cannot capture.

>> ○ **Sequencing read length**

>>> ○ If fragments are too small, adaptor binding begins: Adaptor trimming is performed in this case

<br>

![image](https://github.com/user-attachments/assets/149903cb-995c-4636-a25f-92d9b48b69fa)

**Figure 1.** The reason why adaptors are read when fragments are short

<br>

>>> ○ Advantages of long-read sequencing compared to short-read sequencing: lower cost per nucleotide, more accurate mapping, ability to identify splice junctions, capability to detect allele-specific expression, ability to resolve repetitive sequences.

>>> ○ Disadvantages of long-read sequencing compared to short-read sequencing: higher overall cost, higher cost per read, requires more adaptor trimming. Long-read sequencing (PacBio, Oxford Nanopore) is more likely to contain adapter sequences due to its sequencing method, which involves repetitive reading and single-molecule sequencing.

>> ○ **Adaptor sequence**

>>> ○ **Issue 1:** Since adapter sequences are artificial, they can cause alignment and variant calling to fail or introduce biases.

>>> ○ **Issue 2:** Adapter sequences contain identical sequences, leading to biases in coverage analysis and differential gene expression (DEG) analysis.

>>> ○ To remove adapter sequences, tools such as AdaptorRemoval, Cutadapt, Trimmomatic, and bbduk.sh are used.

>> ○ **Exonic ratio**

>>> ○ For poly-A(+) RNA-seq, exonic region comprises 50 ~ 70% of reads

>>> ○ For rRNA(-) RNA-seq, the proportion of exonic reads is reduced

>> ○ **Paired-end vs Single-end**  

>>> ○ Single end reads: Each library fragment is sequenced only from a single end.

>>> ○ Paired end reads: Each library fragment is sequenced from both ends. 

>>> ○ Paired-end (PE) reads are more accurate than single-end (SE) reads but are approximately twice as expensive.  

>>> ○ If the goal is simply to calculate gene counts for DEG analysis, SE is sufficient.  

>>> ○ SE is recommended when RNA is significantly degraded.  

>>> ○ It's better to avoid using PE on short fragments to prevent inefficiencies caused by sequencing the same nucleotide.  

<br>

<img width="508" alt="스크린샷 2025-01-20 오후 8 30 14" src="https://github.com/user-attachments/assets/70a56c33-3458-4a03-9ed2-d31712ca14d3" />

**Figure 2.** Inefficiencies that can occur in paired-end (PE) sequencing

<br>

>> ○ **Mate pair reads** 

>>> ○ A longer segment of DNA is circularized and reads from the joint region are sequenced (both ends).

>> ○ **Strand-specific** (ssRNA-seq)

>>> ○ Can be either forward or reverse.

>> ○ **Poly-A selection vs Ribo-depletion**

>>> ○ Advantages of a ribo-depletion library

>>>> ○ Works even with RNA degradation: cDNA fragments are uneven and short. Poly-A selection is highly biased toward the 3' end, making it less accurate.

>>>> ○ Suitable for studying non-coding RNA.

>>> ○ Disadvantages of a ribo-depletion library

>>>> ○ Expensive.

>>>> ○ Includes a large number of meaningless reads. 

> ② QC Metrics for ChIP-seq

>> ○ [Mapping ratio](https://www.sciencedirect.com/science/article/pii/S1046202320300591)

>> ○ [Read depth](https://www.sciencedirect.com/science/article/pii/S1046202320300591): ENCODE recommends ≥10 million uniquely mapped reads for TFs (histone modifications).

>> ○ [Library complexity](https://www.sciencedirect.com/science/article/pii/S1046202320300591)

>> ○ [Background uniformity](https://www.sciencedirect.com/science/article/pii/S1046202320300591) (biasedness)

>> ○ [GC summit bias](https://www.sciencedirect.com/science/article/pii/S1046202320300591)

>> ○ qPCR enrichment

>> ○ [Fragment size distribution](https://pmc.ncbi.nlm.nih.gov/articles/PMC3431496/)

>> ○ Input DNA qualuty via NanoDrop

>> ○ [Cross-correlation analysis](https://pmc.ncbi.nlm.nih.gov/articles/PMC3431496/): NSC (normalized strand coefficient), RSC (relative strand correlation)

>> ○ FRiP (fraction of reads in peaks) ([ref1](https://pmc.ncbi.nlm.nih.gov/articles/PMC3431496/), [ref2](https://doi.org/10.1038/nbt.1505)), RUP(reads under peaks): Proportion of reads in a ChIP-seq dataset that falls into a peak. ENCODE recommends FRiP (RUP) ≥ 1%.

>> ○ SPOT(signal portion in tags): Indicates good signal-to-noise ratio.

>> ○ IDR (irreproducibility discovery rate) ([ref1](https://pmc.ncbi.nlm.nih.gov/articles/PMC3431496/), [ref2](https://projecteuclid.org/journals/annals-of-applied-statistics/volume-5/issue-3/Measuring-reproducibility-of-high-throughput-experiments/10.1214/11-AOAS466.full))

>> ○ denQCi, simQCi, QC-STAMP ([ref](https://pmc.ncbi.nlm.nih.gov/articles/PMC3834836/))

>> ○ Motif analysis: What % of peaks contain the TF motif, and does the motif tend to occur in the middle of the peak? Not expected for all peaks, because a TF may bind as part of a protein complex or a heterodimer.

> ③ QC Metrics for ATAC-seq

>> ○ FastQC: For example, "Per base sequence content" can be used to assess the integration bias of Tn5 transposase.

>> ○ The ataqv package provides 35 QC metrics as follows ([ref](https://pmc.ncbi.nlm.nih.gov/articles/instance/8245295/bin/NIHMS1594346-supplement-Supplementary_Figures_and_Table_S1.pdf)): **fragment length distribution**, % reads that are high-quality and autosomal, % reads properly paired end mapped, % reads that aligned to autosomes that were duplicates, short-to-mononucleosomal-ratio, **TSS enrichment**, duplicate fraction in peaks, duplicate fraction outside of peaks, peak duplicate ratio, cumulative fraction of high-quality autosomal reads in peaks, cumulative fraction of the genome that falls within peaks, distribution of mapping qualities, number of total reads, % alignments marked secondary, % alignments marked supplementary, % alignments marked as duplicates, mean mapping quality, median mapping quality, % reads unmapped, % reads with an unmapped mate, % QC-fail reads,% unpaired reads, % reads with mapping quality 0, % reads that paired and mapped but in RF orientation, % reads that paired and mapped but in FF orientation, % reads that paired and mapped but in RR orientation, % reads that paired and mapped but on separate chromosomes, % reads that paired and mapped but too far from mate, % reads that paired and mapped but not properly, % reads that aligned to autosomes, % reads that aligned to mitochondria, % reads that aligned to mitochondria that were duplicate, number of peaks called, fragment length distribution distance, max fraction of reads from a single autosome

> ④ **Method 1.** Other Datasets: 10x Genomics, GEO, ZENODO, etc.

> ⑤ **Method 2.** FastQC: Takes Fastq files as input. Java-based.

>> ○ **2-1.** [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc) and multiQC: Most popular

>>> ○ Base pair quality of reads

>>> ○ Adaptor sequences in reads

>>> ○ PCR duplicates

>>> ○ Overrepresented sequences

>>> ○ GC distribution for each sample

>> ○ **2-2.** QoRTs ([ref1](https://pubmed.ncbi.nlm.nih.gov/26187896/), [ref2](https://github.com/hartleys/QoRTs)): Very good

>>> ○ RNA degradation: Distribution of reads 5’ → 3’

>>> ○ Strandedness check

>>> ○ GC bias

>> ○ **2-3.** [RNASeQC](https://pmc.ncbi.nlm.nih.gov/articles/PMC3356847/): Decent 

>> ○ **2-4.** [RSeQC](https://rseqc.sourceforge.net/): Used to have major bugs

>> ○ **2-5.** Use [conda Fastqc](https://anaconda.org/bioconda/fastqc) command (Linux)

>> ○ **2-6.** Download [SRA](https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit) (Sequence Reads Archive) Toolkit and use fastqc command (Linux)

>> ○ Below is an example of generated files.

<br>

```puython
sudo apt install fastqc
cd sratoolkit.3.0.5-ubuntu64/
cd bin
fastqc DRR016938.fastq
```

<br>

> ⑥ **Method 3.** Trimmomatic: Takes Fastq files as input. 

> ⑦ **Method 4.** Cutadapt 

> ⑧ **Method 5.** FASTX-Toolkit: Takes Fastq files as input. 

> ⑨ **Method 6.** QC after mapping: Takes SAM or BAM files as input. 

>> ○ QC metric

>>> ○ % uniquely mapped reads 

>>> ○ % reads mapping to exons 

>>> ○ Complexity, i.e. x% of read counts being taken up by y% of genes

>>> ○ ○ Consistency across samples 

>>> ○ Sample swap: Match Y chromosome, Xist, Genotype (e.g., SNP) with metadata.

>> ○ **6-1.** Qplot

>> ○ **6-2.** Samtools

> ⑩ **Method 7.** `SnakeMake`: Integrated pipeline providing QC functionality as well

>> ○ `Snakefile`**:** A Snakemake script based on Python. The filename itself is Snakefile.

<br>

```python
# Snakefile

# Define output file
rule all:
    input:
        "results/processed_data.tsv"

# Rule for data processing 
rule process_data:
    input:
        "data/raw_data.tsv"
    output:
        "results/processed_data.tsv"
    shell:
        """
        cat {input} | awk -F'\t' '{{print $1, $2, $3}}' > {output}
        """
```

<br>

>> ○ `config.yaml` (optional): Setting for Snakemake workflow ([ref](https://jb243.github.io/pages/2069))

>> ○ `requirements.txt` (optional): Package dependency 

>> ○ Input file 

>> ○ Output file 

> ⑪ **Method 8.** QuASAR-QC: Applicable for Hi-C data

> ⑫ [Troubleshootings](https://jb243.github.io/pages/2337)

⑵ **Type 1.** Rank-correlation between samples: Used for internal validity confirmation

> ① **Objective 1.** Evaluating sample quality by examining alignment of two variables with alignment characteristics within a single sample

>> ○ **Example 1.** Examining the alignment of expression levels of two genes known to be similar

>> ○ **Example 2.** Investigating whether expression of two genes known to be similar appears in the same cluster

> ② **Objective 2.** Mainly used to observe correspondence between a pair of identical samples

> ③ **Objective 3.** Examining correlation coefficients of two different variables with different data distribution characteristics

>> ○ Somewhat distant from QC analysis

>> ○ Example: Investigating the correlation coefficient between gene A expression in scRNA-seq and gene A expression in ST

> ④ **Method 1.** Pearson Correlation Coefficient

>> ○ Definition: Given the standard deviations σx, σy of X and Y,

<br>

![image](https://github.com/user-attachments/assets/8dfecb2a-1bc4-40ac-ab83-d817d6b02ac1)

<br>

>> ○ [Calculation in RStudio](http://www.sthda.com/english/wiki/correlation-test-between-two-variables-in-r#:~:text=Compute%20correlation%20in%20R-,R%20functions,-Correlation%20coefficient%20can)

>>> ○ `cor(x, y)`

>>> ○ `cor(x, y, method = "pearson")`

>>> ○ `cor.test(x, y)`

>>> ○ `cor.test(x, y, method = "pearson")`

> ⑤ **Method 2.** Spearman's Rank Correlation Coefficient

>> ○ Definition: Defined based on ranks x' = rank(x) and y' = rank(y)

<br>

![image](https://github.com/user-attachments/assets/5cf01c89-f843-4dc5-9e10-9f4313c4d4f9)

<br>

>> ○ [Calculation in RStudio](http://www.sthda.com/english/wiki/correlation-test-between-two-variables-in-r#:~:text=Compute%20correlation%20in%20R-,R%20functions,-Correlation%20coefficient%20can)

>>> ○ `cor(x, y, method = "spearman")`

>>> ○ `cor.test(x, y, method = "spearman")`

> ⑥ **Method 3.** Kendall's Rank Correlation Coefficient

>> ○ Definition: Defined based on concordant pairs and discordant pairs

>> ○ **Step 1.** Sort y values in ascending order for x values: Represent each y value as yi

>> ○ **Step 2.** Count the number of concordant pairs where yj > yi (where j > i) for each yi value

>> ○ **Step 3.** Count the number of discordant pairs where yj < yi (where j > i) for each yi value

>> ○ **Step 4.** Definition of correlation coefficient

<br>

![image](https://github.com/user-attachments/assets/c50afb4f-af76-499d-842c-d09bf9326eb7)

<br>

>>> ○ n<sub>c</sub>: total number of concordant pairs

>>> ○ n<sub>d</sub>: total number of discordant pairs

>>> ○ n: size of x and y

>> ○ [Calculation in RStudio](http://www.sthda.com/english/wiki/correlation-test-between-two-variables-in-r#:~:text=Compute%20correlation%20in%20R-,R%20functions,-Correlation%20coefficient%20can)

>>> ○ `cor(x, y, method = "kendall")`

>>> ○ `cor.test(x, y, method = "kendall")`

> ⑦ **Method 4.** Q-Q plot between empirical CDFs (Cumulative Distribution Functions)

> ⑧ **Method 5.** Q-Q plot between ordered p-values

> ⑨ For Hi-C sequencing, available methods include HiCRep, GenomeDISCO, HiC-Spector, and QuASAR-Rep.

<br>

<br>

## **3. Troubleshooting**

 ⑴ **Method 1.** Website Investigation: Pre-flight errors, in-flight errors, or alerts

> ① **Failing to install bcl2fastq**

> ② **ATAC Sequencing depth per cell is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 10,000. Low ATAC sequencing depth negatively impacts the quality of peak calling, clustering, differential analysis and feature linkage. At very low sequencing depth, < 5000 raw read-pairs per cell, identification of cell barcodes may be unreliable.

> ③ **GEX Sequencing depth per cell is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 5,000. Low GEX sequencing depth negatively impacts the quality of clustering, differential analysis and feature linkage. At very low sequencing depth, < 2,000 raw read-pairs per cell, identification of cell barcodes may be unreliable.

> ④ **ATAC Median fragments per cell is low** ( _Cell Ranger ARC v2.0_ ): A low value is generally caused by low sequence depth, the wrong genome reference, or low library complexity that could be due to a problem during the transposition step or a problem in the library preparation workflow. Low fragment counts negatively impact clustering, differential analysis and feature linkage detection.

> ⑤ **Number of linkages detected is low** ( _Cell Ranger ARC v2.0_ ): The number of detected feature linkage is < 100. This may be caused by a low number of nuclei recovered, low sequencing depth, poor peak calling, or a sample that is relatively homogenous.

> ⑥ **GEX Median UMI counts per cell is low** ( _Cell Ranger ARC v2.0_ ): Observed value < 100. This may be a consequence of very low sequencing depth, poor sample quality, an error in the library preparation workflow, the wrong reference genome, or poor genome annotations. Low UMI counts negatively impact clustering, differential analysis and feature linkage detection.

> ⑦ **GEX Reads mapping to reference is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 80%. This can be caused by the wrong reference genome being used or a poor quality genome assembly. Application performance may be affected.

> ⑧ **GEX Reads mapping to transcriptome is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 50%. This can indicate use of the wrong reference transcriptome, a reference transcriptome with overlapping genes, poor library quality, poor sequencing quality, or reads shorter than the recommended minimum. Application performance may be affected.

> ⑨ **ATAC Reads mapping to reference is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 80%. This can be caused by the wrong reference genome being used or a poor quality genome assembly. Application performance may be affected.

> ⑩ **GEX Transcriptome reads in cells is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 60%. Many of the reads were not assigned to cell-associated barcodes. This is generally indicative of poor sample prep resulting in high levels of ambient RNA. It could also indicate a problem in the cell calling algorithm that could be caused by high RNA or DNA background, exclusion of a large number of barcodes from cell calling due to low targeting, or due to a population of nuclei with low RNA content. The latter case can be addressed by inspecting the data to determine the appropriate cell count and rerunning the pipeline supplying appropriate parameters to override the cell caller. Application performance may be affected.

> ⑪ **Low Fraction Reads Confidently Mapped To Transcriptome **( _Cell Ranger v6.1_ ): Ideal > 30%. This can indicate use of the wrong reference transcriptome, a reference transcriptome with overlapping genes, poor library quality, poor sequencing quality, or reads shorter than the recommended minimum. Application performance may be affected.

> ⑫ **No Cells Detected** ( _Cell Ranger v6.1_ ): Estimated number of cells is expected to be > 100. This usually indicates poor cell handling, poor library, or poor sequencing quality. Application performance is likely to be affected.

> ⑬ **Low Fraction Valid UMIs** ( _Cell Ranger v6.1_ ): Ideal > 75%. This may indicate a quality issue with the Illumina R2 read for Single Cell 3' v1 or the R1 read for Single Cell 3' v2/v3 and Single Cell 5'. Application performance may be affected.

> ⑭ **Fraction of UMI bases with Q-score >= 30 is low** ( _Cell Ranger v6.1_ ): Fraction of UMI bases (Illumina R2 Read for Single Cell 3' v1, R1 for Single Cell 3' v2/v3 and Single Cell 5') with Q-score >= 30 should be above 75%. A lower fraction might indicate poor sequencing quality.

> ⑮ **Fraction of cell barcode bases with Q-score >= 30 is low** ( _Cell Ranger v6.1_ ): Fraction of cell barcode bases (Illumina I7 Read for Single Cell 3' v1, R1 for Single Cell 3' v2/v3 and Single Cell 5') with Q-score >= 30 should be above 55%. A lower fraction might indicate poor sequencing quality.

> ⓐ **Too many detected cells** ( _Cell Ranger ATAC v2.0_ ): Estimated number of cells is expected to be under 10,000. A high value might indicate an overlapping of cells, a problem during library preparation, or unexpected behavior in the cell calling algorithm.

> ⓑ **Average fraction of barcode bases with high sequencing quality is low** ( _Cell Ranger ATAC v2.0_ ): Average fraction of bases in barcode with quality above Q30 should be ideally above 75%. A lower fraction might indicate poor sequencing quality.

> ⓒ **Median fragments per cell is low** ( _Cell Ranger ATAC v2.0_ ): The median number of fragments (that passed all filters) detected in single cells is expected to be above 500. A lower value suggests low sensitivity, potentially due to insufficient sequencing.

> ⓓ ** The percentage of transposition events falling within peaks is low** ( _Cell Ranger ATAC v2.0_ ): It is expected that more than 25% of the transposition events fall within peak regions. A lower value could suggest peak undercalling or low sequencing depth.

> ⓔ **Estimated number of cells is low** ( _Cell Ranger ATAC v2.0_ ): Number of cells detected is expected to be higher than 500. This usually indicates poor cell, library, or sequencing quality.

> ⓕ **Average fraction of barcode bases with high sequencing quality is low** ( _Cell Ranger ATAC v2.0_ ): Average fraction of bases in barcode with quality above Q30 should be above 75%. A lower fraction might indicate poor sequencing quality.

> ⓖ **Fraction of RNA read bases with Q-score >= 30 is low** ( _Space Ranger v1.3_ ): Fraction of RNA read bases with Q-score >= 30 should be above 80%. A lower fraction might indicate poor sequencing quality.

> ⓗ **Low Fraction Reads in Spots** ( _Space Ranger v1.3_ ): Ideal > 50%. Application performance may be affected. Many of the reads were not assigned to tissue covered spots. This could be caused by high levels of ambient RNA resulting from inefficient permeabilization or because of poor tissue detection. The latter case can be addressed by using the manual tissue selection option through Loupe.

⑵ **Method 2.** search for technical note

> ① [Single Cell Gene Expression Assay](https://support.10xgenomics.com/single-cell-gene-expression/sequencing/doc/technical-note-interpreting-cell-ranger-web-summary-files-for-single-cell-gene-expression-assays)

> ② [Single Cell Multiome ATAC + Gene Expression Assay](https://support.10xgenomics.com/single-cell-multiome-atac-gex/sequencing/doc/technical-note-interpreting-cell-ranger-arc-web-summary-files-for-single-cell-multiome-atac-gene-expression-assay)

> ③ [Single ATAC Assay](https://support.10xgenomics.com/single-cell-atac/sequencing/doc/technical-note-interpreting-cell-ranger-atac-web-summary-files-for-single-cell-atac-assay)

> ④ [Visium Assay](https://www.illumina.com/content/dam/illumina/gcs/assembled-assets/marketing-literature/10x-visium-tech-note-m-gl-00177/10x-visium-tech-note-m-gl-00177.pdf)

> ⑤ [Visium Assay2](https://assets.ctfassets.net/an68im79xiti/5LXZH2iIisJHYDqjkKmvYO/10e5b3ad53b380a15c1af7a09a11aa69/CG000241_VisiumImagingGuidelinesTN_RevB.pdf)

<br>

---

_Input: 2023.05.22 11:48_



