## **RNA Sequencing Quality Control (QC)**

Recommended Article: 【Bioinformatics】 [Table of Contents for Bioinformatics Analysis](https://jb243.github.io/pages/836)

---

**1.** [Tissue QC](#1-tissue-qc-quality-control)

**2.** [Data QC](#2-data-qc-quality-control)

**3.** [Trouble-shooting](#3-trouble-shooting)

---

**a.** [Genome Projects and Sequencing Techniques](https://jb243.github.io/pages/75)

**b.** [Transcriptome Analysis Pipeline](https://jb243.github.io/pages/2050)

---

<br>

## **1\. Tissue QC** (sample level quality control)

⑴ Definition: Measure of tissue quality

⑵ **Type 1.** RIN (RNA Integrity Number)

> ① Measured by Agilent 2100 Bioanalyzer

> ② RIN = 10: Intact RNA

> ③ RIN = 1: Completely degraded RNA

> ④ RIN > 7: Generally suitable quality for RNA-seq

⑶ **Type 2.** DV200: For FFPE tissues, measures the percentage of fragments around 200 nt since RNA is fragmented in FFPE tissues

⑷ **Type 3.** Nucleic Acid Purity Quantification via Absorbance Ratio

> ① 260 nm / 280 nm Ratio

>> ○ Pure DNA: ~1.8

>> ○ Pure RNA: ~2.0

>> ○ A low 260 nm / 280 nm ratio suggests the presence of proteins or phenol, which absorb at 280 nm.

> ② 260 nm / 230 nm Ratio

>> ○ Pure DNA/RNA molecules: ~2-2.2

>> ○ A low 260 nm / 230 nm ratio indicates the presence of other contaminants.

<br>

<br>

## **2\. Data QC** (sequence level quality control)

⑴ Definition: Evaluation of data quality and improvement as necessary

⑵ **Type 1.** Data QC Metrics: Used for external validity confirmation by comparing with manuals or other datasets

> ① QC Metrics

>> ○ In [dUTP method](https://jb243.github.io/pages/75), "_1.fastq" represents the first strand (anti-sense), and "_2.fastq" represents the second strand (sense; original RNA sequence)

>> ○ High non-coding RNA ratio and high duplication indicate lower RNA quality

>> ○ GC content

>>> ○ High GC content: Indicates potential rRNA contamination. In this case, filter out 5S, 18S, 28S rRNA

>>> ○ Low GC content: Indicates potential issues with reverse transcription

>> ○ Read duplicate

>>> ○ PCR duplicate: A duplicate that is merely a replication of the same nucleic acid molecule during PCR.

>>> ○ If the sequences are exactly the same, they are considered PCR duplicates. If they are just similar, they are considered biological duplicates.

>>> ○ In paired-end experiments, duplicates occur at the paired-end level.

>>> ○ Generally, DNA-seq involves removing duplicates, while RNA-seq does not: In RNA-seq, the same sequence may repeatedly appear not only due to technical duplicates but also due to high expressing transcripts or short genes. Removing these biological duplicates can reduce the dynamic range of the data or decrease statistical power.

>>> ○ The likelihood of increased duplication rate may rise with the number of cycles during the PCR process: By checking the correlation between the duplicate rate and the number of PCR cycles, one can distinguish between technical and biological duplicates.

>>> ○ Tools for removing duplicates: Samtools, Picard

>> ○ Typically, duplicates are removed in DNA-seq, but not in RNA-seq

>> ○ RNA quality is significantly lower if the proportion of unique molecules is below 10%

>> ○ If fragments are too small, adaptor binding begins: Adaptor trimming is performed in this case

>> ○ For poly-A(+) RNA-seq, exonic region comprises 50 ~ 70% of reads

>> ○ For rRNA(-) RNA-seq, the proportion of exonic reads is reduced

> ② QC Metrics for ChIP-seq

>> ○ [Mapping ratio](https://www.sciencedirect.com/science/article/pii/S1046202320300591)

>> ○ [Read depth](https://www.sciencedirect.com/science/article/pii/S1046202320300591)

>> ○ [Library complexity](https://www.sciencedirect.com/science/article/pii/S1046202320300591)

>> ○ [Background uniformity](https://www.sciencedirect.com/science/article/pii/S1046202320300591) (biasedness)

>> ○ [GC summit bias](https://www.sciencedirect.com/science/article/pii/S1046202320300591)

>> ○ qPCR enrichment

>> ○ [Fragment size distribution](https://pmc.ncbi.nlm.nih.gov/articles/PMC3431496/)

>> ○ Input DNA qualuty via NanoDrop

>> ○ [Cross-correlation analysis](https://pmc.ncbi.nlm.nih.gov/articles/PMC3431496/): NSC (normalized strand coefficient), RSC (relative strand correlation)

>> ○ FRiP (fraction of reads in peaks) ([ref1](https://pmc.ncbi.nlm.nih.gov/articles/PMC3431496/), [ref2](https://doi.org/10.1038/nbt.1505))

>> ○ IDR (irreproducibility discovery rate) ([ref1](https://pmc.ncbi.nlm.nih.gov/articles/PMC3431496/), [ref2](https://projecteuclid.org/journals/annals-of-applied-statistics/volume-5/issue-3/Measuring-reproducibility-of-high-throughput-experiments/10.1214/11-AOAS466.full))

>> ○ denQCi, simQCi, QC-STAMP ([ref](https://pmc.ncbi.nlm.nih.gov/articles/PMC3834836/))

> ③ **Method 1.** Other Datasets: 10x Genomics, GEO, ZENODO, etc.

> ④ **Method 2.** FastQC

>> ○ **2-1.** Use [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc) program

>> ○ **2-2.** Use [conda Fastqc](https://anaconda.org/bioconda/fastqc) command (Linux)

>> ○ **2-3.** Download [SRA](https://github.com/ncbi/sra-tools/wiki/01.-Downloading-SRA-Toolkit) (Sequence Reads Archive) Toolkit and use fastqc command (Linux)

>> ○ Below is an example of generated files.

<br>

```puython
sudo apt install fastqc
cd sratoolkit.3.0.5-ubuntu64/
cd bin
fastqc DRR016938.fastq
```

<br>

> ⑤ **Method 3.** Trimmomatic: Takes Fastq files as input. 

> ⑥ **Method 4.** FASTX-Toolkit: Takes Fastq files as input. 

> ⑦ **Method 5.** Qplot: Takes SAM or BAM files as input. 

> ⑧ **Method 6.** Samtools: Takes SAM or BAM files as input. 

> ⑨ **Method 7.** `SnakeMake`: Integrated pipeline providing QC functionality as well

>> ○ `Snakefile`**:** A Snakemake script based on Python. The filename itself is Snakefile.

<br>

```python
# Snakefile

# Define output file
rule all:
    input:
        "results/processed_data.tsv"

# Rule for data processing 
rule process_data:
    input:
        "data/raw_data.tsv"
    output:
        "results/processed_data.tsv"
    shell:
        """
        cat {input} | awk -F'\t' '{{print $1, $2, $3}}' > {output}
        """
```

<br>

>> ○ `config.yaml` (optional): Setting for Snakemake workflow ([ref](https://jb243.github.io/pages/2069))

>> ○ `requirements.txt` (optional): Package dependency 

>> ○ Input file 

>> ○ Output file 

> ⑩ **Method 8.** QuASAR-QC: Applicable for Hi-C data

> ⑪ [Troubleshootings](https://jb243.github.io/pages/2337)

⑵ **Type 1.** Rank-correlation between samples: Used for internal validity confirmation

> ① **Objective 1.** Evaluating sample quality by examining alignment of two variables with alignment characteristics within a single sample

>> ○ **Example 1.** Examining the alignment of expression levels of two genes known to be similar

>> ○ **Example 2.** Investigating whether expression of two genes known to be similar appears in the same cluster

> ② **Objective 2.** Mainly used to observe correspondence between a pair of identical samples

> ③ **Objective 3.** Examining correlation coefficients of two different variables with different data distribution characteristics

>> ○ Somewhat distant from QC analysis

>> ○ Example: Investigating the correlation coefficient between gene A expression in scRNA-seq and gene A expression in ST

> ④ **Method 1.**

>> ○ Definition: Given the standard deviations σx, σy of X and Y,

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/82e165ee-ae60-44ba-84a7-12ddec0ea92c)

<br>

>> ○ [Calculation in RStudio](http://www.sthda.com/english/wiki/correlation-test-between-two-variables-in-r#:~:text=Compute%20correlation%20in%20R-,R%20functions,-Correlation%20coefficient%20can)

>>> ○ cor(x, y)

>>> ○ cor(x, y, method = "pearson")

>>> ○ cor.test(x, y)

>>> ○ cor.test(x, y, method = "pearson")

> ⑤ **Method 2.** Spearman's Rank Correlation Coefficient

>> ○ Definition: Defined based on ranks x' = rank(x) and y' = rank(y)

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/fa16ea85-4363-400e-b16c-2e1ec212da5d)

<br>

>> ○ [Calculation in RStudio](http://www.sthda.com/english/wiki/correlation-test-between-two-variables-in-r#:~:text=Compute%20correlation%20in%20R-,R%20functions,-Correlation%20coefficient%20can)

>>> ○ cor(x, y, method = "spearman")

>>> ○ cor.test(x, y, method = "spearman")

> ⑥ **Method 3.** Kendall's Rank Correlation Coefficient

>> ○ Definition: Defined based on concordant pairs and discordant pairs

>> ○ **Step 1.** Sort y values in ascending order for x values: Represent each y value as yi

>> ○ **Step 2.** Count the number of concordant pairs where yj > yi (where j > i) for each yi value

>> ○ **Step 3.** Count the number of discordant pairs where yj < yi (where j > i) for each yi value

>> ○ **Step 4.** Definition of correlation coefficient

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/763a6a84-3e61-49a0-be54-00411fa1743d)

<br>

>>> ○ n<sub>c</sub>: total number of concordant pairs

>>> ○ n<sub>d</sub>: total number of discordant pairs

>>> ○ n: size of x and y

>> ○ [Calculation in RStudio](http://www.sthda.com/english/wiki/correlation-test-between-two-variables-in-r#:~:text=Compute%20correlation%20in%20R-,R%20functions,-Correlation%20coefficient%20can)

>>> ○ cor(x, y, method = "kendall")

>>> ○ cor.test(x, y, method = "kendall")

<br>

<br>

## **3. Troubleshooting**

 ⑴ **Method 1.** Website Investigation: Pre-flight errors, in-flight errors, or alerts

> ① **Failing to install bcl2fastq**

> ② **ATAC Sequencing depth per cell is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 10,000. Low ATAC sequencing depth negatively impacts the quality of peak calling, clustering, differential analysis and feature linkage. At very low sequencing depth, < 5000 raw read-pairs per cell, identification of cell barcodes may be unreliable.

> ③ **GEX Sequencing depth per cell is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 5,000. Low GEX sequencing depth negatively impacts the quality of clustering, differential analysis and feature linkage. At very low sequencing depth, < 2,000 raw read-pairs per cell, identification of cell barcodes may be unreliable.

> ④ **ATAC Median fragments per cell is low** ( _Cell Ranger ARC v2.0_ ): A low value is generally caused by low sequence depth, the wrong genome reference, or low library complexity that could be due to a problem during the transposition step or a problem in the library preparation workflow. Low fragment counts negatively impact clustering, differential analysis and feature linkage detection.

> ⑤ **Number of linkages detected is low** ( _Cell Ranger ARC v2.0_ ): The number of detected feature linkage is < 100. This may be caused by a low number of nuclei recovered, low sequencing depth, poor peak calling, or a sample that is relatively homogenous.

> ⑥ **GEX Median UMI counts per cell is low** ( _Cell Ranger ARC v2.0_ ): Observed value < 100. This may be a consequence of very low sequencing depth, poor sample quality, an error in the library preparation workflow, the wrong reference genome, or poor genome annotations. Low UMI counts negatively impact clustering, differential analysis and feature linkage detection.

> ⑦ **GEX Reads mapping to reference is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 80%. This can be caused by the wrong reference genome being used or a poor quality genome assembly. Application performance may be affected.

> ⑧ **GEX Reads mapping to transcriptome is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 50%. This can indicate use of the wrong reference transcriptome, a reference transcriptome with overlapping genes, poor library quality, poor sequencing quality, or reads shorter than the recommended minimum. Application performance may be affected.

> ⑨ **ATAC Reads mapping to reference is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 80%. This can be caused by the wrong reference genome being used or a poor quality genome assembly. Application performance may be affected.

> ⑩ **GEX Transcriptome reads in cells is low** ( _Cell Ranger ARC v2.0_ ): Ideal > 60%. Many of the reads were not assigned to cell-associated barcodes. This is generally indicative of poor sample prep resulting in high levels of ambient RNA. It could also indicate a problem in the cell calling algorithm that could be caused by high RNA or DNA background, exclusion of a large number of barcodes from cell calling due to low targeting, or due to a population of nuclei with low RNA content. The latter case can be addressed by inspecting the data to determine the appropriate cell count and rerunning the pipeline supplying appropriate parameters to override the cell caller. Application performance may be affected.

> ⑪ **Low Fraction Reads Confidently Mapped To Transcriptome **( _Cell Ranger v6.1_ ): Ideal > 30%. This can indicate use of the wrong reference transcriptome, a reference transcriptome with overlapping genes, poor library quality, poor sequencing quality, or reads shorter than the recommended minimum. Application performance may be affected.

> ⑫ **No Cells Detected** ( _Cell Ranger v6.1_ ): Estimated number of cells is expected to be > 100. This usually indicates poor cell handling, poor library, or poor sequencing quality. Application performance is likely to be affected.

> ⑬ **Low Fraction Valid UMIs** ( _Cell Ranger v6.1_ ): Ideal > 75%. This may indicate a quality issue with the Illumina R2 read for Single Cell 3' v1 or the R1 read for Single Cell 3' v2/v3 and Single Cell 5'. Application performance may be affected.

> ⑭ **Fraction of UMI bases with Q-score >= 30 is low** ( _Cell Ranger v6.1_ ): Fraction of UMI bases (Illumina R2 Read for Single Cell 3' v1, R1 for Single Cell 3' v2/v3 and Single Cell 5') with Q-score >= 30 should be above 75%. A lower fraction might indicate poor sequencing quality.

> ⑮ **Fraction of cell barcode bases with Q-score >= 30 is low** ( _Cell Ranger v6.1_ ): Fraction of cell barcode bases (Illumina I7 Read for Single Cell 3' v1, R1 for Single Cell 3' v2/v3 and Single Cell 5') with Q-score >= 30 should be above 55%. A lower fraction might indicate poor sequencing quality.

> ⓐ **Too many detected cells** ( _Cell Ranger ATAC v2.0_ ): Estimated number of cells is expected to be under 10,000. A high value might indicate an overlapping of cells, a problem during library preparation, or unexpected behavior in the cell calling algorithm.

> ⓑ **Average fraction of barcode bases with high sequencing quality is low** ( _Cell Ranger ATAC v2.0_ ): Average fraction of bases in barcode with quality above Q30 should be ideally above 75%. A lower fraction might indicate poor sequencing quality.

> ⓒ **Median fragments per cell is low** ( _Cell Ranger ATAC v2.0_ ): The median number of fragments (that passed all filters) detected in single cells is expected to be above 500. A lower value suggests low sensitivity, potentially due to insufficient sequencing.

> ⓓ ** The percentage of transposition events falling within peaks is low** ( _Cell Ranger ATAC v2.0_ ): It is expected that more than 25% of the transposition events fall within peak regions. A lower value could suggest peak undercalling or low sequencing depth.

> ⓔ **Estimated number of cells is low** ( _Cell Ranger ATAC v2.0_ ): Number of cells detected is expected to be higher than 500. This usually indicates poor cell, library, or sequencing quality.

> ⓕ **Average fraction of barcode bases with high sequencing quality is low** ( _Cell Ranger ATAC v2.0_ ): Average fraction of bases in barcode with quality above Q30 should be above 75%. A lower fraction might indicate poor sequencing quality.

> ⓖ **Fraction of RNA read bases with Q-score >= 30 is low** ( _Space Ranger v1.3_ ): Fraction of RNA read bases with Q-score >= 30 should be above 80%. A lower fraction might indicate poor sequencing quality.

> ⓗ **Low Fraction Reads in Spots** ( _Space Ranger v1.3_ ): Ideal > 50%. Application performance may be affected. Many of the reads were not assigned to tissue covered spots. This could be caused by high levels of ambient RNA resulting from inefficient permeabilization or because of poor tissue detection. The latter case can be addressed by using the manual tissue selection option through Loupe.

⑵ **Method 2.** search for technical note

> ① [Single Cell Gene Expression Assay](https://support.10xgenomics.com/single-cell-gene-expression/sequencing/doc/technical-note-interpreting-cell-ranger-web-summary-files-for-single-cell-gene-expression-assays)

> ② [Single Cell Multiome ATAC + Gene Expression Assay](https://support.10xgenomics.com/single-cell-multiome-atac-gex/sequencing/doc/technical-note-interpreting-cell-ranger-arc-web-summary-files-for-single-cell-multiome-atac-gene-expression-assay)

> ③ [Single ATAC Assay](https://support.10xgenomics.com/single-cell-atac/sequencing/doc/technical-note-interpreting-cell-ranger-atac-web-summary-files-for-single-cell-atac-assay)

> ④ [Visium Assay](https://www.illumina.com/content/dam/illumina/gcs/assembled-assets/marketing-literature/10x-visium-tech-note-m-gl-00177/10x-visium-tech-note-m-gl-00177.pdf)

> ⑤ [Visium Assay2](https://assets.ctfassets.net/an68im79xiti/5LXZH2iIisJHYDqjkKmvYO/10e5b3ad53b380a15c1af7a09a11aa69/CG000241_VisiumImagingGuidelinesTN_RevB.pdf)

<br>

---

_Input: 2023.05.22 11:48_
