## **Collection of Useful Linux Functions**

Recommended Article **:** 【Operating System】 **Chapter 1.** [Linux](https://jb243.github.io/pages/2147)(Linux)

---

**1.** [Search for the directory](#1-search-for-the-directory)

**2.** [Modify the directory](#2-modify-the-directory)

**3.** [Compress files](#3-compress-files)

**4.** [Query process](#4-query-process)

**5.** [Execute process](#5-execute-process)

**6.** [Bioinformatics](#6-bioinformatics)

---

<br>

## **1. Search for the directory**

⑴ Current directory **:** `pwd`

⑵ Code to find the number of files in a given directory

> `ls -l /path/to/directory | grep -v '^d' | wc -l`

⑶ Code to find the number of `.csv` files in a given directory

> `find /path/to/your/directory -type f -name "*.csv" | wc -l`

⑷ Code that outputs the absolute addresses of non-`.h5` files in a given directory

> `find /path/to/directory -type f ! -name '*.h5'`

⑸ Code to find out the number of folders in a given directory.

> `ls -l /path/to/directory | grep '^d' | wc -l`

⑹ Code to determine the size of a given file

> `ls -lh input.bam`

⑺ Code to find out the size of a given directory

> `du -sh /path/to/directory`

⑻ Code to view the contents of a zip file without extracting it

> `unzip -l MyFile.zip`

⑼ Code that determines the number of files in a compressed file without extracting it

> `unzip -l MyFile.zip | wc -l`
 
⑽ Code to display the absolute paths of all files within a certain directory

> `find /path/to/directory -type f`

⑾ Find files containing 'ABC', ignoring case differences.

<br>

```python
grep -ril "ABC" /path/to/directory

# -r is for recursive search.
# -i ignores case (uppercase/lowercase).
# -l lists only filenames.
```

<br>

⑿ Find files and directories containing 'ABC', ignoring case differences

<br>

```python
find /path/to/directory -type d -iname "*ABC*" -o -type f -iname "*ABC*"

# -type d searches for directories, -type f for files.
# -iname searches case-insensitively.
# *ABC* is the search pattern (enclosed in asterisks for partial matching).
```

<br>

⒀ Find a file with the exact file name PTPRC.pickle' in the directory 
 
> `ls /path/to/folder | grep PTPRC.pickle`

⒁  Find files with the exact file name 'ABC' in the directory

> `find /path/to/directory -type f -name "ABC"`

⒂ Code to find all `.loom` files within a specific directory.

> `find /path/to/directory -maxdepth 1 -name "*.loom" -print`

⒃ Find duplicate file names within a particular directory

<br>

```python
#!/bin/bash

# Define the target directory
target_directory="/path/to/your/directory"

# Find and list duplicate files based on their MD5 hash
find "$target_directory" -type f -exec md5sum {} + | sort | uniq -d -w 32
```

<br>

⒄ Output time and file information in the order in which files were created in a given directory

<br>

```python
ls -ltr --time=creation
```

<br>

<br>

## **2. Modify the directory**

⑴ Change 'ABC' to 'ABCD'.

<br>

```python
# Renaming Directories and Files (Caution Advised):
## Renaming directories
find /path/to/directory -depth -type d -iname "*ABC*" -execdir rename 's/ABC/ABCD/' '{}' \;

## Renaming files
find /path/to/directory -type f -iname "*ABC*" -execdir rename 's/ABC/ABCD/' '{}' \;

# Replacing Content Inside Files
find /path/to/directory -type f -exec grep -il 'ABC' '{}' \; -exec sed -i 's/ABC/ABCD/gi' '{}' \;
```

<br>

⑵ Remove all subdirectories within a specific directory: that is, extract all files from the subdirectories.

<br>

```python
#!/bin/bash

# Base directory containing all target directories
base_directory="/path/to/your/directory"

# Iterate over each directory within the base directory
for target_directory in "$base_directory"/*/; do
    # Find all files and move them with a new name to avoid duplicates
    find "$target_directory" -mindepth 2 -type f -print0 | while IFS= read -r -d $'\0' file; do
        # Construct new filename by replacing '/' with '_' and removing the leading path
        new_name=$(echo "$file" | sed -e "s|$target_directory||" -e 's|/|_|g')
        # Move and rename the file
        mv "$file" "$target_directory$new_name"
    done

    # Remove empty directories
    find "$target_directory" -mindepth 1 -type d -empty -delete
done
```

<br>

<br>

## **3. Compress files**

⑴ Compress all files in a specific directory, including subfolders, into test.zip.

> `zip -r test.zip ./*`

⑵ Collecting only files ending with `.merged.png` and compressing them to `.tar.gz`

> `find /path/to/directory -type f -name '*.merged.png' -print0 | tar -czvf merged_png_files.tar.gz --null -T -`

<br>

<br>

## **4. Query process**

⑴ Query for all processes using `app.py`

> `ps -ef | grep app.py`

<br>

<br>

## **5. Execute process**

⑴ Running in the background using `nohup`

> ① **Step 1.** Prepare a `.sh` file or a `.py` file

> ② **Step 2.** Open the terminal

> ③ **Step 3.** After setting up the environment, navigate to the directory containing the `.sh` or `.py` file to be executed (for example: [app.py](https://jb243.github.io/pages/2347))

> ④ **Step 4.** Grant permission: `chmod 755 app.py`

> ⑤ **Step 5.** Execute with nohup: `nohup python app.py \`

> ⑥ **Step 6.** Press Enter once more to execute

> ⑦ **Step 7.** (Optional) To check execution: `ps -ef | grep app.py` → Check nohup.out. The keyword "app.py" should be recognized

> ⑧ **Step 8.** (Optional) To forcibly terminate background execution: `kill {ProcessID}`

⑵ Loops

<br>

<br>

## **6. Bioinformatics**

⑴ `wc -l filename`

> ① Count the number of lines in the file.

⑵ `grep -v ^# gencode.v19.long_noncoding_RNAs.gtf | wc -l`

> ① Output the number of lines in the file excluding those starting with `#`.  

> ② **Role of `^` (caret):** `^` represents the start of a line in regular expressions. `^#` means "lines starting with `#`."  

> ③ **Role of `-v`:** `-v` means the opposite. Without `-v`, only the lines starting with `#` would be output.  

> ④ **Role of `|` (vertical bar):** Passes the output of the previous command as input to the next command (`wc -l`).  

⑶ `grep -w gene gencode.v19.long_noncoding_RNAs.gtf | wc -l`

> ① Output the number of lines in the file that contain the independent word `gene`.  

> ② Without `-w`, using `grep gene` would match any line containing `gene`, even as part of another word, such as `genetics` or `progene`.  

<br>

---

_Input **:** 2023.12.28 14:59_
