## **Understanding COMMOT**

Recommended Reading: 【Bioinformatics】 [Bioinformatics Analysis Index](https://jb243.github.io/pages/836)

---

**1.** [Overview](#1-overview)

**2.** [Theory](#2-theory)

**3.** [Results](#3-results)

**4.** [Discussion](#4-discussion)

---
 
Cang, Z. et al. (2023). Nature methods, 20(2), 218-228. ([ref](https://www.nature.com/articles/s41592-022-01728-4))

---

<br>

## **1. Overview**

⑴ **CCC** (cell-cell communication)

> ① Important for understanding biological heterogeneity: Cell fate determination, cell differentiation patterns, pathological mechanisms

> ② Various forms: Ligand-receptor interactions, cell-cell interactions, organ-level interactions (e.g. insulin), organism-level interactions (e.g. pheromones)

> ③ Clinical biomarkers: ICI (immune-checkpoint inhibitor), tissue regeneration research

>> ○ Example: Significant increase in necroptosis-related hepatocyte and macrophage interactions in APAP-induced AKI (acute liver injury) tissue

<br>

<img width="504" alt="스크린샷 2025-04-05 오후 9 16 13" src="https://github.com/user-attachments/assets/606bd303-8d73-4529-9f69-e951a1ab5a87" />

**Figure 1.** Example of tissue regeneration research

<br>

⑵ [Existing CCC Analysis Algorithms](https://jb243.github.io/pages/841)

> ① **Bulk RNA-seq based**: BulkSignalR, squidpy, IPA, Omnipath

> ② **scRNA-seq based**: CellTalkDB, CellPhoneDB, CellChat, ICELLNET, NicheNet, SoptSC, CytoTalk, scTensor, CCCExplorer, Connectome, Ramilowski, FlowSig, scSeqComm

> ③ **Spatial Transcriptomics (ST) based**: Giotto, spata2, CellPhoneDB v3, stLearn, SVCA, MISTy, NCEM, COMMOT, SCOTIA, STopover, cytosignal, SpatialDM, SpaTalk, stMLnet, HoloNet, DeepLinc

<br>

<img width="665" alt="스크린샷 2025-04-05 오후 9 16 49" src="https://github.com/user-attachments/assets/be05af04-9ce8-4fd6-93b6-2cd29a4b42ec" />

**Figure 2.** Examples of CCC analysis algorithms using ST

<br>

>> ○ **Giotto**: Constructs spatial proximity graph to identify interactions via membrane-bound ligand-receptor pairs

>> ○ **CellPhoneDB v3**: Restricts interactions to cell clusters within the same spatially-defined microenvironment

>> ○ **stLearn**: Associates co-expression of ligand and receptor genes with spatial diversity of cell types

⑶ COMMOT

> ① Necessity: Existing methods cannot handle multiple-species interaction

>> ○ Multiple ligand types can bind multiple receptor types, leading to competition

>> ○ 72% of ligands and 60% of receptors bind to multiple types

<br>

<img width="608" alt="스크린샷 2025-04-05 오후 9 17 27" src="https://github.com/user-attachments/assets/4adf272d-5527-417f-87aa-8cb638e4bba7" />

**Figure 3.** Competition simulation due to multi-species interaction

<br>

> ② Need for [OT](https://jb243.github.io/pages/2386) (optimal transport)

>> ○ OT is defined as the method of optimally allocating resources from origin (supply) to destination (demand)

>> ○ This concept has been studied in economics, physics, astronomy, computer science, etc.

>> ○ Multi-species interaction requires numerical analysis, and the process of distributing gene expression from ligands to receptors is similar, hence OT is needed

<br>

<img width="419" alt="스크린샷 2025-04-05 오후 9 17 46" src="https://github.com/user-attachments/assets/9a0c7fa9-af59-42e1-9de2-06ab9f61d325" />

**Figure 4.** OT analogy with moving sandcastles

<br>

<br>

## **2. Theory**

⑴ Problem Definition of OT

<br>

<img width="616" alt="스크린샷 2025-04-05 오후 9 18 09" src="https://github.com/user-attachments/assets/b38afa14-4068-4d54-82f4-167da32d830a" />

**Figure 5.** Problem Definition of OT

<br>

> ① Can be used for interaction between one ligand and one receptor

> ② To apply to multi-species interaction, the problem needs to be reformulated as COT

⑵ Problem Definition of COT (collective OT)

<br>

<img width="406" alt="스크린샷 2025-04-05 오후 9 18 27" src="https://github.com/user-attachments/assets/2cf2ebc0-d59d-4c02-83c4-55b11bf561b5" />

**Figure 6.** Problem Definition of COT

<br>

> ① Transport plan **P** ∈ ℝ<sub>+</sub><sup>n<sub>l</sub>×n<sub>r</sub>×n<sub>s</sub>×n<sub>s</sub></sup>

>> ○ n<sub>l</sub>: Number of ligands

>> ○ n<sub>r</sub>: Number of receptors

>> ○ n<sub>s</sub>: Number of spots or cells (in case of single cell ST)

> ② Ideally, ∑<sub>j</sub> ∑<sub>l</sub> P<sup>(i,j)</sup>(k,l) = α<sub>i</sub>(k), ∑<sub>i</sub> ∑<sub>k</sub> P<sup>(i,j)</sup>(k,l) = β<sub>j</sub>(l)

>> ○ However, in noisy data this assumption may not hold

>> ○ Instead, define unexplained ligand and receptor quantities **μ**, **ν** and solve optimization by panelizing them

 ⑶ Summary of COT Problem Solving

<br>

<img width="532" alt="스크린샷 2025-04-05 오후 9 21 11" src="https://github.com/user-attachments/assets/91a2d27f-6a3f-4b97-a75d-1f826099fb8b" />

<br>

⑷ **Step 1.** Apply spatial cutoff to cost function C to consider only spatially proximal interactions as valid

<br>

<img width="513" alt="스크린샷 2025-04-05 오후 9 21 31" src="https://github.com/user-attachments/assets/0c0a7c72-60aa-42af-bf6a-dc6633b55396" />

 **Figure 7.** Cost function using spatial information

<br>

 ⑸ **Step 2.** Express _F_(·) term as Shannon entropy term and L2 regularizer term

<br>

<img width="494" alt="스크린샷 2025-04-05 오후 9 21 52" src="https://github.com/user-attachments/assets/e4b44000-1e09-435d-803e-e6c77f6bbb61" />

<br>

 ⑹ **Step 3.** Apply [Lagrange multiplier method](https://jb243.github.io/pages/1813)

> ① Principle of Lagrange multipliers

<br>

<img width="403" alt="스크린샷 2025-04-05 오후 9 22 09" src="https://github.com/user-attachments/assets/36fc384d-7ffb-4422-9988-24685529900a" />

 **Figure 8.** Principle of Lagrange multipliers

<br>

> ② Example of Lagrange multipliers

<br>

<img width="600" alt="스크린샷 2025-04-05 오후 9 22 29" src="https://github.com/user-attachments/assets/0d2325be-8821-4961-a342-98ad1d6bf4e4" />

**Figure 9.** Example of Lagrange multipliers

<br>

> ③ **Step 3-1.** Introduce additional variables for linearity in Lagrange formulation

<br>

<img width="553" alt="스크린샷 2025-04-05 오후 9 22 52" src="https://github.com/user-attachments/assets/b420e228-9064-4e59-a2fb-49fe5fd11e01" />

<br>

> ④ **Step 3-2.** First application of Lagrange method: Apply to 5 out of 9 variables first as it's hard to apply all at once

<br>

<img width="577" alt="스크린샷 2025-04-05 오후 9 23 12" src="https://github.com/user-attachments/assets/a703e5fd-e22c-4c64-9618-668b9172e18d" />

<br>

> ⑤ **Step 3-3.** Second application of Lagrange method: Apply to remaining 4 variables

<br>

<img width="507" alt="스크린샷 2025-04-05 오후 9 23 35" src="https://github.com/user-attachments/assets/6e0514af-8329-4fa0-8728-11096a107d27" />

<br>

> ⑥ **Step 3-4.** Express complex equation as update formula

<br>

<img width="671" alt="스크린샷 2025-04-05 오후 9 23 49" src="https://github.com/user-attachments/assets/7a3cfaad-f325-4d63-aba9-04dbcebf80a4" />

<br>

> ⑦ **Step 3-5.** Final conclusion

<br>

<img width="666" alt="스크린샷 2025-04-05 오후 9 24 13" src="https://github.com/user-attachments/assets/9bf24b95-0f52-447e-96db-9580a1e8fd8e" />

<br>

> ⑧ Numerical process to solve complex equation: Uniqueness or convergence of the solution is not guaranteed

<br>

<img width="317" alt="스크린샷 2025-04-05 오후 9 24 29" src="https://github.com/user-attachments/assets/2413099b-4c73-49e4-a161-b152bebe3520" />

**Figure 10.** Numerical solving of complex equation

<br>

<br>

## **3. Results**

 ⑴ Visium data shows typical skin structure

> ① Mainly shows the lower epidermis and dermis part of the skin, which is composed of epidermis, dermis, and smooth muscle

> ② The inner dermal part being basal cell aligns with expectations

> ③ As seen from pseudotime results, basal cells divide and produce other cells

<br>

<img width="496" alt="스크린샷 2025-04-05 오후 9 24 52" src="https://github.com/user-attachments/assets/cbfd0ea7-f062-4d95-9436-f3147980aa60" />

**Figure 11.** Visium skin data

<br>

 ⑵ **Noteworthy Interactions**

> ① GAS6-TYRO3 interaction inferred by COMMOT

> ② Interaction strength for ligand i, receptor j, and spot l is calculated as follows: Result

<br>

<img width="262" alt="스크린샷 2025-04-05 오후 9 25 09" src="https://github.com/user-attachments/assets/37c93173-af82-404d-8aea-99fa85ade2f1" />

**Figure 12. **Visualization of GAS6-TYRO3 interaction strength

<br>

> ③ Interaction strength between clusters is calculated as follows

<br>

<img width="306" alt="스크린샷 2025-04-05 오후 9 25 27" src="https://github.com/user-attachments/assets/fb6ecf4f-6c8a-4031-8d92-941ea9f72404" />

 **Figure 13.** Cluster-level GAS6-TYRO3 interaction strength

<br>

> ④ GAS6-TYRO3 interaction experimentally validated

<br>

<img width="451" alt="스크린샷 2025-04-05 오후 9 25 42" src="https://github.com/user-attachments/assets/f6d9ec10-0d8f-4598-b7a8-78cb4c4de015" />

**Figure 14.** Experimental validation of GAS6-TYRO3

<br>

> ⑤ GAS-TYRO3 interaction is known as an innate immune suppressor and may relate to the role of unconventional T cells (e.g. γδ T cells, MAIT cells) in skin tissue homeostasis and local immune response

 ⑶ **Noteworthy Pathways**

> ① Interaction direction: Defined as vector sum of relevant transport plan entries limited to ligand-receptor pairs of major pathways. WNT and TGF-β have similar directions due to antagonistic roles in basal cell proliferation

<br>

<img width="592" alt="스크린샷 2025-04-05 오후 9 26 03" src="https://github.com/user-attachments/assets/6ed489a2-d00f-41a3-91bf-0b7b61ba8361" />

**Figure 15.** Interaction directions by major pathways

<br>

> ② Basal cell markers (KRT15, KRT5) and granular cell markers (LOR, FLG) related to basal cell proliferation are prominently observed in signaling pathways like WNT and TGF-β

<br>

<img width="706" alt="스크린샷 2025-04-05 오후 9 26 25" src="https://github.com/user-attachments/assets/83d441e3-5aa2-4c4c-9c79-1ec28be69278" />

**Figure 16. **Interaction strength of major pathways and gene association analysis

<br>

 ⑷ **Noteworthy Visualizations**

<br>

<img width="723" alt="스크린샷 2025-04-05 오후 9 26 41" src="https://github.com/user-attachments/assets/cad2889a-a182-47a1-a069-935e4e081d78" />

**Figure 17. **Other noteworthy visualizations

<br>

<img width="518" alt="스크린샷 2025-04-05 오후 9 27 09" src="https://github.com/user-attachments/assets/f14f3516-04b8-4ca3-930b-638e41443038" />
 
**Figure 18.** Other noteworthy visualizations

(Spot clustering via spot embedding)

<br>

<br>

## **4. Discussion**

 ⑴ Strengths

> ① Enables analysis of multi-species interaction

> ② Provides various downstream analyses

> ③ Validated across five ST platforms and various tissues

> ④ Runs on Python and is relatively easy to use

 ⑵ Limitations

> ① The 1000 μm used as spatial cutoff in cost function is biologically questionable

> ② Convergence and uniqueness of the COT solution are not sufficiently validated

> ③ Protein abundance and modifications are not included in ligand-receptor interactions

> ④ Cannot analyze triple interactions

<br>

<img width="367" alt="스크린샷 2025-04-05 오후 9 27 33" src="https://github.com/user-attachments/assets/a03e0d40-9c2a-4a3b-9dec-965a478c5fa1" />

 **Figure 19.** Triple interaction of FCER2A, ITGAX, ITGB2 using AlphaFold3

<br>

---
 
_Input: 2025.04.05 08:33_
