## Chapter 1. Composition of Living Organisms

Recommended Articles: 【Biology】 [Biology Index](https://jb243.github.io/pages/1457), 【Organic Chemistry】 Biopolymers: [Carbohydrates](https://jb243.github.io/pages/1389),
[Proteins](https://jb243.github.io/pages/1391), [Lipids](https://jb243.github.io/pages/1393), [Nucleic Acids](https://jb243.github.io/pages/1395)

---

**1.** [Definition of Living Organisms](#1-definition-of-living-organisms)

**2.** [Water and Carbon](#2-water-and-carbon)

**3.** [Biopolymers](#3-biopolymers-macromolecules)

**4.** [Carbohydrates](#4-biopolymer-1-carbohydrates-polysaccharide--ch2on)

**5.** [Proteins](#5-biopolymers-2-proteins--polymers-of-amino-acids)

**6.** [Lipids](#6-biomolecules-3-lipids)

**7.** [Nucleic Acids](#7-biomolecules-4-nucleic-acids-polymers-of-nucleotides)

---

**a.** [Hydrophobicity Index of Biopolymers](https://jb243.github.io/pages/1469)

**b.** [Perm hair and Disulfide Bonds](https://jb243.github.io/pages/1417)

**c.** [Biopolymer Library](https://jb243.github.io/pages/1327)

---

<br>

## **1. Definition of Living Organisms**

⑴ **Definition 1.** Organisms grow, move, reproduce, respond to stimuli, and have metabolism.

> ① Is fire a living organism and a donkey not?

 ⑵ **Definition 2.** Defined as an organism if all the following conditions are met:

> ① Possesses a common set of biomolecules.

> ② Maintains homeostasis.

> ③ Capable of evolution.

> ④ Requires water.

<br>

<br>

## **2. Water and Carbon**

⑴ H<sub>2</sub>O

> ① Polarity: Oxygen has high electronegativity, attracting electrons from hydrogen, causing a slight negative charge. Conversely, the hydrogen side has a slight positive charge.

> ② Hydrogen Bonds: Molecular attraction between hydrogen and electronegative atoms (i.e., F, O, N).

>> ○ Water molecules cohere due to hydrogen bonds.

>> ○ A water molecule can form hydrogen bonds at four sites

<br>

<img width="424" alt="스크린샷 2024-12-07 오후 11 02 25" src="https://github.com/user-attachments/assets/05cee0da-ded0-41a3-bc19-48d194905e40">

**Figure 1.** Hydrogen Bonding in Water

<br>

> ③ High heat capacity due to hydrogen bond: Plays an important role in temperature regulation

> ④ Excellent solvent (i.e., universal Solvent) due to its polarity

>> ○ Dissolves other polar substances (e.g., salts, alcohols, acids, bases).

>> ○ Facilitates chemical reactions.

>> ○ Hydration Shell: The layer of water molecules surrounding a solute, making ions appear larger in water.

> ⑤ Expansion upon freezing

>> ○ Reason: Ice forms **hexagonal crystals**.

>> ○ At atmospheric pressure, water's density is always greater than ice.

<br>

![image](https://github.com/user-attachments/assets/5c90037f-d777-42d1-af9d-2785f27feddd)

**Figure 2.** Density of Water and Ice by Temperature

<br>

⑵ C (Carbon)

> ① Carbon can bond with four atoms, used as a backbone for biopolymers.

> ② Carbon constitutes a significant portion of the biomass.

> ③ Organic Chemistry: The chemistry of carbon.

>> ○ Molecules based on carbon are called organic compounds.

>> ○ Compounds made of carbon and hydrogen are called hydrocarbons (e.g., CH<sub>4</sub>).

> ④ Meteorites from Mars contain carbonates and hydrocarbon chains.

>> ○ However, there's no confirmation that these are biopolymers produced by living organisms.

⑶ Functional Groups

<br>

![image](https://github.com/user-attachments/assets/8d97c963-2c1f-4e1c-82ff-a23a7b23e1a6)

**Figure 3.** Types of Functional Groups

<br>

<br>

## 3. Biopolymers (macromolecules)

 ⑴ Earth's organisms contain the same set of macromolecules (e.g., carbohydrates, proteins, lipids, nucleic acids).

 ⑵ Formation of Polymers: All polymers are made through dehydration reactions (water is produced) of monomers.

 ⑶ Degradation of Polymers: Polymers are broken down into monomers through hydrolysis.

 ⑷ **Benefits of Polymers:** Cause less osmotic pressure and maintain a lower concentration of monomers inside cells.

<br>

<br>

## 4. Biopolymer 1. Carbohydrates (polysaccharide) = (CH<sub>2</sub>O)<sub>n</sub>

⑴ Functions: **Primary Energy Source**, **Structural Role**

> ① Energy Source: 4 kcal/g

> ② Water-soluble molecules

⑵ Glycosidic Bond: Dehydration reaction between sugars.

<br>

![image](https://github.com/user-attachments/assets/c0eb1c1e-1edc-4748-98ce-154be1c7d7c3)

**Figure 4.** Glycosidic Bond in Glucose

<br>

 ⑶ [Monosaccharides](https://jb243.github.io/pages/1389): Glucose(6C), Fructose(6C), Galactose(6C), Others(3C ~ 7C)

<br>

![image](https://github.com/user-attachments/assets/d5efe27c-4f36-4dd0-99a6-5f1f481ea729)

**Figure 5.** Structural Isomers of C<sub>6</sub>H<sub>12</sub>O<sub>6</sub>

<br>

⑷ [Disaccharides](https://jb243.github.io/pages/1389)

⑸ [Oligosaccharides](https://jb243.github.io/pages/1389): Referring to C3 ~ C12

⑹ Polysaccharides: **Glycogen**, **Starch**, **Cellulose**, **Chitin**, **Peptidoglycan**

> ① Glycogen

>> ○ A polymer of α-linked glucose with significantly more branches, serving as a relatively short-term energy source in animals.

>> ○ α 1 → 4 linkage, α 1 → 6 linkage

> ② Starch

>> ○ A polymer of α-linked glucose that serves as a long-term energy source in plants.

> ③ Cellulose

>> ○ A polymer of β-linked glucose that forms cell walls and is the most abundant carbohydrate on Earth.

>> ○ β 1 → 4 linkage

> ④ Chitin

>> ○ NAG (N-Acetylglucosamine): The functional group on the second carbon of glucose is substituted with an amine group.

>> ○ Chitin is a polymer of NAG: Forms exoskeletons in insects and crustaceans.

>> ○ β 1 → 4 linkage

<br>

<img width="230" alt="스크린샷 2024-12-07 오후 11 27 51" src="https://github.com/user-attachments/assets/08b434ce-bb25-4854-8004-f42a127214fd">

**Figure 6.** Structure of NAG

<br>

> ⑤ Dextrin

>> ○ α 1 → 4 linkage

>> ○ Keep in mind that Dextran is a brand name.

⑺ α Linkage and β Linkage

<br>

<img width="580" alt="image" src="https://github.com/user-attachments/assets/1afd654b-a44d-4bd2-9838-d3bd1ad49a99" />

**Figure 7.** α Glucose (left) and β Glucose (right)

<br>

> ① α linkage: A type of bond where glucose molecules are connected without flipping upside down, resulting in a slightly twisted three-dimensional structure that forms an α-helix.

> ② α-linked glucose: The -OH group on the first carbon and the -OH group on the fourth carbon are in the same direction, making the structure unstable → Energy storage.

> ③ β linkage

>> ○ The glucose molecule on one side is flipped upside down, aligning the first and fourth carbons, forming a straight linear structure in three dimensions.

>> ○ Chains of β-linked glucose are tightly bound through strong hydrogen bonds.

> ④ β-linked glucose: The -OH group on the first carbon and the -OH group on the fourth carbon are in opposite directions, making the structure stable → Structural formation.

⑻ Starch

> ① Amylose is an α-linked glucose polymer without branches, formed by the bonding of the first carbon to the fourth carbon.

> ② Amylopectin is an α-linked glucose polymer with branches, formed by the bonding of the first carbon to the sixth carbon.

> ③ Starch is insoluble because α-glucose chains form tight crystals through hydrogen bonding.

> ④ Most digestive enzymes can only break down α linkages because the spatial structure is relatively open, allowing easier access for enzymes.

<br>

![image](https://github.com/user-attachments/assets/addb1903-e27d-4aa3-8448-b69c4187114d)

**Figure 8.** Amylose and Amylopectin

<br>

⑼ Cellulose

> ① Most abundant carbohydrate on Earth.

> ② β 1 → 4 linkage

<br>

![image](https://github.com/user-attachments/assets/9c92f17a-c0b0-4b33-bf5a-c1daf2c6d002)

**Figure 9.** Hydrogen Bonding Between β-Glucose Chains

<br>

> ③ Cellulase

>> ○ [Rumen of ruminants](https://jb243.github.io/pages/82): Microbes in the first and second stomachs of ruminants synthesize cellulase.

>> ○ Termites' intestines: Trichonympha, a symbiotic microbe, synthesizes cellulase.

<br>

<br>

## 5. Biopolymers 2. Proteins = Polymers of Amino Acids

⑴ Characteristics

> ① Constituent elements: C, H, O, N, S

> ② Functions

>> ○ Enzymes in metabolic activities.

>> ○ Formation of intra- and extracellular structures: Constitutes half of the body's dry weight.

>> ○ Muscle contraction.

>> ○ Immune function.

>> ○ Hormones or signal proteins.

>> ○ Signal transduction inside cells.

>> ○ Substance transport across membranes.

>> ○ Energy conversion and storage: 4 kcal/g, mainly in fetuses.

>> ○ DNA replication, repair, and recombination.

>> ○ Transcription, translation.

>> ○ Protein transport and secretion.

> ③ Proteins: Composed of one or more polypeptides.

> ④ Polypeptides: Made of peptide bonds between amino acids.

> ⑤ Peptide bond: Dehydration reaction between amino group (-NH<sub>2</sub>) and carboxyl group (-COOH).

<br>

![image](https://github.com/user-attachments/assets/9c36f20c-b10b-4d81-a981-6aad9c12d552)

**Figure 10.** Dehydration reaction between two amino acids

<br>

⑵ Amino Acids

> ① There are a total of 20 amino acids: 21 including selenocysteine (Sec).

<br>

<img width="413" alt="스크린샷 2024-12-07 오후 11 38 47" src="https://github.com/user-attachments/assets/e406bee3-a85f-4268-95f7-8726dddb5c02">

**Table 1.** Symbols of Amino Acids ([ref](https://blogs.oregonstate.edu/psquared/week-2/))

<br>

> ② Structure: Central carbon, amino group, carboxyl group, R group

> ③ All amino acids have the same backbone, but there are 20 types of side chains (denoted as R).

>> ○ Others do not participate in protein synthesis.

>> ○ Including minor amino acids like selenocysteine, the number can exceed 20.

> ④ Zwitterion: Acts as an acid or base. Related to [isoelectric point](https://jb243.github.io/pages/77#footnote_link_67_53).

> ⑤ Amino acids differ in properties depending on their side chains.

>> ○ Nonpolar amino acids: Ala, Ile, Leu, Met, Phe, Pro, Trp, Val, Gly

>> ○ Polar, uncharged amino acids: Asn, Cys, Gln, Ser, Thr, Tyr

>> ○ Positively charged amino acids: Arg, His, Lys

>> ○ Negatively charged amino acids: Asp, Glu

>> ○ Amino acids containing -OH group: Ser, Thr, Tyr. Phosphate groups can be added to the -OH group, making them useful in signal transduction.

>>> ○ Ser, Thr: Involved in phosphorylation cascades in signal transduction.

>>> ○ Tyr: Present in tyrosine kinase receptors in signal transduction.

>> ○ Amino acids containing sulfur: Cys, Met. Only Cys forms disulfide bonds.

>> ○ Amino acids containing phenyl group: Phe, Trp, Tyr

>> ○ Phenyl group shows absorbance at 280 nm. Mechanism for quantifying proteins using a UV spectrometer.

>> ○ Amino acids not used in TCA cycle: Leu, Lys

> ⑥ [Titration of Amino Acids](https://jb243.github.io/pages/1352#footnote_link_67_55)

>> ○ Related to each amino acid's pK<sub>a1</sub>, pK<sub>a2</sub>, pK<sub>aR</sub>, pI, hydrophobicity index, and its proportion in proteins.

⑶ Primary Structure of Proteins

> ① Amino acid sequence.

> ② Planar nature of peptide bonds.

>> ○ Amide plane: Forms one peptide bond. Plane formed by two alpha carbons and the C, O, N, H between them.

>> ○ Naming multiple bonds in one plane.

<br>

![image](https://github.com/user-attachments/assets/6b3119d0-8fca-475e-a138-3f061403045b)

**Figure 11.** Naming multiple bonds in one plane

<br>

>>> ○ Φ bond: N - C<sub>α</sub> bond

>>> ○ ψ bond: C<sub>α</sub> - C bond

>>> ○ ω bond: C - N bond

>>> ○ χ bond: C<sub>α</sub> - R (functional group) bond

>> ○ C=O bond resonates with ω bond, making C=O and C-N bonds into 1.5 bonds, thus fixed and non-rotatable.

<br>

![image](https://github.com/user-attachments/assets/22328d39-320a-42cd-ba91-5ab23c5c1bf7)

**Figure 12.** Planar nature of peptide bonds

<br>

⑷ Secondary Structure of Proteins (Chou-Fasman rules)

> ① A three-dimensional coiling or folding of the amino acid sequence, driven by hydrogen bonding between the backbone atoms (H and O) of amino acids.

> ② Alpha helix (α-helix)

<br>

<img width="290" alt="스크린샷 2024-12-07 오후 11 49 34" src="https://github.com/user-attachments/assets/737ce9a6-7f2b-4ed5-a21f-fafd5aa4ecdd">

**Figure 13.** Alpha helix structure

<br>

>> ○ The C=O bond of the N-th amino acid and N-H bond of the (N+4)-th amino acid form hydrogen bonds.

>>> ○ On average, 3.6 residues per turn (though there is variation).

>> ○ Elastic, right-handed helix.

>> ○ Hydrophilic amino acids have attractions and repulsions that hinder the formation of alpha helices.

>>> ○ Alpha helices are mainly formed by hydrophobic amino acids.

>>> ○ Commonly observed in transmembrane proteins.

>> ○ Gly: The functional group R is a hydrogen atom, making it small and unsuitable for forming an alpha-helix structure.

>> ○ Pro: Forms an imino group, which restricts hydrogen bonding → Does not form an alpha-helix structure.

> ③ Beta pleated sheet (β-pleated sheet): Also known as zigzag conformation.

<br>

<img width="359" alt="스크린샷 2024-12-07 오후 11 51 33" src="https://github.com/user-attachments/assets/18212125-7058-453a-99f6-1373ad1da6fc">

**Figure 14.** Beta pleated sheet structure

<br>

>> ○ Characteristic: Hydrogen bonds exist not only within amino acids but also between amino acid chains.

>> ○ Common structure in fibrous proteins like silk.

>> ○ Spider webs are a structure with β-pleated sheets and added α-helices.

> ④ Ramachandran plot

>> ○ A 2D representation of secondary structures, with the Φ bond on the horizontal axis and the ψ bond on the vertical axis.

>> ○ Alpha helix structures are concentrated around (-60°, -60°).

>> ○ Beta pleated sheet structures are concentrated around (-120°, 120°).

>> ○ Purpose: To see if residues are in appropriate positions. Many islands on the plot indicate inappropriate positions.

>> ○ Example

<br>

![image](https://github.com/user-attachments/assets/668c31f8-73a7-447c-a788-e3746d6ed364)

**Figure 15.** Example of a Ramachandran plot

<br>

⑸ Tertiary Structure of Proteins

> ① Overall three-dimensional structure of a polypeptide.

> ② Interactions between R groups lead to different three-dimensional structures, which determine different properties.

> ③ Ionic bond

> ④ Covalent bond

> ⑤ Disulfide bridge: Makes the tertiary structure more rigid.

>> ○ Reducing agents: Break disulfide bonds. β-mercaptoethanol, DTT (dithiothreitol), etc.

>> ○ Formation: In eukaryotic cells, disulfide bonds are formed in the rough endoplasmic reticulum; in prokaryotic cells, they are formed in the cytoplasm by PDI (protein disulfide isomerase).

>> ○ Anfinsen's experiment

>>> ○ NEM (N-ethylmaleimide): Covalently binds to the -SH group of cysteine not involved in disulfide bonds.

>>> ○ DTT (including β-metcaptoethanol): -S-S- → -SH + HS-. Acts as a reducing agent breaking disulfide bonds. Glutathione (GSH) has a similar function.

>>> ○ NEM* → DTT → NEM: Quantifies the amount that originally existed as SH groups. Note that * represents a radioactive isotope.

>>> ○ NEM → DTT → NEM*: Quantifies the amount that originally existed as -S-S- (disulfide bonds). Note that * represents a radioactive isotope.

>>> ○ **Conclusion 1:** When denaturing factors are removed, the protein restores its original structure.

>>> ○ **Conclusion 2:** Disulfide bonds contribute irreversibly to the formation of the three-dimensional structure and must be carefully regulated.

> ⑥ Hydrophobic interactions: Forces arising because the biological environment is water.

> ⑦ Polar bonds and hydrogen bonds between R groups

> ⑧ Van der Waals forces

> ⑨ Other interactions

>> ○ Pseudo hydrogen bond of C=O bond: n → π* interaction, present in more than 45% of protein residues.

>> ○ π-π stacking: Aromatic ring stacking

>> ○ Steric hindrance

>> ○ C5 hydrogen bonds in β-sheet backbone

>> ○ Cation-π interaction

> ⑩ Specific interactions can be too strong, causing proteins to misfold, necessitating the assistance of protein assembly factors like chaperonins.

>> ○ **Example 1.** [Heat shock protein](https://jb243.github.io/pages/1096): Prevents proteins from denaturing due to temperature.

⑹ The quaternary structure of proteins: Interactions between tertiary structure polypeptides.

> ① Transthyretin (tetramer), hemoglobin (α<sub>2</sub>β<sub>2</sub>), collagen (3 helix)

⑺ Protein structure determination

> ① Determining the primary structure

>> ○ **Method 1.** Edman degradation

>>> ○ Degrades amino acids one at a time by binding to the N-terminus of polypeptides in mild alkaline conditions.

>>> ○ Only about the first 10 amino acids from the N-terminal can be known.

>>> ○ It is a very traditional method that is rarely used nowadays.

>> ○ **Method 2.** Peptidases

>>> ○ Appropriately combining endopeptidases and exopeptidases to appropriately degrade amino acids and determine them one by one.

>>> ○ Endopeptidases: Pepsin, trypsin, chymotrypsin, etc.

>>> ○ Exopeptidases: Carboxypeptidases, etc.

>>> ○ [Mechanism of peptidases](https://jb243.github.io/pages/1022)

>> ○ **Method 3.** Ab array

>> ○ **Method 4.** Mass spectrometry

>>> ○ 1<sup>st</sup>. Electrophoresis & DNA ladder: Can determine whether it's pure or a mixture.

>>> ○ 2<sup>nd</sup>. Trypsin treatment

>>> ○ 3<sup>rd</sup>. First [mass spectroscopy](https://jb243.github.io/pages/1404#footnote_link_67_51): Constructing the MS spectrum.

>>> ○ 4<sup>th</sup>. Second mass spectroscopy: Constructing the MS/MS spectrum. Also known as tandem MS, MS/MS, fragmentation.

>>> ○ 5<sup>th</sup>. Reconstruction through the spectrum: Realm of informatics.

>>> ○ This method is a first-generation technology and becomes much more complicated when considering post-translational modifications.

> ② Determining the secondary structure

>> ○ **Method 1.** [Circular dichroism](https://jb243.github.io/pages/1354#2-coordination-ions) (CD)

>> ○ **Method 2.** Infrared spectroscopy: Useful for investigating flexible peptide and protein structures.

>> ○ **Method 3.** Molecular dynamics simulation

> ③ Determining tertiary and quaternary structures

>> ○ **Method 1.** X-ray crystallography: About 90% are determined by this method. Allows for inference of 3D coordinates of atoms.

>> ○ **Method 2.** NMR: About 9% are determined by this method.

>> ○ **Method 3.** Cryo-EM

>>> ○ Resolution is low but is steadily improving and useful for large protein complexes like capsids and amyloids.

>>> ○ Protein is immobilized by lowering the temperature to -200 ℃ to observe its 3D structure.

⑻ Protein denaturation

> ① Denaturation: The three-dimensional structure changes due to factors such as salt concentration, pH, or high temperature, resulting in loss of function.

<br>

<br>

## 6. Biomolecules 3. Lipids

⑴ Characteristics

> ① Mostly composed of carbon and hydrogen, thus high in energy content.

> ② Hydrophobic: Does not dissolve in water, but dissolves in acetone, alcohol, benzene, etc.

> ③ Insulating effect (e.g. subcutaneous fat layer), insulation, waterproofing.

⑵ Fat: A triglyceride consisting of glycerol linked to three hydrocarbon-rich fatty acid tails via ester bonds.

> ① Energy content: 9 kcal/g

> ② Digestive enzyme, lipase, breaks the ester bonds ([ref](https://jb243.github.io/pages/82)).

> ③ Fatty acids: Composed of an even number of carbons because they are biosynthesized from acetyl-CoA.

> ④ Saturated fatty acid: Fatty acid fully saturated with hydrogen, containing only single bonds. Solid at room temperature, typically of animal origin.

> ⑤ Unsaturated fatty acid: Contains one or more double bonds (cis configuration). Liquid at room temperature, typically of plant or fish origin.

> ⑥ Butter and margarine

>> ○ Butter: Saturated fat obtained from milk.

>> ○ Margarine: Made by adding hydrogen to unsaturated fats derived from plants to convert them into saturated fats.

> ⑦ Trans Fat: Formed during the hydrogenation process when hydrogen is added and then removed, creating trans double bonds.

>> ○ Trans fats have a stacking property, resulting in a higher boiling point.

>> ○ Trans fats have a stacking property, causing them to accumulate in the body, posing health risks.

<br>

![image](https://github.com/user-attachments/assets/e43b87fc-33f9-4010-a63c-92815830f1f9)

 **Figure 16.** _cis_ and _trans_ bonds in fatty acid tails

<br>

⑶ Steroids: A hydrophobic substance with a planar structure composed of three six-membered rings and one five-membered ring.

> ① Derived from lipids without monomers (from cholesterol).

> ② Issues with synthetic steroids

>> ○ Marked mood changes (violent mood swings).

>> ○ Depression, liver damage, high cholesterol, high blood pressure

> ③ **Cholesterol**: Found in some animal cell membranes.

>> ○ Refers to steroids with an -OH group.

>> ○ **Function 1.** Regulating fluidity of animal cell membranes.

>> ○ **Function 2.** Precursor to various substances: Bile salts, steroid hormones (sex hormones, adrenal cortex hormones, etc.), Vitamin D

>> ○ **Function 3.** Constituting the protective sheath of nerves

>> ○ **Function 4.** ROS scavenger 

>> ○ Not oxidized in the body to produce energy.

>> ○ 75% of cholesterol is obtained through synthesis.

>> ○ 25% of cholesterol is obtained through diet.

>> ○ Cholesterol is abundant in animal-based foods such as egg yolks, liver, and squid, and it is recommended to consume no more than 300 mg per day.

>> ○ Can cause arteriosclerosis and heart disease.

> ④ Sterols: Found in some plant cell membranes.

> ⑤ Sex hormones (estradiol, androgen)

<br>

<img width="224" alt="스크린샷 2024-12-08 오전 8 35 58" src="https://github.com/user-attachments/assets/5a5e5245-628f-4152-ae10-fce5cf48e8bf">

**Figure 17.** Examples of sex hormones

<br>

⑷ Phospholipids = 1 × phosphate head group + 1 × glycerol backbone + 2 × fatty acids + ester bonds. Constituting the phospholipid bilayer. Amphipathic.

> ① Phospholipid forms in aqueous solutions

>> ○ Micelle: Spherical shape, single layer. Contains a single fatty acid. Used to package and transport hydrophobic substances through blood vessels.

>> ○ Liposome: Ring-shaped structure, bilayer. Contains two fatty acids. Applied in drug delivery.

> ② Phospholipid Diversity and Membrane Asymmetry: Determined by the types of molecules attached to the phosphate group.

>> ○ Lecithin: The most representative phospholipid. Contains glycerophosphate and adds luster to the skin. Persists in the body for a long time.

<br>

![image](https://github.com/user-attachments/assets/8df7ba19-8e13-4f1b-8a33-a82521f0ac2d)

**Figure 18.** Example of phospholipid: Phosphatidylcholine

 <br>

> ③ Phospholipid bilayer: Forms hydrophobic boundary of cells. Semi-permeable membrane.

> ④ Archaea have adapted to extreme environments by utilizing ether (-ROR-) bonds instead of ester (-RCOO-) bonds in the fatty acid linkages of their phospholipids.

<br>

![image](https://github.com/user-attachments/assets/65953ad2-bdce-4238-929a-dad6fd4064b3)

 **Figure 19.** Differences Between archaeal phospholipids and those of bacteria and eukaryotes

<br>

⑸ Wax

> ① Ester bond between fatty acids and long-chain alcohols.

> ② Consists of 40 ~ 60 CH<sub>2</sub> units,

> ③ Prevents the ingress and egress of water.

> ④ The Casparian strip, located in the cell walls of the endodermis in plant roots and rich in wax, regulates water transport.

⑹ Fat-soluble vitamins: A, D, E, K

⑺ Carotenoids

> ① Auxiliary pigments capturing light energy.

> ② Main reaction: β-Carotene + O<sub>2</sub> → 2 Vitamin A

<br>

<br>

## 7. Biomolecules 4. Nucleic Acids: Polymers of nucleotides

⑴ Nucleotide = Phosphate + Nucleoside (e.g., cytidine, uridine, thymidine) = Phosphate + Sugar + Nitrogenous Base

> ① Sugar: 5-carbon sugar (pentose)

>> ○ Ribose: 2' carbon has -OH group. Highly reactive.

>> ○ Deoxyribose: 2' carbon has -H. Less reactive.

> ② Nitrogenous base 

>> ○ Basic ring structure made of carbon-nitrogen covalent bonds.

>>> ○ Unshared electron pairs of nitrogen act as H<sup>+</sup> acceptors.

>>> ○ "Base" here refers to the base in acid-base reactions.

>> ○ Aromatic, thus has an absorbance at 260 nm.

>>> ○ Note that phenyl groups in proteins have an absorbance at 280 nm.

>> ○ Purity of nucleic acids := Absorbance at 260 nm / Absorbance at 280 nm = Amount of nucleic acids / Total protein

>> ○ N-glycosidic bond: Bond between the nitrogen atom of a base and 5-carbon sugar. The 1st carbon of the sugar participates in bonding.

>> ○ Nitrogenous bases are located inside the double helix to avoid the hydrophilic environment, as they contain hydrophobic aromatic rings.

>>> ○ RNA, which cannot form a double helix as DNA does, is less stable.

> ③ Triphosphate (3 phosphate)

>> ○ A structure containing two high-energy phosphate bonds, providing energy.

>> ○ Triphosphate is located on the outside of the double helix.

>>> ○ **Reason 1.** DNA undergoes polymerization reactions through diphosphate ester bonds, thus DNA polymerase must be accessible.

>>> ○ **Reason 2.** Triphosphate is hydrophilic, thus tends to be on the outside of the double helix.

> ④ [Types ofnucleotides](https://jb243.github.io/pages/1395#footnote_link_67_50)

>> ○ **Pyrimidine**: Single-ring base. Three types

>>> ○ Cytosine (C)

<br>

<img width="161" alt="스크린샷 2024-12-08 오전 9 05 43" src="https://github.com/user-attachments/assets/5dc34456-ab5b-4221-b37e-0c4f08dd69ff">

**Figure 20.** Structure of cytosine

<br>

>>> ○ Thymine (T)

<br>

<img width="193" alt="스크린샷 2024-12-08 오전 9 06 25" src="https://github.com/user-attachments/assets/3c0fff7a-2e50-4970-8524-fb00cd0c376c">

**Figure 21.** Structure of thymine

<br>

>>> ○ Uracil 

<br>

<img width="159" alt="스크린샷 2024-12-08 오전 9 07 04" src="https://github.com/user-attachments/assets/621aee94-4a5b-4964-a661-53fde2ed8a71">

**Figure 22.** Structure of uracil

<br>

>> ○ **Purine**: Double-ring base. Three types

>>> ○ Guanine (G)

<br>

<img width="254" alt="스크린샷 2024-12-08 오전 9 07 29" src="https://github.com/user-attachments/assets/fbdfd36c-e90b-4e18-bbb5-97831716c5f2">

**Figure 23.** Structure of guanine

<br>

>>> ○ Adenine (A)

<br>

<img width="196" alt="스크린샷 2024-12-08 오전 9 08 00" src="https://github.com/user-attachments/assets/24002b14-a06f-4315-9ea3-8a5a5fec8b00">

**Figure 24.** Structure of adenine

<br>

>>> ○ Inosinic acid (I)

⑵ Polynucleotide

> ① Phosphodiester bond: Dehydration condensation between the 3'-OH of the first sugar and the 5'-ⓟⓟⓟ of the second sugar in nucleotides.

> ② Polymerization occurs in the 5'-3' direction and is an endothermic reaction. Energy is supplied as two molecules of pyrophosphate (PPi) are released from the triphosphate.

> ③ The 5' end has a free phosphate group, known as the γ phosphate, exposed.

⑶ DNA (deoxyribonucleic acid)

> ① Function: Storing genetic information, RNA transcription

> ② Increase in pH: When the pH exceeds 10, the hydrogen involved in hydrogen bonding dissociates, causing the double strand to separate into single strands.

> ③ Increase in temperature: Increased kinetic energy of polynucleotides breaks hydrogen bonds between strands, leading to separation into single strands.

> ④ T<sub>m</sub>: Temperature at which 50% of double-stranded DNA becomes single-stranded.

>> ○ Higher G≡C ratio increases bonding strength, thus increasing T<sub>m</sub>.

>> ○ Higher amounts of added NaCl counteract repulsion between phosphate groups, thus increasing T<sub>m</sub>.

>> ○ Extremely low pH: H<sup>+</sup> interferes with hydrogen bonds, decreasing T<sub>m</sub>. DNA and RNA degrade in strong acids.

>> ○ Extremely high pH: Separates double-stranded DNA into single strands.

> ⑤ Renaturation kinetics: The study of the extent to which a DNA double helix is restored after denaturation.

>> ○ Regions with high frequency of repetitive sequences denature (ds DNA → ss DNA) first and renature first.

>> ○ High-frequency repetitive sequences: Centromere, etc.

>> ○ Moderate-frequency repetitive sequences: Telomeres, etc.

> ⑥ DNA Absorbance Experiment

>> ○ At 260 nm, absorbance of 1 corresponds to 50 μg / mL dsDNA concentration.

>> ○ At 260 nm, absorbance of 1 corresponds to 40 μg / mL ssRNA concentration.

> ⑦ Thermophiles: High G≡C content, increased histones, and supercoiled structures.

⑷ RNA (ribonucleic acid)

> ① Function: Transmission of genetic information and polypeptide synthesis.

> ② Increase in pH: The H+ of the 2'-OH group in RNA dissociates → Hydration → Hydrolysis, leading to RNA breaking into fragments.

> ③ Circular RNA is more stable than linear RNA.

 ⑸ Comparison of DNA and RNA

<br>

| Item         | DNA                  | RNA            |
| ------------ | -------------------- | -------------- |
| Complementarity | O (Yes)               | X (No)         |
| Sugar        | Deoxyribose          | Ribose         |
| Bases        | A, T, G, C           | A, U, G, C     |
| Strand Number | Double-stranded       | Single-stranded|
| Hydrogen Bonds| A=T, G≡C             | A=U, G≡C       |
| Sugar-Phosphate Backbone | Antiparallel strands |          |

**Table 2.** Comparison of DNA and RNA

<br>

> ① Complementarity: The property that allows restoration of information using the other strand when one strand is damaged.

> ② Sugar: Deoxyribose has a -H group at the 2nd carbon. Ribose has a -OH (more reactive) group at the 2nd carbon.

> ③ Instability of RNA

>> ○ Ribose: Increased reactivity due to 2'-OH group.

>> ○ Single-strand: Cannot be corrected or repaired, easily affected by external substances.

>> ○ [Deamination process](https://jb243.github.io/pages/1386#2-imines): Amino bases A and C can undergo deamination.

>>> ○ Loss of amino group in C base: Becomes U base. A base becomes I.

>>> ○ DNA can be corrected, but RNA cannot.

<br>

<img width="408" alt="스크린샷 2025-02-16 오전 11 53 36" src="https://github.com/user-attachments/assets/a3c04af1-3bd3-487c-ad1e-0b9d857e0007" />

**Figure 25.** Deamination of cytosine

<br>

>> ○ RNA undergoes hydrolysis with increased pH.

⑹ Eukaryotic Chromatin

> ① DNA as genetic material

>> ○ In 1871, Fredrich Miescher first reported DNA in the nucleus.

>> ○ **Evidence 1.** Chargaff's rules: [A] = [T], [G] = [C]

>> ○ **Evidence 2.** X-ray diffraction analysis: Rosalind Elsie Franklin's diffraction research played a key role in revealing the structure of DNA.

<br>

![image](https://github.com/user-attachments/assets/0adb6a90-0a41-4334-8ead-f31b7dbf9377)

**Figure 26.** DNA's 3D diffraction pattern

<br>

>> ○ Conclusions by Watson and Crick

<br>

![image](https://github.com/user-attachments/assets/af479e3d-72c8-4085-b7e1-67459d9628f7)

**Figure 27.** DNA structure sketched by Francis Crick

<br>

![image](https://github.com/user-attachments/assets/e2a19270-94d6-4d45-b9d3-05eefffff80e)

**Figure 28.** Watson and Crick's paper

<br>

> ② Structure of DNA

>> ○ Major groove: Phosphate groups are not exposed, making it accessible to transcription factors or enzymes.

>> ○ Minor groove: Phosphate groups are exposed, allowing access to histone proteins.

>> ○ B-form DNA: Watson-Crick model (i.e., DNA in the body). High relative humidity (92 %).

>> ○ A-form DNA: Dehydrated environment (75 %).

>>> ○ Example: Double-stranded RNA, DNA-RNA double helix (hairpin, stem loop)

>> ○ Z-form DNA: Zigzag, slender

>>> ○ Example: Non-expressed promoters, regions with high G≡C ratio (e.g., centromere)

>> ○ Comparative summary of DNA forms

<br>

|   | Type A | Type B | Type Z |
| --- | --- | --- | --- |
| Helix Direction | Right-handed | Right-handed | Left-handed |
| Diameter | 2.37 nm | 2.55 nm | 1.84 nm |
| Base Pairs per Turn | 11 bp | 10 bp | 12 bp |
| Spacing Between Base Pairs | 0.26 nm | 0.34 nm | 0.37 nm |
| Base tilt relative to the helical axis | 20° | 6° | 7° |

**Table 3.** Comparative Summary of DNA Forms

<br>

![image](https://github.com/user-attachments/assets/5727917c-ca8a-4865-9f72-11efa6472efc)

**Figure 29.** A-form DNA (left), B-form DNA (middle), Z-form DNA (right)

㉠ Minor groove, ㉡ Major groove

<br>

> ③ RNA structure: Most RNA with 3D structure is less than 150 nt in length.

> ④ Nucleosome (beads-on-a-string, bead-on-string): Also called 10 nm fiber.

>> ○ Structure

>>> ○ 146 bp of DNA wraps around a histone octamer (composed of one pair each of H2A, H2B, H3, and H4) with 1.65 turns.

>>> ○ Linker DNA: Refers to DNA from H1 to H1. About 200 nucleotides.

>>> ○ When treated with endonuclease, H1 to H1 gets cut → visible in electrophoresis.

>> ○ H1: Adjusts nucleosome structure. One per nucleosome. Involved in chromatin condensation.

>> ○ Histone and DNA Binding

>>> ○ Over 20% of the amino acids in histones are basic amino acids with a positive charge (e.g., Lys, Arg).

>>> ○ Ionic bonds form between the negatively charged phosphate groups of DNA and the positively charged histone proteins.

>> ○ Histone Tail: The N-terminal of the histone molecule within the nucleosome that protrudes outward.

>> ○ [Histone Modification](https://jb243.github.io/pages/74): Histone acetylation, histone deacetylation, histone methylation, histone phosphorylation

> ⑤ 30 nm Fiber: The structure formed when the 10 nm fiber is wound and folded.

<br>

![image](https://github.com/user-attachments/assets/e44b1397-ec7b-4c34-9c5f-a2404121c5b6)

**Figure 30.** 30 nm Fiber

<br>

> ⑥ 300 nm Fiber: A structure in which 30 nm fibers are attached in a looped form around a protein scaffold.

> ⑦ Chromatin

>> ○ Types of Chromatin

>>> ○ **Type 1.** Euchromatin: Uncondensed chromatin.

>>> ○ **Type 2.** Heterochromatin: Condensed chromatin. Typically centromeres, telomeres exist in heterochromatin form.

>> ○ Structure of Chromatin

>>> ○ TAD (topologically associating domain): Chromatin forming loops.

>>> ○ The actual shape of TAD is complex and tangled like a ball of yarn.

>>> ○ CTCF (11-zinc finger protein, CCCTC-binding factor): A transcription factor encoded by the _CTCF_ gene.

>>> ○ CTCF and PDS5A/B determine the boundaries and loop anchors of TAD.

>>> ○ WAPL (responsible for releasing cohesin from chromosomes) and PDS5A/B determine the length of the loops.

> ⑧ Chromosome: The width of a single chromatid is 700 nm.

<br>

![image](https://github.com/user-attachments/assets/92a8dde3-d2e2-4c23-a53b-03442e67c4da)

**Figure 31.** From nucleosome to chromosome

<br>

⑺ Polytene Chromosome (giant chromosome)

> ① **Requirement 1.** Related to Mitosis.

> ② **Requirement 2.** A giant chromosome formed by repeated DNA replication (only the S phase occurs) without nuclear or cytoplasmic division (no division phase), maintaining a folded state.

> ③ Endoreduplication

>> ○ Definition: The process of repeatedly synthesizing DNA without the division phase (M phase). Does not form sister chromatids.

>> ○ Location: Ovary nurse cell, follicle cell, abdominal histoblast, fat body cell, gut cell, prepupal salivary gland cell in Drosophila.

>> ○ Result: Formation of polyploidy

> ④ Composed of more than 1,000 identical DNA molecules.

> ⑤ Staining reveals alternating dark and light bands.

>> ○ Less condensed light bands are areas of active transcription.

>> ○ Conversely, dark areas are where transcription is suppressed.

> ⑥ Puff: An area in the polytene chromosome where the light bands locally unravel and expand.

>> ○ Puffs are observed in regions with vigorous gene expression.

>> ○ The location of puffs varies depending on developmental stages and internal or external signals.

> ⑦ Chromocenter: Not all DNA in chromosomes is polytenic.

>> ○ Exists as highly condensed heterochromatin.

>> ○ Non-polytenic and exhibits minimal transcriptional activity.

>> ○ Salivary gland chromosomes cluster at the chromocenters.

>> ○ DNA in these regions accounts for about 30% of the genome and has very low gene density.

>> ○ Puffs are observed in regions with active gene expression.

> ⑧ Example: Salivary chromosomes of dipteran insects

<br>

![image](https://github.com/user-attachments/assets/d9abb43f-c8df-442c-a262-ecef3b9a692b)

**Figure 32.** Sketch of salivary gland chromosomes by Calvin B. Bridges

<br>

>> ○ Facilitates the production of glue needed for pupation.

>> ○ Salivary gland chromosomes have a polyteny level of 1024.

>> ○ Salivary glands degenerate during metamorphosis.

⑻ Lampbrush Chromosome

> ① **Requirement 1.** Related to Meiosis.

> ② **Requirement 2.** Formed when bivalent chromosomes fail to separate during the first meiotic division.

> ③ Feature: Lateral Loops

<br>

![image](https://github.com/user-attachments/assets/7c3f199a-96b7-482f-b205-6f579258b3f5)

**Figure 33.** Lateral Loops

<br>

⑼ Multinucleate

> ① **Requirement 1.** Related to Mitosis.

> ② **Requirement 2.** Nuclear division occurs but cytoplasmic division does not.

⑽ Polyploid

> ① **Requirement 1.** Related to Meiosis.

> ② **Requirement 2.** Treated with colchicine to inhibit separation of homologous chromosomes and then combined with normal gametes.

> ③ Example: Seedless watermelon (3n)

⑾ Comparison of polytene chromosome, lampbrush chromosome, Multinucleate, and polyploid

<br>

|   | **Polytene chromosome** | **Lampbrush chromosome** | **Multinucleate** | **Polyploid** |
| --- | --- | --- | --- | --- |
| Amount of DNA per cell | ⇑ | ⇑ | ⇑ | ⇑ |
| Number of nuclei per cell | Normal | Normal | ⇑ | Normal |
| Number of chromosomes per nucleus | ⇑ | Normal | Normal | Normal |

**Table 4.** Comparison of polytene chromosome, lampbrush chromosome, multinucleate, and polyploid

<br>

---

_Input: 2015.06.22 22:36_

_Edit: 2020.03.21 10:16_
