## Chapter 1. Composition of Living Organisms

Recommended Articles: 【Biology】 [Biology Index](https://jb243.github.io/pages/1457), 【Organic Chemistry】 Biopolymers: [Carbohydrates](https://jb243.github.io/pages/1389),
[Proteins](https://jb243.github.io/pages/1391), [Lipids](https://jb243.github.io/pages/1393), [Nucleic Acids](https://jb243.github.io/pages/1395)

---

**1.** [Definition of Living Organisms](#1-definition-of-living-organisms)

**2.** [Water and Carbon](#2-water-and-carbon)

**3.** [Biopolymers](#3-biopolymers-macromolecules)

**4.** [Carbohydrates](#4-biopolymer-1-carbohydrates-polysaccharide-CH2On)

**5.** [Proteins](#5-biopolymers-2-proteins-polymers-of-amino-acids)

**6.** [Lipids](#6-biomolecules-3-lipids)

**7.** [Nucleic Acids](#7-biomolecules-4-nucleic-acids-polymers-of-nucleotides)

---

**a.** [Hydrophobicity Index of Biopolymers](https://jb243.github.io/pages/1469)

**b.** [Biopolymer Library](https://jb243.github.io/pages/1327)

**c.** [Perm hair and Disulfide Bonds](https://jb243.github.io/pages/1417)

---

<br>

## **1. Definition of Living Organisms**

⑴ **Definition 1.** Organisms grow, move, reproduce, respond to stimuli, and have metabolism.

> ① Is fire a living organism and a donkey not?

 ⑵ **Definition 2.** Defined as an organism if all the following conditions are met:

> ① Possesses a common set of biomolecules.

> ② Maintains homeostasis.

> ③ Capable of evolution.

> ④ Requires water.

<br>

<br>

## **2. Water and Carbon**

⑴ H<sub>2</sub>O

> ① Polarity: Oxygen has high electronegativity, attracting electrons from hydrogen, causing a slight negative charge. Conversely, the hydrogen side has a slight positive charge.

> ② Hydrogen Bonds: Molecular attraction between hydrogen and electronegative atoms (i.e., F, O, N).

>> ○ Water molecules cohere due to hydrogen bonds.

>> ○ A water molecule can form hydrogen bonds at four sites

<br>

<img width="424" alt="스크린샷 2024-12-07 오후 11 02 25" src="https://github.com/user-attachments/assets/05cee0da-ded0-41a3-bc19-48d194905e40">

**Figure 1. Hydrogen Bonding in Water**

<br>

> ③ High heat capacity due to hydrogen bond: Plays an important role in temperature regulation

> ④ Excellent solvent (i.e., universal Solvent) due to its polarity

>> ○ Dissolves other polar substances (e.g., salts, alcohols, acids, bases).

>> ○ Facilitates chemical reactions.

>> ○ Hydration Shell: The layer of water molecules surrounding a solute, making ions appear larger in water.

> ⑤ Expansion upon freezing

>> ○ Reason: Ice forms **hexagonal crystals**.

>> ○ At atmospheric pressure, water's density is always greater than ice.

<br>

![image](https://github.com/user-attachments/assets/5c90037f-d777-42d1-af9d-2785f27feddd)

**Figure 2. Density of Water and Ice by Temperature**

<br>

 ⑵ C (Carbon)

> ① Carbon can bond with four atoms, used as a backbone for biopolymers

> ② Carbon constitutes a significant portion of the biomass

> ③ Organic Chemistry: The chemistry of carbon

>> ○ Molecules based on carbon are called organic compounds

>> ○ Compounds made of carbon and hydrogen are called hydrocarbons (e.g., CH4)

> ④ Meteorites from Mars contain carbonates and hydrocarbon chains

>> ○ However, there's no confirmation that these are biopolymers produced by living organisms

 ⑶ Functional Groups

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/c683628d-3b5b-43b8-b5ea-1060aa73b0b7)

**Figure 3. Types of Functional Groups**

<br>

<br>

## 3. Biopolymers (macromolecules)

 ⑴ Earth's organisms contain the same set of macromolecules (e.g., carbohydrates, proteins, lipids, nucleic acids)

 ⑵ Polymer Formation: All polymers are made through dehydration reactions (water is produced) of monomers

 ⑶ Polymer Degradation: Polymers are broken down into monomers through hydrolysis

 ⑷ **Benefits of Polymers:** Cause less osmotic pressure and maintain a lower concentration of monomers inside cells

<br>

<br>

## 4. Biopolymer 1. Carbohydrates (polysaccharide) = (CH<sub>2</sub>O)<sub>n</sub>

 ⑴ Functions: **Primary Energy Source**, **Structural Role**

> ① Energy Source: 4 kcal/g

> ② Water-soluble molecules

 ⑵ Glycosidic Bond: Dehydration reaction between sugars

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/7541acaf-676f-4317-adb4-258aa357b992)

**Figure 4. Glycosidic Bond in Glucose**

<br>

 ⑶ [Monosaccharides](https://jb243.github.io/pages/1389): Glucose(6C), Fructose(6C), Galactose(6C), Others(3C ~ 7C)

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/b5debe48-f3e6-4ec9-8998-e4b8535c36bb)

**Figure 5. Structural Isomers of C<sub>6</sub>H<sub>12</sub>O<sub>6</sub>**

<br>

 ⑷ [Disaccharides](https://jb243.github.io/pages/1389)

 ⑸ [Oligosaccharides](https://jb243.github.io/pages/1389): Referring to C3 ~ C12

 ⑹ Polysaccharides: **Glycogen**, **Starch**, **Cellulose**, **Chitin**, **Peptidoglycan**

> ① Glycogen

>> ○ α linkage glucose polymer, more branched, short-term energy source in animals

>> ○ α 1 → 4 linkage, α 1 → 6 linkage

> ② Starch

>> ○ α linkage glucose polymer, long-term energy source in plants

> ③ Cellulose

>> ○ β linkage glucose polymer, forms cell walls, most abundant carbohydrate on Earth

>> ○ β 1 → 4 linkage

> ④ Chitin

>> ○ NAG(N-Acetylglucosamin): The 2nd carbon functional group of glucose is replaced with an amino group

>> ○ Chitin is a polymer of NAG: Forms exoskeletons in insects and crustaceans

>> ○ β 1 → 4 linkage

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/857b0224-20e0-4f2d-8254-979dad5dbe7d)

**Figure 6. Structure of NAG**

<br>

> ⑤ Dextrin

>> ○ α 1 → 4 linkage

>> ○ Dextran is a trademark name

 ⑹ α Linkage and β Linkage

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/3b5ce60a-70e0-46e9-b4ed-589e1fce99b9)

**Figure 7. α Glucose (left) and β Glucose (right)**

<br>

> ① α Linkage: Glucose units are bonded without flipping upside down, creating an α helix structure

> ② α Linkage Glucose: The -OH group of the 1st carbon and the -OH group of the 4th carbon are in the same direction, unstable → Energy Storage

> ③ β Linkage

>> ○ One glucose flips upside down, so the 1st carbon and 4th carbon face each other, forming a straight line

>> ○ β linkage glucose chains cohere tightly due to strong hydrogen bonds

> ④ β Linkage Glucose: The -OH group of the 1st carbon and the -OH group of the 4th carbon are in opposite directions, stable → Structural Formation

 ⑺ Starch

> ① Amylose: α linkage glucose polymer without branches. 1st carbon and 4th carbon are bonded

> ② Amylopectin: Branched α linkage glucose polymer. 1st carbon and 6th carbon are bonded

> ③ Starch is insoluble because α glucose chains form tight crystals through hydrogen bonding

> ④ Most digestive enzymes can only break down α linkages due to the spacious arrangement that allows enzyme accessibility

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/6c934f01-11e4-4b29-bcda-85b094baad01)

**Figure 8. Amylose and Amylopectin**

<br>

 ⑻ Cellulose

> ① Most abundant carbohydrate on Earth

> ② β 1 → 4 linkage

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/4a2fa0ba-3c3c-43b8-9948-53aa76f0859a)

**Figure 9. Hydrogen Bonding Between β-Glucose Chains**

<br>

> ③ Cellulase

>> ○ [Rumen of ruminants](https://jb243.github.io/pages/82): Microbes in the first and second stomachs of ruminants synthesize cellulase

>> ○ Termites' intestines: Trichonympha, a symbiotic microbe, synthesizes cellulase

<br>

<br>

## 5. Biopolymers 2. Proteins = Polymers of Amino Acids

 ⑴ Characteristics

> ① Constituent elements: C, H, O, N, S

> ② Functions

>> ○ Enzymes in metabolic activities

>> ○ Formation of intra- and extracellular structures: Constitutes half of the body's dry weight

>> ○ Muscle contraction

>> ○ Immune function

>> ○ Hormones or signal proteins

>> ○ Signal transduction inside cells

>> ○ Substance transport across membranes

>> ○ Energy conversion and storage: 4 kcal/g, mainly in fetuses

>> ○ DNA replication, repair, and recombination

>> ○ Transcription, translation

>> ○ Protein transport and secretion

> ③ Proteins: Composed of one or more polypeptides

> ④ Polypeptides: Made of peptide bonds between amino acids

> ⑤ Peptide bond: Dehydration reaction between amino group (-NH<sub>2</sub>) and carboxyl group (-COOH)

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/742c2e4c-8078-40fc-8266-93694796ee48)

**Figure 10.** Dehydration reaction between two amino acids

<br>

 ⑵ Amino Acids

> ① There are a total of 20 amino acids: 21 including selenocysteine (Sec)

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/b990c26c-06e5-4705-94a1-61d28c1d6314)

**Table. 1.** Symbols of Amino Acids ([ref](https://blogs.oregonstate.edu/psquared/week-2/))

<br>

> ② Structure: Central carbon, amino group, carboxyl group, R group

> ③ All amino acids have the same backbone, but there are 20 types of side chains (denoted as R)

>> ○ Others do not participate in protein synthesis

>> ○ Including minor amino acids like selenocysteine, the number can exceed 20

> ④ Zwitterion: Acts as an acid or base. [Isoelectric point](https://jb243.github.io/pages/77#footnote_link_67_53)

> ⑤ Amino acids differ in properties depending on their side chains

>> ○ Nonpolar amino acids: Ala, Ile, Leu, Met, Phe, Pro, Trp, Val, Gly

>> ○ Polar, uncharged amino acids: Asn, Cys, Gln, Ser, Thr, Tyr

>> ○ Positively charged amino acids: Arg, His, Lys

>> ○ Negatively charged amino acids: Asp, Glu

>> ○ Amino acids containing -OH group: Ser, Thr, Tyr - Can add phosphate for signal transduction

>>> ○ Ser, Thr: Involved in phosphorylation cascades in signal transduction

>>> ○ Tyr: Present in tyrosine kinase receptors in signal transduction

>> ○ Sulfur-containing amino acids: Cys, Met. Only Cys forms disulfide bonds

>> ○ Phenyl group containing amino acids: Phe, Trp, Tyr

>> ○ Phenyl group shows absorbance at 280 nm. Mechanism for quantifying proteins using a UV spectrometer

>> ○ Amino acids not used in TCA cycle: Leu, Lys

> ⑥ [Titration of Amino Acids](https://jb243.github.io/pages/1352#footnote_link_67_55)

>> ○ Each amino acid's pKa1, pKa2, pKaR, pI, hydrophobicity index, and its proportion in proteins

 ⑶ Primary Structure of Proteins

> ① Amino acid sequence

> ② Planar nature of peptide bonds

>> ○ Amide plane: Forms one peptide bond. Plane formed by two alpha carbons and the C, O, N, H between them

>> ○ Naming multiple bonds in one plane

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/34d217eb-0072-4a69-b003-59d948e78b58)

 **Figure 11.** Naming multiple bonds in one plane

<br>

>>> ○ Φ bond: N - Cα bond

>>> ○ ψ bond: Cα - C bond

>>> ○ ω bond: C - N bond

>>> ○ χ bond: Cα - R (functional group) bond

>> ○ C=O bond resonates with ω bond, making C=O and C-N bonds into 1.5 bonds, thus fixed and non-rotatable

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/65d3296f-778a-42d3-b6bd-b7d0913cd4f3)

**Figure 12.** Planar nature of peptide bonds

<br>

 ⑷ Secondary Structure of Proteins

> ① Structure formed by the coiling or folding of the amino acid sequence due to hydrogen bonds between the amino acid backbone (H and O)

> ② Alpha helix (α-helix)

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/74d1c397-ae32-4e11-85a8-f0c7311b860d)

**Figure 13.** Alpha helix structure

<br>

>> ○ The C=O bond of the Nth amino acid and N-H bond of the (N+4)-th amino acid form hydrogen bonds

>>> ○ On average, 3.6 residues per turn (though there is variation)

>> ○ Elastic, right-handed helix

>> ○ Hydrophilic amino acids have attractions and repulsions that hinder the formation of alpha helices

>>> ○ Alpha helices are mainly formed by hydrophobic amino acids

>>> ○ Commonly observed in transmembrane proteins

>> ○ Gly: Inappropriate for alpha helix structure due to small size as R group is hydrogen

>> ○ Pro: Forms an imino group, restricting hydrogen bonding → Does not form alpha helix

> ③ Beta pleated sheet (β-pleated sheet): Also known as zigzag conformation

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/016f51d5-8328-413b-a469-e4a4c83bb69c)

**Figure 14.** Beta pleated sheet structure

<br>

>> ○ Characteristic: Not only intra-amino acid hydrogen bonds but also inter-amino acid chain hydrogen bonds

>> ○ Common structure in fibrous proteins like silk

>> ○ Spider webs are a structure with β-pleated sheets and added α-helices

> ④ Ramachandran plot

>> ○ Φ bonds on the x-axis and ψ bonds on the y-axis, a two-dimensional representation of tertiary structure

>> ○ Alpha helix structures are concentrated around (-60°, -60°). Beta pleated sheet structures are concentrated around (-120°, 120°)

>> ○ Purpose: To see if residues are in appropriate positions. Many islands on the plot indicate inappropriate positions

>> ○ Example

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/de6796c5-36d3-435c-9237-b435a084b889)

 **Figure 15.** Example of a Ramachandran plot

<br>

 ⑸ Tertiary Structure of Proteins

> ① Overall three-dimensional structure of a polypeptide

> ② Interactions between R groups lead to different three-dimensional structures, which determine different properties

> ③ Ionic bond

> ④ Covalent bond

> ⑤ Disulfide bridge: Makes the tertiary structure more rigid

>> ○ Reducing agents: Break disulfide bonds. β-mercaptoethanol, DTT (dithiothreitol), etc.

>> ○ In eukaryotic cells, formed in the endoplasmic reticulum. In prokaryotic cells, in the cytoplasm by PDI (protein disulfide isomerase)

>> ○ Anfinsen's experiment

>>> ○ NEM (N-ethylmaleimide): Covalently binds to the -SH group of cysteine not involved in disulfide bonds

>>> ○ DTT (including β-metcaptoethanol): -S-S- → -SH + HS-. Acts as a reducing agent breaking disulfide bonds. Glutathione (GSH) has a similar function

>>> ○ NEM* → DTT → NEM: Initially existing as SH groups quantified, * indicates radioactive isotope

>>> ○ NEM → DTT → NEM*: Initially existing as -S-S- quantified, * indicates radioactive isotope

>>> ○ **Conclusion 1:** If denaturing factors of a protein are removed, the protein reverts to its original structure

>>> ○ **Conclusion 2:** Since disulfide bonds contribute irreversibly to the creation of tertiary structure, they must be finely regulated

> ⑥ Hydrophobic interactions: Forces arising because the biological environment is water

> ⑦ Polar bonds and hydrogen bonds between R groups

> ⑧ Van der Waals forces

> ⑨ Other interactions

>> ○ Similar hydrogen bonds in C=O bonds: n → π*, more than 45% of protein residues

>> ○ π-π stacking: aromatic ring stacking

○ Steric hindrance

○ C5 hydrogen bonds in β-sheet backbone

○ cation-π interaction

> ⑩ Due to certain interactions being strong, proteins can fold incorrectly, necessitating protein assembly proteins like chaperones

>> ○ **Example 1.** [heat shock protein](https://jb243.github.io/pages/1096): Prevents proteins from denaturing due to temperature

⑹ The quaternary structure of proteins: Interactions between tertiary structure polypeptides

> ① transthyretin (tetramer), hemoglobin (α2β2), collagen (3 helix)

⑺ Protein structure determination

> ① Determining the primary structure

>> ○ **Method 1.** Edman degradation

>>> ○ Degrades amino acids one at a time by binding to the N-terminus of polypeptides in mild alkaline conditions

>>> ○ Only about the first 10 amino acids from the N-terminal can be known

>>> ○ A very traditional method, currently almost unused

>> ○ **Method 2.** Peptidases

>>> ○ Appropriately combining endopeptidases and exopeptidases to appropriately degrade amino acids and determine them one by one

>>> ○ Endopeptidases: Pepsin, trypsin, chymotrypsin, etc.

>>> ○ Exopeptidases: Carboxypeptidases, etc.

>>> ○ [Mechanism of peptidases](https://jb243.github.io/pages/1022)

>> ○ **Method 3.** Ab array

>> ○ **Method 4.** Mass spectrometry

>>> ○ 1<sup>st</sup>. electrophoresis & DNA ladder: Can determine whether it's pure or a mixture

>>> ○ 2<sup>nd</sup>. Trypsin treatment

>>> ○ 3<sup>rd</sup>. First [mass spectroscopy](https://jb243.github.io/pages/1404#footnote_link_67_51): Constructing the MS spectrum

>>> ○ 4<sup>th</sup>. Second mass spectroscopy: Constructing the MS/MS spectrum. Also known as tandem MS, MS/MS, fragmentation

>>> ○ 5<sup>th</sup>. Reconstruction through the spectrum: Realm of informatics

>>> ○ This method is a first-generation technology and becomes much more complicated when considering post-translational modifications

> ② Determining the secondary structure

>> ○ **Method 1.** Circular dichroism (CD)

>> ○ **Method 2.** Infrared spectroscopy: Useful for investigating flexible peptide and protein structures

>> ○ **Method 3.** Molecular dynamics simulation

> ③ Determining tertiary and quaternary structures

>> ○ **Method 1.** X-ray crystallography: About 90% are determined by this method. Allows for inference of 3D coordinates of atoms

>> ○ **Method 2.** NMR: About 9% are determined by this method

>> ○ **Method 3.** Cryo-EM

>>> ○ Resolution is low but is steadily improving and useful for large protein complexes like capsids and amyloids

>>> ○ Protein is immobilized by lowering the temperature to -200 ℃ to observe its 3D structure

⑻ Protein denaturation

> ① Denaturation: Change in tertiary structure due to concentration of salts, pH, high temperatures, resulting in loss of function

<br>

<br>

## 6. Biomolecules 3. Lipids

⑴ Characteristics

> ① Mostly composed of carbon and hydrogen, thus high in energy content

> ② Hydrophobic: Does not dissolve in water, but dissolves in acetone, alcohol, benzene, etc.

> ③ Insulating effect (e.g. subcutaneous fat layer), insulation, waterproofing

⑵ Fat: Triglyceride with three hydrocarbon-rich fatty acid tails linked by ester bonds to glycerol

> ① Energy content: 9 kcal/g

> ② Digestive enzyme lipase breaks the ester bonds ([ref](https://jb243.github.io/pages/82))

> ③ Fatty acids are synthesized from acetyl coA, so they consist of an even number of carbons

> ④ Saturated fatty acid: Hydrogen-saturated fatty acids with only single bonds. Solid at room temperature, animal-based

> ⑤ Unsaturated fatty acid: At least one cis double bond. Liquid at room temperature, plant-based + fish

> ⑥ Butter and margarine

>> ○ Butter: Saturated fat obtained from milk

>> ○ Margarine: Unsaturated fat obtained from plants, hydrogenated to become saturated fat

> ⑦ Trans fat: Created during hydrogenation, where hydrogen is added and then removed, creating trans double bonds

>> ○ Trans fats accumulate and have a high boiling point

>> ○ Trans fats accumulate in the body, thus posing a health risk

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/9b8c015a-f537-44e9-a906-d4ce4b4ca96a)

 **Figure 16.** _cis_ and _trans_ bonds in fatty acid tails]

<br>

⑶ Steroids: Flat structure composed of three hexagonal rings and one pentagonal ring, hydrophobic substance

> ① Derived from lipids without monomers (from cholesterol)

> ② Issues with synthetic steroids

>> ○ Marked mood changes (violent mood swings)

>> ○ Depression, liver damage, high cholesterol, high blood pressure

> ③ **Cholesterol**: Found in some animal cell membranes

>> ○ Refers to steroids with an -OH group

>> ○ **Function 1.** Regulating fluidity of animal cell membranes

>> ○ **Function 2.** Precursor to various substances: Bile salts, steroid hormones (sex hormones, adrenal cortex hormones, etc.), Vitamin D

>> ○ **Function 3.** Constituting the protective sheath of nerves

>> ○ Not oxidized in the body to produce energy

>> ○ 75% of cholesterol is obtained through synthesis. 25% of cholesterol is obtained through diet

>> ○ Cholesterol is abundant in egg yolks, liver, squid, etc., and it's advisable to consume less than 300 mg per day

>> ○ Can cause arteriosclerosis and heart disease

> ④ Sterols: Found in some plant cell membranes

> ⑤ Sex hormones (estradiol, androgen)

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/d1e7326a-cd60-4c11-a093-717615d3f840)

**Figure 17.** Examples of sex hormones

<br>

⑷ Phospholipids = **** 1 × phosphate head group + 1 × glycerol backbone + 2 × fatty acids + ester bonds, constituting the phospholipid bilayer, amphipathic

> ① Phospholipid forms in aqueous solution

>> ○ Micelle: Spherical shape. Single layer. Packs 1 fatty acid, transports hydrophobic substances through blood vessels as micelles

>> ○ Liposome: Ring shape. Double layer. Packs 2 fatty acids, applied in drug delivery

> ② Diversity and membrane asymmetry of phospholipids: Type of molecule attached to the phosphate group

>> ○ Lecithin: Most representative phospholipid. Contains glycerol phosphate and adds luster to the skin. Remains in the body for a long time

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/799ef790-49e4-4ddf-a908-03334ba544c8)

 **Figure 18.** Example of phospholipid: Phosphatidylcholine

 <br>

> ③ Phospholipid bilayer: Forms hydrophobic boundary of cells, semi-permeable membrane

> ④ In extremophiles, the fatty acid bonds in phospholipids are ether (-ROR-) bonds instead of ester (-RCOO-) bonds

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/7d8a2fac-58a3-4805-ac34-6cef0f4e2352)

 **Figure 19.** Difference between extremophiles' phospholipids and those of true bacteria and eukaryotes

<br>

⑸ Wax

> ① Ester bond between fatty acids and long-chain alcohols

> ② Consists of 40 ~ 60 CH2 units

> ③ Prevents moisture ingress

> ④ The Casparian strip, deposited with wax in plant root endodermis cells, regulates water transport

⑹ Fat-soluble vitamins: A, D, E, K

⑺ Carotenoids

> ① Auxiliary pigments capturing light energy

> ② Main reaction: β-Carotene + O2 **→** 2 Vitamin A

<br>

<br>

## 7. Biomolecules 4. Nucleic Acids: Polymers of nucleotides

 ⑴ Nucleotide = Phosphate + Nucleoside (e.g.,: cytidine, uridine, thymidine) = Phosphate + Sugar + Nitrogenous Base

> ① Sugar: 5-carbon sugar (pentose)

>> ○ Ribose: 2' carbon has -OH group, highly reactive

>> ○ Deoxyribose: 2' carbon has -H, less reactive

> ② Nitrogenous Base (base)

>> ○ Basic ring structure made of carbon-nitrogen covalent bonds, unshared electron pairs of nitrogen act as H+ acceptors

>>> ○ "Base" here refers to the base in acid-base reactions

>> ○ Aromatic, thus has an absorbance at 260 nm

>>> ○ Note that phenyl groups in proteins have an absorbance at 280 nm

>> ○ Purity of nucleic acids: = Absorbance at 260 nm / Absorbance at 280 nm = Amount of nucleic acids / Total protein

>> ○ N-glycosidic bond: Bond between the nitrogen atom of a base and 5-carbon sugar, the 1st carbon of the sugar participates in bonding

>> ○ Nitrogenous bases contain hydrophobic aromatic rings, thus are located **inside the double helix**

>>> ○ RNA, which cannot form a double helix as DNA does, is less stable

> ③ Triphosphate (3 phosphate)

>> ○ Contains two high-energy phosphate bonds, provides energy

>> ○ Triphosphate is located on the outside of the double helix

>>> ○ **Reason 1.** DNA undergoes polymerization reactions through diphosphate ester bonds, thus DNA polymerase must be accessible

>>> ○ **Reason 2.** Triphosphate is hydrophilic, thus tends to be on the outside of the double helix

> ④ [Types of Nucleotides](https://jb243.github.io/pages/1395#footnote_link_67_50)

>> ○ **Pyrimidine**: Single-ring base, three types

>>> ○ Cytosine (C, cytosine)

>>> ○ Thymine (T, thymine)

>>> ○ Uracil (uracil)

>> ○ **Purine**: Double-ring base, three types

>>> ○ Guanine (G, guanine)

>>> ○ Adenine (A, adenine)

>>> ○ Inosinic acid (I, inosinic acid)

>> ○ Structural variations of nucleotides

 ⑵ Polynucleotide

> ① Phosphodiester bond: Dehydration condensation between the 3'-OH of the first sugar and the 5'-ⓟⓟⓟ of the second sugar in nucleotides

> ② Polymerization occurs from 5' to 3', an endothermic reaction, energy is supplied by the release of two molecules of pyrophosphate (pi) from triphosphate

> ③ The 5' end has a free γ phosphate exposed

 ⑶ DNA (deoxyribonucleic acid)

> ① Function: Stores genetic information, RNA transcription

> ② Increase in pH: At pH above 10, hydrogen bonds forming between bases dissociate, leading to separation of double strands into single strands

> ③ Increase in temperature: Increased kinetic energy of polynucleotides breaks hydrogen bonds between strands, leading to separation into single strands

> ④ Tm: Temperature at which 50% of double-stranded DNA becomes single-stranded

>> ○ Higher G≡C ratio increases bonding strength, thus increasing Tm

>> ○ Higher amounts of added NaCl counteract repulsion between phosphate groups, thus increasing Tm

>> ○ Extremely low pH: H+ interferes with hydrogen bonds, decreasing Tm. DNA and RNA degrade in strong acids

>> ○ Extremely high pH: Separates DNA into single strands

> ⑤ Renaturation kinetics: Study of the extent of double helix restoration after denaturation

>> ○ Regions with high frequency of repetitive sequences denature (ds DNA → ss DNA) first and renature first

>> ○ High frequency repetitive sequences example: Satellites

>> ○ Medium frequency repetitive sequences example: Telomeres

> ⑥ DNA Absorbance Experiment

>> ○ At 260 nm, absorbance of 1 corresponds to 50 μg / mL dsDNA concentration

>> ○ At 260 nm, absorbance of 1 corresponds to 40 μg / mL ssRNA concentration

> ⑦ Thermophiles: High G≡C content, high histone content, supercoiled structure

 ⑷ RNA (ribonucleic acid)

> ① Function: Transmits genetic information, creates polypeptides

> ② Increase in pH: RNA 2'-OH group's H+ dissociates → hydration → hydrolysis, RNA fragments

> ③ Cyclic RNA is more stable than linear RNA

 ⑸ Comparison of DNA and RNA

<br>

| Item         | DNA                  | RNA            |
| ------------ | -------------------- | -------------- |
| Complementarity | O (Yes)               | X (No)         |
| Sugar        | Deoxyribose          | Ribose         |
| Bases        | A, T, G, C           | A, U, G, C     |
| Strand Number | Double-stranded       | Single-stranded|
| Hydrogen Bonds| A=T, G≡C             | A=U, G≡C       |
| Sugar-Phosphate Backbone | Antiparallel strands |          |

**Table. 2.** Comparison of DNA and RNA

<br>

> ① Complementarity: If one strand is damaged, the information from the other strand can be restored

> ② Sugar: Deoxyribose has a -H group at the 2nd carbon. Ribose has a -OH (more reactive) group at the 2nd carbon

> ③ Instability of RNA

>> ○ Ribose: Increased reactivity due to 2'-OH group

>> ○ Single-strand: Cannot be corrected or repaired, easily affected by external substances

>> ○ Deamination process: Amino bases A and C can undergo deamination

>>> ○ Loss of amino group in C base: Becomes U base. A base becomes I

>>> ○ DNA can be corrected, but RNA cannot

>> ○ RNA undergoes hydrolysis with increased pH

 ⑷ Eukaryotic Chromatin

> ① DNA as genetic material

>> ○ 1871, Fredrich Miescher first reported DNA in the nucleus

>> ○ **Evidence 1.** Chargaff's rules: [A] = [T], [G] = [C]

>> ○ **Evidence 2.** X-ray diffraction analysis: Rosalind Elsie Franklin's diffraction research played a key role in revealing the structure of DNA

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/1e776bf0-b4b9-4bed-af56-d7bd66460ff7)

**Figure 20.** DNA's 3D diffraction pattern

<br>

>> ○ Conclusions by Watson and Crick

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/b4466516-81bd-44cf-a800-50a54059e4fb)

**Figure 21.** DNA structure sketched by Francis Crick

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/851e4317-ee8f-424c-ba8c-0f53ee8a4d43)

**Figure 22.** Watson and Crick's paper]

<br>

> ② Structure of DNA

>> ○ Major groove: Phosphate groups not exposed. Accessible to transcription factors and enzymes

>> ○ Minor groove: Phosphate groups exposed. Accessible to histone proteins

>> ○ B-form DNA: Watson-Crick model (i.e., DNA in the body). High relative humidity (92 %)

>> ○ A-form DNA: Dehydrated environment (75 %)

>>> ○ Example: Double-stranded RNA, DNA-RNA double helix (hairpin, stem loop)

>> ○ Z-form DNA: Zigzag, slender

>>> ○ Example: Non-expressed promoters, regions with high G≡C ratio (e.g., satellites)

>> ○ Comparative summary of DNA forms

<br>

|   | Type A | Type B | Type Z |
| --- | --- | --- | --- |
| Helix Direction | Right-handed | Right-handed | Left-handed |
| Diameter | 2.37 nm | 2.55 nm | 1.84 nm |
| Base Pairs per Turn | 11 bp | 10 bp | 12 bp |
| Spacing Between Base Pairs | 0.26 nm | 0.34 nm | 0.37 nm |
| Tilt of Bases Relative to Axis | 20° | 6° | 7° |

**Table. 3.** Comparative Summary of DNA Forms

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/7c290dc5-fe9e-42cf-bca1-ac2969728b05)

**Figure 23.** A-form DNA (left), B-form DNA (middle), Z-form DNA (right)

㉠ Minor groove, ㉡ Major groove]

<br>

> ③ RNA structure: Most RNA with 3D structure is less than 150 nt in length

> ④ Nucleosome: Also called 10 nm fiber (bead-on-string)

>> ○ Structure

>>> ○ 146 bp DNA wraps 1.65 times around an octamer of histones (i.e., H2A, H2B, H3, H4 pairs)

>>> ○ Linker DNA: Refers to DNA from H1 to H1. About 200 nucleotides

>>> ○ When treated with endonuclease, H1 to H1 gets cut → visible in electrophoresis

>> ○ H1: Adjusts nucleosome structure. One per nucleosome. Involved in chromatin condensation

>> ○ Histone and DNA Binding

>>> ○ Over 20% of the amino acids in histones are basic amino acids (e.g., Lys, Arg) with a positive charge

>>> ○ Ionic bond between the negatively charged DNA with phosphate groups and the positively charged histone proteins

>> ○ Histone Tails: N-terminal ends of histones in the nucleosome protruding outward

>> ○ [Histone Modification](https://jb243.github.io/pages/74)

> ⑤ 30 nm Fiber: The structure formed when the 10 nm fiber is wound and folded

> ⑥ 300 nm Fiber: The structure in which the 30 nm fiber is attached in a loop form around a protein scaffold

> ⑦ Chromatin

>> ○ Types of Chromatin

>>> ○ **Type 1.** Euchromatin: Uncondensed chromatin

>>> ○ **Type 2.** Heterochromatin: Condensed chromatin. Typically centromeres, telomeres exist in heterochromatin form

>> ○ Structure of Chromatin

>>> ○ TAD (topologically associating domain): Chromatin forming loops

>>> ○ The actual shape of TAD is complex and tangled like a ball of yarn

>>> ○ CTCF (11-zinc finger protein, CCCTC-binding factor): A transcription factor encoded by the _CTCF_ gene

>>> ○ CTCF and PDS5A/B determine the boundaries and loop anchors of TAD

>>> ○ WAPL and PDS5A/B determine the length of the loop

> ⑧ Chromosome: The width of a single chromonema is 700 nm

 ⑸ Polytene Chromosome (giant chromosome)

> ① **Requirement 1.** **Related to Mitosis**

> ② **Requirement 2.** ** Without nuclear division, cytoplasmic division** (No division phase) DNA repeatedly replicates ( **only S phase** ) and remains in a folded state

> ③ Endoreduplication

>> ○ Definition: The process of repeatedly synthesizing DNA without the division phase (M phase). Does not form sister chromatids

>> ○ Location: In Drosophila's ovary nurse cell, follicle cell, abdominal histoblast, fat body cell, gut cell, prepupal salivary gland cell

>> ○ Result: Formation of polyploidy

> ④ Composed of more than 1,000 identical DNA molecules

> ⑤ Staining reveals alternating dark and light bands

>> ○ Less condensed light bands are areas of active transcription

>> ○ Conversely, dark areas are where transcription is suppressed

> ⑥ Puff: An area in the polytene chromosome where the light bands locally unravel and expand

>> ○ Puffs are observed in regions with vigorous gene expression

>> ○ The location of puffs varies depending on the developmental stage and external signals

> ⑦ Chromocenter: Not all DNA in chromosomes is polytene

>> ○ Part exists as strong heterochromatin

>> ○ Do not have polytene nature and almost no transcription occurs

>> ○ Salivary chromosomes are clustered around the chromocenter

>> ○ DNA in these areas makes up about 30% of the total genome and has very low gene density

>> ○ Puffs are observed in regions with vigorous gene expression

> ⑧ Example: Salivary chromosomes of dipteran insects

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/1e2ff42e-16b5-447b-8d53-2259bce4cddf)

**Figure 24.** Sketch of Calvin B. Bridge's Salivary Chromosome]

<br>

>> ○ Beneficial for producing glue necessary for becoming a pupa

>> ○ Salivary chromosomes have a polytene degree of 1024

>> ○ Salivary glands degenerate during metamorphosis

 ⑹ Lampbrush Chromosome

> ① **Requirement 1.** **Related to Meiosis**

> ② **Requirement 2.** **Created when bivalents do not separate during the first meiotic division**

> ③ Feature: Lateral Loops

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/87a00d82-32bf-491b-84bb-9953ed0f8445)

 **Figure 25.** Lateral Loops

<br>

 ⑺ Multinucleate

> ① **Requirement 1. Related to Mitosis**

> ② **Requirement 2.** **Nuclear division occurs** but cytoplasmic division does not

 ⑻ Polyploid

> ① **Requirement 1.** **Related to Meiosis**

> ② **Requirement 2.** **Treated with colchicine** to inhibit separation of homologous chromosomes and then combined with normal gametes

> ③ Example: Seedless watermelon (3n)

 ⑼ Comparison of Polytene Chromosome, Lampbrush Chromosome, Multinucleate, Polyploid

<br>

|   | **Polyploidy** | **Aneuploidy** | **Multinucleated** | **Polycaryotic** |
| --- | --- | --- | --- | --- |
| Amount of DNA per cell | ⇑ | ⇑ | ⇑ | ⇑ |
| Number of nuclei per cell | Normal | Normal | ⇑ | Normal |
| Number of chromosomes per nucleus | ⇑ | Normal | Normal | Normal |

**Table. 4.** Comparison of Polytene Chromosome, Lampbrush Chromosome, Multinucleate, Polyploid

<br>

---

_Input: 2015.06.22 22:36_

_Edit: 2020.03.21 10:16_
