## **Chapter 4. Cell and Energy Metabolism**

Recommended Reading: 【Biology】 [Biology Index](https://jb243.github.io/pages/1457)

---

**1.** [Classification of Organisms Based on Energy Metabolism](#1-classification-of-organisms-based-on-energy-metabolism)

**2.** [ATP and Electron Carriers](#2-atp-and-electron-carriers)

**3.** [Cellular Respiration](#3-cellular-respiration)

**4.** [Photosynthesis](#4-photosynthesis)

---

**a.** [ATP Synthase](https://jb243.github.io/pages/1411)

**b.** [Glycolysis](https://jb243.github.io/pages/1410)

**c.** [Glucose Synthesis](https://jb243.github.io/pages/1413)

**d.** [Protein Degradation](https://jb243.github.io/pages/1416)

**e.** [Protein Synthesis](https://jb243.github.io/pages/1418)

**f.** [Lipid Degradation](https://jb243.github.io/pages/1420)

**g.** [Lipid Synthesis](https://jb243.github.io/pages/1422)

**h.** [Alcohol Breakdown and Hangover](https://jb243.github.io/pages/1408)

---

<br>

## **1. Classification of Organisms Based on Energy Metabolism**

⑴ Classification based on energy source

> ① **Class 1.** Phototrophs: Organisms that use light as an energy source. Divided into photoautotrophs and photoheterotrophs.

> ② **Class 2.** Chemotrophs: Organisms that use chemical energy as an energy source. Divided into chemoautotrophs and chemoheterotrophs.

>> ○ Cyanobacteria _O. limnetica_ uses H<sub>2</sub>O as the initial electron donor when H<sub>2</sub>S is absent, and H<sub>2</sub>S when it is present.

>> ○ For Cyanobacteria _O. limnetica_, the energy source is H<sub>2</sub>S when it is used, otherwise it is light.

 ⑵ Classification based on carbon source

> ① **Class 1.** Autotrophs: Organisms with CO<sub>2</sub> as their carbon source. Divided into photoautotrophs and chemoautotrophs.

> ② **Class 2.** Heterotrophs: Organisms with organic compounds as their carbon source. Divided into photoheterotrophs and chemoheterotrophs.

 ⑶ Summary: When listed by energy source, carbon source,

> ① Photoautotrophs: Light + CO<sub>2</sub>

> ② Photoheterotrophs: Light + Organic Compounds

> ③ Chemoautotrophs: Inorganic Substances + CO<sub>2</sub>

> ④ Chemoheterotrophs: Organic Compounds + Organic Compounds

<br>

<br>

## 2. ATP and Electron Carriers

 ⑴ ATP (Adenosine Triphosphate)

> ① Causes conformational changes in proteins through phosphorylation to perform mechanical work, substance transport, and chemical reactions (coupled reaction molecules).

> ② Structure

>> ○ Adenine + Ribose + 3 Phosphate Molecules

>> ○ High-Energy phosphate bond: Due to electrostatic repulsion of negatively charged phosphate, it has high energy content and is unstable, thus easily hydrolyzed releasing a large amount of energy (7.3 kcal/mol).

>> ○ ATP ⇄ ADP + P<sub>i</sub> (phosphate), K = 1.5 × 10<sup>5</sup>

> ③ Types of Phosphorylation

>> ○ **Type 1.** Substrate-level phosphorylation: [Glycolysis](https://jb243.github.io/pages/1410) 

>> ○ **Type 2.** Photophosphorylation: [Bacteriorhodopsin](https://jb243.github.io/pages/69) 

>> ○ **Type 3.** Oxidative phosphorylation: Phosphorylation by the electron transport chain

 ⑵ NAD<sup>+</sup> (Nicotinamide Adenine Dinucleotide): Coenzyme in oxidation reactions, involved in cellular respiration.

> ① Adenine + 2 Riboses + Nicotinamide Group

>> ○ Derived from Niacin (Vitamin B<sub>2</sub> derivative)

> ② NAD<sup>+</sup> + H<sub>2</sub> ⇄ NADH + H<sup>+</sup>

>> ○ 1 molecule of NAD<sup>+</sup> carries 1 hydrogen atom and 1 electron.

>> ○ Thus, it carries 1 hydrogen ion and 2 electrons.

> ③ Mechanism: Nucleophilic aromatic addition reaction and hydride shift

<br>

![image](https://github.com/user-attachments/assets/b5f8db18-d94a-4a94-b987-1710f4158cdb)

**Figure 1.** The process of reducing NAD<sup>+</sup> to NADH during glycerol metabolism

<br>

⑶ NADP<sup>+</sup> (Nicotinamide Adenine Dinucleotide Phosphate): Coenzyme in anabolic reactions, involved in photosynthesis

> ① In liver cell cytoplasm, [NADPH] / [NADP<sup>+</sup>] is higher than [NADH] / [NAD<sup>+</sup>].

⑷ FAD (Flavin Adenine Dinucleotide)

> ① FAD + 2H ⇄ NADH<sub>2</sub>

<br>

<br>

## 3. Cellular Respiration

⑴ Overview of cellular respiration

> ① External respiration: Breathing in O<sub>2</sub> and releasing CO<sub>2</sub> through the lungs.

> ② Internal Respiration: Tissue cells use O<sub>2</sub> for cellular respiration and release CO<sub>2</sub>.

> ③ In cellular respiration, glucose oxidizes to CO<sub>2</sub>, O<sub>2</sub> reduces to H<sub>2</sub>O, synthesizing ATP.

⑵ Aerobic Respiration: O<sub>2</sub> is used as the **final electron acceptor** in the electron transport chain.

<br>

![image](https://github.com/user-attachments/assets/c67df71b-3f78-42d7-9601-43fcf965c607)

<br>

> ① **Step 1.** [Glycolysis](https://jb243.github.io/pages/1410): An anaerobic process that does not require oxygen.

>> ○ Overall Reaction: Glucose (6-C) + 2 ATP + 2NAD<sup>+</sup> → 2 Pyruvate (3-C) + 4 ATP + 2 NADH

>> ○ A series of 10 sequential reactions occurring in the cytosol.

>> ○ CNADH in the cytoplasm utilizes shuttle molecules to move into the mitochondrial matrix.

>>> ○ Malate-aspartate shuttle: Cytoplasmic NADH → Mitochondrial matrix NADH

>>> ○ Glycerol-phosphate shuttle: Cytoplasmic NADH → Mitochondrial matrix FADH<sub>2</sub>

>>> ○ The glycerol-3-phosphate shuttle allows cytoplasmic NADH to directly activate FADH2 without additional steps, enabling faster ATP production.

<br>

![image](https://github.com/user-attachments/assets/8f61cb5b-16db-44f4-895a-6f9d5d317ed2)

**Figure 2.** Shuttle systems located in the mitochondrial inner membrane

(A) Malate-aspartate shuttle, (B) Glycerol-phosphate shuttle

<br>

![image](https://github.com/user-attachments/assets/728ab834-7c5e-4b94-85e4-b39e00db002f)

**Figure 3.** Overall process of glycolysis

<br>

> ② **Step 2.** TCA Cycle: Also known as the Citric Acid Cycle, Krebs Cycle, Tricarboxylic Acid Cycle, etc.

<br>

![image](https://github.com/user-attachments/assets/ecc96bd4-73a5-4438-93f7-868da0b0b2b8)

**Figure 4.** TCA Cycle

<br>

>> ○ Inference of the overall reaction equation

>>> ○ Half oxidation reaction equation

<br>

![image](https://github.com/user-attachments/assets/369b046a-d129-4804-bb64-e4f624ed5958)

<br>

>>> ○ Half reduction reaction equation

<br>

![image](https://github.com/user-attachments/assets/3e6e0696-8182-4b2d-88ce-0319d7afcb08)

<br>

>>> ○ Incomplete reaction equation

<br>

![image](https://github.com/user-attachments/assets/fd517fb6-13a6-4df7-b3b1-42c430954836)

<br>

>>> ○ Conclusion: 2C<sub>3</sub>H<sub>4</sub>O<sub>3</sub> + 6H<sub>2</sub>O + 5O<sub>2</sub> + ADP + P<sub>i</sub> → 6CO<sub>2</sub> + 10H<sub>2</sub>O

>> ○ Oxidation of pyruvate

>>> ○ Occurs in the inner membrane of mitochondria.

>>> ○ 2 Pyruvate (3-C) → 2 CO<sub>2</sub> + 2NADH + 2 Acetyl CoA (2-C)

>> ○ Citric acid cycle

>>> ○ Occurs in the mitochondrial matrix.

>>> ○ Acetyl CoA → 2 CO<sub>2</sub> + 3 NADH + FADH<sub>2</sub> + GTP (by 8 enzymes)

>>> ○ GTP produced in the citric acid cycle is generally considered equivalent to ATP.

>>> ○ Oxaloacetate (OAA) continuously changes its components.

>> ○ During the process of carbon dioxide release, C-C bonds are broken, and the energy released is used to produce NADH.

>> ○ **Memory Tip:** Pyruvate - Acetyl CoA - Citrate - α-Ketoglutarate - Succinate - Fumarate - Malate - Oxaloacetate

> ③ **Step 3.** Electron Transport System (ETS) in Mitochondria

>> ○ Complex I (NADH-Q Oxidoreductase): Releases 1 H<sup>+</sup> as electrons pass through.

>> ○ Complex II (Succinate Dehydrogenase)

>>> ○ Refers to membrane-bound protein FAD.

>>> ○ Does not pump H<sup>+</sup> due to low energy.

>> ○ Ubiquinone 

>>> ○ A hydrophobic molecule in the quinone family, located inside the inner membrane.

>>> ○ Coenzyme A (related to Acetyl CoA)

>>> ○ Fat-soluble vitamin (the only lipid component)

>> ○ Complex III (Cytochrome C Reductase, Cytochrome B): Releases 1 H<sup>+</sup> as electrons pass through.

>> ○ Cytochrome C

>>> ○ Attached to the inner membrane, facing the intermembrane space.

>>> ○ Leads to programmed cell death when released into the cytoplasm.

>> ○ Complex IV (Cytochrome C Oxidase, Cytochrome A)

>>> ○ Releases 1 H<sup>+</sup> as electrons pass through.

>>> ○ Transfers electrons to oxygen at the end, forming water.

>> ○ ATP Synthase: H<sup>+</sup> ions pass through a protein channel called [ATP Synthase](https://jb243.github.io/pages/1411) in the inner membrane, leading to ATP production.

<br>

![image](https://github.com/user-attachments/assets/31757abc-b8ec-4895-b71e-827c8b82e8ce)

**Figure 5.** Electron transport system in mitochondrial inner membrane

<br>

>> ○ Redox potential of the electron transport chain

>>> ○ Reducing power: The tendency to reduce other substances. Low reducing power implies a tendency to reduce by itself.

>>> ○ Reduction potential: A higher standard reduction potential value for chemical species that tend to be reduced.

>>> ○ Gibbs free energy change: The more a chemical species prefers electrons, the more negative the change in free energy becomes.

>>> ○ Conclusion: As the reaction process progresses, reducing power decreases, reduction potential increases, and the free energy change ΔG becomes more negative.

>>> ○ For a chemical reaction to occur, ΔG ＜ 0, so ΔG must become more negative as the process progresses: ΔG<sub>earlier_step</sub> + ΔG<sub>latter_step</sub> ＜ 0

>>> ○ By using ΔG = -nFE, the relationship between reduction potential and ΔG can be easily understood.

<br>

![image](https://github.com/user-attachments/assets/17511a68-9ec2-4847-8a86-813d0b9829dd)

**Figure 6.** Redox potential of the electron transport chain

<br>

> ④ Cellular respiration calculation

>> ○ **Yield 1.** Approximately 1 NADH → 3 ATP, 1 FADH<sub>2</sub> → 2 ATP

>> ○ **Yield 2.** For prokaryotes, fungi, algae, etc., the yield is lower: 1 NADH → 2.5 ATP, 1 FADH<sub>2</sub> → 1.5 ATP

>>> ○ 4H<sup>+</sup> are released when electrons move through complexes I and III, and 2H<sup>+</sup> are released when electrons move through complex IV.

>>>> ○ Complex IV, being closer to ATP synthase, has a difficulty in structural change and therefore has a less yield.

>>> ○ 4H<sup>+</sup> are required to synthesize 1 ATP.

>>>> ○ ATP synthase makes one turn when 3H<sup>+</sup> move.

>>>> ○ H<sup>+</sup> is consumed when pyruvate moves into the inner membrane.

>>> ○ NADH yield = Total H<sup>+</sup> moved ÷ (4 H<sup>+</sup> / 1 ATP) = (4 × 2 + 2) ÷ 4 = 2.5

>>> ○ FADH<sub>2</sub> yield = Total H+ moved ÷ (4 H<sup>+</sup> / 1 ATP) = (4 × 1 + 2) ÷ 4 = 1.5

>> ○ One molecule of glucose is converted into 2 pyruvate molecules through glycolysis, producing 2 ATP and 2 NADH.

>> ○ 2 Pyruvates enter mitochondria, undergo oxidation and citric acid cycle, producing 2 ATP + 8 NADH + 2 FADH<sub>2</sub>.

>>> ○ **Tip:** CO<sub>2</sub> release produces NADH.

>>> ○ **Tip:** Except for the conversion of malate to OAA, NADH formation accompanies CO<sub>2</sub> production.

>> ○ Cytosolic NADH must pass through a shuttle in the mitochondrial inner membrane, and may be converted to FADH<sub>2</sub>.

>> ○ Liver, heart, kidneys: A total of **32 ATP** produced (38 ATP, calculated with yield 1). Related to malate-aspartate shuttle.

>> ○ Skeletal muscle, brain, etc.: A total of **30 ATP** produced (36 ATP, calculated with yield 1). Related to glycerol phosphate shuttle.

>>> ○ Previously calculated as 32 ATP.

>>> ○ Skeletal muscle and brain use the glycerol phosphate shuttle for rapid energy despite lower efficiency.

>> ○ In reality, 1 H<sup>+</sup> is also used when free phosphate moves across the mitochondrial inner membrane.

> ⑤ Cellular respiration inhibitors

>> ○ Complex I inhibitors: Rotenone, Barbiturates, Piericidin

>> ○ Complex II inhibitors: Malonate

>> ○ Complex III inhibitors: Antimycin

>> ○ Complex IV inhibitors: CN<sup>-</sup>, CO

>> ○ ATP synthase inhibitors: Oligomycin

>> ○ Uncoupling agents

>>> ○ DNP (2,4-Dinitrophenol), Valinomycin, Thermogenin ([brown fat](https://jb243.github.io/pages/81) is related)

>>> ○ Uncoupling agents are a type of ionophore for H<sup>+</sup> that rapidly decrease the proton concentration gradient, generating a lot of heat.

<br>

|   | **Electron Transport** | **ATP Synthesis** |
| --- | --- | --- |
| Electron Transport Inhibitor | Stops | Decreases then Stops |
| ATP Synthase Inhibitor | Decreases then Stops | Stops |
| Uncoupling Agent | Increases | Stops |

**Table 1.** Summary of inhibitors

<br>

> ⑥ Global warming and cellular respiration

>> ○ Temperature increase accelerates the metabolic rate of organisms, speeding up growth, development, and reproduction.

>> ○ Example: Acceleration of the life cycle of beetles (2 years → 1 year) → Destruction of spruce forests

 ⑶ [Gluconeogenesis](https://jb243.github.io/pages/1413)

 ⑷ Anaerobic respiration and fermentation

> ① Electron transfer interruption and continuation of glycolysis → Accumulation of pyruvate, NADH, depletion of NAD<sup>+</sup> → Reaction stop

> ② Reason for anaerobic respiration: To supply NAD<sup>+</sup> for glycolysis

>> ○ Reduces pyruvate to lactate, ethyl alcohol, etc., while oxidizing NADH to NAD<sup>+</sup>.

>> ○ A pair of glycolysis and ab anaerobic respiration produce 2 ATP.

> ③ **Type 1.** Lactic acid fermentation: Occurs in muscle cells, lactic acid bacteria, etc.

<br>

![image](https://github.com/user-attachments/assets/5b39edc0-2f65-461d-8dcf-6ec680c671e6)

**Figure 7.** Lactic Acid Fermentation

<br>

>> ○ **Reaction 1.** Glycolysis: Glucose → 2 Pyruvates

>> ○ **Reaction 2.** Reduction of Pyruvate: 2 Pyruvates → 2 Lactic Acids

>> ○ Lactic acid is decomposed in the liver.

>> ○ During the reduction from pyruvate to lactic acid, lactate dehydrogenase (gene name: _LDHA_) is involved.

>> ○ Warburg effect: Cancer cells prefer lactic acid fermentation over efficient cellular respiration.

>> ○ The brain also performs lactic acid fermentation and is known to [excrete lactate](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5315230/#:~:text=to%20export%20lactate-,at%20rest,-%5BPaulson%2C%202002) during rest.

> ④ **Type 2.** Alcohol fermentation: Occurs in yeast, etc.

>> ○ **Reaction 1.** Glycolysis: Glucose → 2 Pyruvates

>> ○ **Reaction 2.** Decarboxylation: 2 Pyruvates → 2 Acetaldehydes + 2CO<sub>2</sub>

>> ○ **Reaction 3.** Reduction of Acetaldehyde: 2 Acetaldehydes → 2 Ethanol

>> ○ The decarboxylation reaction from pyruvate to acetaldehyde involves pyruvate carboxylase.

>> ○ The reduction from acetaldehyde to ethanol involves alcohol dehydrogenase.

<br>

![image](https://github.com/user-attachments/assets/7434ae39-729e-4f25-8a2c-9ce0976ffca3)

**Figure 8.** Alcohol fermentation

<br>

> ⑤ **Type 3.** Acetic acid fermentation: Oxygen is involved. Not a complete fermentation in the true sense.

<br>

![image](https://github.com/user-attachments/assets/fbd786aa-e7ff-4cde-925d-a5de9df3eba3)

**Figure 9.** Acetic acid fermentation

<br>

⑸ Other glucose catabolic pathways

> ① EMP (Embden-Meyerhof-Parnas) pathway: [Glycolysis](https://jb243.github.io/pages/1410)

>> ○ Reaction that generates ATP.

>> ○ Chemical reaction formula: C<sub>6</sub>H<sub>12</sub>O<sub>6</sub> + 2ADP + 2P<sub>i</sub> + 2NAD<sup>+</sup> → 2C<sub>3</sub>H<sub>4</sub>O<sub>3</sub> + 2ATP + 2NADH + 2H<sup>+</sup>

> ② TCA cycle

>> ○ Reaction that generates ATP.

>> ○ Chemical reaction formula: C<sub>3</sub>H<sub>4</sub>O<sub>3</sub> + 3H<sub>2</sub>O + ADP + P<sub>i</sub> + 4NAD<sup>+</sup> + FAD → 3CO<sub>2</sub> + 4(NADH + H<sup>+</sup>) + FADH<sub>2</sub> + ATP

> ③ HMP (hexose monophosphate pathway)

>> ○ Also known as PPP (pentose phosphate pathway), pentose shunt, Warburg-Dickens pathway.

>> ○ Reaction to produce NADPH required for anabolic processes such as fatty acid synthesis.

>> ○ Chemical reaction formula: C<sub>6</sub>H<sub>12</sub>O<sub>6</sub> + 12NADP<sup>+</sup> + 3O<sub>2</sub> + ATP → 6CO<sub>2</sub> + 12NADPH + ADP + P<sub>i</sub>

>>> ○ Glucose-6-phosphate can be completely reduced to carbon dioxide to generate 12 molecules of NADPH.

>>> ○ Sugars with different carbon numbers can be synthesized in this reaction.

>>> ○ The pentose sugars obtained from the breakdown of ingested nucleic acids are also metabolized through the HMP pathway.

>> ○ Mechanism

<br>

![image](https://github.com/user-attachments/assets/92d12b91-7e2a-4e44-87e1-07dad8860a90)

**Figure 10.** HMP pathway mechanism

<br>

>>> ○ Oxidative phase: Generates NADPH.

>>> ○ Non-oxidative phase: Generates pentoses etc.

>> ○ Reaction location

>>> ○ Animal cells: Cytoplasm

>>> ○ Plant cells: Chloroplasts, cytoplasm; dominant in chloroplasts.

>>> ○ Most active in the liver, mammary glands, and adrenal glands, accounting for over 60% of reducing power.

>> ○ Basic role: Provides carbon skeletons needed for biosynthetic reactions and reduction power to support anabolic processes

>>> ○ Abundant of NADPH: Supplies the reducing power required for the biosynthesis of biomolecules.

>>> ○ Ribose-5-phosphate: Material for nucleotide bases, coenzymes, and a major precursor of histidine.

>>> ○ Ribulose-5-phosphate: Participates in the Krebs cycle.

>>> ○ Erythrose-4-phosphate: Material for aromatic amino acids.

>>> ○ Glyceraldehyde-3-phosphate: Participates in the EMP and TCA cycles.

> ④ ED (Entner-Doudoroff) pathway

>> ○ Chemical reaction formula: C<sub>6</sub>H<sub>12</sub>O<sub>6</sub> + ADP + P<sub>i</sub> → C<sub>2</sub>H<sub>5</sub>OH + 2CO<sub>2</sub> + ATP

>> ○ Found in _Zymomonas mobilis, Pseudomonas saccharophilia_, etc.

>> ○ _Zymomonas mobilis_: Gram-positive. Produces high concentrations of ethanol in a short time.

> ⑤ PK (phosphoketolase) pathway

>> ○ Feature: Involvement of phosphateketolase.

>> ○ Phosphateketolase: Breaks down into glyceraldehyde-3-phosphate, acetyl phosphate, and pentose phosphate.

⑹ Metabolism of carbohydrates, proteins, and lipids

> ① Overview

>> ○ 1<sup>st</sup> priority energy source: Carbohydrates

>> ○ 2<sup>nd</sup> priority energy source: Lipids

>> ○ 3<sup>rd</sup> priority energy source: Proteins. Since proteins have many unique functions, it's a very dangerous situation if proteins are being broken down.

> ② Glycogen breakdown process

>> ○ Almost all tissues can contain glycogen.

>> ○ Glycogen storage tissues: Liver, muscles, kidneys. Glycogens exist in the cytoplasm in granular form.

>> ○ Glycogen breakdown: Non-reducing end's α 1→4 bond phosphorolysis enzyme → transferase → α 1→6 glucosidase hydrolase

<br>

![image](https://github.com/user-attachments/assets/5d1ca793-d2af-4c2b-b04b-b0df5eb1e8a6)

**Figure 11.** Glycogen Breakdown Process

<br>

> ③ [Protein breakdown](https://jb243.github.io/pages/1416)

> ④ [Protein synthesis](https://jb243.github.io/pages/1418)

> ⑤ [Lipid breakdown](https://jb243.github.io/pages/1420)

> ⑥ [Lipid synthesis](https://jb243.github.io/pages/1422)

⑺ Tissue-specific glucose catabolism

> ① Liver

>> ○ Absorbed by GLUT2.

>> ○ **Pathway 1.** PPP

>> ○ **Pathway 2.** TCA cycle

> ② Adipose tissue

>> ○ GLUT4 is conditionally expressed in response to insulin.

>> ○ **Pathway 1.** Glucose absorption → PPP pathway → DNA skeleton formation

>> ○ **Pathway 2.** Glucose → Acetyl coA → TCA cycle

>> ○ **Pathway 3.** Glucose → Fatty acids → Stored as triacylglycerol.

> ③ Muscle tissue

>> ○ GLUT4 is conditionally expressed in response to insulin.

>> ○ PPP pathway does not occur.

> ④ Brain

>> ○ GLUT1, which is the most efficient glucose transporter, is expressed because the brain requires a high amount of glucose.

>> ○ Glucose can pass through the BBB, and insulin does not affect glucose metabolism in the brain.

⑻ Response index

> ① P:O ratio

>> ○ The ratio of P<sub>i</sub> to oxygen. It represents the amount of ATP produced by oxidative phosphorylation when one oxygen atom is reduced to water.

>> ○ 2O<sub>2</sub> + 2H<sup>+</sup> → H<sub>2</sub>O

> ② Respiratory quotient (RQ): Moles of produced carbon dioxide / Moles of consumed oxygen

> ③ Oxygen transfer rate (OTR): The rate at which oxygen in the air dissolves in solution.

> ④ Oxygen uptake rate (OUR): Oxygen respiration rate of microorganisms contained in a unit volume of culture medium.

<br>

<br>

## **4. Photosynthesis**

 ⑴ Overview

> ① Photosynthetic organisms: Bacteria, Algae, Plants

>> ○ **Type 1.** Bacteria: Purple bacteria, Green sulfur bacteria, Cyanobacteria

>> ○ **Type 1-1.** Purple bacteria

>>> ○ Absorbs 870 nm wavelength. 1 photosystem

>>> ○ Cyclic photophosphorylation only

>>> ○ H2S as electron donor

>>> ○ Reverse electron transfer ( **∵** Cannot harvest light)

>> ○ **Type 1-2.** Green sulfur bacteria

>>> ○ Absorbs 840 nm wavelength. 2 photosystems

>>> ○ Cyclic photophosphorylation + primitive non-cyclic photophosphorylation

>>> ○ H<sub>2</sub>S as electron donor

>> ○ **Type 1-3.** Cyanobacteria: Ancestor of chloroplasts

>>> ○ Absorbs 670 nm wavelength. 2 photosystems

>>> ○ Cyclic photophosphorylation + non-cyclic photophosphorylation

>>> ○ H<sub>2</sub>O as electron donor

>> ○ **Type 2.** Algae

>>> ○ Red algae: Chlorophyll a, d

>>> ○ Diatoms: Chlorophyll a, c

>>> ○ Brown algae: Chlorophyll a, c

>>> ○ Green algae: Chlorophyll a, b

>> ○ **Type 3.** Bryophytes, Ferns, Gymnosperms, Angiosperms

>>> ○ Bryophytes: Exchange substances through diffusion, chlorophyll a, b

>>> ○ Ferns, Gymnosperms, Angiosperms: Chlorophyll a, b

>>> ○ Shade plants: Lack carotenoid pigments

> ② Photosynthesis equation: 6 CO<sub>2</sub> + 12 H<sub>2</sub>O + Light → Glucose(C6H12O6)+ 6O2 + 6H<sub>2</sub>O (ΔG = 686 kcal / mol)

>> ○ Achieves the opposite result of cellular respiration

>> ○ Tracing the photosynthesis equation

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/ca7d41f9-1931-4e8b-be18-821042119aff)

**Figure 12.** Experiment results using radioactive isotopes in photosynthesis

<br>

>> ○ Van Niel's research: Used green sulfur bacteria to prove that oxygen in photosynthesis comes from water

>> ○ Ruben's research: Proved Van Niel's hypothesis using radioactive isotope 18O

> ③ Photosynthesis experiments

>> ○ Engelmann's experiment

>>> ○ Experiment on photosynthetic efficiency using Spirogyra and aerobic bacteria

>>> ○ Chlorophyll absorbs light well at 400 ~ 500 nm and 600 ~ 700 nm wavelengths but not at 500 ~ 600 nm

>>> ○ Thus, photosynthesis is efficient under blue-violet and red-orange light wavelengths

>> ○ Emerson experiment - Emerson red drop

>>> ○ Presented a graph of wavelength against quantum yield

>>> ○ Quantum yield: A function of absorption spectrum and photosynthetic efficiency

>>> ○ Absorption spectrum peaks at 680 nm and drops sharply after 700 nm: Due to Photosystem II not functioning, reducing photosynthetic efficiency

>> ○ Emerson experiment - Emerson enhancement effect

>> ○ Duysens' experiment

>> ○ Benson's experiment

> ④ Principles of photosynthesis

>> ○ CO<sub>2</sub> and O2 are transported through stomata

>> ○ Light-dependent reactions

>>> ○ Utilizes light energy for electron transfer

>>> ○ Reduction reactions: NADP+ → NADPH, ADP → ATP

>>> ○ Water splitting, oxygen release

>> ○ Light-independent reactions

>>> ○ Reduce CO<sub>2</sub> to produce C6H12O6

>>> ○ Oxidation reactions: NADPH → NADP+, ATP → ADP

> ⑤ Factors affecting photosynthesis

>> ○ Law of the minimum

>>> ○ The growth of plants is governed by the scarcest essential nutrient, not by the abundant ones

>>> ○ Proposed by the German botanist Justus von Liebig in 1840

>> ○ **Dependent variable:** CO<sub>2</sub>

>> ○ **Independent variable 1.** Light: Has a light saturation point

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/5ead4b86-203a-4d9e-86a4-da8aea694880)

**Figure 13.** Light intensity and photosynthesis

<br>

>>> ○ Apparent photosynthetic rate: Visible photosynthesis rate

>>> ○ Total photosynthetic rate: Total amount of CO<sub>2</sub> used in actual photosynthesis

>>> ○ Respiratory rate = Total photosynthetic rate - Apparent photosynthetic rate

>>> ○ Compensation point(B): Light intensity where total photosynthesis rate equals respiratory rate

>>> ○ Light saturation point(D): Light intensity where photosynthesis rate starts to maximize

>>> ○ Sun plants have higher light saturation and compensation points as they are bigger and more metabolically active than shade plants

>>> ○ Sun plants have carotenoids, hence a higher light saturation point

>> ○ **Independent variable 2.** Temperature: Has an optimal temperature

 ⑵ [Chloroplast](https://jb243.github.io/pages/1409#:~:text=%E2%91%B5-,%EC%97%BD%EB%A1%9D%EC%B2%B4,-\(chloroplast\)%20%3A)

 ⑶ Pigments

> ① Photosynthetic pigment separation experiment: Paper chromatography (Toluene solvent)

> ② Chlorophyll: Divided into chlorophyll a and chlorophyll b, absorbs red and blue light

>> ○ Porphyrin ring

>>> ○ Aromatic, functions as a pigment

>>> ○ Structure: Magnesium coordinating group + tetrapyrrole (pyrrole) + phytol

>> ○ Hydrocarbon tail(phytol): Interacts with the hydrophobic parts of proteins (like photosystems) located in the thylakoid membrane of chloroplasts

>> ○ Chlorophyll a: Methyl group

>> ○ Chlorophyll b: Aldehyde group

> ③ Carotenoids: Accessory pigments. Absorb green light. Lower photosynthetic efficiency than chlorophyll. Present only in sun plants

>> ○ **Function 1.** Spectrum expansion

>> ○ **Function 2.** Protecting photosynthetic machinery from intense light

>> ○ β-carotene: A type of carotenoid

>>> ○ Mainly absorbs light at 400 ~ 500 nm but also green light

>>> ○ Can be converted into Vitamin A

>>> ○ Retinal, a derivative of Vitamin A, is used as a component of visual pigments in the rod and cone cells of the retina

> ④ Xanthophyll

> ⑤ Phycobilin: Pigment molecules of Purple bacteria (870 nm), Green sulfur bacteria (840 nm), Cyanobacteria (670 nm)

 ⑷ Photosystems

> ① Reaction center: The site in thylakoid membranes where light energy is converted into chemical energy

>> ○ Composed of 2 chlorophyll a, a primary electron acceptor, proteins, etc.

>> ○ Photosystem I: Absorbs 700 nm wavelength. Exposed to the stroma and historically discovered first

>> ○ Photosystem II: Absorbs 680 nm wavelength. Hidden from the stroma and historically discovered later

>> ○ Chlorophyll a absorbs enough energy → Excited electron leaves its orbit → High-energy electron is transferred to the electron transport chain

> ② Antenna complex(light-harvesting complex): Surrounds the reaction center, contains chlorophyll a, chlorophyll b, carotenoids, xanthophylls, etc.

 ⑸ Light-dependent reactions: Occur on the disc-shaped membrane structures covered with chlorophyll molecules in the thylakoid(thylakoid) membrane.

> ① 1<sup>st</sup>. Antenna: Photons are absorbed by pigment molecules and transferred to adjacent pigment molecules by RET(resonance energy transfer)

>> ○ RET: When two molecules are close enough, photons are transferred more directly rather than through fluorescence

> ② 2<sup>nd</sup>. Reaction center: Electrons move between pigment molecules until they reach P680 of Photosystem II or P700 of Photosystem I

>> ○ P680 shows maximum absorption at 680 nm, P700 at 700 nm

> ③ 3<sup>rd</sup>. P680 of Photosystem II oxidizes to P680+ and transfers to the primary acceptor

>> ○ The primary acceptor is also called pheophytin

>> ○ P680+ is **the strongest oxidizing agent on Earth**

> ④ 4<sup>th</sup>. P680 has a structure with Mn in the center of pheophytin; oxidized P680+ oxidizes H<sub>2</sub>O to gain electrons (O2 production)

>> ○ Water photolysis occurs inside the thylakoid

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/db70140c-e4e1-4ac8-bb16-15e1572d9703)

**Figure 14.** Pheophytin and water photolysis

<br>

> ⑤ 5<sup>th</sup>. The electron moved to the first electron acceptor reaches P700 through PQ (plastoquinone), ATP synthase (cytochrome b0f complex, CF<sub>0</sub>CF<sub>1</sub> complex), PC (plastocyanin)

>> ○ In this process, 3H<sup>+</sup> moves, generating 1 ATP

>> ○ ATP synthase consists of the CF<sub>0</sub> channel, which facilitates the facilitated diffusion of protons, and CF<sub>1</sub>, which contains ATP synthase, creating ATP by chemiosmotic phosphorylation similar to mitochondria's F<sub>0</sub>F<sub>1</sub>

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/bdfa1e82-beed-4b46-8db2-81a949ffb370)

**Figure 15.** Structure of the CF0CF1 complex

<br>

> ⑥ 6<sup>th</sup>. P700, having received electrons from ② and ⑤, becomes the excited P700*, which then transfers the electron to the primary electron acceptor

>> ○ The primary electron acceptor is also known as phylloquinone

>> ○ P700* is one of the strongest reductants in the world

> ⑦ 7<sup>th</sup>. The electron is transferred from phylloquinone to ferredoxin

> ⑧ 8<sup>th</sup> \- 1<sup>st</sup>. Cyclic photophosphorylation: When the electron moves from ferredoxin to the cytochrome complex. 1 ATP is generated

>> ○ Product: ATP

> ⑨ 8<sup>th</sup> \- 2<sup>nd</sup>. Non-cyclic photophosphorylation: Ferredoxin transfers the electron to ferredoxin-NADP+

>> ○ Reduction of ferredoxin-NADP+ results in the production of 1 ATP + 1 NADPH

>> ○ Products: ATP, NADPH, O<sub>2</sub>

>> ○ NADP<sup>+</sup> is the final electron acceptor

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/afb3d410-c87e-46a9-a0b4-6d49c452927f)

**Figure 16.** Cyclic and non-cyclic photophosphorylation

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/d75ccc9e-f5df-40f9-a4f2-e92275cd2715)

**Figure 17.** Standard reduction potentials of the light reactions

<br>

> ⑨ Inhibitors

>> ○ Photosystem I inhibitor: Paraquat

>> ○ Photosystem II inhibitor: DCMU

 ⑹ Dark reactions (Calvin Cycle)

> ① Stroma: The matrix of chloroplasts. **Site of dark reactions**

>> ○ The reaction synthesizing sugar from glucose takes place in the cytosol

> ② Calvin Cycle: Uses ATP and NADPH to reduce CO<sub>2</sub> and synthesize carbohydrates (12 NADPH, 18 ATP)

>> ○ Part of the [glucose neosynthesis process](https://jb243.github.io/pages/1413): 3PGA → 1,3-BPG → G3P

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/010af421-0471-42b2-8f4a-909f9c57dae3)

**Figure 18.** Calvin Cycle

PGA is 3PG, DPGA is 1,3-BPG, PGAL is G3P

<br>

> ③ Rubisco (ribulose-1,5-bisphosphate carboxylase oxygenase)

>> ○ Accounts for 50% of all proteins in leaves, the most abundant protein on Earth

>> ○ CO<sub>2</sub> **↑** **→** Carboxylase activity, O<sub>2</sub> **↑** **→** Oxygenase activity

> ④ G3P: Becomes starch in the stroma or sugar in the cytosol of mesophyll cells

> ⑤ Dark reaction enzymes function more efficiently under light conditions

>> ○ Rubisco

>>> ○ Light conditions → Alkalization of stroma → Increase in Rubisco activity

>>> ○ pH changes due to light reactions move Mg2+ stored in the thylakoid lumen to the stroma → Activates Rubisco

 ⑺ C3, C4, and CAM plants

> ① C3 plants: Common plants (e.g., roses, rice, and most temperate plants)

>> ○ First carbon fixation product: 3-PGA (3C)

>> ○ Reactions at high temperatures

>>> ○ Guard cells: Regulate the opening and closing of stomata and transpiration

>>> ○ Open stomata: Allows CO<sub>2</sub> intake, but also water loss

>>> ○ Temperature↑ → Stomata close → CO<sub>2</sub> deficiency → Decrease in photosynthesis rate

>>> ○ Temperature **↑** **→** Stomata close → CO<sub>2</sub> deficiency → Excess O2 leads to photorespiration

> ② Photorespiration

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/3fa56647-e747-4031-8b37-3e7d95dc83eb)

**Figure 19.** Photorespiration

<br>

>> ○ Rubisco can use either CO<sub>2</sub> or O2 as substrates

>> ○ Rubisco acts as an oxygenase

>> ○ 1<sup>st</sup>. RuBP + O<sub>2</sub> → 3PG + Glycolate (2C)

>> ○ 2<sup>nd</sup>. 3PG + Glycolate (2C) + ATP **→** RuBP + CO<sub>2</sub>

>> ○ Glycolate is decomposed, losing carbon sources

>> ○ Photorespiration involves chloroplasts, peroxisomes, and mitochondria

>> ○ Photorespiration uses energy, so cyclic photophosphorylation is dominant in photorespiratory conditions

>> ○ **Glycine** and **Serine** are involved as intermediates

> ③ C4 plants: Tropical and subtropical plants (e.g., corn, sugarcane, sorghum, tropical plants)

>> ○ Lack of Photosystem II: Specifically, mesophyll cells have both Photosystems I and II, but bundle-sheath cells only have Photosystem I ([ref](https://www.pnas.org/doi/10.1073/pnas.67.1.18))

>> ○ Only cyclic photophosphorylation, low compensation point, spatial separation of carbon fixation

>> ○ Krantz anatomy: Bundle-sheath cells surround the vascular bundle, and mesophyll cells surround them

>> ○ 1<sup>st</sup>. Stomata nearly closed, mesophyll cells ( **mesophyll cells** ) fix carbon to PEP, which has a high affinity for CO<sub>2</sub>

>>> ○ Mesophyll cells of C4 plants lack chloroplasts

>> ○ 2<sup>nd</sup>. PEP carboxylase binds CO<sub>2</sub> to PEP (3C) to produce oxaloacetate (OAA, 4C). **First carbon fixation product**

>>> ○ **** PEP carboxylase has a higher CO<sub>2</sub> affinity than Rubisco

>>> ○ PEP carboxylase does not exist in bundle-sheath cells

>>> ○ PEP carboxylase (PEP carboxykinase)

>> ○ 3<sup>rd</sup>. Reduced malate (4C) moves to the interior ( **bundle-sheath cells** ), releasing CO<sub>2</sub>

>> ○ 4<sup>th</sup>. Bundle-sheath cell Rubisco uses the supplied CO<sub>2</sub> and oxaloacetate in the stroma for the Calvin cycle

>>> ○ Rubisco does not exist in mesophyll cells

>> ○ 5<sup>th</sup>. Malate releasing CO<sub>2</sub> generates NADPH and produces pyruvate (3C): why Photosystem II is unnecessary

>> ○ 6<sup>th</sup>. Pyruvate uses ATP to produce PEP

>> ○ 7<sup>th</sup>. PEP moves to mesophyll cells

>> ○ Alternative pathway: Oxaloacetate converted to malate via aspartate

> ④ CAM plants: Plants in extremely hot and dry environments, succulents (e.g., jade plants, cacti, pineapple, desert plants)

>> ○ Photosystem II exists, but its activity decreases under conditions where photorespiration occurs readily
([ref](https://www.frontiersin.org/articles/10.3389/fpls.2019.01012/full))]

>> ○ Only cyclic photophosphorylation, low compensation point, carbon fixation **temporally** separated

>> ○ 1<sup>st</sup>. During the day, stomata are completely closed to conserve water, dark reactions occur but with lower efficiency

>> ○ 2<sup>nd</sup>. At night, stomata open to produce malate from CO<sub>2</sub> in the C4 cycle

>>> ○ 2<sup>nd</sup> - 1<sup>st</sup>. PEP carboxylase binds CO<sub>2</sub> to PEP to produce oxaloacetate. **The initial carbon fixation product**

>>> ○ 2<sup>nd</sup> - 2<sup>nd</sup>. Oxaloacetate is reduced to malate

>>> ○ PEP carboxylase (PEP carboxykinase)

>> ○ 3<sup>rd</sup>. The produced malate is stored in the vacuole

>> ○ 4<sup>th</sup>. During the day, malate is decomposed to operate the Calvin cycle with the CO<sub>2</sub> obtained

>>> ○ 4<sup>th</sup> - 1<sup>st</sup>. Malate → CO<sub>2</sub> + NADPH + Pyruvate (3C)

>>> ○ 4<sup>th</sup> - 2<sup>nd</sup>. Pyruvate (3C) is used to produce ATP or used in the TCA cycle to produce ATP

>>> ○ 4<sup>th</sup> - 3<sup>rd</sup>. CO<sub>2</sub> moves to the stroma to participate in the Calvin cycle with oxaloacetate and Rubisco

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/ff65044e-bd67-4d6c-adac-628ea4878ea8)

**Figure 20.** CAM Plants

<br>

> ⑤ Comparison of C3, C4, CAM Plants

<br>

|   | C3 Plants | C4 Plants | CAM Plants |
| --- | --- | --- | --- |
| Primary CO<sub>2</sub> Acceptor | RuBP | PEP | PEP |
| CO<sub>2</sub> Fixation Enzyme | Rubisco | PEP Carboxylase | PEP Carboxylase |
| Minimum CO<sub>2</sub> Product | 3PGA | OAA | OAA |
| Initial Process | CO<sub>2</sub> + H<sub>2</sub>O + RuBP → 2PGA | CO<sub>2</sub> + H<sub>2</sub>O + PEP → OAA |  |
| Light Reaction | Mesophyll cells | Mesophyll cells | Mesophyll cells |
| Calvin-Benson Cycle | Mesophyll cells | Bundle Sheath Cells | Mesophyll cells |
| CO<sub>2</sub> Fixation Site | Mesophyll cells | Mesophyll cells | Mesophyll cells |
| CO<sub>2</sub> Fixation Time | Day | Day | Night |
| Photorespiration | Strong | Weak | Weak |
| Water Needed for 1g of Photosynthesis | About 500 g | About 300 g | About 50 g |

**Table. 2.** Comparison of C3, C4, CAM Plants

<br>

 ⑻ Global Warming and Photosynthesis

> ① Temperature rise affects the relative proportions and distribution of C3, C4, CAM plants

> ② Carbon dioxide is being released faster than current photosynthesis can remove

> ③ 25% of the additional CO<sub>2</sub> in the atmosphere is due to deforestation (logging and fires) of tropical forests

> ④ Reforestation helps; young trees have a faster rate of net photosynthesis than old trees

<br>

---

_Input: 2015.06.25 12:57_

_Edit: 2021.02.11 10:54_
