## **Chapter 5. Cell division and Cancer**

Higher category: 【Biology】 [Biology Index](https://jb243.github.io/pages/1457) 

---

**1.** [DNA and Chromosomes](#1-dna-and-chromosome)

**2.** [Vertical transfer of genes](#2-vertical-transfer-of-genes-cell-division)

**3.** [Horizontal transfer of genes](#3-horizontal-transfer-of-genes)

**4.** [Cell cycle regulation](#4-cell-cycle-regulation)

**5.** [Mutation](#5-mutation)

**6.** [Cancer](#6-cancer)

---

**a.** [Horizontal Transfer of Genes](https://jb243.github.io/pages/554)

**b.** [Statistics of Cancer](https://jb243.github.io/pages/730)

---

<br>

## **1\. DNA and Chromosomes**

⑴ DNA binds to proteins to form a dye (= chromatin) and contains many genes.

> ① Cationic amino acids, such as histidine, condense negatively charged DNA.

> ② Heterochromatin: Condensed chromatin

> ③ Euchromatin: Loose chromatin

⑵ Chromosome: Chromatin condenses into a form that is visible under an optical microscope.

> ① Number of chromosomes = Number of chunks acting as one = Number of chunks that are dyed together

> ② The number of chromosomes is equal to the number of centromeres, except in the case of tetravalent chromosomes.

> ③ The short arm and long arm of a chromosome are referred to as the P arm and Q arm, respectively.

⑶ Karyotype: Refers to the shape, size and number of chromosomes. Species-specific.

<br>

![image](https://github.com/user-attachments/assets/6457d3f8-2c82-42cf-9d83-6529d39c7252)

**Figure 1.** Karyotype picture of male (Giemsa banding)

<br>

> ① Chemicals are used to halt cells in metaphase during cell division for observation.

⑷ Sister chromatids: Chromatids that are bound together at the centromere after the replication of an unreplicated chromosome.

> ① The number of sister chromosomes is the number of DNA.

> ② Centromere: Possesses a binding site for dynein motor proteins and a kinetochore.

> ③ Bivalent Chromosome (Tetrad): A chromosome containing four sister chromatids.

>> ○ In this case, four sister chromatids are attached to the centromere.

>> ○ Homologous chromosomes do not share centromeres: The reason why the number of chromosomes does not always equal the number of centromeres.

>> ○ 1 bivalent chromosome = 1 tetrad = 2 chromosomes = 4 sister chromatids.

> ④ Univalent Chromosome (Dyad): A chromosome containing two sister chromatids.

>> ○ In this case, two sister chromatids are attached to the centromere.

>> ○ 1 univalent chromosome = 1 dyad = 1 chromosome = 2 sister chromatids.

> ⑤ Nullivalent Chromosome (Monad): A chromosome containing a single chromatid, which corresponds to a double-stranded DNA.

>> ○ This is the form of chromosomes in gametes.

>> ○ 1 nullivalent chromosome = 0 chromosomes = 1 sister chromatid.

⑸ Homologous and sex chromosomes

> ① Homologous chromosome (autosome): Same size, shape, centroid position, relative locus 

> ② Sex chromosome: Different positions, genetic makeup, partial homology

> ③ Example: Humans have 22 pairs of autosomal and 1 pair of sex chromosomes.

> ④ Example: Human germ cells have 22 autosomal and 1 sex chromosomes.

⑹ Proteins involved in cell division 

> ① Condensin: Condenses chromatin into chromosomes

> ② Cohesin: Binds individual chromatids together; during metaphase, cohesin is present only near the centromere.

> ③ Shugoshin: Prevents homologous chromosomes from separating during the first meiotic division.

> ④ Securin: Inhibits separase, preventing the separation of sister chromatids until anaphase.

>> ○ Functions in mitosis, meiosis I, and meiosis II.

>> ○ APC (Anaphase Promoting Complex) inhibits securin.

> ⑤ Separase: Degrades cohesin near the centromere, allowing sister chromatids to separate during anaphase.

>> ○ Functions in mitosis, meiosis I, and meiosis II.

⑺ Mitotic spindle

> ① The spindle fibers are composed of [microtubules](https://jb243.github.io/pages/1409#4-structure-3-broad-meaning-of-cytoskeleton).  

> ② Kinetochore: A protein-bound structure on the centromere where spindle fibers attach.  

> ③ Kinetochore microtubules: Spindle fibers attached to the centromere of a chromosome.  

>> ○ Involved in the movement of chromosomes during anaphase.  

>> ○ Protein involved in microtubule attachment at the kinetochore microtubules: Ndc80

> ④ Polar spindle fibers (nonkinetochore microtubules): Spindle fibers that interact with spindle fibers from the opposite pole, contributing to cell elongation during telophase.

>> ○ Each polar spindle fiber is pushed apart by kinesins, which propel them in opposite directions.  
   
>> ○ Also referred to as interpolar mitotic spindle.  
   
>> ○ Kinesin-5: A pair of motor proteins that move toward the plus end.  
   
>> ○ Kinesin-14: Connects polar spindle fibers and moves toward the minus end.  
   
>> ○ Kinesin-4 and Kinesin-10: Move chromosomes toward the plus end.  

> ⑤ Astral microtubules: Extend toward the cell membrane, aiding in moving centrosomes to the both poles during anaphase.  
   
>> ○ Dynein: A motor protein that moves toward the minus end of astral microtubules.  

> ⑥ Chromosome-arm spindle fibers: Spindle fibers attached to the arms of chromosomes, rotating them to proper orientations during anaphase.  

⑻ Reagent

> ① Giemsa: Stains A-T rich regions on chromosomes with lower binding number, creating banding patterns.  

> ② Acetic acid alcohol: Fixes cells to prevent progression of cell division.

> ③ Hydroxyurea: Synchronizes all cells in the G1 phase. 

> ④ Acridine orange: Causes breaks in double-stranded DNA.  

> ⑤ PHA (Phytohemagglutinin): Converts leukocytes into lymphoblasts, inducing division and differentiation again; used in karyotype analysis.  

<br>

<br>

## **2. Vertical transfer of genes**: Cell division

⑴ Type of cell division

> ① Mitosis: Somatic cell division

>> ○ Occurs during embryonic development, wound healing, and the formation and growth of tissues and organs.

>> ○ Daughter cells are genetically identical to the parent cell, assuming no mutations occur.

>> ○ Organs with active mitosis: Bone marrow, blood, spleen, liver, skin.

>> ○ Organs with less active mitosis: Bladder, muscles, adipose tissue.

>> ○ The method of propagation in asexual reproduction.

> ② Meiosis: Germ cell division

>> ○ Division in diploid organisms to produce gametes required for sexual reproduction.

>> ○ The number of chromosomes is halved (reductional division), resulting in haploid (n) daughter cells.

>> ○ The method of propagation in sexual reproduction.

> ③ Number of chromosomes and sister chromosomes (based on meiosis)

<br>

![image](https://github.com/user-attachments/assets/1a11cfab-cd21-4e5e-8ceb-8fb76f3dc8f1)

**Table 1.** Number of chromosomes and sister chromosomes

<br>

⑵ Mitotic Phase and Interphase

> ① Interphase: The majority of the cell cycle

>> ○ G stands for "gap" or "growth," and S stands for "synthesis."

>> ○ G1 phase: Cell growth and proliferation of organelles.

>> ○ S phase: DNA replication, centriole replication, and histone protein synthesis.

>>> ○ 46 chromosomes, 92 sister chromatids per cell.

>>> ○ In animal cells, 2 centrioles form one centrosome, resulting in a total of 4 centrioles.

>> ○ In plant cells, a total of 2 centrioles.

>> ○ G2 phase: Synthesis of proteins required for cell division (e.g., tubulin).

>> ○ PCNA assay: Detects cells in the G1 and S phases.

> ② Mitotic Phase (M phase)

>> ○ Nuclear division: Chromosomes move to opposite poles.

>> ○ Cytokinesis: Cytoplasm divides, forming two daughter cells.

> ③ Cell fusion experiment

<br>

| Pre-Fusion Cell Cycle State |       | Post-Fusion Nuclear State       |       |
|-----------------------------|-------|---------------------------------|-------|
| **Cell A**                 | **Cell B** | **Nucleus A**                 | **Nucleus B** |
| S phase                    | G1 phase | DNA replication proceeds normally | DNA replication begins immediately |
| S phase                    | G2 phase | DNA replication proceeds normally | Remains in G2 phase until DNA replication in A is completed, followed by division |
| M phase                    | G1 phase | Division                       | Chromosomes undergo early condensation |
| M phase                    | G2 phase | Division                       | Chromosomes undergo early condensation |
| G1 phase                   | G2 phase | Proceeds normally to S phase   | Proceeds normally to M phase |

**Table 2.** Cell Fusion Experiment Results

<br>

>> ○ The S-phase cells contain substances that promote the transition of G1-phase nuclei to the S-phase.

>> ○ The S-phase cells contain substances that inhibit the transition of G2-phase nuclei to the M-phase.

>> ○ The G2-phase cells do not contain substances that promote the transition of G1-phase nuclei to the S-phase.

>> ○ The M-phase cells contain mitosis-promoting factors.

⑶ Mitosis = 1 round of nuclear division + cytokinesis

> ① GFP (Green Fluorescent Protein)

>> ○ A reporter protein that indicates the location and level of gene expression within a cell.

>> ○ Derived from jellyfish and widely used, as it allows live-cell observations.

<br>

![image](https://github.com/user-attachments/assets/5368ef52-6c42-4ee8-a6a0-013ee2faa3dd)

![image](https://github.com/user-attachments/assets/f6f68ce9-f02a-48d5-86b2-312c511782e5)

**Figure 2.** Mitosis process

<br> 

> ② Prophase: 46 chromosomes, 92 sister chromatids per cell  

>> ○ Nucleolus disappears, chromatin condenses, and chromosomes form.  

>> ○ Sister chromatids are held together by cohesin, but cohesin dissolves except at the centromeres.  

>> ○ Spindle fibers (mitotic spindle) are generated from centrosomes.  

> ③ Prometaphase: 46 chromosomes, 92 sister chromatids per cell  

>> ○ Nuclear envelope breaks down; prometaphase begins after the breakdown.  

>> ○ Spindle microtubules reach the chromosomes.  

>>> ○ The nuclear envelope must dissolve for the spindle fibers to attach to microtubules.  

>> ○ Chromosome movement occurs.  

>>> ○ Spindle fibers must attach to microtubules to enable chromosome movement.  

> ④ Metaphase: 46 chromosomes, 92 sister chromatids per cell  

>> ○ The spindle apparatus is fully formed.  

>> ○ All chromosomes align along the equatorial plate of the cell.  

>> ○ The centromeres of the two sister chromatids face opposite poles.

> ⑤ Anaphase: 92 chromosomes, 92 sister chromatids per cell  

>> ○ Separase cleaves the centromeres.  

>> ○ Sister chromatids move toward opposite poles along the kinetochore mitotic spindles using dynein.  

>> ○ Kinetochore microtubules: Due to the characteristics of tubulin, the (-) end undergoes microtubule disassembly, while rapid disassembly occurs at the (+) end facilitated by dynein.

>> ○ As the length of the polar spindle fibers continues to increase, the cell elongates.

> ⑥ Telophase: 92 chromosomes, 92 sister chromatids per cell  

>> ○ Cell elongation continues.  

>> ○ A nuclear envelope forms around chromosomes at each pole, re-establishing the nucleus.  

>> ○ Nucleoli reappear, and spindle fibers disappear.  

>> ○ Chromosomes decondense into chromatin.  

> ⑦ Cytokinesis: Cytokinesis occurs independently of nuclear division.  

<br>

![image](https://github.com/user-attachments/assets/79a3485b-effd-48b4-acde-e852ee05bdbb)

**Figure 3.** The process of cytokinesis

<br>

>> ○ The cytoplasm divides into two halves.

>> ○ Cytokinesis in plant cells: Centrifugal division.  

>>> ○ Golgi-derived vesicles secrete cell plate components at the metaphase plate position to form the cell plate.

>>> ○ Cell plate: Composed of pectin, a non-cellulosic carbohydrate fiber.  

>>> ○ Vesicles move along the phragmoplast, which consists of microtubules, microfilaments, and parts of the endoplasmic reticulum.  

>> ○ Cytokinesis in animal cells: Centripetal division.  

>>> ○ 1<sup>st</sup>. Condensation of a microfilament ring between the two daughter cells.  

>>> ○ 2<sup>nd</sup>. Formation of the cleavage furrow.  

>>> ○ 3<sup>rd</sup>. Cytoplasm separation from the outside inward.  

>>> ○ This contractile ring is regulated by RhoA.  

>> ⑧ End of cell division: 46 chromosomes and 46 sister chromatids per cell.  

>> ⑨ Substances influencing cell division

>>> ○ Mitogen: A substance that induces DNA synthesis and cell division.  

>>> ○ Cell division inhibitors: Refer to "induced mutations" below.  

> ⑩ Marker genes for each phase

>> ○  G1/S phase: _CEP57_, _CDCA7L_

>> ○ S phase: _ABHD10_, _CCDC14_, _CDKN2AIP_, _NT5DC1_, _SVIP_, _PTAR1_ 

>> ○ G2 phase: _ANKRD36C_, _YEATS4_, _DCTPP1_  

>> ○ G2/M phase: _SMC4_, _TMPO_, _LMNB1_, _HINT3_  

>> ○ M phase: _HMG20B_, _HMGB3_, _HPS4_  

>> ○ G0 phase: _CDKN1A_, _CDKN1B_, _CDKN1C_  

⑷ Meiosis = First meiotic division (separation of homologous chromosomes) + Second meiotic division (mitosis of haploid cells).  

<br>

![image](https://github.com/user-attachments/assets/37e083b6-3fdd-4e9d-a1a8-5f88c8852ac0)

![image](https://github.com/user-attachments/assets/45ebd8a0-d61c-453b-a55e-91628610f044)
  
**Figure 4.** Meiosis process

<br> 

> ① Prophase I: 46 chromosomes, 92 sister chromatids per cell  

>> ○ 1<sup>st</sup>. Chromosomes condense and become more compact; the nuclear envelope disappears.  

>> ○ 2<sup>nd</sup>. Homologous chromosomes are loosely connected through crossing over between them.

>> ○ 3<sup>rd</sup>. Synapsis: Homologous chromosomes are tightly connected via the synaptonemal complex (SP).

>> ○ 4<sup>th</sup>. Genetic information is exchanged between homologous chromosome pairs through crossing over.  

> ② Metaphase I: 46 chromosomes, 92 sister chromatids per cell  

>> ○ Tetrads align along the equatorial plate.  

> ③ Anaphase I: 46 chromosomes, 92 sister chromatids per cell  

>> ○ Tetrads are separated into pairs of homologous chromosomes (dyads), which migrate to opposite poles.  

>> ○ Tetrad: 2 chromosomes

>> ○ Dyad: 1 chromosome

> ④ Telophase I + Cytokinesis: 46 chromosomes, 92 sister chromatids per cell  

>> ○ In some organisms, the nuclear envelope reforms, but not in humans.  

>> ○ Each nucleus becomes haploid (n).  

> ⑤ End of Meiosis I: 23 chromosomes, 46 sister chromatids per cell  

> ⑥ Prophase II: 23 chromosomes, 46 sister chromatids per cell  

>> ○ Meiosis II begins without an interphase (e.g., no DNA replication).  

>> ○ Chromosomes condense and become compact.  

>> ○ If the nuclear envelope is formed, it disappears again.  

> ⑦ Metaphase II: 23 chromosomes, 46 sister chromatids per cell  

>> ○ Dyads align along the equatorial plate.  

> ⑧ Anaphase II: 46 chromosomes, 46 sister chromatids per cell  

>> ○ Sister chromatids are separated.  

>> ○ A pair of homologous chromosomes moves toward opposite poles.

> ⑨ Telophase II: 46 chromosomes, 46 sister chromatids per cell  

>> ○ Chromosomes reach each pole, and the nuclear envelope forms around the chromosomes.  

> ⑩ End of Meiosis II: 23 chromosomes, 23 sister chromatids per cell  

> ⑪ Crossing Over

>> ○ Definition: The phenomenon of partial genetic exchange occurring between chromatids of homologous chromosomes, resulting in the formation of new types of chromosomes.

>> ○ Chiasma: The site where crossing over occurs.  

>> ○ Crossover Rate: Classical genetics assumes that the likelihood of crossing over is proportional to the distance between genes.  

>> ○ Approximately 30 crossovers occur during one meiotic division.  

> ⑫ Random Alignment

>> ○ During meiosis I, the independent alignment and separation of homologous chromosome pairs result in diverse gametes.  

> ⑬ Genetic Diversity in Humans

>> ○ Chromosomal diversity in gametes produced from meiosis I: 2<sup>23</sup> + crossing over.  

>> ○ Chromosomal diversity in zygotes: (2<sup>23</sup> + crossing over) × (2<sup>23</sup> + crossing over).  

<br>

<br>

## 3. Horizontal transfer of genes

⑴ [Virus](https://jb243.github.io/pages/554)

⑵ [Bacterial Recombination](https://jb243.github.io/pages/554)

⑶ [Mobile DNA](https://jb243.github.io/pages/554)

⑷ [Plasmodesmata](https://jb243.github.io/pages/554)

<br>

<br>

## **4. Cell cycle regulation**

⑴ Regulatory proteins controlling the mitotic phase

> ① Cyclin: Its levels fluctuate throughout the cell cycle, and all cyclins associated with a specific stage are degraded once that stage is completed.

>> ○ Degraded by the proteasome.  

> ② CDK (Cyclin-Dependent Kinase): Protein levels remain constant throughout the cell cycle, but activity changes depending on cyclin levels.  

>> ○ CDK1: Regulates the M phase.  

>> ○ CDK2 ~ 8: Regulate the G1 and S phases.  

> ③ MPF (M-phase Promoting Factor)  

>> ○ Refers to CDK1 and cyclin B.

>> ○ Present in the cytoplasm of dividing mammalian cells.  

>> ○ Functions: Chromosome condensation, spindle elongation, and nuclear envelope breakdown.  

>> ○ When spindles attach to kinetochores, APC is activated, which inhibits MPF.  

> ④ CDC (Cell Division Cycle): Substances in yeast that function similarly to MPF.  

>> ○ CDC2: Regulates the M phase.  

>> ○ CDC2 kinase: Regulates the M phase.  

>> ○ Other CDC kinases: Regulate G and S phases.  

>> ○ CDI (CDK Inhibitor): Inhibits CDK activation by acting on CDK.  

> ⑤ APC (Anaphase Promoting Complex)  

>> ○ APC activation during anaphase inhibits securin, thus promotes separase.  

>> ○ Mechanism: APC attaches ubiquitin to securin, leading to the degradation of securin.

>> ○ APC activation inhibits MPF.  

>> ○ When APC is activated, it works with cdc20 to promote the degradation of M cyclin via the proteasome.

>> ○ Plays a critical role in tumor suppression.  

⑵ Checkpoint 

> ① The cell cycle progresses only after passing the checkpoint.

> ② Checkpoints: A total of five checkpoints exist.  

> ③ G1 Checkpoint: The point for transitioning to the S phase.  

>> ○ Factors: Necessity for cell division, growth factors, cell size, sufficient nutrients, and DNA damage.  

>> ○ Mechanism: CDK4, 6 + Cyclin D

>> ○ If this checkpoint is not passed, cells remain in the non-dividing G<sub>0</sub> phase (e.g., nerve and muscle cells).  

>> ○ CDK4/6 inhibitors: Palbociclib, Abemaciclib, Ribociclib

> ④ G1-S boundary

>> ○ Mechanism: CDK2 + Cyclin E

> ⑤ S checkpoint

>> ○ Mechanism: CDK2, 1 + Cyclin A

> ⑥ G2 checkpoint

>> ○ Factors: Accuracy of DNA replication and cell size 

>> ○ Mechanism: MPF (CDK1 + Cyclin B) → Promotes DNA replication

> ⑦ M checkpoint: Metaphase checkpoint

>> ○ Factors: Attachment of microtubules to all chromosomes 

>> ○ Reason why hybrid organisms cannot reproduce

⑶ G1 checkpoint mechanism: Promotion of transition to the S phase (e.g., PDGF from platelets).

> ① 1<sup>st</sup>. Growth factor → Ras

> ② 2<sup>nd</sup>. Signaling

>> ○ 2<sup>nd</sup>\-1<sup>st</sup>. Ras: GDP → GTP

>> ○ 2<sup>nd</sup>\-2<sup>nd</sup>. GTP → Raf

>> ○ 2<sup>nd</sup>\-3<sup>rd</sup>. Raf → MEK

>> ○ 2<sup>nd</sup>\-4<sup>th</sup>. MEK → MAP kinase

>> ○ 2<sup>nd</sup>\-5<sup>th</sup>. MAP kinase moves inside the nucleus.

> ③ 3<sup>rd</sup>. Synthesis of cyclin D and cyclin E 

>> ○ 3<sup>rd</sup>\-1<sup>st</sup>. MAP kinase → Myc

>> ○ 3<sup>rd</sup>\-2<sup>nd</sup>. Myc → Transcription of Cyclin D and Cyclin E genes.

⑷ G1 Checkpoint Mechanism: Inhibition of S phase transition. p53 is primarily involved.  

> ① Since the G1 phase is the starting point of all cell cycles, multiple inhibitory mechanisms exist.  

> ② 1<sup>st</sup>. UV radiation → Mdm protein is removed from p53, converting it to p21.  

> ③ 2<sup>nd</sup> - 1<sup>st</sup>: p21 inhibits CDK4-Cyclin D and CDK2-Cyclin E, thereby preventing the S phase transition (e.g., inhibition of E2F).  

> ④ 2<sup>nd</sup> - 2<sup>nd</sup>: If cell damage is severe, p21 activates Bax → Cytochrome C is released into the cytoplasm → Caspase activation → Apoptosis.  

>> ○ Cytochrome C: Involved in the [mitochondrial electron transport chain](https://jb243.github.io/pages/70#3-cellular-respiration) and evolutionarily conserved. Located in the intermembrane space and attached to the inner membrane.  

⑸ S checkpoint mechanism: G1 Cdk inactivates Rb, enabling the S phase to proceed.

> ① 1<sup>st</sup>. CDK4-Cyclin D and CDK2-Cyclin E increase the concentration of the transcription factor E2F.

> ② 2<sup>nd</sup>. Rb binds to E2F → Cyclin-CDK complexes are inactivated.

>> ○ Rb protein: Suppresses E2F.

> ③ 3<sup>rd</sup>. Cyclin-CDK complexes use 2 ATP to phosphorylate Rb.

> ④ 4<sup>th</sup>. Phosphorylated Rb detaches from E2F.

> ⑤ 5<sup>th</sup>. E2F translocates to the nucleus → Promotes the synthesis of Cyclin A.

> ⑥ 6<sup>th</sup>. CDK2 binds with Cyclin A → Acts as a transcription factor to complete the S phase and transcribe the Cyclin B gene.

⑹ G2 checkpoint mechanism (yeast)

> ① 1<sup>st</sup>. CDK1 binds with Cyclin B (M Cyclin) to form an inactive Cyclin-CDK complex.

> ② 2<sup>nd</sup>. The Cyclin-CDK complex is phosphorylated and inhibited by WEE1.

>> ○ WEE1: A type of kinase.

> ③ 3<sup>rd</sup>. The complex is phosphorylated and activated by CAK (CDK Activating Kinase), but its activity is suppressed by WEE1.

> ④ 4<sup>th</sup>. CDC25 removes the phosphate group from the Cyclin-CDK complex, allowing the activation of cyclin-CDK complex by CAK.

>> ○ CDC25: A type of phosphatase.

> ⑤ 5<sup>th</sup>. The activated Cyclin-CDK complex further activates CDC25 and inactivates WEE1.

⑺ Cell division inhibitors ([ref](http://www.mycancergenome.org/content/drugs/#chemotherapies))

> ① M-phase specific inhibitors

>> ○ Antimicrotubule agents: Epothilones, Halichondrin B analogues, Taxanes, Vinca alkaloids

>> ○ Topoisomerase II inhibitors: Anthracemedione, Anthracycline, Epipodophyllotoxin

> ② S-phase specific inhibitors

>> ○ DNA Antimetabolites: Folate antagonists, Purine analogues, Pyrimidine analogues, Hydroxyurea

>> ○ Anti-folate Agent: [Methotrexate](https://jb243.github.io/pages/69).  

>> ○ Topoisomerase II inhibitors: Anthracemedione, Anthracycline, Epipodophyllotoxin

> ③ Inhibitors affecting multiple stages of cell division

>> ○ Antitumor antibiotics: Bleomycin, Dactinomycin, Mitomycin

> ④ Inhibitors unrelated to specific cell division stages

>> ○ Alkylating agents: Alkyl sulfonates, Ethylenimines, Nitrogen mustards, Nitrosoureas, Platinum analogues, Triazenes

<br>

<br>

## **5. Mutation**

⑴ **Cause 1.** Spontaneous mutation

> ① Tautomerization: Formation of isomers leading to bonds with different bases.

> ② Deamination process

>> ○ [Types of nucleotides](https://jb243.github.io/pages/1395) and their structures

>> ○ Amino bases: Bases with amino groups (-NH<sub>2</sub>) such as A, C.

>> ○ Keto bases: Bases with keto groups (C=O) such as T, G, U, I.

>> ○ Deamination of cytosine (C): Conversion of C base to uracil (U) and then to thymine (T), ultimately changing G≡C to A=T.

>> ○ Deamination of adenine (A): Conversion of A base to hypoxanthine, leading to inosine acid (I), ultimately changing A=T to G≡C.

> ③ Depurination: The bond between purines and the carbon backbone is relatively weak.

> ④ Repeat sequences (e.g., CAG repeats and [Huntington's disease](https://jb243.github.io/pages/72))

> ⑤ Replication errors: Mutation rates during DNA replication are similar in E. coli and eukaryotes at about 1 mutation per 10<sup>10</sup> nucleotides.

>> ○ When comparing two populations, A and B, the statement that the mutation frequency is higher or lower is almost always incorrect.

⑵ **Cause 2.** Induced mutation

> ① DNA-modifying agents

>> ○ Nitrous acid: Induces deamination reactions.

>> ○ Aflatoxin: A fungal toxin that modifies G bases, leading to liver cancer.

> ② Alkylating agents: Substances that donate CH<sub>3</sub> or CH<sub>3</sub>CH<sub>2</sub> to the amino or keto groups of nucleotides.

>> ○ EMS (ethyl methyl sulfonate): A representative alkylating agent.

>> ○ Nitrogen mustard: Includes bendamustine, cyclophosphamide, estramustine, ifosfamide, mechlorethamine, melphalan

>> ○ Platinum analogs: Includes carboplatin, cisplatin, oxaliplatin, etc.

>> ○ Triazene: Includes dacarbazine, procarbazine, temozolomide, etc.

>> ○ Antidote: MGMT (O6-methylguanine-DNA methyltransferase): Antidote for alkylating agents like EMS.

> ③ DNA analogs

>> ○ 2-AP (2-amino purine): Forms base pairs with thymine or cytosine.

>> ○ 5-BU (5-bromouracil): Thymine analog; forms base pairs with adenine or guanine.

>> ○ BrdU (bromodeoxyuridine): Thymidine analog used for labeling S-phase cells.

>> ○ Hydroxylamine: Adds hydroxyl groups to cytosine, enabling it to bind with adenine.

>> ○ ENU (N-ethyl N-nitrosourea): Induces random point mutation.

>> ○ 5-FU (5-fluorouracil): Uracil analog

>>> ○ Fluorine replaces hydrogen at the 5th carbon of uracil.

>>> ○ 5-FU (5-fluorouracil) is converted to DHFU (dihydrofluorouracil) by DPD (dihydropyrimidine dehydrogenase).

>>> ○ 80% of administered 5-FU is broken down in the liver by DPD.

>> ○ 6-MP (6-mercaptopurine)

>> ○ Capecitabine (Xeloda®), Cytarabine (Ara-C®), Floxuridine, Fludarabine

>> ○ Gemcitabine (Gemzar®)

>>> ○ A first-line drug for pancreatic cancer.

>>> ○ Pyrimidine antimetabolite

>> ○ Hydroxycarbamide

>> ○ Pemetrexed (Alimta®), phototrexate

> ④ Antifolates

>> ○ [Methotrexate](https://jb243.github.io/pages/69)

> ⑤ Insert material

>> ○ Acridine orange: A flat ring molecule that intercalates between base pairs, causing frameshift mutations.

>> ○ EtBr: Intercalates between base pairs, causing frameshift mutations.

>> ○ Benzopyrene: Inserted between base pairs in the major groove of DNA → lung cancer

>> ○ MCA (methylcholanthrene): A chemical mutagen

> ⑥ UV irradiation: Induces pyrimidine dimers, leading to deletion mutations during replication, and causes skin cancer

> ⑦ Radiation: Free radicals generated by radiation attack bases or the sugar-phosphate backbone of DNA.

⑶ **Type 1.** Point Mutations: Gene-level mutations

> ① Base substitution mutations

>> ○ Definition: Mutations caused by the replacement of one base of DNA with another.

>> ○ Classification based on phenotype

>>> ○ Silent mutation: Due to the redundancy of the genetic code, a base substitution does not alter the encoded amino acid, resulting in no change in phenotype.

>>> ○ Missense mutation: Base substitution changes the codon to encode a different amino acid.

>>> ○ Nonsense mutation: Codon changes to a stop codon, resulting in premature termination.

>>> ○ Neutral mutation: Changes to a similar amino acid with no significant phenotypic change.

>>> ○ Mutations preventing transcription termination.

>>> ○ Translation of stop codon into a meaningful amino acid, potentially elongating the protein.

>>> ○ Point mutations in non-coding regions of DNA: Reduces splicing efficiency of introns.

>>> ○ Mutations in promoters: Prevent transcription.

>> ○ Types of base substitution mutations

>>> ○ Transition mutation: A mutation where a purine is replaced by another purine, or a pyrimidine is replaced by another pyrimidine.

>>> ○ Transversion mutation: A mutation where a purine is replaced by a pyrimidine, or a pyrimidine is replaced by a purine.

>>> ○ Transition mutations are less likely to result in amino acid changes and have a higher probability of reverting to the original state through an additional mutation at the same site.

<br>

![image](https://github.com/user-attachments/assets/2b15eff2-1a81-43d5-bebd-d3c6341cbf67)

**Figure 5.** Transition (㉠) and transversion (㉡) mutations

<br>

>>> ○ Whole genome: Transition mutation / Transversion mutation ≃ 2

>>> ○ Protein-coding gene: Transition mutation / Transversion mutation > 3

> ② Frameshift mutations

>> ○ Mutations caused by insertions or deletions of bases, leading to a shift in the reading frame.

> ③ Reversion mutations

>> ○ Reapplication of the same mutagen restores normal offspring from mutated organisms.

>> ○ Typically limited to base substitution mutations.

⑷ **Type 2.** Chromosomal mutations

> ① Deletion

>> ○ Deletion loop: Used in creating [genome maps](https://jb243.github.io/pages/75).

>> ○ Nearby genes are often deleted together.

> ② Duplication

>> ○ Compensation loop

> ③ Inversion

>> ○ Paracentric inversion: Results in 50% normal, 25% acentric, and 25% dicentric chromosomes.

>> ○ Pericentric inversion: Results in 100% normal chromosomes.

> ④ Translocation

>> ○ Reciprocal translocation: Exchange of genetic material between two chromosomes, forming a cross-shaped translocation.

>>> ○ [Holliday Model](https://jb243.github.io/pages/1460)

>>> ○ Example: Chromosome 8 → Chromosome 14

>> ○ Non-reciprocal translocation: Genetic material is transferred unidirectionally from one chromosome to another.

>> ○ **Example 1.** Robertsonian translocation: Causes familial Down syndrome.

>>> ○ Found in 1/1000 births.

>>> ○ 1<sup>st</sup>. Long arms of chromosomes 14 and 21 fuse: (14, 14), (21, 21) → (14, 14-21), (21, ×)

>>>> ○ The centromere in 14-21 is derived from chromosome 14, so it behaves like chromosome 14.

>>> ○ 2<sup>nd</sup>. Short arms of chromosomes 14 and 21 are lost.

>>> ○ 3<sup>rd</sup>. Gametes: <span>(14, 21) || (14, ×) || (14-21, 21) || (14-21, ×)</span>

>>> ○ 4<sup>th</sup>. Normal gametes (14, 21) and resulting outcomes

>>>> ○ (14, 14), (21, 21): Normal, 25%

>>>> ○ (14, 14), (21, ×): Miscarriage, 25%

>>>> ○ (14, 14-21), (21, 21): Familial Down syndrome, capable of reproduction unlike typical Down syndrome.

>>>> ○ (14, 14-21), (21, ×): Inherited condition, 25%

<br>

![image](https://github.com/user-attachments/assets/5fcdf90b-867e-4a3f-8bde-400207d763e3)

**Figure 6.** Robertsonian translocation

A translocation occurs between non-homologous chromosomes 14 and 21.

<br>

>> ○ **Example 2.** Chronic Myelogenous Leukemia (CML): Reciprocal translocation

>>> ○ abl gene: Encodes tyrosine kinase.

>>> ○ bcr gene: Highly active due to the presence of promoter.

>>> ○ Philadelphia chromosome: Recombinant chromosome containing abl and bcr genes.

<br>

![image](https://github.com/user-attachments/assets/7eab3400-520b-4e8f-a747-5d3c3c07322c)

**Figure 7.** Chronic myelogenous leukemia

<br>

⑸ **Type 3.** **Chromosomal Non-disjunction**

> ① Chromosomal non-disjunction: Causes aneuploidy in autosomes and sex chromosomes.

<br>

![image](https://github.com/user-attachments/assets/7bbe6946-7c05-4148-ba22-d9049f30d558)

**Figure 8.** Non-disjunction Process

<br>

>> ○ Non-disjunction in Meiosis I: (n+1, n+1, n-1, n-1). No normal gametes. No duplicated genes in all gametes.

>> ○ Non-disjunction in Meiosis II: (n+1, n-1, n, n). 50% normal gametes. Duplicated genes present in n+1 gametes.

> ② Autosomal aneuploidy: Gain or loss of one or two chromosomes.

>> ○ Nullisomy: Loss of a homologous pair; lethal before implantation.

>> ○ Monosomy: Loss of a single chromosome; embryonic lethal.

>> ○ Trisomy: Embryonic or fetal lethal.

>> ○ Patau syndrome (Trisomy of chromosome 13)

>> ○ Edward syndrome (Trisomy of chromosome 18)

>> ○ Down syndrome (Trisomy of chromosome 21): Occurs in 1/750 births; characterized by distinctive facial features, intellectual disability, short stature, heart defects, shorter lifespan, and susceptibility to respiratory diseases. Frequency increases with maternal age.

>> ○ Haplo IV: Chromosome 4 deletion in Drosophila

> ③ Sex chromosomal aneuploidy: Having one or two extra or missing chromosomes. The effects are not severe, and normal lifespan is maintained.

>> ○ Reason: The Y chromosome contains a small amount of genetic information, and only one X chromosome is active (the other forms a Barr body).

>> ○ XXX (Triple X syndrome): 1/1000 births; normal phenotype with slightly increased risk of intellectual disability.

>> ○ XYY (Double Y syndrome): 1/2000 births; tendency for tall stature, normal phenotype with mild intellectual disability, predisposition to petty thief.

>> ○ XXY, XXXY (Klinefelter syndrome): 1/2000 births; males with tall stature, sexual immaturity, small testes, infertility, and intellectual disability.

>> ○ XO (Turner syndrome): 1/5000 births; 99% spontaneous abortion; short stature and incomplete sexual maturation.

>> ○ XYY, XYYY (Jacob's syndrome)

> ④ Polyploidy: A mutation where the number of chromosomes increases as a multiple of n.

>> ○ Autopolyploidy: Occurs when a 2n (plant) organism becomes 4n, then reproduces with a 2n organism of the same species to produce a 3n organism.  

>>> ○ Colchicine: A spindle fiber polymerization inhibitor that binds to tubulin dimers, halting cell division at metaphase. 

>>> ○ Colchicine is frequently used to induce autopolyploidy for crop improvement purposes.  

>>> ○ Taxol: A spindle fiber depolymerization inhibitor used as an anticancer drug.

>> ○ Allopolyploidy: Polyploidy resulting from hybridization between closely related species (plants).

>>> ○ Mechanism of evolution.  

>>> ○ 1<sup>st</sup>. An AB organism is formed through hybridization between an AA organism and a BB organism.

>>> ○ 2<sup>nd</sup>. If the AB organism has an odd number of chromosomes, it is infertile. Fertility is restored through polyploidy, resulting in an AABB organism.

>>> ○ 3<sup>rd</sup>. Fertile AABB organisms reproduce to create more fertile AABB organisms.  

>> ○ **Example 1.** Seedless watermelon (3n).  

>> ○ **Example 2.** Triploids (3n) occur in 1–3% of all pregnancies → Most are not viable for birth or survival.  

⑹ **Type 4.** Temperature-sensitive mutations

> ① Permissive temperature

> ② Restrictive temperature

⑺ **Type 5.** [Transposons](https://jb243.github.io/pages/554) and [retrotransposons](https://jb243.github.io/pages/554)

<br>

<br>

## **6. Cancer**

⑴ Definition

> ① Tumor: A mass of non-functional cells formed through uncontrolled cell division.

> ② Benign tumor: A tumor that does not invade surrounding tissues (≠ cancer).

> ③ Malignant tumor: A tumor that invades surrounding tissues (cancer).

> ④ Metastasis: When cancer cells spread and form new tumors in other locations.

> ⑤ Normal cells → Potential tumor cells (divide without signals) → Benign tumor → Malignant tumor → Metastasis

> ⑥ Oncology: The study of cancer.

⑵ Characteristics of cancer cells

> ① Normal cells

>> ○ Divide in response to signals and survive only as long as necessary.

>> ○ Perform specialized differentiation.

>> ○ Remain in their designated location due to cell-cell and cell-matrix adhesion. Movement and invasion into other areas are prohibited.

> ② Cancer cells

>> ○ Lose normal regulatory functions due to genetic mutations, exhibiting abnormal behavior.

>> ○ **Step 1.** Unlimited division potential → Benign tumor

>>> ○ Cancer cells have significantly larger nuclei compared to normal cells.

>> ○ **Step 2.** Invasion to other cell territories → Metastasis

>>> ○ Spread through lymphatic or blood vessels to other parts of the body.

>>> ○ Cancer as small as 2–3 mm can metastasize.

>> ○ **Step 3.** Formation of secondary tumors → Malignant tumor

>> ○ **Step 4.** Metastatic cancer cells circulate throughout the body via the circulatory or lymphatic systems.

> ③ Distribution of cancer cells

>> ○ 4–5% of cancer cells are stem cells.

>> ○ Normal immune systems can destroy up to 10 million tumor cells.

>> ○ Clinically detectable cancer contains at least 1 billion tumor cells.

⑶ Characteristics of Cancerous Tissues

> ① Formation of cancerous tissues

>> ○ Takes 5–10 years for cancer tissue to grow to 5 mm.

>> ○ Faster cancer growth in younger patients.

> ② **Feature 1.** Angiogenesis begins when cancer tissue exceeds 2 mm<sup>3</sup>.

> ③ **Feature 2.** Cancer tissues are more acidic (pH ≤ 7.2) than normal tissues (pH 7.4) due to active glycolysis.

> ④ **Feature 3.** Cancer tissues are slightly warmer than normal tissues due to inefficient heat dissipation from poorly formed blood vessels.

> ⑤ **Feature 4.** Warburg effect (aerobic glycolysis): Increased glycolysis and lactate production in cancer cells regardless of oxygen levels.

> ⑥ **Feature 5.** Metastasis: Cancer cells lose intercellular adhesion due to reduced expression of E/P-Cadherin, ultimately leading to increased metastasis.

> ⑦ **Feature 6.** Hypercalcemia: Elevated blood calcium levels causing lethargy due to sodium channel closure.

⑷ Causes of cancer

> ① Cancer risk factors: Factors that increase the likelihood of cancer development. Approximately 80-90% of all cancers are caused by lifestyle and environmental factors.

>> ○ Smoking: Cigarettes contain various carcinogens and increase the production of free radicals, leading to DNA damage.

>> ○ High-fat, low-fiber diet: Poor dietary habits contribute to cancer risk.  

>> ○ Lack of exercise: Weakens the immune system and increases the risk of cancer associated with obesity.  

>> ○ Alcohol consumption: Alcohol combined with smoking multiplies cancer risk.  

>> ○ Aging: The immune system deteriorates, and mutations accumulate over time.  

>> ○ Frequent cell division: Leads to an increased chance of mutations.  

>>> ○ Damage from ovarian ovulation and subsequent repair processes.  

>>> ○ Damage from epithelial movement in the digestive tract and its repair.  

>> ○ Exposure to radiation and high-voltage lines: Increases the frequency of cancers such as leukemia.

> ② **Gene Group 1.** Proto-oncogenes

>> ○ Genes that regulate the cell cycle, often encoding growth factors or receptor proteins.

>> ○ Dominant mutations: Mutation in one homologous chromosome is sufficient for cancerous effects.

>> **○ Example 1.** EGFR (epidermal growth factor receptor)

>>> ○ Also called ErbB1 or HER-1.

>>> ○ When EGF binds to its receptor, it induces a structural change that activates the kinase domain.

>>> ○ SH2 domain recognizes tyrosine sequences.

>>> ○ Inhibitors

>>>> ○ Erlotinib

>>>> ○ Gefitinib (Iressa): FDA-approved EGFR tyrosine kinase inhibitor (EGFR-TKI).

>>>> ○ Cetuximab (Erbitux): FDA-approved antibody.

>>>>> ○ IgG1 isotype

>>>>> ○ Binding site: Q384, Q408, H409, K443, K465, I467, S468, F352, D355, P387

>>>>> ○ Has an immunogenic activity.

>>>> ○ Panitumumab (Vectibix): FDA-approved antibody.

>>>>> ○ IgG2 isotype

>>>>> ○ Binding site: P349, P362, D355, F412, I438

>>>>> ○ No immunogenic activity.

>>>> ○ Lapatinib (Tyverb): Directly inhibits downstream signaling by targeting the tyrosine kinase domain.

>>>> ○ Afatinib: Directly inhibits downstream signaling by targeting the tyrosine kinase domain.

>>>> ○ Sapitinib

>> **○ Example 2.** HER2 (human epidermal growth factor receptor type 2): Also called ErbB-2.

>>> ○ Structure: Extracellular environment - I - II - III - IV - Transmembrane region - Tyrosine kinase domain - Intracellular environment

>>>> ○ Domain II: Dimerization domain

>>> ○ Function

>>>> ○ When HER2 is overexpressed, autophosphorylation occurs, leading to excessive cell proliferation, aggressiveness, and tumorigenesis.

>>>> ○ Found to be overexpressed in 15–40% of ovarian cancer cases.

>>> ○ Mechanism: Utilizes PI3-kinase signaling.

>>> ○ Inhibitors

>>>> ○ Pertuzumab: Blocks domain II to inhibit receptor dimerization; FDA-approved.

>>>> ○ Trastuzumab, Margetuximab: Blocks domain IV; involved in ADCC (antibody-dependent cellular cytotoxicity); FDA-approved.

>>>> ○ Trastuzumab emtansine (T-DM1), Trastuzumab deruxtecan: Trastuzumab conjugated with anticancer drugs; FDA-approved.

>>>> ○ Lapatinib (Tyverb): Directly inhibits downstream signaling by targeting the tyrosine kinase domain; FDA-approved.

>>>> ○ Afatinib: Directly inhibits downstream signaling by targeting the tyrosine kinase domain; FDA-approved.

>>>> ○ Neratinib: Directly inhibits downstream signaling by targeting the tyrosine kinase domain; FDA-approved.

>>>> ○ Sapitinib

>>>> ○ CI-1033

>> **○ Example 3.** HER3: Also called ErbB-3.

>>> ○ Inhibitor: Sapitinib

>> **○ Example 4.** IGF-1R (insulin-like growth factor 1 receptor)

>>> ○ Structure

>>>> ○ Extracellular ligand-binding domain: Contains two α subunits.

>>>> ○ Transmembrane domain

>>>> ○ Cytoplasmic domain: Contains two β subunits.

>>>> ○ A total of three disulfide bonds are present: α-α, α-β, α-β.

>>>> ○ The β subunits function as tyrosine kinases.

>>> ○ IGF-1R binds IGF-1 and IGF-2 ligands.

>>> ○ Upon activation, IGF-1R triggers the following signaling pathways:

>>>> ○ STAT3

>>>> ○ PI3K → AKT (inhibited by PTE) → mTOR → S6K

>>>> ○ GRB2 → Ras / Raf → MEK → MAPK

>>> ○ IGF-1R induces malignant tumors.

>>> ○ Inhibitors: AXL1717, Linsitinib (in Phase III trials)

>> **○ Example 5.** IR (insulin receptor)

>> **○ Example 6.** Ras (rat sarcoma)

>>> ○ A GPCR activated by tyrosine kinase, involved in signal transduction pathways.

>>> ○ Plays a key role in MAPK activation.

>>> ○ Normal function: Regulates cell proliferation, differentiation, and survival.

>>> ○ Types of Ras include H-Ras, K-Ras, and N-Ras.

>>> ○ Loss of GTPase-activating protein function or accumulation of degradation-resistant proteins leads to excessive Ras activation.

>>> ○ This results in growth factor-independent cell division, leading to transformation into cancer cells.

>>> ○ Most cases of growth factor-independent cell division in cancer are caused by Ras.

>> **○ Example 7.** myc

>>> ○ **Type 1.** c-myc (MYC)

>>>> ○ The first discovered myc, as it is similar to v-myc found in viruses.

>>>> ○ c-myc expression is generally elevated in tumors.

>>>> ○ Located on chromosome 8, and it is known to regulate the expression of 15% of all genes.

>>>> ○ The c-myc gene recombines with chromosome 14, which encodes antibody genes → increased expression → leads to Burkitt's lymphoma.

>>> ○ **Type 2.** l-myc (MYCL)

>>> ○ **Type 3.** n-myc (MYCN)

>> **○ Example 8.** abl-bcr

>> **○ Example 9.** c-erbB, c-mx, c-kit, RARa, Eb, Cyclin, CDK 2, 4

>> **○ Example 10.** src

>> **○ Example 11.** β-catenin

>> **○ Example 12.** BRAF

>>> ○ At the 600th amino acid, a substitution from valine (V) to glutamic acid (E) converts it into an oncogene.

>>> ○ Once it becomes an oncogene, it continuously activates kinases.

>>> ○ Frequency of B-RAF mutations by tumor type: melanoma (60%), colon (15%), ovarian (30%), thyroid (30%).

>>> ○ Inhibitors: dabrafenib, vemurafenib (Zelboraf), PLX4720.

>> **○ Example 13.** MDM2, MITF, PPM1D, KAT7

>> **○ Example 14.** YAP (Yes-associated protein)

>> **○ Example 15.** ALK

> ③ **Gene Group 2.** Oncogenes: Mutations in proto-oncogenes cause DNA alterations.

>> ○ When a proto-oncogene mutates into an oncogene, it excessively promotes cell division even in the absence of growth factors.  

>> ○ **Mechanism 1.** Hyperactivation due to structural or functional changes in encoded proteins caused by DNA base mutations.  

>> ○ **Mechanism 2.** Overexpression due to gene rearrangement (e.g., when a gene is positioned directly in front of a promoter).  

>> ○ **Mechanism 3.** Overexpression caused by an increase in the copy number of gene.  

>> ○ **Example 1.** When a mutation occurs in a proto-oncogene.  

>> ○ **Example 2.** When a virus behaves like an oncogene.  

<br>

| Virus Type                     | Virus Family                          | Related Cancer Diseases                                    |
|--------------------------------|---------------------------------------|----------------------------------------------------------|
| HTLV-I (human T-cell leukemia/lymphoma virus) | Retrovirus (RNA virus)               | T-cell leukemia/lymphoma                                 |
| Hepatitis B                    | Hepadnavirus (hepatotropic DNA virus) | Hepatocellular carcinoma                                 |
| Hepatitis C                    | Hepadnavirus                          | Hepatocellular carcinoma                                 |
| Epstein-Barr                   | Herpesvirus (DNA virus)               | Nasopharyngeal carcinoma, Burkitt lymphoma, immunoblastic lymphoma, Hodgkin disease |
| HHV-8 (KSHV) (human herpesvirus-8) (Kaposi sarcoma herpesvirus) | Herpesvirus                          | Kaposi sarcoma, body cavity lymphoma                    |
| HPV serotypes 16, 18, 33, 39   | Papillomavirus (DNA virus)            | Cervical carcinoma, anal carcinoma                      |
| HPV serotypes 5, 8, 17         | Papillomavirus                        | Skin cancer                                              |

**Table 3.** Viruses acting like oncogenes

<br>

> ④ **Gene Group 3.** Tumor suppressor genes

>> ○ Genes encoding proteins that stop cell division or repair damaged DNA at cell cycle checkpoints.

>> ○ Recessive mutations: Both homologous chromosomes must carry mutations to exhibit cancerous effects.

>> ○ **Example 1.** BRCA2 gene

>>> ○ Located on chromosome 13, and encodes a protein involved in DNA damage repair.  

>>> ○ BRCA 1/2 mutations can help assess whether an individual is at high or low risk of developing breast cancer (e.g., Angelina Jolie).

>>> ○ BRCA 1/2 mutations can be used to predict whether ionizing radiation will be effective or not.

>>> ○ BRCA 1/2 mutations can help determine the efficacy of PARP (poly(ADP-ribose) polymerase) inhibitors.

>>>> ○ PARP is an enzyme that repairs single-strand breaks (SSBs).  

>>>> ○ BRCA is an enzyme that repairs DNA through homologous recombination.  

>>>> ○ In the case of a BRCA mutation combined with a PARP inhibitor, the inability to bypass repair leads to cancer cell death.  

>>> ○ BRCA 1/2 mutations can be used to predict the prognosis of breast cancer metastasis.  

>> ○ **Example 2.** p53 gene: Regulates the entire cell cycle.

>>> ○ **Function 1.** Increases transcription of p21, leading to cell cycle arrest.  

>>> ○ **Function 2.** Associated with the expression of specific miRNAs that inhibit cell division.  

>>> ○ **Function 3.** Activates the expression of genes directly involved in DNA repair.  

>>> ○ **Function 4.** Inhibits cyclin-CDK complexes in response to DNA damage.  

>>> ○ **Function 5.** Can induce apoptosis by increasing the expression of genes encoding BH3-only proteins.  

>>> ○ Structure: Composed of 393 amino acids.  

>>> ○ Active research is being conducted to determine which mutations of p53 regions lose their tumor-suppressing function.

>>> ○ About 86% of p53 mutations occur between 120th and 290th amino acids.

>>> ○ Approximately 50% of tumor cells exhibit the above-mentioned mutations.  

>>> ○ In April 1989 in Science, ir was reported that p53, previously known as an oncogene, is a tumor suppressor gene.  

>>> ○ Over the following 20 years, more than 50,000 research papers on this topic were published.  

>> ○ **Example 3.** pRb1: Acts in G1 phase, inhibits E2F.

>> ○ **Example 4.** bcl2: Inhibits Bax.

>>> ○ bcl2 and bcl-xl both inhibit apoptosis and are overexpressed in solid tumors and blood cancers.  

>>> ○ ABT263: Inhibits bcl2 and bcl-xl.  

>>> ○ Navitoclax: Inhibitor of bcl2, bcl-xl, and bcl-w.

>> ○ **Example 5.** [APC](https://jb243.github.io/pages/71)

>> ○ **Example 6.** Rb: Blocks cell cycle progression at the G1 checkpoint.

>> ○ **Example 7.** DCC 

>> ○ **Example 8.** p16 gene

>> ○ **Example 9.** VHL (von Hippel-Lindau tumor suppressor): Widely used in [PROTAC](https://jb243.github.io/pages/74#7-post-translational-modification-ptm) technology.

>> ○ **Example 10.** ATM

> ⑤ **Gene Group 4.** Mechanisms of cancer cell immortalization

>> ○ Cancer cells possess [telomerase](https://jb243.github.io/pages/74).  

>> ○ Telomerase protects the telomere DNA at the ends of chromosomes, allowing cells to continue dividing beyond the typical division limit of 50 times in normal cells. 

> ⑥ **Gene Group 5.** Hereditary cancer syndromes: The following is a list of mutated genes associated with cancer syndromes.

>> ○ Familial retinoblastoma: RB1

>> ○ Li-Fraumeni: TP53 (p53 protein)

>> ○ Familial adenomatous polyposis: APC

>> ○ Hereditary nonpolyposis colorectal cancer: MLH1, MSH2, MSH6, PMS1, PMS2

>> ○ Wilms' tumor: WT1

>> ○ Breast and ovarian cancer: BRCA1, BRCA2

>> ○ von Hippel-Lindau: VHL

>> ○ Cowden syndrome: PTEN

>> ○ Diffuse gastric cancer and ILC: CDH1

>> ○ Peutz-Jeghers syndrome: STK11

> ⑦ **Gene Group 6.** Other genes overexpressed in cancer

>> ○ Transferrin

> ⑧ **Multi-hit model:** Cancer development requires multiple mutations.

>> ○ Reasons for frequent occurrence in epithelial tissues with high cell division rate.

>> ○ Proto-oncogenes and tumor suppressor genes

>>> ○ In normal cells, mutations in proto-oncogenes result in benign tumors (≠ cancer).  

>>> ○ In benign tumors, mutations in both maternal and paternal copies of tumor suppressor genes lead to malignant tumors (= cancer).  

>> ○ Mutations that must accumulate for a benign tumor to progress into cancer

>>> ○ Angiogenesis: Tumors induce blood supply to sustain themselves.  

>>> ○ Loss of contact inhibition and anchorage dependence: Cells pile up, and cancer cells can migrate to other locations.

>>> ○ Loss of density-dependent inhibition of growth

>>> ○ **Immortalization**: Activation of telomerase prevents the limitation on the number of cell divisions.  

>>> ○ Mutations are inheritable, but most mutations occur during an individual's lifetime.  

>> ○ At least 5–6 independent mutations are required in a single cell.  

>>> ○ This process typically takes about 10–20 years.

>>> ○ Whole-genome sequencing studies reveal thousands to tens of thousands of mutations in a single cancer.  

>>> ○ Among these mutations, approximately five are driver mutations, while the rest are passenger mutations.  

>>> ○ Passenger mutations are much more numerous but are only expressed under the influence of driver mutations.  

> ⑨ **Example**: Colorectal cancer is one of the most common cancers.  

>> ○ APC tumor suppressor gene deficiency: Leads to excessive epithelial cell proliferation (loss of polarity and contact inhibition), resulting in polyp (adenoma) formation.  

>>> ○ Colorectal cancer frequently involves mutations in the APC gene, known as familial adenomatous polyposis (FAP).

>>> ○ APC gene mutations act in a recessive manner.  

>> ○ Ras oncogene: Promotes cell proliferation without signals, forming small benign tumors (class II adenomas).  

>>> ○ Most growth factor-independent cell division in cancer is caused by Ras.  

>> ○ Tumor suppressor gene DCC (Deleted in Colon Cancer).  

>> ○ Loss of tumor suppressor gene p53: Fails the DNA damage-related G1/S checkpoint, rapidly accumulating mutations and leading to malignant tumors.  

>>> ○ p53 mutations are observed in 35% of cancer patients.

>> ○ Loss of anti-metastasis genes

>> ○ Angiogenesis, invasion, and metastasis occur.  

>>> ○ Angiogenesis: Blood vessels and lymphatic vessels grow toward the tumor mass.  

>>> ○ Invasion: Cancer cells infiltrate blood vessels or other organs.  

>>> ○ Metastasis: Cancer cells spread to other organs.  

>> ○ The sequence is not important; cancer can develop as long as the conditions are met.

⑸ Types of cancer: Classified by tissue of origin.

> ① Carcinoma: Epithelial tissue; most common, fast-growing, and invasive; occurs in older age groups.

> ② Adenocarcinoma: Glandular epithelial tissue, secretory glands, etc.

> ③ Squamous cell carcinoma: Epidermal cells; treated with docetaxel.

> ④ Large cell carcinoma: Treated with pemetrexed.

> ⑤ Sarcoma: Muscle and connective tissues; rare but unaffected by external factors; occurs in younger individuals.

>> ○ Osteogenic sarcoma (osteosarcoma): Destruction of bone tissue in children, with metastasis throughout the body.

>> ○ Myeloma: Disrupts hematopoiesis in the bone marrow, classified as a malignant tumor.

>> ○ Chondrosarcoma: Cartilage tumor

>> ○ Muscle sarcoma

>> ○ Liposarcoma

> ⑥ Hematological cancer

>> ○ Leukemia: Immature white blood cells trigger autoimmune responses, destroying normal tissues. The number of normal blood cells drastically decreases, impairing immune function as well as basic blood functions such as oxygen transport and nutrient supply.  

>> ○ Lymphoma: Representative types include Hodgkin's lymphoma and Burkitt's lymphoma.  

> ⑦ Others

>> ○ Solid tumors

>> ○ Glioblastoma: A tumor originating from glial cells in the brain, with the highest incidence and malignancy among glial cell tumors.  

>> ○ Teratoma: A cancer in which tissues from all three germ layers (endoderm, mesoderm, and ectoderm) are observed.  

⑹ Diagnosis methods

> ① CA125 (Cancer Antigen 125): Cancer-specific protein 

>> ○ Amino acid sequence: 22,000 aa; also known as Mucin 16 (MUC16).

>> ○ Recognized by monoclonal antibody OC-125.

>> ○ A glycoprotein on the cell membrane of epithelial cancers.

>> ○ Normal serum CA125 levels: Below 35 U/mL.

>> ○ The only tumor marker used for evaluating the treatment efficacy and early detection of recurrence in ovarian cancer.  

>> ○ Associated with prognosis, as well as the size, stage, and survival rate of ovarian cancer.  

>> ○ CA125 levels can also be elevated in ovarian cancer, endometrial cancer, pancreatic cancer, lung cancer, breast cancer, colorectal cancer, and gastrointestinal cancers, making it less valuable for screening purposes.  

> ② CA15-3

>> ○ Measuring the blood levels of MUC1 to predict cancer recurrence: An FDA-approved method.  

>> ○ MUC1 consists of MUC1-N (N-terminal direction) and MUC1-C (C-terminal direction), with MUC1-N being the form released into the bloodstream.  

>> ○ On the cell membrane, MUC1 expression has polarity, but this polarity is lost in cancer cells.  

>> ○ MUC1-N is hyperglycosylated, but glycosylation is reportedly reduced in cancer cells.  

> ③ PSA (Prostate-Specific Antigen), PSMA(Prostate-specific Membrane Antigen): A marker for prostate cancer

> ④ Biopsy: Surgical collection of cells or tissues

>> ○ Needle biopsy: Tissue collected using a needle.

>> ○ Laparoscopy: Uses lighted instruments, cameras, and small cutting tools for sample collection.

> ⑤ X-ray diagnosis

>> ○ General tumor detection: Requires approximately 10<sup>8</sup> cells.

>> ○ Palpable tumors (10 mm): Approximately 10<sup>9</sup> cells

>> ○ Fatal tumors (10 cm): Approximately 10<sup>12</sup> cells

> ⑥ Tumor mutation burden (TMB)

>> ○ Measures the number of mutations in a tumor.

>> ○ Used to predict prognosis, metastasis, and efficacy of immune checkpoint inhibitors.

> ⑦ ICB (Immune Checkpoint Blockade): Targets PD-1, PD-L1, CTLA4, etc.

> ⑧ Lymphocyte Markers: Includes markers for NK cells, B cells, T cells, etc.

> ⑨ Neoantigens

>> ○ Concept

>>> ○ Neoantigen: A completely new protein formed in cancer cells due to specific mutations in tumor DNA.

>>> ○ Tumor-Associated Antigen (TAA): Partially expressed in normal cells as well.

>>> ○ Cancer-Germline Antigen (CGA): Also partially expressed in normal cells.  

>>> ○ Neoantigens are also referred to as Tumor-Specific Antigens (TSA).

>>> ○ Neoantigens are categorized into shared neoantigens and personalized neoantigens.

>> ○ History

>>> ○ The first neoantigen was discovered in 1988 by De Plaen's research team through cDNA library screening in a mouse tumor model.  

>>> ○ The first clinical trial using a neoantigen vaccine in humans was conducted in 2015.  

>> ○ Characteristics

>>> ○ Not expressed in normal tissues, thus avoiding off-target effects.  

>>> ○ Primarily caused by non-synonymous mutations.

>>> ○ Even a single amino acid change can be recognized by T cells.  

>>> ○ Neoantigens are immunogenic; note that they are often defined as neoantigens only if they exhibit immunogenicity.

>> ○ Examples

>>> ○ KRAS G12D

>>> ○ BRAF V600E

>>> ○ EGFRvIII

>>> ○ CD19

>>> ○ NY-ESO-1

>>> ○ MUC1

⑺ Treatment methods

> ① Overview

>> ○ Easily treatable cancers: Thyroid cancer, breast cancer, prostate cancer

>> ○ Difficult-to-treat cancers: Pancreatic cancer, lung cancer, liver cancer, brain cancer

> ② **Method 1.** Surgery: Localized tumors can often be removed surgically.

> ③ **Method 2.** Chemotherapy: Uses anticancer drugs to kill dividing cells.

>> ○ Cocktail therapy: A combination of various cell division-inhibiting chemotherapy drugs.  

>> ○ Resistant cells: Occur at a rate of 1 in 1 million cells. In palpable tumors, approximately 1,000 resistant cells are present.  

>> ○ Side effects: Kills normally dividing cells (all cells, including those in the bone marrow and stomach wall), leading to side effects such as hair loss, vomiting, and diarrhea.  

>> ○ TNBC (Triple Negative Breast Cancer Cells)  
  
>>> ○ Breast cancer cells that do not express estrogen, progesterone, or epithelial growth factors.  

>>> ○ Among epithelial growth factors, particularly ERBB2 (Her2/neu) is important.  

>>> ○ Accounts for 10-20% of all breast cancer cases diagnosed in the U.S.  

>>> ○ Lack of expression of these three receptors reduces the effectiveness of hormone-based therapies (e.g., Tamoxifen) and receptor-targeted therapies (e.g., Herceptin).  

>> ○ **Example 1.** Standard pancreatic cancer treatment: nab-paclitaxel + gemcitabine  

> ④ **Method 3.** Radiation Therapy: Uses high-energy radiation.

>> ○ Mechanism: Damages DNA to induce cancer cell death (e.g., BRCA2, p53).

>> ○ Typically used for cancers located nearby.

>> ○ Generally involves 10-20 sessions of radiation therapy after surgically removing the tumor.

>> ○ Effects on normal cells: Normal cells undergo normal cell division (via p53 and DNA repair) or apoptosis.

>> ○ Effects on tumor cells: Results in cell death due to increased mutations or further malignancy progression.

>> ○ **Example 1.** Standard prostate cancer treatment: PSMA-targeting radionuclide.

> ⑤ Standard treatment methods by cancer type

>> ○ Thyroid cancer: Surgery or I-131 is the primary treatment.  

>> ○ Skin cancer: Typically treated with cryo-surgery.  

>> ○ Blood cancer: Among blood cancer antigens, CD19 is the most commonly targeted.  

>> ○ Solid tumors: No established treatment methods known to date. The most commonly targeted antigen in solid tumors is TAA (tumor-associated antigen).  

>> ○ Glioblastoma: Difficult to treat due to unclear boundaries with normal tissue. No established treatment methods known to date.  

>> ○ Lymphoma: Generally treated with chemotherapy.  

⑻ [Cancer Statistics](https://jb243.github.io/pages/730): In 2015, cancer caused 8.8 million deaths worldwide.

<br>

---

*Input: 2015.6.27 18:10*

*Modify: 2019.2.20 10:14*
