## **Chapter 8. Central Dogma**

Higher category: 【Biology】 [Biology Index](https://jb243.github.io/pages/1457) 

---

**1.** [Overview](#1-overview)

**2.** [Genes and Gene Expression](#2-genes-and-gene-expression)

**3.** [DNA replication](#3-dna-replication)

**4.** [Transcription](#4-transcription)

**5.** [Translation](#5-translation)

**6.** [Delivery of Proteins After Translation](#6-delivery-of-proteins-after-translation)

**7.** [Post-translational Modification](#7-post-translational-modification)

**8.** [Inhibitor](#8-inhibitor)

---

**a.** [DNA proofreading and repair](https://jb243.github.io/pages/1472)

**b.** [Central dogma of microbiology](https://jb243.github.io/pages/828)

---

<br>

## **1\. Overview** 

⑴ Central dogma refers to the flow of genetic information (DNA → RNA → protein).

⑵ After the discovery of reverse transcriptase in retroviruses (RNA viruses), the central dogma was revised.

⑶ Information transfer relationship

> ① DNA replication: DNA → DNA

> ② Transcription: DNA → RNA

> ③ Translation: RNA → Amino acid sequence (primary structure)

> ④ Triplet code (DNA) → Codon (mRNA) → Anticodon (tRNA) → Amino acid

<br>

<br>

## **2\. Genes and Gene Expression**

⑴ Genes and Chromosomes

⑵ One gene, multiple polypeptides theory

> ① One gene, one enzyme hypothesis: Beadle and Tatum's experiment

>> ○ Inferring biosynthetic pathways using mutants: ornithine, citrulline, arginine

<br>

<img width="644" alt="스크린샷 2024-11-30 오후 2 01 54" src="https://github.com/user-attachments/assets/cb8c9531-b894-4fa7-aa45-5162ec2bc218">

**Table 1.** Beads and Tatum's Experiments

<br>

>> ○ **Tip 1.** The more growing the material, the later stages of the biosynthetic pathway.

>> ○ **Tip 2.** The more a mutant grows, the earlier the step it is involved in the metabolic pathway.

> ② One gene, one protein hypothesis: Proteins that do not function as enzymes are found.

> ③ One gene, one polypeptide hypothesis: Proteins forming quaternary structures are revealed.

> ④ One gene, multiple polypeptides hypothesis: It has been revealed that multiple polypeptides can be produced from a single gene.

⑶ Regulation of gene expression

> ① Overview of gene expression

>> ○ First, the somatic cells of our body have the same DNA.

>> ○ Second, each tissue cell transcribes only specific genes, resulting in different gene expression patterns, which lead to variations in structure and function.

> ② Regulation of Transcription**  

>> ○ Regulation through transcription repression (prokaryotic cells)  

>> ○ Regulation through transcription activation (prokaryotic and eukaryotic cells)  

>> ○ Regulation through chromatin condensation: preventing RNA polymerase access (Barr bodies)  

> ③ Regulation through mRNA degradation

> ④ Regulation of Translation

> ⑤ Regulation through protein degradation (proteases)

<br>

<br>

## **3\. DNA replication**

⑴ 1<sup>st</sup>. Untangling DNA helix

> ① Replication proceeds in both directions from the replication origin (ori) (= the replication fork moves in both directions from the replication origin)

>> ○ Replication origin: A = T rich site with weak hydrogen bonds

>> ○ Replication fork: The junction between the strand where the DNA is hydrogen-bonded and the strand that is not.

>> ○ Prokaryotes: DnaA recognizes one replication origin.

>>> ○ DnaA

>>>> ○ Consensus sequence: 9 base pairs × 4 + 13 base pairs × 3  

>>>> ○ 20-50 monomers

>>>> ○ Binds simultaneously to multiple AT-rich sites.

>>>> ○ Melts DNA by using one molecule of ATP.

>> ○ Eukaryotes: ORC (origin recognition complex) recognizes origin of replication.

>>> ○ The replication speed is slow, so there are multiple origins of replication.

>>> ○ Yeast: 250 to 400 replication origins.

>>> ○ Mammals: 25,000 origins of replication; approximately 6,500 in humans.

>>> ○ Rich in A-T sequences.

>>> ○ The unit of replication determined by the origin of replication is called a replicon.

>> ○ ARS (autonomic replicatoin sequence)

>> ○ DnaA and ORC create replication bubbles, exposing several dozen nucleotides of single-stranded DNA (ssDNA).

> ② Enzyme: Helicase, DNA topoisomerase, single chain binding protein

>> ○ Helicase: Releases hydrogen bonds in two complementary strands.

>>> ○ Prokaryotes: DnaB, PcrA

>>> ○ Eukaryotes: MCM (Minichromosome Maintenance Complex)  

>>> ○ DnaB

>>>> ○ [Helicase characteristic experiment](https://jb243.github.io/pages/1468): Two conclusions can be drawn  

>>>> ○ **Conclusion 1**: Requires replication bubbles created by DnaA or ORC to bind to DNA.  
  
>>>> ○ **Conclusion 2**: Typically binds to the template of the lagging strand at the replication fork and moves in the 5' → 3' direction (exceptions exist).  

>>>> ○ ATP is used.

>>>> ○ Minimum ssDNA length for a replication bubble: 16 nucleotides or more.  
  
>>>> ○ 100 turns/second = 6,000 rpm

>>> ○ PcrA: Consists of A1, A2, B1, B2 domains, and a P-loop.  

>> ○ The helicase acts to intensify the DNA's torsion, resulting in unstable structure.

<br>

![image](https://github.com/user-attachments/assets/2b315eb3-1742-4908-9b3c-0c0e857e5cd9)

**Figure 1.** Topological isomers of circular DNA with a size of 210 bp

The second double helix has fewer nucleotides per turn (lower torsional stress), while the third double helix has more nucleotides per turn (higher torsional stress).

<br>

>>> ○ L (Linking number): The number of loops. Conserved during torsional stress calculations, with counterclockwise (+) rotation.

>>> ○ T (Twist number): The number of helical turns. DNA is inherently right-handed, so it has counterclockwise (+) torsional stress.

>>> ○ W (Writhe number): The number of supercoils.

>>> ○ Supercoil: A structure where the DNA molecule forms loops on itself, related to the enhancement or relaxation of torsional stress.

>>> ○ Positive supercoil: A left-handed supercoil (the <span style="color: red;">third</span> in the figure), with a positive writhe number.

>>> ○ Negative supercoil: A right-handed supercoil (the <span style="color: red;">second</span> in the figure), with a negative writhe number.

>>> ○ Positive supercoils form left-handed supercoils, thereby increasing the right-handed torsional stress of the double helix.  

>>> ○ Negative supercoils form right-handed supercoils, thereby relaxing the right-handed torsional stress of the double helix.

>>> ○ Most organisms that are not thermophilic bacteria have DNA with a negative supercoil structure.

>> ○ DNA topoisomerase: An enzyme that relieves torsional stress in DNA.

>>> ○ Acts at a region far ahead of the replication fork.

>>> ○ **Type 1 DNA Topoisomerase**

>>>> ○ An enzyme that cuts one strand of the DNA, allows it to unwind naturally, and then reseals it.  

>>>> ○ Strand-cutting and strand-releasing activities: Nuclease and ligase activities.  

>>>> ○ Does not require ATP.  

>>>> ○ Relieves negative supercoils in E. coli. 

>>>> ○ **Step 1.** Tyrosine attacks the phosphate backbone, creating a nick in the DNA strand.  

>>>> ○ **Step 2.** The DNA rotates one turn under torsional stress.  

>>>> ○ **Step 3.** Resealing occurs, ultimately resolving the supercoil.  

>>> ○ **Type 2 DNA Toposiomerase** (DNA gyrase)

>>>> ○ An enzyme that cuts both strands of the DNA, rotates (topologically rearranges) them, and reseals them.

>>>> ○ Converts positively supercoiled DNA molecules into negatively supercoiled forms.

>>>> ○ Requires ATP hydrolysis.

>>>> ○ Targeted by antibiotics such as fluoroquinolones (e.g., ciprofloxacin).

>>> ○ **Single-Stranded Binding Protein**  

>>>> ○ Prokaryotes: Single-Strand Binding Protein (SSBP); a monomer with two domains.  

>>>> ○ Eukaryotes: Replication Protein A (RPA).  

>>>> ○ **Function 1.** Temporarily binds to each single strand of DNA after the hydrogen bonds are broken, preventing the strands from pairing back together.

>>>> ○ **Function 2.** Stabilizes the DNA.  

>>>> ○ **Function 3.** Protects DNA from twisting.  

>>>> ○ SSBPs are required to activate the helicase function in experimental settings.  

> ③ Prokaryotic cells: Circular DNA, one replication origin.

> ④ Eukaryotic cells: Linear DNA, multiple replication origins.

>> ○ Eukaryotic cells have larger genomes than prokaryotic cells, so they have many origins of replication.

⑵ 2<sup>nd</sup>. Attaching primer

> ① Primer: Short RNA fragments underlying DNA polymerization

>> ○ PCR primers are DNA fragments.

>> ○ The primer is approximately 10 nucleotides in length.

> ② Primers provide the first 3'-OH group ends needed for DNA polymerization.

> ③ Enzyme: Primase is involved. Primase is responsible for synthesizing and attaching primers.  

>> ○ Since helicase operates at a speed of 100 turns/second, primase also works at a remarkably fast pace.  

>> ○ Eukaryotes: Pol α/primase complex. 

> ④ Primosome: A prepriming complex formed by several proteins before the synthesis of RNA primers.

⑶ 3<sup>rd</sup>. DNA polymerization

> ① Enzyme: DNA polymerase is involved.

>> ○ E. coli has three DNA polymerases: DNA pol I, II, and III.

>> ○ DNA pol I: Nick translation. Removes primers and replaces RNA primers with dNTP

>>> ○ Speed: 10 nt/s.

>>> ○ T4 ligase performs functions similar to the DNA pol I subunit.

>> ○ DNA pol II: Repairs DNA damaged by external factors and is activated when DNA synthesis is interrupted.

>> ○ DNA pol III: Involved in DNA elongation. Functions as a dimer and does not move itself; instead, the DNA strand moves.

>>> ○ Characteristics: Very high potency, fidelity, and processivity.

>>> ○ 1000 nt/s

>>> ○ Asymmetry: Proceeds differently when elongating the lagging strand and the leading strand.

>>> ○ DNA pol III is composed of about 10 different polypeptide chains.

>>> ○ Overall structure: αεθβ<sub>2</sub>τ<sub>2</sub> - αεθβ<sub>2</sub>γ<sub>2</sub>(δδ'χψ)<sub>2</sub>.

>>> ○ α subunit: Polymerase.

>>> ○ ε subunit: 3' → 5' exonuclease.

>>> ○ β<sub>2</sub>τ<sub>2</sub>: Processivity, star-shaped ring, 35 Å hole.

>>> ○ β<sub>2</sub>: Related to sliding DNA clamp.

>> ○ Sliding clamp 

<br>

![image](https://github.com/user-attachments/assets/c20a7d5d-2604-4b4c-a06d-da8dfd6a0008)

**Figure 2.** Sliding clamp

<br>

>>> ○ Ensures proper alignment of the template strand and the daughter strand, allowing DNA polymerase to synthesize DNA accurately and efficiently, enhancing both precision and efficiency.

>>> ○ Eukaryotes: The sliding clamp is associated with PCNA (Proliferating Cell Nuclear Antigen), which is related to Ki-67 staining.  

>>> ○ Prokaryotes: The sliding clamp is temporarily opened by the clamp loader to allow the insertion of the closed circular structure of E. coli DNA into the central channel. It then closes again, enabling polymerase activity during DNA replication.  

>> ○ Klenow fragment

>>> ○ Recall DNA pol I: 5' → 3' polymerization, 3' → 5' exonuclease, 5' → 3' exonuclease.

>>> ○ Recall the large fragment of DNA pol I: 5' → 3' polymerization, 3' → 5' exonuclease.

>>> ○ Recall the small fragment of DNA pol I: 5' → 3' exonuclease.

>>> ○ Recall 5' → 3' exonuclease: Removes terminal nucleic acids.

>>> ○ Recall 3' → 5' exonuclease: Proofreading function.  

>>> ○ The Klenow fragment is the large fragment of DNA pol I and is identical to DNA pol III.

>>> ○ The Klenow fragment is DNA polymerase of T4 phage.  

>>> ○ Converts sticky ends of restriction fragments to blunt ends in [DNA recombination](https://jb243.github.io/pages/77).

>>> ○ Lacks 5' → 3' exonuclease activity, so it cannot degrade the primer of template DNA.  

>>> ○ Example of an in vitro polymerization reaction using Klenow fragment.  

<br>

<img width="584" alt="image" src="https://github.com/user-attachments/assets/4b4ffaf4-776e-4746-af0e-accc2d358c8d" />

**Figure 3.** Example of an in vitro polymerization using the Klenow fragment

<br>

>>> ○ In eukaryotes, DNA pol α, β, γ, δ, and ε exist.

>>> ○ DNA pol α: Contains primase, and lacks proofreading.

>>> ○ DNA pol β: Repair function, and lacks proofreading.

>>> ○ DNA pol δ: Extends the Okazaki fragments of the lagging strand, and has proofreading.

>>> ○ DNA pol ε: Extends the leading strand, and has proofreading.

>>> ○ DNA pol γ: Replicates mitochondrial DNA, and has proofreading.

> ② Substrate: dNTP (dATP, dTTP, dCTP, dGTP), with 3 phosphate groups on 5 'carbon.

> ③ Release 2 pyrophosphates by attaching the 5' phosphate group of nucleotides complementary to the template to the 3'-OH group of the synthetic strand.

>> ○ 3'-OH group in DNA double helix structure accurately attack α phosphate group.

>> ○ Pyrophosphate

> ④ A pairs with T, and G pairs with C in a complementary manner (purine-pyrimidine pairing → the distance between the two strands of DNA remains constant).

> ⑤ Composite direction: 5' → 3'

> ⑥ Leading strand

>> ○ A strand synthesized continuously.  

>> ○ Occurs when the direction of replication is the same as the movement of the replication fork.  

> ⑦ Lagging strand

>> ○ A strand synthesized in discontinuous fragments (Okazaki fragments).  

>> ○ Occurs when the direction of replication is opposite to the movement of the replication fork.  

⑷ 4<sup>th</sup>. Chain closing

> ① Termination mechanisms exist but are not important because they replicate 100% by default.

⑸ 5<sup>th</sup>. Primer removal

> ① RNA primer removal enzymes

>> ○ Prokaryotes: DNA polymerase I  

>> ○ Eukaryotes: RNAase H, FEN1 (flap endonuclease)  

> ② After RNA primers are removed, DNA ligase (DnaG) performs the DNA strand-joining reaction.

>> ○ Ligase uses ATP.  

>> ○ Collectively, DnaB and DnaG are referred to as the primosome.

>> ○ **Step 1.** E + ATP (or NAD<sup>+</sup>) → E-AMP + PPi (or NMN)  

>>> ○ Activated AMP from ATP forms a phosphoamide bond with the e-amino group of a lysine residue in DNA ligase.  

>> ○ **Step 2.** E-AMP + ⓟ-5'-DNA → E + AMP-ⓟ-5'-DNA

>> ○ **Step 3.** DNA-3'-OH + AMP-ⓟ-5'-DNA → DNA-3'-O-ⓟ-5'-DNA + AMP

>>> ○ The activated 5' phosphorous atom undergoes a nucleophilic attack by the 3'-OH group.  

> ③ Removal rate

>> ○ 1 per second

>> ○ 1 error per 10,000,000

> ④ The RNA primer at the 5' end of the daughter strand is not replicated.

>> ○ This is why DNA becomes shorter with successive rounds of replication.  

>> ○ Related to aging.  

⑹ 6<sup>th</sup>. Correction and Modification of DNA replication: Exonuclease activity

> ① Original marker: Marks the original template strand by methylating the A bases.

>> ○ Dna methylase is involved. Methylation of 5'-GATC-3' sequence at the ori C site.

> ② Proofreading: DNA polymerase has proofreading functionality.

>> ○ Acts like replacing a single nucleotide.  

>> ○ 3' → 5' exonuclease activity

>>> ○ Performed by DNA pol I, II, and III.

>>> ○ Reduces replication error rate to 1/10<sup>7</sup> during polymerization.

>>> ○ **Step 1.** During the DNA synthesis process, an incorrect nucleotide is incorporated, leaving the 3' end of that nucleotide exposed at this stage.

>>> ○ **Step 2.** Structural fluctuation occurs significantly. 

>>> ○ **Step 3.** Weakened hydrogen bonds allow the incorrect nucleotide to move to the exonuclease site.  

>>> ○ **Step 4.** The incorrect nucleotide is removed through hydrolysis.

>> ○ 5' → 3' exonuclease activity

>>> ○ Performed only by DNA pol I.

>>> ○ It replaces single nucleotides one at a time, but ultimately replaces an RNA primer approximately 10 nucleotides long in a single process.

>>> ○ **Step 1.** The polymerase positions itself between the 3' end of the Okazaki fragment and the 5' end of the adjacent RNA primer.  

>>> ○ **Step 2.** Proceeds nucleotide by nucleotide:  

>>> ○ **Step 2-1.** Removes the RNA primer through hydrolysis.  

>>> ○ **Step 2-2.** Replaces each RNA nucleotide with dNTP using 5' → 3' polymerase activity and performs proofreading using 3' → 5' exonuclease activity.

>>> ○ **Step 3.** Repeats these steps until the gap is completely filled with DNA.  

>> ○ The HIV virus lacks a repair mechanism.  

> ③ [Repair](https://jb243.github.io/pages/1472): If the repair enzyme corrects, the error rate of replication is 1/10<sup>9</sup>.

>> ○ Like replacing multiple nucleotides.  

>> ○ **Type 1.** Dimer repair.  

>> ○ **Type 2.** Excision repair: Removes abnormal bases caused by chemical damage.  

> ④ If DNA proofreading and repair fail, mutations occur frequently.

⑺ Characteristics of DNA replication

> ① Conservative replication: After replication, one original DNA molecule and one entirely new DNA molecule are present (**rejected**).  

> ② Dispersive replication: After replication, two completely new DNA molecules are generated (**rejected**).  

> ③ Semi-conservative replication: After replication, each DNA molecule contains one template strand and one newly synthesized strand (**accepted**).  

> ④ Complementarity: The template strand and the newly synthesized strand contain complementary information.  

> ⑤ Meselson-Stahl experiment: Validated the semi-conservative model of DNA replication.  

<br>

![image](https://github.com/user-attachments/assets/dee0740d-df30-4cc3-8e51-54a709000858)

**Figure 4.** Meselson-Stahl's experiment

<br> 

>> ○ Density gradient centrifugation using cesium chloride → Separates DNA based on density.  

>> ○ E. coli labeled with <sup>15</sup>N was cultured in a medium containing normal nitrogen (<sup>14</sup>N), and the DNA density was analyzed at each generation.  

⑻ Telomerase: 2009 Nobel Prize in Physiology or Medicine

> ① The RNA primer at the 5' end of the daughter strand cannot be replicated → DNA shortens → Aging

>> ○ Hayflick limit: The number of cell divisions until the cell can no longer divide.  

> ② Telomere

>> ○ Protects DNA to some extent from shortening.  

>> ○ Consists of meaningless repetitive sequences. 6-nt long repeats. Repeated approximately 300 to 5000 times.  

>> ○ Tetra-G on the telomere forms a t-loop, transforming the exposed 3' end of the telomere into a hairpin structure to protect the ends. 

>> ○ Humans, mice, birds, red bread mold, silkworms: TTAGGG

>> ○ Arabidopsis thaliana: TTTAGGG

>> ○ Chlamydomonas: TTTTAGGG

>> ○ Yeast: TTAC(A)(C)G(1-8)

> ③ Telomerase

>> ○ A reverse transcriptase and RNA-dependent DNA polymerase.  

>> ○ Extends telomeres using a short RNA template within the molecule.  

>> ○ The internal short RNA: 3'-CCCAAUCCC-5' RNA template

<br>

![image](https://github.com/user-attachments/assets/5c1b0efd-9499-4b79-9809-87127080e9bd)

**Figure 5.** Mechanism of exposed 3' terminus in telomeres and telomerase

<br>

> ④ Found in germ cells, proliferating normal cells, and cancer cells.

> ⑤ In mice, telomerase is commonly found, unlike in humans.

> ⑥ Circular DNA (e.g., mitochondrial DNA) does not require telomerase.

⑼ RCR(rolling-circle replication): Only observed in circular DNA.

<br>

![image](https://github.com/user-attachments/assets/9cbad033-b29b-45b1-a6fa-cc4854239224)

**Figure 6.** rolling-circle replication

<br>

⑽ Multi-fork replication

> ① Situation: Under extremely favorable conditions, the reproduction cycle of E. coli is 20 minutes.  

> ② Question: Since the DNA replication speed is 1000 nucleotides per second, it takes longer to replicate the entire chromosome than the reproduction cycle.  

> ③ Solution: Multi-fork replication. This phenomenon allows replication to start on already replicating daughter strands, preparing for the next cell cycle before the current cycle is completed.  

<br>

<br>

**4\. Transcription**

<br>

![image](https://github.com/user-attachments/assets/6f706fb1-c798-4639-b849-bbaa57431ca2)

**Figure 7.** Gene transcription

<br>

⑴ DNA vs. RNA

> ① Strands: DNA is double stranded, RNA is single stranded.

> ② Sugar: DNA contains deoxyribose (a 5-carbon sugar), while RNA contains ribose (a 5-carbon sugar).

>> ○ Deoxyribose has a -H group on the 2nd carbon, while ribose has a -OH group on the 2nd carbon.

> ③ Base: A, T, G, C (DNA) ↔ A, U, G, C (RNA)

> ④ DNA type

>> ○ gDNA (Genomic DNA): Typical DNA.

>> ○ [Mobile DNA](https://jb243.github.io/pages/554#3-mobility-dna)

>> ○ Satellite DNA: Non-coding DNA that randomly repeats within eukaryotic genomes.

>> ○ Plasmid: Only found in bacteria.

>> ○ ecDNA (Extrachromosomal DNA): Circular DNA that exists outside of chromosomes, contributing to high copy numbers of specific genes in cancer cells.

>> ○ Pseudogene: A gene-like group that lacks protein-coding ability.

>>> ○ **Type 1:** A gene duplicated by a retrotransposon but missing introns and a promoter.

>>> ○ **Type 2:** A gene disabled due to accumulated mutations.

> ⑤ RNA type

>> ○ mRNA (messenger RNA)

>> ○ tRNA (transfer RNA)

>> ○ rRNA (ribosome RNA)

>> ○ SRP RNA

>> ○ snRNA (small nuclear RNA)

>> ○ miRNA, siRNA

>> ○ lncRNA and ncRNA: Non-coding RNA (e.g., pseudo genes are transcribed but not translated). lncRNAs have fewer exons, have fewer isoforms, are lowly expressed, and exhibit high tissue specificity.

>> ○ sciRNA

>> ○ ASO(antisense oligonucleotide)

>> ○ snoRNA (small nucleolar RNA)

>> ○ glycoRNA

>> ○ piRNA

>> ○ eRNA

>> ○ lincRNA

⑵ Prokaryotic and Eukaryotic mRNAs  

> ① Polycistronic and Monocistronic

>> ○ Prokaryotes: Polycistronic mRNA (= operon)
  
>>> ○ A single mRNA molecule encodes several different polypeptide chains.  
  
>>> ○ Each polypeptide has a start site containing a ribosome binding site (RBS).  
  
>>> ○ Advantage: Allows multiple genes to be regulated by a single promoter.  
  
>>> ○ Polyribosome (polysome): A structure where multiple ribosomes are bound to a single mRNA.  

>> ○ Eukaryotes: Monocistronic mRNA
  
>>> ○ Polyribosomes are observed but are extremely rare.  
  
>>> ○ rRNA: Polyribosomes are also observed in eukaryotic cells for rRNA.  

> ② Presence of introns: Found only in eukaryotes.  

>> ○ Histone DNA: Lacks introns.  

> ③ Simultaneity of transcription and translation: Occurs only in prokaryotes.  

> ④ Post-transcriptional modification: Present only in eukaryotes.  

⑶ 1<sup>st</sup>. RNA chain initiation: RNA polymerase binds to gene promoter.

> ① RNA releases closed promoter complexes into open promoter complexes.

> ② Both strands of the DNA can encode genes, but any given gene uses only one strand as the template.

>> ○ In the narrow range, only one strand of DNA can be considered as a template.

>> ○ Sense strand (coding strand, (+) strand, non-template strand): The strand to which RNA polymerase does not bind, making it convenient for codon analysis.

>> ○ Anti-sense strands (non-coding strands, (-) strands, template strands): The strand to which RNA polymerase binds.

>> ○ Example: Yeast galactose genes (6 total)

>>> ○ 4 on chromosome 2 (GAL7, GAL10, GAL1, MEL1), 1 on chromosome 12 (GAL2), 1 on chromosome 4 (GAL3).

>>> ○ GAL7, GAL10, GAL1 are on one strand, MEL1 is on the other strand.

> ③ Promoter

>> ○ About 40 base pairs of DNA indicating transcription initiation site.

>> ○ Site where RNA polymerase binds.

>> ○ Prokaryotic promoter consensus

>>> ○ **Example 1.** Pribnow box (-10 box): TATAAT (or TATGTT, ⋯)

>>> ○ **Example 2.** -35 box: TTGACA (or TTTACA, ⋯)

>> ○ Eukaryotic promoter consensus

>>> ○ **Example 1.** TATA box (Goldberg-Hogness box, -25 box): TATA

>>> ○ **Example 2.** CAAT box (-75 box): CAAT

>>> ○ **Example 3.** GC box

<br>

![image](https://github.com/user-attachments/assets/10e54ade-5b47-4a9c-9dff-3220b1318fb2)

**Figure 8.** TATA box motif 

<br>

>> ○ Core promoter: The region of approximately 300 bp from the promoter to just before the start of transcription.

> ④ Upstream / downstream

>> ○ The first nucleotide transcribed into RNA is designated as +1.

>> ○ Based on +1, the 5' direction is referred to as upstream (-), and the 3' direction as downstream (+).

> ⑤ Sigma factors in E. coli: All recognize the -35 upstream region.  

>> ○ σ<sup>70</sup> (σ<sup>D</sup>): Recognizes TTGACA. The primary sigma factor involved in most genes during normal growth.  

>> ○ σ<sup>54</sup>: Recognizes TTGGCACA. Involved in nitrogen anabolism.  

>> ○ σ<sup>38</sup>: Recognizes CCGGCG. The major sigma factor involved during stationary phase, oxidative stress, and osmotic stress.  

>> ○ σ<sup>32</sup>: Recognizes TNTCNCCTTGAA (N represents any nucleotide). Involved during heat shock stress.  

>> ○ σ<sup>28</sup>: Recognizes TAAA. Involved in the synthesis of flagellar genes.  

>> ○ σ<sup>24</sup>: Recognizes GAACTT. Involved in the response to misfolded proteins in the periplasm.  

>> ○ σ<sup>19</sup>: Recognizes AAGGAAAAT. Involved in genes responsible for iron ion transport.  

⑷ 2<sup>nd</sup>. Elongation of the RNA chain: RNA polymerase polymerizes RNA base complementary to DNA base (template strand) of gene into 5 '→ 3'.

> ① Transcription bubble = RNA transcript + DNA + RNA polymerase

> ② RNA-DNA hybrid

> ③ RNA polymerase: No primer required. No topoisomerase. No exonuclease activity leading to a higher transcription error rate.

> ④ E. coli has one type of DNA-dependent RNA polymerase.

>> ○ Holoenzyme form: α2ββ'ωσ

>> ○ Core enzyme form: α2ββ'ω. Binds to the promoter with the help of the σ<sup>70</sup> factor, which recognizes the promoter.  

>> ○ σ<sup>70</sup>: Recognizes the promoter and initiates synthesis. Can be reused.  

>> ○ β: Forms phosphodiester bonds and binds to rNTPs.  

>> ○ β': Binds to the DNA template. 

>> ○ Core enzyme: Only involves the α subunit.

>>> ○ No specific binding occurs.  

>>> ○ Tight non-specific DNA binding.  

>>> ○ K<sub>d</sub> ≈ 5 × 10<sup>-12</sup> M

>> ○ Holoenzyme: Includes the σ<sup>70</sup> subunit

>>> ○ Specific promoter binding occurs.  

>>> ○ Weak non-specific DNA binding.  

>>> ○ K<sub>d</sub> ≈ 10<sup>-7</sup> M  

>>> ○ Finds the promoter 10,000 times faster.

> ⑤ Eukaryotes have three types of DNA-dependent RNA polymerases.

>> ○ RNA pol I: Functions in the nucleolus. Synthesizes 28S rRNA, 18S rRNA, and 5.8S rRNA.

>> ○ RNA pol II: Functions in the nucleoplasm. Synthesizes mRNA.

>> ○ RNA pol III: Functions in the nucleoplasm. Synthesizes tRNA, 5S rRNA, ncRNA, and snRNA.

⑸ 3<sup>rd</sup>. Termination of the RNA chain: In the terminator sequence, RNA polymerase leaves DNA and DNA twists again.

> ① Prokaryotic Terminators

>> ○ Intrinsic Terminator: Transcription is terminated by forming a hairpin (hairpin-oligo-U structure).

>>> ○ A hairpin is formed due to an inverted repeat sequence.

>>> ○ An A-T rich site is located near the inverted repeat sequence, where U bases are transcribed using A as the template.

>>> ○ The transcription product naturally dissociates due to the unstable A=U base pair chain.

>> ○ Rho-dependent Terminator: The Rho factor acts as a DNA-RNA helicase to promote the detachment of the RNA being synthesized.

> ② Eukaryotic Terminators

>> ○ When the poly-A signal sequence (5'-AAUAAA-3') appears on the transcript, endonuclease activity is triggered.

>> ○ The endonuclease cleaves the transcript 10 to 35 nucleotides downstream, leading to its detachment.

>> ○ A poly-A tail, consisting of multiple A bases, is added to the cleaved 3' end.

⑹ 4<sup>th</sup>. mRNA processing (eukaryotic cells only): Mature mRNA moves to the cytoplasm through the nuclear pore. Due to the maturity, the degradation is decreased and the translation is increased.

> ① 1<sup>st</sup>. Capping

<br>

<img width="607" alt="스크린샷 2024-11-30 오후 2 08 24" src="https://github.com/user-attachments/assets/c637a9b3-76bb-46fa-aaad-a1da6f6d6e2b">
 
**Figure 9.** 5'-cap

<br>

> ○ Capping begins simultaneously with transcription.

> ○ 7-MeG, a methylated guanosine derivative, is linked to the 5’ end of pre-mRNA via a 5’-5’ bond (5' carbon - 3 phosphate groups - 5' carbon).  

> ○ The 5'-MeG binds to CBC (cap binding complex) to form a 5’-cap: 5'-cap is also known as m7GpppN.

> ○ Cap: Helps nuclear export, protects mRNA from nucleases, and serves as a recognition site for protein synthesis machinery.  

> ○ Artificial 5’-cap for IVT (in vitro transcription)
  
>> ○ 1<sup>st</sup> generation: mCap (Cap 0) derived from yeast. The disadvantage is the lack of directionality.  

>> ○ 2<sup>nd</sup> generation: ARCA (anti-reverse cap analog) (Cap 0 mimic) Derived from yeast. Introduces a methyl group for directionality.  

>> ○ 3<sup>rd</sup> generation: Trinucleotide cap analog (e.g., CleanCap) (Cap 1). It is excellent because it closely resembles the 5'-cap found in vivo.

> ② 2<sup>nd</sup>. Poly A tail formation

>> ○ Poly A tail is related to transcription terminators in eukaryotes.  

>> ○ PBP (poly A binding protein) adds A bases following the 5’-AAUAAA-3’ sequence.  

>> ○ Poly A tail aids mRNA transport from the nucleus to the cytoplasm and increases mRNA stability.  

> ③ 3<sup>rd</sup>. Splicing

>> ○ Performed by spliceosomes, connecting exon regions only. Requires ATP.  

>> ○ Exons: Encode practical genetic information and untranslated regions (UTRs; for translational regulation). 

>> ○ Introns: Encode non-practical information.  
  
>>> ○ May exist to maintain distances between DNA in chromatin.  

>>> ○ Speculated to have deeper significance.  

>> ○ Spliceosome (snRNP): Composed of snRNA.  
 
>> ○ u<sub>1</sub> snRNP + u<sub>2</sub> snRNP + u<sub>4</sub> snRNP + u<sub>5</sub> snRNP + u<sub>6</sub> snRNP → Removal of u<sub>1</sub> snRNP and u<sub>4</sub> snRNP.  

>> ○ Conserved sequences in pre-mRNA for splicing  

>>> ○ Splice donor, splice acceptor: The splice junction start and end are frequently observed with GT and AG motifs, respectively.

>>> ○ That is, 5'-GU---AG-3': GU at the 5' splice site and AG at the 3' splice site.  
  
>>> ○ Splice acceptor: Pyrimidine rich site, Branch point A

<br>

![image](https://github.com/user-attachments/assets/c82008aa-da58-407a-bdc1-a118a7288a7d)

**Figure 10.** splice donor, splice acceptor

<br>

>> ○ Group I self-splicing

<br>

![image](https://github.com/user-attachments/assets/9c179b7f-0fb3-4e17-8132-75f743359228)

**Figure 11.** Group I self-splicing

<br>

>>> ○ **Step 1.** External guanosine molecule attacks the 5' splice site: As a result, a guanosine molecule binds to the intron, and the 5' exon obtains a 3' OH group.

>>> ○ **Step 2.** The 3' end of the 5' exon attacks the 3' splice site, joining the 5' exon to the 3' exon and excising the intron.  

>>> ○ In this way, the exons gradually become connected.

>> ○ Group II self-splicing 

<br>

![image](https://github.com/user-attachments/assets/3249518c-f399-46a4-b7fc-3e1c08e94b96)

**Figure 12.** Group II self-splicing

<br>

>>> ○ **Step 1.** Internal adenosine molecule attacks the 5' splice site, forming a lariat structure.  

>>> ○ **Step 2.** The 3' end of the 5' exon attacks the 3' splice site, joining the 5' exon to the 3' exon and excising the intron.  

>>> ○ In this way, the exons gradually become connected.

>> ○ **Difference 1.** Initial nucleophile: Group I uses an external guanosine molecule, while Group II uses an internal adenosine molecule.

>> ○ **Difference 2.** Structure: Group I is characterized by a complex loop and helix structure, while Group II is characterized by a lariat structure.

>> ○ **Difference 3.** Distribution  
   
>>> ○ Group I: rRNA, mRNA, and tRNA of bacteria; mitochondrial genome and chloroplast genome of lower eukaryotes; rRNA of the nuclear genome in lower eukaryotes; and some tRNA and mRNA in the chloroplast and mitochondria of higher plants.
  
>>> ○ Group II: rRNA, tRNA, and mRNA of cellular organelles in fungi, plants, and protists; mRNA of bacteria.

>>> ○ Summary: Group I is more common in lower organisms (e.g., bacteria), and Group II in higher organisms (e.g., fungi, plants).  

> ④ 4<sup>th</sup>. mRNA editing: Altering the RNA sequence

>> ○ Rare compared to capping, poly A tail formation, and splicing.  

>> ○ Guide RNA serves as a template.  

>> ○ **Type 1.** Protein length remains unchanged.  

>> ○ **Type 2.** Protein length changes (e.g., apolipoprotein B protein).  
  
>>> ○ In humans, apolipoprotein B protein becomes 100 kDa Apo-B100 in the liver.  

>>> ○ In humans, apolipoprotein B protein becomes 48 kDa Apo-B48 in the small intestine.  

>>> ○ Cytidine deaminase deaminates 5'-CAG-3' in RNA to 5'-UAG-3', creating a stop codon at a new position.

> ⑤ 5<sup>th</sup>. Alternative splicing

>> ○ Some exons on a single pre-mRNA are selectively included, forming multiple mRNA combinations.  

>> ○ >95% of intron-containing human genes undergo alternative splicing.

>> ○ **Result 1.** Improper splicing removes normal exons → May cause cancer.  

>> ○ **Result 2.** Premature stop codons appear as a result of splicing → Removal of abnormal mRNA by NMD (nonsense-mediated mRNA decay).

>> ○ Alternative splicing events

<br>

![image](https://github.com/user-attachments/assets/c29131ad-ad78-4854-aab1-cd47e10077bf)

**Figure 13.** Types of mRNA splicing

<br>

>>> ○ **SE (Skipped Event)**: ~40%. When an entire specific exon is either included or excluded.

>>> ○ **A5SS (Alternative 5' Splice Site)**: ~7.9%. The splice junction at the 5' end of the exon is used differently.

>>> ○ **A3SS (Alternative 3' Splice Site)**: ~18.4%. The splice junction at the 3' end of the exon is used differently.

>>> ○ **RI (Retained Intron)**: <5%. Introns that do not code for amino acid sequences are either retained or spliced.  

>>> ○ **MXE (Mutually Exclusive Exon)**: When one exon is included, the other is spliced out, and vice versa. This exclusive splicing is mainly observed in brain tissue.  

>>> ○ **Alternative Promoter**  

>>> ○ **Alternative Poly-A**  

>> ○ Note that the various types of hemoglobin are not due to alternative splicing but rather time-specific expression.

> ⑥ 6<sup>th</sup>. Exon shuffling

>> ○ A gene is formed with a new combination of exons due to crossover events between different genes.

> ⑦ 7<sup>th</sup>. Trans-splicing

<br>

![image](https://github.com/user-attachments/assets/af0a14f0-5f96-401b-99b3-e38bdf624ba0)

**Figure 14.** Cis/Trans splicing

<br>

>> ○ **Type 1.** Cis-splicing: The process in which exons within a single pre-mRNA are joined through 5' ss (splice site) to 3' ss reactions, resulting in the formation of mRNA.
  
>> ○ **Type 2.** Trans-splicing: The process in which two different pre-mRNAs are joined into a single mRNA molecule.

>>> ○ **2-1.** The process in which the spliced leader (SL) from the 5' tss (trans splice site) is joined with the pre-mRNA at the 3' tss.

>>> ○ **2-2.** Trans-splicing between different genes

>>> ○ **2-3.** Trans-splicing within the same gene: The direction of the template may differ, or the origin may be maternal or paternal. Exon duplication is also possible.

>> ○ SL trans-splicing process
  
>>> ○ **Step 1.** The 5' tss of SL RNA binds to the 3' tss of pre-mRNA.
  
>>> ○ **Step 2.** Mature mRNA contains an SL exon and a TMG cap at the 5' end and is translated in the cytoplasm.
  
>>> ○ **Step 3.** As a result of splicing, a Y-shaped product (SL intron + pre-mRNA outron) is formed, resembling the lariat intron product generated in cis-splicing.
  
>>> ○ **Step 4.** Y-shaped product rapidly degrades.  

<br>

![image](https://github.com/user-attachments/assets/1c2269e0-1a57-4fd9-aa3c-04cf56b6aa2a)

**Figure 15.** SL trans-splicing process

<br>

> ⑧ 8<sup>th</sup>. mRNA stability regulation 

>> ○ Longer mRNA lifespan enables more protein translation.  

>> ○ Prokaryotic mRNA has a lifespan of about 2 to 3 minutes.

>> ○ Eukaryotic mRNA has a lifespan ranging from several days to several weeks.

>> ○ **Example 1.** Hemoglobin mRNA remains intact and is not degraded as long as the red blood cell is alive.  

>>> ○ Globin mRNA is highly expressed in red blood cells, accounting for about two-thirds of total blood RNA.  

>> ○ **Example 2.** Transferrin: An iron-importing and transport protein.  

>>> ○ Iron Response Element (IRE): Located downstream of the coding region. An AT-rich sequence.  

>>> ○ When Fe<sup>3+</sup> increases: IRE-binding protein (IRE-BP) becomes inactive → The poly-A sequence of transferrin mRNA is removed → mRNA is degraded.  

>>> ○ When Fe<sup>3+</sup> decreases: IRE-BP becomes active → The poly-A sequence of transferrin mRNA is protected → mRNA is stabilized → Expression increases.

⑺ 5<sup>th</sup>. rRNA, tRNA transcription

> ① rRNA is transcribed in the nucleolus by RNA polymerase I.  

> ② mRNA and microRNA are transcribed in the nucleoplasm by RNA polymerase II.  

> ③ tRNA is transcribed in the nucleoplasm by RNA polymerase III.  

> ④ The promoter for rRNA is located far upstream of the gene.  

> ⑤ The promoter for mRNA and microRNA is located close upstream of the gene.  

> ⑥ The promoter for tRNA is located close downstream of the gene.  

> ⑦ TBP (TATA-binding protein) is required for all.

⑻ 6<sup>th</sup>. rRNA processing, tRNA processing

> ① rRNA processing

>> ○ **Type 1.** 45S rDNA in the nucleolus → 18S rRNA + 5.8S rRNA + 28S rRNA

>> ○ **Type 2.** Cleaving unmethylated bases

> ② tRNA processing: Slight cleavage of 3' and 5' from pre-tRNA.

⑼ Regulation of Transcription in Prokaryotes

> ① Regulatory elements
   
>> ○ Common sequences in the promoter (P) and operator gene (O) (repressor binding site)

>> ○ Activator binding site

>>> ○ Example: E. coli cAMP-CAP (Catabolite Activator Protein) complex recognition site

> ② Operon: A mechanism for coordinated regulation by transcribing multiple genes into a single mRNA.  

> ③ Regulon: A mechanism where multiple operons are regulated by a single protein.  
 
>> ○ Example: Maltose regulon in E. coli. 

> ④ Regulation by transcription repressors 

>> ○ When a repressor binds to the operator gene, it prevents RNA polymerase from binding to the promoter.  

>> ○ Inducible operon: In the presence of an inducer, the repressor bound to the operator is removed, allowing RNA polymerase to bind to the promoter and initiate transcription.  

>>> ○ **Example 1.** Catabolic operon: Lactose operon (3 genes)

>> ○ Repressor operon: In the absence of an inducer, the repressor bound to the operator is removed, enabling RNA polymerase to bind to the promoter and initiate transcription.  

>>> ○ **Example 1.** Biosynthetic operon: Tryptophan operon

> ⑤ Regulation by transcription activators

⑽ Operon: Having ORFs for several genes under one promoter / operating region and being expressed simultaneously

> ① Summary

>> ○ Inducible operon: The substance of interest finally promotes the expression of the operon.

>> ○ Repressor operon: The substance of interest finally inhibits the expression of the operon.

>> ○ Positive regulation: Regulation involving activators. In inducible operons, the activator is activated by the target molecule, while in repressor operons, the activator is inhibited by the target molecule.  

>> ○ Negative regulation: Regulation involving repressors. In inducible operons, the target molecule inhibits the repressor, while in repressor operons, the target molecule activates the repressor.

> ② Lactose operon (lac operon): Inducible operon

>> ○ Composition of lac operon: (Regulator) - CAP binding site - Promoter - Operating region - lac Z - lac Y - lac A

<br>

![image](https://github.com/user-attachments/assets/0a420dde-a936-47e6-aeda-b2c55dd57bda)

**Figure 16.** Composition of lactose operon 

<br>

>>> ○ Regulator (lac I): Sites that encode inhibitors. It is not included in the operon because it has its own promoter (constantly expressed).

>>> ○ Operating site

>>> ○ **Gene 1.** Beta-galactosidase (lac Z)

>>>> ○ β-galactosidase: Lactate lyase. Converts lactose into allolactose.

>>>> ○ lac Z actually produces the α peptide, which is the N-terminal part of β-galactosidase.

>>>> ○ α complementation: The process by which the α peptide combines with another peptide from the host to form β-galactosidase.

>>> ○ **Gene 2.** Beta-galactosid permease (lac Y): Lactose transporter. Expressed on cell membrane.

>>> ○ **Gene 3.** Beta-galactoside transacetylase (lac A)

>>>> ○ Beta-galactoside transacetylase removes by-products.

>> ○ Negative regulation of lac operon

>>> ○ The lac operon repressor can bind to the operator site.  

>>> ○ The lac operon repressor is produced by the **lac I gene** and inhibits RNA polymerase from binding to the promoter.  

>>> ○ When the lac repressor binds to allolactose, a derivative of lactose, it can no longer bind to the operator site (lactose acts as an inducer).

>> ○ Positive regulation of lac operon

>>> ○ **Background**: The promoter of the lac operon has low affinity for RNA polymerase, so transcription does not occur efficiently even in the absence of the repressor.  

>>> ○ Adenylyl cyclase (AC) converts ATP into cAMP.  

>>> ○ Catabolite Activator Protein (CAP) binds to cAMP to form a complex.  

>>> ○ The CAP-cAMP complex facilitates the binding of RNA polymerase to the promoter.  

>>> ○ Glucose can inhibit adenylyl cyclase through an allosteric inhibition site, preventing the above mechanism from functioning.  

>>> ○ **Result**: Glucose, being more efficient, is prioritized over the operation of the lac operon, avoiding the unnecessary synthesis of enzymes from the lac operon.  

>> ○ The lac operon provides a mechanism for diauxic growth in media containing both glucose and lactose.

>>> ○ Glucose (+), Lactose (+): cAMP decreases → Positive regulation decreases → Lactose-degrading enzymes are not produced.

>>> ○ Glucose (+), Lactose (-): cAMP decreases → Positive regulation decreases → Lactose-degrading enzymes are not produced.

>>> ○ Glucose (-), Lactose (+): Allolactose increases → Repressor binding to the operator decreases → Lactose-degrading enzymes are produced.

>>> ○ Glucose (-), Lactose (-): Allolactose decreases → Repressor binding to the operator increases → Lactose-degrading enzymes are not produced.

>> ○ Operons derived from the lac operon

>>> ○ tac operon

>>> ○ lac I<sup>-</sup>: The repressor is defective and cannot bind to the operator site, resulting in constant activation.

>>> ○ lac I<sup>S</sup>: Lactose cannot bind to the repressor, resulting in constant inactivation.

>>> ○ lac O<sup>c</sup>: The operator site is defective, preventing the repressor from binding to the operator, resulting in constant activation.

>>> ○ lac Z<sup>-</sup>: Encodes a defective lac Z gene. β-galactosidase is not expressed, regardless of operon activity.

> ③ Tryptophan operon (trp operon): Inhibitory operon  

>> ○ Composition of tryptophan operon: (Regulator) - Promoter - Operator site - Structural genes

<br>

<img width="830" alt="스크린샷 2025-01-31 오전 11 00 47" src="https://github.com/user-attachments/assets/e6e20546-39f6-46b4-b356-a4227767487c" />

**Figure 17.** Composition of tryptophan operon

<br>

>>> ○ Regulator: Encoding the suppressor. Not included in operon because it has its own promoter (always expressed).

>>> ○ Five structural genes (trp E, D, C, B, A) encoding the five enzymes necessary for the biosynthesis process of tryptophan.

>>> ○ ξ polypeptide, δ polypeptide, indole glycerolphosphate synthase, β polar peptide, α polar peptide, respectively.

>> ○ **Adjusting mechanism 1.** Negative control

>>> ○ The trp promoter has a relatively high binding affinity for RNA polymerase → No need for an activator protein.

>>> ○ By the regulatory gene (trpR), the trp repressor binds to trp and attaches to the operator site, acting as a corepressor.

>> ○ **Adjusting mechanism 2.** Attenuation regulator mechanism

>>> ○ trp mRNA has four regions that can form complementary base pairs, each located near the trp coding region.

>>> ○ Leader sequence: Translated into a protein, but unrelated to tryptophan biosynthesis.

>>> ○ The term "leader" in leader peptide simply means that it is synthesized before the tryptophan-related enzymes.

<br>

<img width="664" alt="스크린샷 2025-01-31 오전 11 03 19" src="https://github.com/user-attachments/assets/bf169e61-9c79-4de9-9800-37d503f591d7" />

**Figure 18.** DNA sequence of the trp operon

<br>

<img width="797" alt="스크린샷 2025-01-31 오전 11 05 58" src="https://github.com/user-attachments/assets/9e58f066-37ba-430e-aa6a-1f8913e466c4" />

**Figure 19.** Leader peptide mRNA sequence of the trp operon

Following the poly-U sequence, trp E, D, C, B, A sequences are present.

<br>

>> ○ **Case 1.** When tryptophan is scarce

>>> ○ Even if RNA polymerase transcribes up to region 4, the shortage of tryptophan tRNA slows down the translation of the trp genetic code.  

>>> ○ The ribosome remains at sequence 1 (during the synthesis of the leader peptide).  

>>> ○ This allows sequences 2 and 3 to form a hairpin structure.  

>>> ○ Since the RNA polymerase is positioned far from the terminator sequence, intrinsic transcription termination does not occur.  

>>> ○ As a result, trp mRNA 2 is produced.

>> ○ **Case 2.** When tryptophan is abundant

>>> ○ When RNA polymerase transcribes up to region 4, the translation speed is fast, positioning the ribosome at sequences 1 and 2.  

>>> ○ This leads to the formation of a hairpin structure between sequences 3 and 4.  

>>> ○ Transcription termination occurs (intrinsic transcription termination).  

>>> ○ As a result, trp mRNA 1 is produced.

>> ○ **Case 3.** When tryptophan is completely absent

>>> ○ Even if RNA polymerase transcribes up to region 4, no ribosome is bound to the mRNA.  

>>> ○ This results in the formation of a hairpin structure between sequences 3 and 4.  

>>> ○ Transcription termination occurs (intrinsic transcription termination).  

>>> ○ As a result, trp mRNA 1 is produced.

<br>

<img width="532" alt="스크린샷 2025-01-31 오전 11 33 28" src="https://github.com/user-attachments/assets/93640c15-30ff-4736-a30f-565fea985906" />

**Figure 20.** Hairpin formation in the attenuation regulator mechanism of trp operon

<br>

>>> ○ If the tryptophan codon in trp L is deleted, the ribosome doesn’t need to wait for the tryptophan amino acid. In this case, since the ribosome doesn’t pause (similar to **Case 2**, where the waiting time for tryptophan is short), trp mRNA 1 is produced.

> ④ tac operon  

>> ○ Isopropylthiogalactoside (IPTG): Artificial substances that activate lac and tac promoters, such as allolactose. It remains undegraded, maintaining a constant concentration and continuously activating the promoter. 

>> ○ An artificial promoter that requires IPTG for expression, like the lac promoter.

>> ○ Stronger expression than trp promoter.

>> ○ Inhibitors are made from lac I or lac I<sup>Q</sup>.

>>> ○ lac I<sup>Q</sup>: A repressor gene mutated to be expressed more strongly than lac I.

> ⑤ P<sub>L</sub> operon

>> ○ cI repressor (cI 857) gene: Inactivate P<sub>L</sub> promoter by making cI repressor at low temperature.

> ⑥ T7 Operon

> ⑦ Partial diploid: Cis-trans diploid

>> ○ Partial diploid: If the F plasmid carries only a part of the E. coli genome, the genes on the F plasmid exist in a diploid state.  

>> ○ Partial dimer: Conjugation through sex pili. Considered a partial diploid.  

>> ○ Cis-acting element: A promoter in a partial diploid that affects only its own operon. Functions constitutively. 

>>> ○ Example: O<sup>c</sup> in lac-derived operons.  

>> ○ Trans-acting element: A promoter in a partial diploid that affects both copies of the diploid. Functions conditionally.  

>>> ○ Example: I<sup>-</sup> and I<sup>s</sup> in lac-derived operons.

> ⑧ Recombinant protein and inducible promoter

>> ○ Cells that continuously produce recombinant proteins experience a metabolic burden, leading to growth inhibition, plasmid instability, and reduced production yield.

>> ○ Expression strategy: Increase cell density to a certain level before adding an inducer to initiate expression.

>> ○ Conclusion: Inducible promoters with controllable expression are widely used to enhance production yield.

⑾ Regulation of eukaryotic transcription

> ① Transcription factor

>> ○ Overview

>>> Eukaryotic RNA polymerase cannot bind to promoter alone.

>>> ○ Transcription begins when general and specific transcription factors bind to various regulatory sequences upstream of the gene (e.g., TATA box, CAAT box), allowing RNA polymerase to initiate transcription.

>>> ○ The number of transcription factors is approximately 1,600.  

>>> ○ Pioneer factor: A special type of transcription factor that can bind even when chromatin is in a closed state. It is the first transcription factor to bind and is much less abundant.  

>>> ○ The CTCF transcription factor binds to DNA for a longer duration (~several minutes) compared to other transcription factors.

>> ○ General transcription factor: Universally present in each cell (e.g., TFIIA, TFIIB, TFIID, TFIIE, TFIIF, TFIIH, TFIIJ, etc.)

>>> ○ TFIID: A protein complex consisting of 13 different proteins.  

>>> ○ TFIIH: A protein complex consisting of 10 different proteins.

>> ○ Special transcription factor: Specifically present 

>>> ○ **Example 1.** DBD(DNA-binding domain)

>>> ○ **Example 2.** AD(activation domain)

>>> ○ **Example 3.** SP1: Zinc finger. Recognizes 5'-GGGCGG-3'. Monomer 

>>> ○ **Example 4.** AP-1: Basic zipper. Recognizes 5'-TGA(G/C)TCA-3'. Dimer

>>> ○ **Example 5.** C/EBP: Basic zipper. Recognizes 5'-ATTGCGCAAT-3'. Dimer

>>> ○ **Example 6.** Heat shock factor: Basic zipper. Recognizes 5'-XGAAX-3'. Trimer 

>>> ○ **Example 7.** ATF/CREB: Basic zipper. Recognizes 5'-TGACGTCA-3'. Dimer 

>>> ○ **Example 8.** c-Myc: Basic helix-loop-helix. Recognizes 5'-CACGTG-3'. Monomer 

>>> ○ **Example 9.** Oct-1: Helix-turn-helix. Recognizes 5'-ATGCAAAT-3'. Monomer 

>>> ○ **Example 10.** NF-1: Novel structure. Recognizes 5'-TTGGCXXXXXGCCAA-3'. Dimer 

>> ○ Proximal regulatory element: Located near the promoter.

>>> ○ 1<sup>st</sup>. TFIID binds to the TATA box through the TBP subunit of TFIID.  

>>> ○ 2<sup>nd</sup>. TFIIA and TFIIB stabilize the entire complex.  

>>> ○ 3<sup>rd</sup>. TFIIF and RNAPII (RNA polymerase II) bind to the complex.  

>>>> ○ GTF is involved in recruiting RNAPII.  

>>> ○ 4<sup>th</sup>. TFIIE and TFIIH additionally bind to the complex.  

>>>> ○ TFIIH phosphorylates the CTD (C-terminal domain) of RNAPII in the nucleus → Releases TFIID and activates helicase activity.  

>>>> ○ RNAPII, unlike α<sub>2</sub>ββ'ω, does not have helicase activity.  

>>> ○ 5<sup>th</sup>. Transcription is initiated: PIC (pre-initiation complex) is formed.  

>>>> ○ GTF is also involved in transcription initiation.  

<br>

![image](https://github.com/user-attachments/assets/6f613d1d-d74c-4300-b187-237780c3ec58)

**Figure 21.** PIC formation process

<br>

>> ○ Distal regulator: Mainly distributed upstream of the promoter.

>>> ○ Enhancer (Amplification Sequence): Facilitates RNA polymerase binding to the promoter. It can be located upstream or downstream. The length ranges from approximately 50 to 1500 bp.

>>> ○ Silencer (Repressive Sequence): Unconditionally inhibits RNA polymerase. Located upstream of the promoter.

>>> ○ Insulator: Suppresses the enhancer to regulate transcription. Positioned between the enhancer and the promoter.

>>> ○ Stress Response Element (SRE)

>> ○ DNA binding protein

>>> ○ HMG protein: Bends the DNA so that RNA pol II can function.

>> ○ Common structures of DNA-binding proteins (DNA binding motifs)

>>> ○ Helix-turn-helix motif: Consists of two α-helices; found in CAP and operon operator regions. 

>>> ○ Zinc finger motif: Observed in steroid hormone receptors.  

>>> ○ Leucine zipper motif (e.g., AP-1, which is involved in mammalian cell growth and division)

>>> ○ Helix-loop-helix motif

>> ○ The mouse and human genomes are folded into more than 10,000 loops.

> ② Regulation of the modification step (pre-mRNA → mature mRNA)

> ③ DNA Methylation: A key concept in epigenetics

>> ○ Methyl-CpG binding domain containing protein (MBD)

>>> ○ MBD1 ~ MBD6

>>> ○ MeCP2: Related to Rett syndrome.

>>> ○ The CpG dinucleotide is almost always methylated.

>>> ○ Approximately 70% of promoters near the transcription start site (TSS) overlap with CpG islands: When the gene is expressed, the CpG island remains unmethylated.

>>> ○ [The deamination of cytosine](https://jb243.github.io/pages/1386#2-imines): The methyl group attaches to the C base → The C base is prone to becoming a T base through deamination → As a result, CpG is relatively rare in the genome.

<br>

<img width="406" alt="스크린샷 2025-02-16 오전 11 48 40" src="https://github.com/user-attachments/assets/b0595257-4311-4102-9b8d-bdfbda7e9764" />

**Figure 22.** The deamination of cytosine 

<br>

>> ○ 70-80% of the human genome is methylated.

>> ○ Global methylation map

<br>

![image](https://github.com/user-attachments/assets/4ea4f66a-aa0e-4f73-86a4-32ab60646cef)

**Figure 23.** Global methylation map

<br>

>> ○ Chromatin with attached methyl groups condenses, leading to transcriptional inactivation.

>>> ○ Expression level: Euchromatin ＞ Heterochromatin

>>> ○ Promoters and transcription factor binding sites have a high content of cytosine bases (e.g., GC box).

<br>

<img width="408" alt="스크린샷 2025-01-31 오후 2 48 06" src="https://github.com/user-attachments/assets/480a1105-233c-4745-b575-650b76d715cc" />

**Figure 24.** DNA methylation and transcription inactivation 

<br>

> ④ Histone modification

>> ○ [The binding of histones and DNA](https://jb243.github.io/pages/67#:~:text=%E2%91%A2-,Nucleosome%C2%A0,-%3A%C2%A0Also%20known)

>> ○ Among histones, **H3** undergoes the most modifications, while **H2A** and **H2B** are modified much less frequently.

>> ○ Histone modifications occur on the histone tail.

>> ○ Histone acetylation: Acetyl groups bind exclusively to lysine residues on the histone tail.  

>>> ○ The NH<sub>2</sub> group of lysine acts as a base, carrying a positive charge, which plays a crucial role in histone condensation.  

>>> ○ When an acetyl group (-COCH<sub>3</sub>) binds to the NH<sub>2</sub> group of lysine, the histone becomes neutralized.  

>>> ○ Histone neutralization → Weakens electrostatic interactions between DNA and histones → Leads to euchromatin formation and transcription activation.  

>>> ○ Acetylation of lysine 9 on histone H3 is a representative example.

>> ○ Histone acetyltransferase (HAT)

>>> ○ **Type 1.** HAT A

>>>> ○ HAT A, located in the nucleus, acetylates histones within nucleosomes.

>>>> ○ Acetylation of histones neutralizes the NH<sub>2</sub> group of lysine, which carries a **positive charge**, leading to the relaxation of nucleosome condensation.  

>>> ○ **Type 2.** HAT B

>>>> ○ HAT B, located in the cytoplasm, acetylates free histones required for chromatin assembly during cell division.  

>>>> ○ HAT B interacts with newly synthesized histone H4 lysine residues in the cytoplasm: HAT1/HAT2/H4

>>>> ○ Afterward, the newly synthesized histone H3 binds to histone H4, and with the help of karyopherin, they are transported into the nucleus: HAT1/HAT2/H3/H4  

>>>> ○ It has not yet been determined whether the HAT1/HAT2/H3/H4 structure exists as a heterodimer or a heterotetramer.  

>>>> ○ The HAT1/HAT2/H3/H4 complex facilitates the transfer of the H3/H4 complex onto DNA.

>> ○ Histone deacetylation: Opposition of histone acetylation

>>> ○ Valproic acid: An inhibitor of histone deacetylase

>> ○ Histone methylation

>>> ○ Occurs on lysine or arginine residues of the histone tail, in contrast to histone acetylation, which occurs only on lysine.  

>>> ○ Increases the positive charge of histones → Enhances electrostatic interactions between DNA and histones → Leads to heterochromatin formation and transcriptional inactivation.

>>> ○ Methylation can occur up to three times at the same site, resulting in mono-methylation, di-methylation, and tri-methylation.  

>>> ○ Histone methylation can also activate gene transcription in certain contexts.

>> ○ Histone Phosphorylation: The effect on transcription (activation or inactivation) varies depending on the phosphorylation site.

>> ○ Histone modifications and their effects on activation/inhibition

>>> ○ Only lysine amino acids can undergo mono-, di-, tri-methylation.

>>> ○ As a result, transcription activation or repression is not always straightforward.

<br>

| Type              | H3K4        | H3K9        | H3K14       | H3K27       | H3K79               | H4K20       | H2BK5       |
|-------------------|-------------|-------------|-------------|-------------|---------------------|-------------|-------------|
| mono-methylation | activation  | activation  |             | activation  | activation          | activation  | activation  |
| di-methylation   |             | repression  |             | repression  | activation          |             |             |
| tri-methylation  | activation  | repression  |             | repression  | activation, repression |             | repression  |
| acetylation      |             | activation | activation  | activation  |                     |   |             |

**Table 2.** Effects of histone modifications on activation / inhibition

For example, H3K27 refers to the 27th lysine amino acid (Lys) in the H3 histone.

<br>

> ⑤ Chromatin remodeling protein

>> ○ Remodeling protein a: Enable ATP.

>> ○ Remodeling protein b: ATP not used.

> ⑥ [X chromosome inactivation](https://jb243.github.io/pages/72#:~:text=%E2%91%A4-,X%20inactivation,-%3A%20Observed%20only)

> ⑦ RNA interference (RNAi): Also referred to as RNA silencing.

>> ○ **Type 1.** miRNA (micro RNA): A small single-stranded RNA processed through the nucleus and cytoplasm.

>>> ○ Overview

>>>> ○ Starts with single-stranded one.

>>>> ○ Small non-coding RNA (~22 nucleotides)

>>>> ○ Transcribed by RNA polymerase II.

>>>> ○ Generated by endonucleases.

>>>> ○ Well conserved.

>>>> ○ Humans have over 2,000 types of miRNA.

>>>> ○ Simple ternary interaction and its known determinants explain only 10-20% of miRNA targeting.

>>>> ○ More than 60% of human protein-coding genes are direct targets of miRNA.

>>> ○ 1<sup>st</sup>. Gene → pri-miRNA: RNA pol II is involved.

>>> ○ 2<sup>nd</sup>. pri-miRNA → pre-miRNA: Drosha, Dgcr8, and Pasha are involved. pre-miRNA has a hairpin structure.

>>> ○ 3<sup>rd</sup>. pre-miRNA is exported out of the nucleus: RNA-GTP and Exportin-5 are involved.

>>> ○ 4<sup>th</sup>. pre-miRNA → miRNA/miRNA* duplex: Dicer is involved. miRNA/miRNA is dsRNA.

>>>> ○ miRNA/miRNA* duplex is about 20 bp.

>>> ○ 5<sup>th</sup>. miRNA/miRNA* duplex → mature miRNA: Ago and RISC are involved. miRNA is ssRNA.

>>>> ○ Ago(Argonaute)

>>>> ○ RISC(RNA-induced silencing complex)

>>> ○ 6<sup>th</sup>. Mature miRNA binds to mRNA: Ago is involved.

>>> ○ When miRNA and mRNA are fully complementary, mRNA degradation** occurs.

>>> ○ When miRNA and mRNA are partially complementary, translation is inhibited.

>>> ○ miRNA targeting

>>>> ○ CST(canonical site type): incomplete sequence

>>>>> ○ 6mer site: NNNNNN-5'

>>>>> ○ 7mer-A1 site: NNNNNNA-5'

>>>>> ○ 7mer-m8 site: NNNNNNN-5'

>>>>> ○ 8mer site: NNNNNNNA-5'

>>>> ○ NST(noncanonical site type)

>>>>> ○ Offset 6mer: 6mer site, 7mer-A1 site, 7mer-m8 site, 8mer site 

>>>>> ○ Centered site 

>>>>> ○ Pivot pairing site

>>>>> ○ Single mismatch site  

>>>>> ○ AGO CLIPSeq-based analysis

>>>>> ○ MIRZA site 

>>>> ○ miRNA targeting enhancer: Pumilio, PCBP2, FUS, PTBP1

>>>> ○ miRNA targeting suppressor: Dnd1, RBM38, HuR, IGF2BP1, PTBP1

>> ○ **Type 2.** siRNA (small interfering RNA, short intefering RNA)

>>> ○ First introduced in 1998.

>>> ○ Similar to miRNA, but starts as a double-stranded RNA.

>>> ○ Also called small non-coding RNA: Artificially synthesized (approximately 20-27 nucleotides).

>>> ○ siRNA: Refers to viral RNA or **experimentally introduced RNA. pre-miRNA (shRNA) or dsRNA.

>>> ○ Blood half-life: 4 minutes. With a vector, the blood half-life can increase to several hours.

>>> ○ 1<sup>st</sup>. If introduced as a pre-miRNA: pre-miRNA → miRNA/miRNA* duplex. Dicer is involved.  

>>> ○ 1<sup>st</sup>. If introduced in the form of dsRNA: it is cleaved by Dicer without passing through the pre-miRNA stage.

>>> ○ 2<sup>nd</sup>. miRNA/miRNA* duplex → miRNA: Ago and RISC are involved.  

>>> ○ 3<sup>rd</sup>. miRNA binds to mRNA: Ago is involved.  

>>> ○ 4<sup>th</sup>. Can suppress the expression of any gene.

>> ○ **Type 3.** lncRNA (Long non-coding RNA)

>>> ○ 200 nucleotides or longer

>>> ○ Transcribed by RNA polymerase II.

>>> ○ Mostly undergo 5'-capping, partial polyadenylation, and splicing. 

>>> ○ Poorly conserved.

>>> ○ Humans have over 30,000 types of lncRNA.

>> ○ **Type 4.** sciRNA: small circular interfering RNA

>> ○ **Type 5.** ASO(antisense oligonucleotide)

>>> ○ First introduced in 1978.

>>> ○ **Function 1.** RNaseH recruitment → target mRNA degradation

>>> ○ **Function 2.** splicing modification → exon inclusion / exclusion

>>> ○ **Function 3.** miRNA targeting → miRNA sequestration

>> ○ The comparison of miRNA and siRNA

>>> ○ miRNA has a hairpin structure, whereas siRNA does not.

>>> ○ miRNA has both partial and perfect complementarity mechanisms, whereas siRNA has only the perfect complementarity mechanism. The following table can be filled only with the complementarity.

<br>

| Category                        | miRNA                                     | siRNA                                       |
|--------------------------------|-------------------------------------------|---------------------------------------------|
| Biogenesis                     | ①        | ②      |
| Complementarity with target    | ③                  | ④                    |
| Mechanism of action            | ⑤  | ⑥                             |
| Number of targets              | ⑦                         | ⑧             |


**Table 3.** The comparison of miRNA and siRNA

<br>

>> ○ Domain-specific RNA interference

>>> ○ Animals: RNAi is present.  

>>> ○ Plants: Exhibit PTGS (post-transcriptional gene silencing).  

>>> ○ Prokaryotes and Archaea: Do not have RNA interference but instead possess **restriction enzymes**.

<br>

<br>

## **5\. Translation**

⑴ tRNA

> ① tRNA: 20 types. Binds to the **mRNA codon** via the anticodon.

>> ○ Composed of approximately 80 nucleotides.

> ② Aminoacyl tRNA synthetase: 20 types

>> ○ A specific amino acid is bound to the corresponding tRNA to produce an active amino acid (aa-tRNA).

>> ○ Key regions recognized by aminoacyl-tRNA synthetase on tRNA:

>>> ○ tRNA anticodon loop (mRNA-side element): Recognizes the loop containing the anticodon of tRNA to ensure the selection of the correct tRNA.  

>>> ○ 3' acceptor stem base (Amino acid-side element): The 3' end of tRNA where a specific tRNA binds to its corresponding amino acid.

>> ○ Amino acid binds to the 3' end of tRNA (using ATP): 3'-CCA serves as the amino acid attachment site.

>>> ○ A dehydration condensation reaction occurs between the 3'-OH group of the pentose at the 3' end and the -COOH group of the amino acid.

>>> ○ **Step 1.** -COO<sup>-</sup> → -CO(AMP): The carboxyl (-COOH) end of the amino acid binds to AMP using energy from ATP hydrolysis.

>>> ○ **Step 2.** -CO(AMP) → -CO(tRNA): The amino acid is transferred from AMP to tRNA. (Enzyme: aminoacyl-tRNA synthetase)

>> ○ Double sieve model: The mechanism by which aminoacyl-tRNA synthetase selects a specific amino acid.  

>>> ○ First Sieve (activation site): 

>>>> ○ Excludes amino acids larger than isoleucine (e.g., phenylalanine). 

>>>> ○ Involves an enzyme that specifically binds based on size and structure.  

>>> ○ Second Sieve (editing site):

>>>> ○ Excludes amino acids smaller than isoleucine (e.g., alanine).  

>>>> ○ Ensures that only the correct amino acid-AMP complex binds to tRNA.

>> ○ Mismatch between the amino acid and anticodon can be corrected, but no correction is possible in later translation stages.

>> ○ Only L-form amino acids are used. For reference, only D-form glucose exists in the body.

<br>

![image](https://github.com/user-attachments/assets/1d7bdaa2-d7bd-4224-9c63-a7ffcc1a1f71)

**Figure 25.** Structure of tRNA

<br>

> ③ Primary transcripts of tRNAs become mature tRNAs through post-transcriptional modification.

>> ○ Mature tRNA: Single stranded RNA consisting of about 80 to 90 nucleotides.

>> ○ Some of the bases in the tRNA's specific site are chemically modified (methylguanin, ribothymidine, dihydrouridine, etc.) and these modifications give stability.

>> ○ Add three nucleotides (-CCA) to the 3' end of tRNA.

>> ○ RNase D, a type of exonuclease, is involved.

> ④ Mature tRNA has three rings.

>> ○ Because hydrogen bonds occur between the base and the OH group of ribose.

>> ○ Two-dimensional shape is planar clover leaf shape.

>> ○ Three-dimensional shape is L-shaped.

>> ○ 5' end is shorter because amino acid binds to 3'-OH group of tRNA.

> ⑤ An experiment on the specificity of tRNA recognition by the protein synthesis machinery.

⑵ rRNA

> ① Eukaryotes: Four different rRNAs (5S rRNA, 5.8S rRNA, 18S rRNA, 28S rRNA) are needed.

>> ○ 45S rDNA in nucleolus → 45S rRNA (by RNA pol I) → 18S rRNA + 5.8S rRNA + 28S rRNA

>> ○ 45S rDNA in nucleolus → 5S rRNA (by RNA pol III)

> ② Prokaryotes: Requires three rRNAs (5S rRNA, 16S rRNA, 23S rRNA). There is one RNA pol.

> ③ Ribosome assembly

>> ○ Proteins synthesized from ribosomes in the cytoplasm are delivered to the nucleolus through the nuclear pores.

>> ○ Protein and rRNA assemble in nucleolus → Two ribosomal subunits

>> ○ Ribosome small subunit and large subunit are assembled after migration to cytoplasm (translation initiation complex).

>> ○ Microorganisms produce approximately 100,000 ribosomes per hour.

> ④ Eukaryotic ribosomes: 80S

>> ○ 60S large subunit: 28S, 5.8S, 5S rRNA + 46 different proteins. There is tRNA binding site.

>>> ○ A site (amino site)

>>> ○ P site (peptide site)

>>> ○ E site (exit site)

>> ○ 40S small subunit: 18S rRNA + 33 proteins. There is mRNA binding site.

> ⑤ Prokaryotic ribosome: Contains approximately 55 proteins. 70S (Archaea also have 70S ribosomes).

>> ○ 50S large subunit: 5S, 23S rRNA

>> ○ 30S small subunit: 16S rRNA

> ⑥ snoRNA is involved in rRNA processing.

>> ○ Bacteria: rRNA processing via rRNA cleavage. RNase III, RNase P, RNase E are involved.

⑶ 1<sup>st</sup>. Initiation: Ribosomes bind to the ribosome binding site (RBS).

> ① Initiation of prokaryotes

>> ○ 5'-UTR (5'-Untranslated Region): The untranslated region from the 5' end up to just before the start codon.

>> ○ SD (Shine-Dalgarno) sequence: Purine-rich sequence of 5'-AGGAGGU-3'.

>>> ○ The 30S subunit (16S rRNA + proteins) binds to the Shine-Dalgarno sequence and moves until it reaches the AUG start codon on mRNA.

>>> ○ First discovered in E. coli in the mid-1970s as a sequence interacting with 16S rRNA.

>>> ○ Located upstream of the TSS (start codon): Within 50 bp of the TSS. Typically positioned within 5-10 bp of the TSS.

>> ○ The initiator tRNA (formyl-Met-tRNA) recognizes the start codon on mRNA and binds to the P site (using GTP). 

>> ○ Initiation factors (IF-1, IF-2, IF-3) and the large ribosomal subunit bind to form the translation initiation complex.

> ② Initiation of eukaryotes

>> ○ The small subunit binds to the 5'-cap end of the mRNA to initiate translation: 18S rRNA binds.  

>> ○ 5'-UTR: The untranslated region from the 5'-cap to just before the start codon.  

>> ○ Just before the start codon, there is a common sequence called the Kozak sequence.

>>> ○ Marilyn Kozak first discovered the Kozak sequence.  

>>> ○ Kozak sequence: 5'-A/GCCACC-3'

>>> ○ Immediately after the Kozak sequence, it follows as 5'-AUGG-3'.

>>> ○ That is, the 5'-UTR of eukaryotes contains the Kozak sequence.

>> ○ The ribosome requires a helicase that hydrolyzes ATP as it moves from the 5'-cap to the 5'-AUG-3'.

>> ○ Met-tRNA recognizes the start codon of mRNA and binds to the P site using GTP.

>> ○ Initiation factors and the large subunit bind to form the translation initiation complex.

⑷ 2<sup>nd</sup>. Elongation

> ① A site, P site, E site

>> ○ A site: The site where **aa-tRNA enters** the translation initiation complex.  

>> ○ P site: The site where **peptide bonds form**, allowing the polypeptide chain to elongate.  

>> ○ E site: The site where **tRNA exits** the translation initiation complex.  

>> ○ The order follows 5' - E - P - A - 3'.  

> ② Elongation factor: EF-Tu, EF-Ts, EF-G

> ③ Codon recognition: A complementary tRNA recognizes the codon at the A site, requiring 2 GTP and involving EF-Tu and EF-Ts.

> ④ Peptide bond formation

>> ○ The carboxyl group of the peptide chain at the P site forms a peptide bond with the amino group of the amino acid at the A site.  

>> ○ The nitrogen of the amino group acts as a nucleophile, attacking the carbonyl carbon of the carboxyl group.  

>> ○ This process is catalyzed by the peptidyl transferase activity of the large ribosomal subunit.  

>> ○ Direction of polypeptide synthesis: N → C  

> ⑤ Through the previous step (④ Peptide bond formation), the entire amino acid chain from the P site is transferred to the tRNA at the A site.

> ⑥ Translocation

>> ○ The ribosome moves to the next codon, shifting positions:

>>> ○ tRNA at the P site moves to the E site and is released.

>>> ○ tRNA at the A site moves to the P site.

>> ○ This process requires 1 GTP and involves EF-G.

⑸ 3<sup>rd</sup>. Termination

> ① When a stop codon (UAA, UAG, UGA) reaches the A site, releasing factors bind, leading to termination of translation.

>> ○ Mycoplasma code (Code 4): Stop codons are UAA and UAG.

>> ○ Ciliate code (Code 6): Stop codon is UGA.

> ② Releasing factor (RF): Hydrolyzes the bond between the **tRNA at the P site** and the **last amino acid of the polypeptide**.

>> ○ Releasing factors are recycled.

>> ○ RF-1: Recognizes UAA and UAG.

>> ○ RF-2: Recognizes UAA and UGA.

> ③ 3'-UTR: The untranslated region from immediately after the stop codon to just before the 3'-poly(A) tail.

⑹ Genetic code

> ① Triplet codon: Starting from the start codon, codons are grouped in sets of three nucleotides, forming the fundamental framework for translation.  

>> ○ This framework is called the reading frame.

> ② No redundant translation: Each codon is translated only once.

> ③ Codon directionality: Codons are read in the **5' to 3' direction** based on mRNA, which is interpreted based on the codon table.

> ④ Start codon: Translation always begins at the AUG (Met) start codon** on the mRNA.

> ⑤ Nonsense codon: Among the 64 codons, three codons (UAA, UAG, UGA) are stop codons, meaning they do not specify an amino acid.

> ⑥ Wobble hypothesis: There are 61 tRNAs in theory but about 45 in reality.

>> ○ Associated with the [deaminoation process](https://jb243.github.io/pages/71#5-mutation) where the A base turns into inosine acid (I).

>> ○ There are cases where a specific codon forms a pairing with a non-complementary tRNA.

>> ○ The sequence of the third codon is called the wobble sequence, and the binding with tRNA is called wobble pairing.  

>> ○ Hypothesis: The first base of the anticodon and the third base of the codon do not pair strictly.

> ⑦ Degeneration: Since there are only about 20 amino acids, each amino acid corresponds to two or more codons (a total of 45 codons).

>> ○ Codon usage: Even if the same amino acid is produced, the translation speed varies depending on the tRNA.

>> ○ Some studies identify the amount of tRNA in the pool and replace it with an appropriate codon to speed up translation.

> ⑧ Unambiguity: By ⑥ and ⑦, one codon corresponds to one amino acid.

>> ○ Exception: There are cases where one codon corresponds to multiple anticodons.

> ⑨ Universal codon

>> ○ The genetic code evolved early in the history of life.  

>> ○ It has remained well preserved without changes because even minimal alterations can be fatal (frozen accident hypothesis).  

>> ○ Exceptions have been found in some protozoa and mitochondria.

> ⑩ Overlapping gene

>> ○ Viruses have small genomes, so overlapping genetic information is observed, meaning the reading frame is violated.  

>> ○ A single nucleotide can be read up to six times: three times per reading frame and twice depending on whether it's the sense or anti-sense strand.

>> ○ A single mutation can have a significant impact.

⑺ Deciphering the genetic code: Awarded the 1968 Nobel Prize in Physiology or Medicine.

> ① Amino acid: 20 types

>> ○ <sub>4</sub>∏<sub>2</sub> = 16 < 20 < <sub>4</sub>∏<sub>3</sub> = 64

>> ○ Early scientists deduce the unit of code is three nucleotides.

> ② Types of genetic codes

>> ○ Meaningless codons: The three stop codons serve as termination signals for amino acid sequences.  

>> ○ Meaningful codons: Specify 20 amino acids.

>> ○ Start codon: Methionine (AUG).

> ③ Decoding genetic code

>> ○ Utilizes an in vitro translation system.  

>>> ○ Identical to the prokaryotic expression system.  

>>> ○ Does not require translation to start from AUG.  

>>> ○ Can be synthesized with only polynucleotide phosphorylase.

>> ○ **Method 1.** Marshall W. Nirenberg and J. H. Mattaei experiment (1961)

>>> ○ Artificial synthetic RNA and cell-free protein synthesis system (in vitro translation system)

>>> ○ Amino acid decoding for UUU, AAA, CCC, and GGG.

>> ○ **Method 2.** H. Gobind Khorana ㄷxperiment

>>> ○ Further decoding of the genetic code using synthetic RNA.  

>>> ○ Example: When using a 5'-CACACACACAC-3' sequence with an ATP:CTP ratio of 1:1, the resulting amino acid mixture is threonine:histidine = 1:1.

>> ○ **Method 3.** Marshal W. Nirenberg and Ochora experiment

>>> ○ Mix UTP and GTP in a 3:1 ratio.

>>> ○ The probability of UUU codon is 0.75<sup>3</sup> → The detection rate tells you what amino acid matches.

>>> ○ If two amino acids have similar probabilities, it is uncertain which codons they correspond to.

>>> ○ Example: When using a 5'-CACACACACAC-3' sequence with an ATP:CTP ratio of 5:1, the resulting amino acid distribution is:  

>>>> ○ Lysine: 100  

>>>> ○ Threonine: 26  

>>>> ○ Asparagine: 24  

>>>> ○ Glutamine: 24  

>>>> ○ Proline: 7  

>>>> ○ Histidine: 6  

>> ○ **Method 4.** tRNA hybridization + filtration method

>>> ○ Enables the creation of an accurate codon table.  

>>> ○ This method is actually used in practice.

> ④ Genetic code table

<br>

![image](https://github.com/user-attachments/assets/8f524ea3-7362-4782-bd2c-5e83fd913735)

**Table 4.** Genetic code table

<br>

⑻ Reading frame (RF)

> ① Definition: A frame separated by 3 nucleotides that can be translated into an amino acid sequence.

> ② Composition: Approximately 100 amino acids starting from the start codon (ATG; met).

> ③ Open Reading Frame (ORF): Multiple reading frames generated based on how certain codons are considered as start and stop codons.

> ④ Reading frames do not include stop codons (there are 3 types).

> ⑤ Position of reading frames within mRNA.

<br>

![image](https://github.com/user-attachments/assets/0ee65416-a98a-4a8f-bf9a-35f6a340bfc4)

**Figure 26.** Position of reading frames within mRNA.

<br>

<br>

## **6\. Delivery of Proteins After Translation** 

⑴ Overview

> ① Synthesized proteins contain signal sequences composed of specific amino acid sequences, directing the protein to specific locations within the cell.

> ② Field awarded the 2013 Nobel Prize in Physiology or Medicine.

⑵ Intracellular compartments

> ① Intracellular chemical reactions are often opposed to each other and require intracellular compartments.

> ② The assembly of multiple enzymes into a large protein complex to jointly catalyze a specific reaction process (in both prokaryotes and eukaryotes).

> ③ The method of compartmentalization within a single membrane structure (eukaryotes): The significance of membrane-bound organelles.

> ④ The method of transporting newly synthesized proteins to their corresponding membrane-bound organelles (protein targeting).

⑶ Signal sequences

> ① A sequence of approximately 20 amino acids typically located at the N-terminal region.

> ② Endoplasmic reticulum (ER) signal sequence

>> ○ Contains about 10 hydrophobic amino acids.

>> ○ Reason for hydrophobicity: If the signal peptide were hydrophilic, it would be difficult to pass through the membrane.  

> ③ Nuclear localization signal (NLS)

>> ○ Composed of at least four consecutive positively charged amino acids.  

> ④ Chloroplast targeting signal

>> ○ Involves two different signal sequences.  

> ⑤ Example: Signal sequence of Hsp90

<br>

![image](https://github.com/user-attachments/assets/5e4b600b-73f8-49c2-9cae-6f860be1d43a)

**Figure 27.** Signal sequence of Hsp90

<br>

>> ○ All Hsp90 proteins include a signal sequence composed of 10 amino acids.

>> ○ Y (1), F (7), R (9) are conserved in all species.

>> ○ The size of each amino acid symbol in the graph below is proportional to its frequency.

>> ○ Theoretically, the second amino acid can be any amino acid, but Ser is the most commonly found.

⑷ Free ribosome

> ① Floating in the cytoplasm and synthesizing proteins acting inside the cell.

> ② Characteristic

>> ○ Generally, no protein delivery mechanism.

>> ○ Proteins synthesized by free ribosomes lack disulfide bonds.

>> ○ Post-translation: Proteins move after translation is completed.

> ③ **Example 1.** Nuclear proteins

>> ○ Protein transport through the nuclear pore (in → out): Requires a nuclear export signal (NES) + exportin.  

>> ○ Protein transport through the nuclear pore (out → in): Requires a nuclear localization signal (NLS) + importin.

>> ○ A receptor protein binds to the signal sequence of the protein and transports it together, preserving both the signal sequence and the tertiary structure.

>> ○ Proteins pass through the nuclear pore in a folded state.

>> ○ GTP is used to return importin and exportin to their original locations.

>> ○ About 45% of yeast proteins contain an amino acid sequence matching the classic NLS pattern.

> ④ **Example 2.** Peroxisomal proteins

>> ○ Transported by peroxins, maintaining both their tertiary structure and signal sequence.  

> ⑤ **Example 3.** Mitochondrial and chloroplast proteins

>> ○ Lose their tertiary structure when passing through the membrane.  

>> ○ Signal sequence is cleaved after transport.  

>> ○ Chloroplast-targeted proteins contain two signal sequences.

> ⑥ **Example 4.** Surface proteins

> ⑦ **Example 5.** Cytoplasmic proteins (e.g., cytoskeletal proteins)

⑸ Bound ribosome

> ① Examples

>> ○ Membrane proteins

>> ○ Secretory proteins

>> ○ Lysosomal proteins

>> ○ Vacuolar proteins

> ② Protein delivery mechanisms

>> ○ 1<sup>st</sup>. Start translation.

>> ○ 2<sup>nd</sup>. Co-translation: Signal peptide on the amino acid sequence being translated binds to SRP in the cytoplasm.

>> ○ 3<sup>rd</sup>. Removal before transport: The signal sequence is cleaved by signal peptidase before transport.

>> ○ 4<sup>th</sup>. Removal before transport: Imported into the endoplasmic reticulum (ER) through the translocation complex.

>> ○ 5<sup>th</sup>. Resume translation.

<br>

<br>

## **7\. Post-translational Modification**

⑴ Overview

> ① **Purpose 1.** Regulation of activity

>> ○ Post-translational modifications (PTMs) can increase or decrease protein activity.

>> ○ PTMs can confer various functions.

> ② **Purpose 2.** Protein-protein interaction

>> ○ Sites modified by PTMs can be binding interfaces.

> ③ **Purpose 3.** Subcellular interaction

>> ○ PTMs can serve as targeting signals or membrane anchors.

> ④ **Purpose 4.** Lifespan regulation

>> ○ PTMs can promote protein degradation or scavenging.

⑵ Chemical modifications

> ① Glycosylation: Most membrane proteins and signal proteins are glycoproteins.

>> ○ **Classification 1.** Differences in glycosylation processes by domain

>>> ○ E. coli: No glycoprotein due to the absence of glycosylation process.

>>> ○ Yeast: Glycoproteins appear in a form with a high mannose content.

>>>> ○ Yeast structure: Cytosol - Cell membrane - Chitin - β-Glucan - Mannose

>>> ○ Animal cells: Glycoproteins exhibit a form with sialic acid.

>> ○ **Classification 2.** Glycosylation process in organelles: Divided into N-glycosylation and O-glycosylation.
  
>> ○ **2-1.** N-glycosylation
 
>>> ○ Mainly targets cell membrane proteins, attaching N-glycan.

>>> ○ Location: N-linked glycosylation occurs in the endoplasmic reticulum (ER) on asparagine (Asn) residues.

>>> ○ Attaches multiple sugars.

>>> ○ Albumin does not undergo N-glycosylation.

>>> ○ PNGase F: Degrades N-glycosylation.
  
>> ○ **2-2.** O-glycosylation

>>> ○ Mainly targets nuclear and cytoplasmic proteins, attaching O-glycan.

>>> ○ Location: O-linked glycosylation occurs in the Golgi apparatus on serine (Ser) and threonine (Thr) residues.

>>> ○ Attaches a single sugar.

>>> ○ **Type 1.** O-N-acetylgalactosamine (O-GalNAc)

>>> ○ **Type 2.** O-N-acetylglucosamine (O-GlcNAc)

>>>> ○ Formation of N-acetylglucosamine (GlcNAc)

>>>>> ○ Glc

>>>>> ○ Glc-6-ⓟ

>>>>> ○ Fruc-6-ⓟ

>>>>> ○ GlcN-6-ⓟ

>>>>> ○ GlcNAc-6-ⓟ

>>>>> ○ GlcNAc-1-ⓟ

>>>>> ○ UDP-GlcNAc

>>>> ○ Characterized by a β-glycosidic bond between the -OH group of serine (Ser) or threonine (Thr) and N-acetylglucosamine (GlcNAc).

>>> ○ **Type 3.** O-mannose (O-Man)

>>> ○ **Type 4.** O-galactose (O-Gal)

>>> ○ **Type 5.** O-fucose (O-Fuc)

>>> ○ **Type 6.** O-glucose (O-Glc)

>> ○ Glycocapture method: Used to identify N-glycosylation sites.

>>> ○ 1<sup>st</sup>. Oxidize the glycan chain.

>>> ○ 2<sup>nd</sup>. Covalently attach it to hydrazide resin.  

>>> ○ 3<sup>rd</sup>. Remove N-glycosylation using PNGase F.

>>> ○ 4<sup>th</sup>. Asparagine (Asn) is simultaneously converted to aspartic acid (Asp).  

>>> ○ 5<sup>th</sup>. Analyze with LC/ESI-MS/MS, where a 0.984 Da mass shift is observed.  

> ② Phosphorylation

>> ○ Plays a crucial role in signal transduction.

>> ○ Example: Serine → Phosphorylated serine (ATP is used).  

>> ○ Example: Threonine → Phosphorylated threonine (ATP is used).  

>> ○ Example: Tyrosine → Phosphorylated tyrosine (ATP is used).

> ③ Disulfide bond formation: Occurs in the rough endoplasmic reticulum (RER) and involves PDI (protein disulfide isomerase).

> ④ Prenylation: Includes farnesylation, geranyl-geranylation, and others.

> ⑤ Ubiquitination

> ⑥ Acetylation

>> ○ Example: Lysine → Acetyl-lysine

> ⑦ Sulfation

<br>

| **phosphorylation** | **sulfation** |
| --- | --- |
| HPO<sub>3</sub><sup>-</sup> (+79.9663 Da) | SO<sub>3</sub><sup>-</sup> (+79.9568 Da) |
| Ser, Thr, Tyr | Tyr |
| loss of 98 Da (pST), 80 Da (pY) | loss of 80 Da (sY) |
| reversible by kinase and phosphatase | sulfotransferase |
| 70-80% of proteome | secreted or membrane proteins   (all proteins: 487, mammalian: 155, human: 54) |
| intracellular signaling | extracellular signaling |
| triester   (pK<sub>a1</sub> = 12, pK<sub>a2</sub> = 7.2, pK<sub>a3</sub> = 2.1) | monoester   (pK<sub>a</sub> = 2.1) |

**Table 5.** Comparison between phosphorylation and sulfation 

<br>

> ⑧ Other modifications

>> ○ Methylation 

>> ○ Nitration 

>> ○ Amidation 

>> ○ Formylation 

>> ○ Palmitoylation 

>> ○ Hydroxylation 

>> ○ SUMOylation

>> ○ Lipid anchoring

⑶ Chaperones

> ① Translated protein precursors acquire the appropriate folding structure, forming tertiary and quaternary structures with the help of chaperones.

> ② **Example 1.** Partial degradation
  
>> ○ **Example:** Cleavage of signal sequences, protein activation by endopeptidases, etc.

>> ○ **Example:** Amino acid cleavage is required for insulin activation.

>>> ○ In the rough endoplasmic reticulum, the N-signal of prepro-insulin is cleaved, forming pro-insulin.

>>> ○ In the Golgi apparatus, pro-insulin undergoes cleavage.

>>> ○ Insulin and C-peptide are secreted.

>>> ○ Insulin and C-peptide are clinically useful for diagnosis.

>> ○ **Example:** Enteropeptidase attached to the intestinal barrier.

>>> ○ Trypsinogen is converted into trypsin by cleaving specific bonds.

>>> ○ Trypsin converts chymotrypsinogen into chymotrypsin by cleaving specific bonds.

> ③ **Example 2.** Calnexin: Precise tertiary structure. Associated with [cystic fibrosis](https://jb243.github.io/pages/72).

> ④ **Example 3.** [Heat shock proteins](https://jb243.github.io/pages/1096) (HSPs)

⑷ Zymogen

> ① Inactive enzyme precursors.

> ② Acquire activity through post-translational processing.

> ③ Most digestive enzymes are in zymogen form.

⑸ Protein synthesis regulation

> ① **Example 1.** Ferritin: Iron storage protein.

>> ○ The iron response element (IRE) is located upstream of the coding region.

>> ○ When Fe levels ↑: IRE-BP is inactivated → Protein translation initiates → Expression occurs.

>> ○ When Fe levels ↓: IRE-BP is activated → IRE-BP binds to the mRNA RBS → Protein translation is inhibited.
  
>> ○ Dps: Functions as ferritin in prokaryotes. Discovered in E. coli. (Almiron et al., 1992)

⑹ Lifespan regulation

> ① Purpose

>> ○ Proteins with incorrect or denatured structures may be toxic in the body.

>> ○ Even normal proteins are frequently degraded in the body to regulate protein activity.

> ② **Example 1.** Ubiquitin

>> ○ A small protein composed of 76 amino acids.

>> ○ 1<sup>st</sup>. Ubiquitin attaches to the protein to be degraded with chaperone (e.g., hsp70) assistance.

>> ○ 2<sup>nd</sup>. The protein tagged with ubiquitin is recognized by proteasome and then undergoes hydrolysis.

> ③ **Example 2.** Lysosome and mannose pathway

>> ○ Digestion of target molecules, damaged receptors, and damaged organelles.

> ④ **Example 3.** PROTAC (PROteolysis TAgeting Chimeras)

>> ○ Overview

>>> ○ Definition: A technology that utilizes the ubiquitin-proteasome system to degrade proteins, including those not previously targeted for inhibition.

>>> ○ A chemical compound that binds to E3 ligase is artificially linked to a chemical compound that binds to the target protein for degradation.

>>> ○ Molecular glue components forming PROTAC were accidentally discovered while searching for drugs with physiological activity.

>> ○ E3 ubiquitin ligase

>>> ○ Attaches ubiquitin to the protein targeted for degradation.

>>> ○ Although there are around 600 types, only 2-3 are actively used as PROTAC.

>> ○ Composition

>>> ○ Compound binding to E3 ubiquitin ligase.

>>> ○ Compound binding to the protein targeted for degradation.

>>> ○ Linker connecting the two compounds.

>> ○ Examples

>>> ○ VHL (von Hippel-Lindau tumor suppressor): Widely used in PROTAC technology.

>>> ○ CRLs·Cullin-RING E3 ligase: Induces degradation by attaching ubiquitin to various proteins.

>>> ○ SD-36: A PROTAC selectively degrading STAT3, a transcription factor involved in the growth, proliferation, invasion, and metastasis of cancer cells.

>>> ○ Fulvestrant: Not only inhibits the estrogen receptor (ER) but also induces ER protein degradation by attaching hydrophobic residues.

>>> ○ Thalidomide: A representative example of molecular glue. Thalidomide binds to CRBN, and CRBN attaches ubiquitin to the target protein.

>>> ○ Lenalidomide, Pomalidomide: Both thalidomide derivatives and molecular glue.

<br>

<br>

## **8\. Inhibitor**

⑴ DNA polymerase inhibitors

> ① Acyclovir (Acycloguanosine): An antiviral drug that interacts with viral thymidine kinase (tk) and inhibits viral DNA polymerization.

> ② AraC (Cytosine arabinoside, Cytarabine): Anticancer drug.

> ③ AraA (Adenine arabinoside, Vidarabine): Antiviral drug.

> ④ Antinomycin D: Binds to the groove, preventing DNA replication; can act on both prokaryotic and eukaryotic cells.

> ⑤ Fluoroquinolones (Ciprofloxacin): Inhibits DNA gyrase.

> ⑥ Carboplatin: Anticancer drug that inhibits DNA synthesis.

> ⑦ Topoisomerase I Inhibitors: Inhibitors of type IB topoisomerase.

>> ○ Irinotecan

>> ○ Topotecan

>> ○ Camptothecin (CPT-11: Camtosar)

>> ○ Diflomotecan

>> ○ Lamellarin D

> ⑧ Topoisomerase II Inhibitors

>> ○ Etoposide

>> ○ Teniposide

>> ○ Doxorubicin

>> ○ Daunorubicin

>> ○ Mitoxantrone

>> ○ Amsacrine

>> ○ Ellipticines

>> ○ Aurintricarboxylic acid

>> ○ HU-331

>> ○ Ofloxacin (Trade name: Tarivid)

>> ○ Levofloxacin (Trade name: Cravit)

⑵ Transcription inhibitors

> ① Antinomycin D: Transcription inhibitor.

⑶ Translation inhibitors

> ① Cycloheximide: Translation inhibitor in eukaryotic cells.

> ② Translation inhibitors acting on bacteria: Can also function as antibiotics.

>> ○ Chloromycetin: Inhibits peptide bond formation.

>> ○ Erythromycin: Inhibits ribosomal movement on mRNA.

>> ○ Neomycin: Interferes with tRNA-mRNA interaction.

>> ○ Streptomycin: Inhibits translation initiation.

>> ○ Tetracycline: Inhibits tRNA binding to the ribosome.

⑷ Nucleoside reverse transcriptase inhibitors (NRTI)

> ① AZT (Zidovudine)

> ② Combivir (AZT + Epivir)

> ③ Emtriva (Emtricitabine)

> ④ Epivir (Lamivudine)

> ⑤ Epzicom (Abacavir + Epivir)

> ⑥ Hivid (ddC)

> ⑦ Trizivir (Abacavir + AZT + Epivir)

> ⑧ Videx & Videx EC (ddI): 2', 3'-dideoxynosine. Removes 3'-OH.

> ⑨ Zerit (D4T)

> ⑩ Ziagen (Abacavir)

<br>

---

*Input: 2015.7.02 17:56*

*Modified: 2025.11.27 13:25*
