## Chapter 10. Genome Project and Sequencing Technology

Recommended Article: 【Biology】 [Biology Table of Contents](https://jb243.github.io/pages/1457)

---

**1.** [Genome Project](#1-genome-project)

**2.** [Sequencing Technology](#2-sequencing-technology)

---

**a.** [Pyrosequencing](https://jb243.github.io/pages/76)

**b.** [Whole Genome Sequencing](https://jb243.github.io/pages/1431)

**c.** [Biological Informatics Analysis Table of Contents](https://jb243.github.io/pages/836)

**d.** [Transcriptome Analysis Pipeline](https://jb243.github.io/pages/2050)

---

<br>

## 1\. **Genome Project**

 ⑴ Overview

> ① Started under the command of Watson in 1990: Initiated as a 15-year project by a coalition of 6 countries

> ② Collaborative research involving over 350 research institutions

>> ○ 84.5% completion announced on June 11, 2000, with the draft

>> ○ 99.99% accuracy final version released on April 15, 2003

>> ○ Involvement of over 2,800 researchers over 13 years, costing 2.7 trillion won

> ③ Side effects of human genome research

>> ○ Birth of [biological informatics](https://jb243.github.io/pages/75#footnote_link_67_53)

>> ○ Promotion of the development of the human protein production process

>>> ○ Insulin: The first protein with a determined sequence

>> ○ Promotion of the development of automated DNA sequencing devices

>> ○ Promotion of the genome analysis of other applicable organisms

 ⑵ **Methodology 1:** Stepwise Sequencing Method (Scientists' Approach)

> ① **Step 1:** Determining Restriction Enzyme Recognition Sites

>> ○ Cutting DNA with restriction enzymes and electrophoresis reveals the sizes of each fragment

>> ○ Treating two restriction enzymes in various ways reveals the relative distances between their recognition sites

> ② **Step 2:** Constructing a Gene Map

>> ○ Determining the relative distances of genes on the chromosome

>> ○ Inferring the distance between genes through recombination rates

> ③ **Step 3:** Physical Map (DNA Map) Construction

>> ○ Meaning of determining restriction enzyme recognition sites: Using the information of fragments with known sequences based on each restriction enzyme recognition site to cumulatively construct a physical map

>> ○ Meaning of constructing a gene map: Once a physical map is constructed, it can be compared with a gene map. Introns exist between genes

>> ○ Approach using a single library

 ⑶ **Methodology 2:** Shotgun Sequencing (Celera, J. Craig Venter) (Entrepreneurial Approach)

> ① Cutting one DNA in multiple ways

> ② Determining the sequence of fragments cut by one method

>> ○ The length of the analysis sample is limited, so the DNA sequence cannot be determined all at once

> ③ Randomly arranging the sequences of bases in each method until a common result is obtained

> ④ Method based on computer science

 ⑷ Scientists' Approach vs. Entrepreneurial Approach

> ① Scientists' approach is dissatisfied with entrepreneurs' appropriation of their contributions and investments

> ② Entrepreneurial approach is dissatisfied with scientists' failure to disclose information, leading to the development of new methodologies

> ③ The final completion of the genome map is a joint effort of both.

<br>

![image](https://github.com/user-attachments/assets/252def50-89f3-45ea-b7fd-aab1a1f03125)

**Figure 1.** Stepwise Sequencing Method and Shotgun Sequencing Method

<br>

<br>

## **2\. Sequencing Technology**

 ⑴ Overview

> ① DNA Sequencing: Applying the principle of DNA replication

>> ○ Template: Each strand of DNA

>> ○ Substrate: dNTP (dATP, dCTP, dGTP, dTTP)

>>> ○ NTP has an -OH group at the 2' carbon and is a material for RNA synthesis

>> ○ DNA polymerase: Phosphate of the next nucleotide binds to the 3'-OH of deoxyribose

>> ○ Synthesis direction: 5' → 3', forming complementary base pairs with the template

>> ○ ddNTP: DNA polymerization is terminated because the 3<sup>rd</sup> carbon lacks an OH group

> ② RNA Sequencing: Applying the principle of RNA transcription

> ③ Sequencing by Synthesis

>> ○ Definition: A method of determining nucleotide sequences by detecting signals emitted when a polymerase incorporates nucleotides into the primer sequence.  

>> ○ **Type 1.** CRT (Cyclic Reversible Termination): A cyclic process consisting of incorporation, imaging, and deprotection.  

>> ○ **Type 2.** SNA (Single Nucleotide Addition): Adds only one nucleotide at a time.  

>> ○ Examples: Ion semiconductor sequencing (PGM), Pyrosequencing (Roche 454)  

> ④ Sequencing by Ligation

>> ○ Definition: A method of determining nucleotide sequences by detecting signals emitted when a ligase connects a hybridized sequence to the primer sequence.  

>> ○ 1 or more nucleotide can be incorporated at a time (e.g., dibase). 

>> ○ Examples: Nanoball sequencing, Thermo Fisher SOLiD  

⑵ _in vitro_ cloning: The very first sequencing method

⑶ Dideoxy Chain Termination Method (= Sanger sequencing): Reported in 1977, Sanger's second Nobel Prize

> ① Substrate: dNTP + ddNTP (in small quantities) + buffer (pH stabilized)

>> ○ ddNTP lacks a 3'-OH group, so it terminates the polymerization reaction

>> ○ If ddNTP is added in large quantities, all template DNAs are quickly terminated

> ② Primer

>> ○ Example: p32-primer (CTAG)

> ③ 1<sup>st</sup>. Addition of template DNA and polymerase

> ④ 2<sup>nd</sup>. Heating to separate the complementary strand after polymerization

> ⑤ 3<sup>rd</sup>. Electrophoresis followed by reading the sequence on X-ray film or fluorescence examination

<br>

![image](https://github.com/user-attachments/assets/e6c2ebfd-734a-4cf5-99a8-4b8c3471b692)

**Figure 2.** Process of Dideoxy Chain Termination Method

<br>

> ⑥ Advantages: Can read very long strands, still used in laboratories

> ⑦ Disadvantages: Requires a large amount of the same DNA strand

⑷ Dye-dideoxy chain termination method: Using **laser**

> ① Add a small amount of ddNTP to 4-color fluorescent dNTP.

> ② Automatic DNA sequencing is possible.

<br>

![image](https://github.com/user-attachments/assets/67614b86-e2fa-40b2-9b83-d0c739437f08)

**Figure 3.** Process of the Dye-dideoxy chain termination method 

<br>

⑸ Ion Semiconductor Sequencing (PGM)

> ① **Sequencing by Synthesis**

> ② Prone to insertion/deletion errors

⑹ [Pyrosequencing](https://jb243.github.io/pages/76) (Roche 454)

> ① Definition: A DNA sequencing method that relies on the proportional luminescence produced based on the amount of pyrophosphate generated during DNA synthesis. ~400 bp per read.

> ② **Sequencing by Synthesis:** Light is emitted when a nucleotide is incorporated, appearing as peaks in a pyrogram.

> ③ Prone to insertion/deletion errors.

> ④ No longer in use.

<br>

![image](https://github.com/user-attachments/assets/caa87c91-072b-41e4-908b-fe30aeceab77)

**Figure 4.** Pyrosequencing diagram 

<br>

![image](https://github.com/user-attachments/assets/25ae6674-7cab-4a41-8c12-252170f53ba4)

![image](https://github.com/user-attachments/assets/49395354-3ca0-46bd-a7a5-9d5f0cd7a618)

![image](https://github.com/user-attachments/assets/be09e07d-0cb2-4ec1-9196-a1139d27b2b1)

![image](https://github.com/user-attachments/assets/56f079bf-4f27-40ce-aec6-4eef5fb53b5c)

![image](https://github.com/user-attachments/assets/4526c92b-7391-44b2-9967-b97a6046fb4b)

![image](https://github.com/user-attachments/assets/621300f2-db77-46a0-b435-35fc44b6ff4a)

**Figure 5.** Pyrosequencing process

<br>

⑺ Nanoball sequencing 

> ① **Sequenching by Ligation**

⑻ Thermo Fisher SOLiD 

> ① **Sequenching by Ligation**

> ② No longer in use.

⑼ Illumina solid-phase amplification ([ref](https://www.illumina.com/content/dam/illumina-marketing/documents/products/illumina_sequencing_introduction.pdf))

<br>

![image](https://github.com/user-attachments/assets/0eea3d97-e794-4bdc-aa96-06d7ab01a220)

**Figure 6.** Illumina solid-phase amplification 

<br>

![image](https://github.com/user-attachments/assets/a0c4c61f-7fa4-4b8e-a8d9-4ab83cd92703)

**Figure 7.** Fluorescent color distribution photo 

<br>

> ① 1<sup>st</sup>. Fragmentation: Randomly cut the given DNA sample.

> ② 2<sup>nd</sup>. Gel-based size selection: Size of each DNA fragment can be limited if necessary.

> ③ 3<sup>rd</sup>. Adaptor binding: Attach an adapter to both ends of all DNA sample fragments.

> ④ 4<sup>th</sup>. Amplification

>> ○ 4<sup>th</sup> - 1<sup>st</sup>. Denature DNA into single-strands.

>> ○ 4<sup>th</sup> - 2<sup>nd</sup>. Attach single-stranded DNA to the Illumina flow cell.

>> ○ 4<sup>th</sup> - 3<sup>rd</sup>. Add enzymes to allow single-stranded DNA to form bridges on a solid-phase substrate.

>> ○ 4<sup>th</sup> - 4<sup>th</sup>. After adding primers to the single-stranded DNA bridges, primers can bind to the bridges.

>> ○ 4<sup>th</sup> - 5<sup>th</sup>. Add unlabeled single-stranded DNA and induce DNA synthesis: Forms double-stranded DNA bridges.

>> ○ 4<sup>th</sup> - 6<sup>th</sup>. Denature to turn double-stranded DNA bridges into anchored single-stranded DNA.

>> ○ 4<sup>th</sup> - 7<sup>th</sup>. Repeat the above six steps to create anchored single-stranded clusters with the same base sequence.

>> ○ **Feature: Anchored single-stranded clusters form millions of clusters.

> ⑤ 5<sup>th</sup>. Only the forward strand is retained to prevent unwanted priming.

> ⑥ 6<sup>th</sup>. Sequencing by synthesis (SBS)

>> ○ 6<sup>th</sup> - 1<sup>st</sup>. Add 4 types of labeled reversible terminators, primers, and DNA polymerase according to the base type.

>> ○ 6<sup>th</sup> - 2<sup>nd</sup>. When labeled reversible terminators form phosphodiester bonds, fluorescence is emitted.

>> ○ 6<sup>th</sup> - 3<sup>rd</sup>. Obtain a fluorescent color distribution image of each cluster.

>> ○ 6<sup>th</sup> - 4<sup>th</sup>. Washing

>> ○ 6<sup>th</sup> - 5<sup>th</sup>. Repeat the above four steps to determine the entire base sequence.

>> ○ **Type 1.** **Single-end sequencing** (SES): Sequencing with only one adapter.

>> ○ **Type 2.** **Paired-end sequencing** (PES): Sequencing with both adapters.

>>> ○ Initially, sequence with one adapter (Read1 acquisition), then sequence with the opposite adapter (Read2 acquisition).

>>> ○ Read1 and Read2 from the same DNA fragment can be easily matched since they come from the same cluster.

>>> ○ Advantages: Higher accuracy (due to Read1 and Read2 comparison), easy detection of DNA variations, easy analysis of repetitive sequences, and easy mapping between different species.

>>> ○ Disadvantages: Higher cost and more steps required than SES.

⑽ WGS (Whole Genome Sequencing)

> ① SNV, insertion, deletion, structural variant, CNV

> ② Sequencing depth > 30X

⑾ WES (Whole Exon Sequencing)

> ① Only SNV, insertion, deletion, SNP in protein-coding genes

> ② Sequencing depth > 50X ~ 100X

> ③ Cost-effective

⑿ RNA-seq

> ① Overview

>> ○ Microarray data has raw data in the form of continuous values, whereas RNA-seq has raw data in the form of count data.

>> ○ Compared to microarray data, RNA-seq can robustly capture signals even for genes with very low or very high expression levels.

> ② 1<sup>st</sup>. Microdissection: Separating specific tissues for RNA extraction.

>> ○ LCM (Laser Capture Microdissection): Cutting specific tissues with a laser beam. Robust but labor-intensive.

>> ○ TOMO-seq: Using cryosection and computer-based 3D sectioning. Not suitable for clinical purposes.

>> ○ Transcriptome _in vivo_ analysis

>> ○ ProximID

>> ○ STRP-seq

> ③ 2<sup>nd</sup>. Attach poly T recognizing the poly A tail of RNA.

> ④ 3<sup>rd</sup>. Fragment RNA: 200-400 nt.

> ⑤ 4<sup>th</sup>. Attach primers to RNA.

> ⑥ 5<sup>th</sup>. First cDNA synthesis.

> ⑦ 6<sup>th</sup>. Second cDNA synthesis.

> ⑧ 7<sup>th</sup>. Process the 3' and 5' ends of RNA.

> ⑨ 8<sup>th</sup>. Ligate DNA sequencing adapters.

> ⑩ 9<sup>th</sup>. Amplify ligated fragments with PCR.

> ⑪ **Application 1.** dUTP method: A representative method for strand-specific sequencing.

>> ○ Background: Used for studying biological functions based on RNA orientation (e.g., regulation of antisense miRNA).

>> ○ **Step 1.** _DNA &RNA hybrid_: Synthesize cDNA (first or anti-sense strand) using dT primers and reverse transcriptase, targeting mRNA poly-A tails.

>>> 5'-//-U-//-AAAAAA-3'

>>> 3'-//-A-//-TTTTTT-5'

>> ○ **Step 2.** __ds__ _ cDNA_: Use dUTP instead of dTTP to synthesize cDNA (first strand) as the template for cDNA (second or sense strand).

>>> 3'-//-A-//-TTTTTT-5'

>>> 5'-//-U-//-AAAAAA-3'

>> ○ **Step 3.** _ligated _ds_ cDNA_: Connect Y-adaptors to both ends of ds cDNA.

>> ○ **Step 4.** Treatment with UDG (uracil-DNA glycosylase) breaks down the second strand, which contains uracil.

>> ○ **Step 5.** Amplify the remaining reverse antisense strand (first strand) to create the library.

>> ○ In the library raw data, "_1.fastq" represents the first strand, while "_2.fastq" represents the second strand.

>> ○ Thus, _2.fastq represents the original RNA profile.

⒀ **Single-cell sequencing**

> ① Types: scDNA-seq, scRNA-seq (2013 [Technology of the Year](https://jb243.github.io/pages/1059)), single-cell epigenetics sequencing

> ② **Step 1.** Isolation of single cells

>> ○ **Method 1.** Simple isolation: Very early method.

>> ○ **Method 2.** Based on FACS or LCM (laser microdissection)

>> ○ **Method 3.** Acoustic separation

>>> ○ Separates single cells hydrodynamically, causing minimal impact on cells.

>>> ○ CyTOF (cytometry by time of flight) is a representative method.

>> ○ **Method 4.** Immuno-magnetic separation

>>> ○ Attach magnets to cells.

>>> ○ Can obtain a large number of cells.

>>> ○ Divided into cases with and without centrifugation requirements.

>> ○ Droplet-based platform and plate-based platform have different library size.

> ③ **Step 2.** Reverse transcription

> ④ **Step 3.** cDNA amplification

> ⑤ **Step 4.** Library construction: e.g., Drop-seq

> ⑥ Single-cell genomics (scDNA-seq) + Single-cell transcriptomics (scRNA-seq)

>> ○ Allows understanding the relationship between genomic mutation patterns and gene expression in transcriptomes.

>> ○ Technologies for separating DNA and RNA: G&T seq, SIDR-seq, DNTR-seq

⒁ **Single Nucleus RNA Sequencing (snRNA-seq)**

> ① **Purpose 1.** Muscles are multinucleated cells, so they need to be analyzed at the nuclear level as they are not captured by scRNA-seq.

> ② **Purpose 2.** snRNA-seq captures more various RNA, including introns, pre-mRNA, non-coding RNA, compared to scRNA-seq.

> ③ **Purpose 3.** In snRNA-seq, nuclear RNA is primarily captured, while cytoplasmic RNA is also captured (although in small amounts).

⒂ [Spatial Sequencing](https://www.nature.com/articles/s41576-022-00515-3?fbclid=IwAR0T71CxITX9xDJk0n41irqduCpyCTc25yd28ISdWSt3nlGpSNMS5Mr10xU&mibextid=7GMMY6&fs=e&s=cl) (<span style="background-color: #ef5369; color: #ffffff; font-family: Times New Roman;">▶</span> <a href="https://www.youtube.com/watch?v=6zKpSD1ZxAk" target="_blank" rel="noopener">Supplement</a>)

<br>

![image](https://github.com/user-attachments/assets/9dedeaa7-9fc8-4d07-966c-ef5b581321be)

**Figure 8.** Overview of spatial sequencing

<br>

![image](https://github.com/user-attachments/assets/a9bb1611-2f36-4d4c-ad1b-b071d69ce29f)

**Table 1.** Comparison of different spatial transcriptomic technologies

<br>

> ① **Type 1.** Spatial genomics

>> ○ **Example 1.** Tumor research: Tumors are heterogeneous.

>> ○ **Example 2.** Spleen research: Mature immune cells have diverse genetic compositions.

> ② **Type 2.** Spatial transcriptomics: 2020 [Technology of the Year](https://jb243.github.io/pages/1059)

> ③ **2-1.** Spot-based spatial transcriptomics: Many genes + few spots

<br>

![image](https://github.com/user-attachments/assets/b12cd9eb-1b54-4084-9042-59590ae3edc7)

**Figure 9.** Betchmark Study of spot-based spatial transcriptomics

<br>

>> ○ [**ST**](https://pubmed.ncbi.nlm.nih.gov/30154148/) (Spatial Transcriptomics)

>>> ○ Barcoded oligos are randomly arranged on a functionalized surface, capturing mRNA released from the mounted tissues and/or cells.

>> ○ **10X Visium**

>>> ○ Principle: Attach spot-specific oligonucleotides to each spot to hybridize with tissue-derived RNA, obtaining spotwise transcriptomes.

>>> ○ Surface area: 6.5 mm × 6.5 mm

>>> ○ Thickness: 10 ~ 20 μm

>>> ○ Number of spots: Up to 4992 (Based on previous version of Visium HD)

>>> ○ Distance between spots: 100 μm

>>> ○ Diameter of spots: 55 μm

>>> ○ Sensitivity: 10,000 transcripts per spot

>>> ○ **Type 1.** Direct Visium (oligo-dT based method): Captures mRNAs with poly dT. Only applicable to FF (fresh-frozen) samples.

<br>

![image](https://github.com/user-attachments/assets/55acb52a-799f-48aa-b3f7-fbb97844cf1c)

**Figure 10.** Principle of Visium FF

<br>

>>> ○ **Type 2.** probe-based Visium

>>>> ○ It can be done in both FF (Fresh Frozen) and FFPE (Formalin-Fixed Paraffin-Embedded) samples. In particular, FFPE (formalin-fixed paraffin-embedded) samples cannot undergo direct Visium due to RNA degradation, where mRNA molecules are fragmented into various pieces.

>>>> ○ To identify the target mRNA, all three pairs of LHS and RHS must be ligated together: each probe's length is 25 base pairs. RTL (probe-based RNA-templated ligation chemistry) is utilized for this purpose.

<br>

![image](https://github.com/user-attachments/assets/784bd6c7-004b-4e25-93dd-dc018dd14b1f)

**Figure 11.** Principle of probe-based Visium

<br>

>>>> ○ Advantage: Superior data quality compared to direct Visium.

>>>> ○ Disadvantage: Limited freedom in analysis compared to Visium FF, as only genes specified by the probe are detected.

>>>> ○ For Visium FFPE, starting from June 2024, 10x will discontinue the Visium FFPE service, not using [CytAssist](https://www.10xgenomics.com/library/c04eca).

>>>> ○ The CytAssist images represent the distribution of gene expression and are used for image alignment.

>> ○ 10X Visium HD

>>> ○ The basic data consists of spots with a diameter of 2 μm, and additional data binned at 8 μm and 16 μm are also provided.

>> ○ **Slide-seq** and [Slide-seq V2](https://www.nature.com/articles/s41587-020-0739-1)

>>> ○ Employs random spatial bead spreading and in situ sequencing decoding.

>>> ○ 97% of spots consist of one or two cell types.

>> ○ [HDST](https://www.nature.com/articles/s41592-019-0548-y) 

>>> ○ Deposits beads with combinatorial barcodes on patterned wafers which are then decoded with serial hybridization.

>> ○ NanoString GeoMx

>>> ○ Nanostring lost a patent dispute with 10x Genomics as of Nov '23 ([ref](https://www.businesswire.com/news/home/20231119485930/en/NanoString-Comments-on-Delaware-District-Court-Verdict)) → The bankruptsy of Nanostring ([ref](https://www.genengnews.com/news/nanostring-files-for-chapter-11-bankruptcy-launches-strategic-review/))

>> ○ Stereo-seq: Higher spatial resolution than Visium

>>> ○ Utilizes Illumina or MGI sequencing for oligo patterning on flow cells, and barcode calling is performed directly on the sequencer.

>>> ○ Diameter: 220 nm

>>> ○ Distance between spots: 500 or 715 nm

>> ○ Seq-Scope: Higher spatial resolution than Visium

>>> ○ Utilizes Illumina or MGI sequencing for oligo patterning on flow cells, and barcode calling is performed directly on the sequencer.

>> ○ PIXEL-seq

>> ○ [XYZeq](https://www.science.org/doi/10.1126/sciadv.abg4755)

>>> ○ Tissue is placed on a spatially barcoded microwell array for an initial round of reverse transcription, after which whole cells are removed and undergo single-cell sequencing.

>> ○ [sci-Space](https://www.science.org/doi/10.1126/science.abb9536)

>>> ○ Tissue is placed on a glass slide bearing spatially gridded hashing oligos; tissue is then permeabilized to enable oligo transfer and then imaged; nuclei are then extracted, fixed and sequenced.

>> ○ sci-RNA-seq

>> ○ TIVA-seq

>> ○ NICHE-seq

>> ○ [ZipSeq](https://www.nature.com/articles/s41592-020-0880-2)

>>> ○ It uses patterned illumination and photocaged oligonucleotides to serially print ‘zipcodes’ onto live cells in intact tissues in real time.

>> ○ [DBiT-seq](https://www.cell.com/cell/fulltext/S0092-8674(20)31390-8?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867420313908%3Fshowall%3Dtrue)

>>> ○ Delivers barcoded oligos directly to tissue through orthogonal microfluidics in a predetermined spatial distribution.

>> ○ CITE-seq ([ref1](https://www.nature.com/articles/nmeth.4380), [ref2](https://cite-seq.com/)): Enables parallel comparison of spatial transcriptomics and antibody distribution

<br>

![image](https://github.com/user-attachments/assets/4340a02a-9e93-4a66-9cfc-f048c858a1ca)

**Figure 12.** Diagram of CITE-seq

<br>

>>> ○ Connect the 5' end of oligonucleotide to an antibody using streptavidin-biotin.

>>> ○ The oligonucleotide can hybridize complementarily with the oligo-dT primer.

>>> ○ Streptavidin-biotin bond can dissociate under reducing conditions.

>>> ○ Recently, perturb-CITE-seq was also developed.

>> ○ SPOTS

>>> ○ Spatial PrOtein and Transcriptome Sequencing

>>> ○ Indirectly assess protein level on Visium using polyadenylated DNA-barcoded antibody

>> ○ Open-ST 

>> ○ MAGIC-seq

> ④ **2-2.** [FISH](https://jb243.github.io/pages/77) based spatial transcriptomics: Few genes + many spots

>> ○ ISS( _in situ_ sequencing): Technique to sequence RNA at its original location in tissue. Sequencing by ligation

>>> ○ **Type 1.** [The first ISS](https://pubmed.ncbi.nlm.nih.gov/23852452/)

>>> ○ **Type 2.** ISS with Padlock probe

>>>> ○ Reverse transcriptase creates cDNA of the RNA target

>>>> ○ Padlock probe can hybridize to two regions of the cDNA

>>>> ○ Target sequence amplification occurs through RCA (rolling-circle amplification)

>>>> ○ RCA product is sequenced in situ by ligation

>>> ○ **Type 3.** [ISS using fluorescent probes and cross-linking](https://pubmed.ncbi.nlm.nih.gov/24578530/)

>>> ○ **Type 4.** [barcode based methods](https://pubmed.ncbi.nlm.nih.gov/29190363/)

>>> ○ **Type 5.** gap-filled ISS

>> ○ [smFISH](https://pubmed.ncbi.nlm.nih.gov/18806792/)(single molecule FISH) (2008)

>> ○ [seqFISH](https://pubmed.ncbi.nlm.nih.gov/24681720/)(sequential FISH) (2014): DNAse I-based digestion and sequential staining and imaging rounds to decode transcripts

>> ○ [seqFISH+](https://www.nature.com/articles/s41586-019-1049-y): Genome-scale transcriptome investigation separating individual transcripts into fluorescence spectra, employing 20 probes per encoding round.

>> ○ **Vizgen - MERSCOPE** (Technology name: **[MERFISH](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4662681/)** (multiplexed error-robust FISH))

>>> ○ Direct probe hybridization without separate amplification mechanism.

>>> ○ Each FISH probe corresponds 1:1 with each gene (though this assumption may not always hold).

>>> ○ Employing error correction in barcode assignment for robust barcode calling in noisy FISH-based images.

>>> ○ **Step 1.** Photograph multiple times with fluorescence varying over time for each FISH probe

>>> ○ **Step 2.** Reverse identify genes based on binary code read from each RNA

<br>

![image](https://github.com/user-attachments/assets/dcb98e51-c723-4b9b-8f4a-e131baf3c225)

**Figure 13.** Principle of MERFISH

<br>

>> ○ **10x - Xenium**

>>> ○ Small amount of padlock probe + rolling circle amplification

>>> ○ **Step 1.** Padlock probe binds complementary RNA transcript in a pincer shape, forming a loop

>>> ○ **Step 2.** RCA (rolling circle amplification): RNA transcript amplified after loop formation

>>> ○ **Step 3.** Hybridize each RNA transcript with a fluorescent probe, then perform fluorescent imaging → washing

>>> ○ **Step 4.** Repeat **Step 3** and decode labels for each gene from the generated images

<br>

![image](https://github.com/user-attachments/assets/c0294746-8102-4b65-948c-201ebb59e8e8)

**Figure 14.** Principle of Xenium

<br>

>> ○ **Nanostring - CosMx**

>>> ○ Small amount of probe + branch chain hybridization

>>> ○ Nanostring won in the U.S. against 10x for violating antitrust laws in July '23
([ref](https://www.businesswire.com/news/home/20230711643008/en/Delaware-District-Court-Permits-NanoString%E2%80%99s-Counterclaims-that-10x-Genomics-and-Harvard-Violated-the-Antitrust-Laws)) → The bankruptsy of Nanostring ([ref](https://www.genengnews.com/news/nanostring-files-for-chapter-11-bankruptcy-launches-strategic-review/))

<br>

![image](https://github.com/user-attachments/assets/c3f261da-2ff3-4ff2-a21c-2f57a1e818f2)

**Figure 15.** Principle of CosMx

<br>

>> ○ FISSEQ and oligoFISSEQ

>> ○ Veranome

>> ○ Rebus

>> ○ BOLORAMIS

>> ○ STARmap: Sequencing by ligation

>> ○ SEDAL sequencing

>> ○ ExSeq

>> ○ BaristaSeq: Sequencing by synthesis

>> ○ BARSeq and BARSeq2

>> ○ HybISS

>> ○ SABER

>> ○ clampFISH

>> ○ split-FISH

>> ○ SCRINSHOT

>> ○ PLISH

>> ○ osmFISH

>> ○ ExFISH

>> ○ par-seqFISH

>> ○ EASI-FISH

>> ○ SGA

>> ○ corrFISH

> ⑤ **Type 3.** Spatial proteomics: Broadly classified into mass spectrometry-based and imaging-based methods

>> ○ SWITCH

>> ○ MxIF

>> ○ t-CyCIF

>> ○ IBEX

>> ○ DEI

>> ○ CODEX

>> ○ immuno-SABER

>> ○ TSA

>> ○ Opal IHC

>> ○ MIBI

>> ○ IMC

>> ○ HD-MIBI

>> ○ GeoMx Digital Spatial Profiler (DSP): 100 mm scale

>>> ○ GeoMX DSP stains tissues with suites of antibodies or gene probes fused to UV-cleavable DNA barcodes. 

>> ○ [4i multiplexed imaging](https://www.science.org/doi/10.1126/science.aar7042?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub%20%200pubmed)

⒃ Other sequencing technologies

> ① TCR-seq (T cell receptor sequencing): Sequencing used to track T cell subtypes and clones.

> ② Invade-seq: A sequencing technique for analyzing the host-microbiome.

> ③ long-read sequencing: 2022 Technology of the Year [Technology of the Year](https://jb243.github.io/pages/1059) ([Reference](https://www.nature.com/articles/s41592-022-01759-x))

>> ○ Less sequencing gap compared to short-read sequencing

<br>

![image](https://github.com/user-attachments/assets/0b674c1c-9cad-4f78-9c38-c7fe59c75611)

**Figure 16.** long-read sequencing and short-read sequencing

<br>

>> ○ **Advantage 1.** [AS Analysis](https://jb243.github.io/pages/2050#footnote_link_67_56)(alternative splicing analysis): Can identify alternative splicing events, isoforms, etc.

>> ○ **Advantage 2.** Easier integration of epigenetics and transcriptomics

>> ○ **Example 1.** Pacific Biosciences SMRT (single molecule real-time) sequencing: Average read length is ~20 kb

>> ○ **Example 2.** Oxford Nanopore Sequencing: Average read length is ~100 kb

> ④ non-invasive sequencing

>> ○ A technology that allows sequencing without breaking cells

> ⑤ [Halo-seq](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9097300/): A technique for obtaining the transcriptome of RNAs adjacent to a specific protein.

>> ○ **Step 1.** Attach a HaloTag domain to a specific target.

>> ○ **Step 2.** This HaloTag generates an alkyne handle radical by ejecting a hydrogen radical H· from a radical-producing Halo ligand injected with an alkyne handle.

>>> ○ R-H → R· + H·

>> ○ **Step 3.** Similarly, the HaloTag generates an RNA radical by ejecting a hydrogen radical H· from RNA.

>>> ○ RNA-H → RNA· + H·

>> ○ **Step 4.** The alkyne handle radical combines with the RNA radical.

>> ○ **Step 5.** React alkyne-RNA with biotin azide to produce biotinylated RNA.

>> ○ **Step 6.** Separate only the biotinylated RNA using affinity chromatography with streptavidin.

>> ○ **Step 7.** RNA-seq allows for the detection of RNAs close to the specific target.

>>> ○ Reason: Radicals are unstable and cannot travel long distances.

<br>

![image](https://github.com/user-attachments/assets/e1bce731-d487-49aa-b74a-9ec68d5b4a0f)

**Figure 17.** Principle of Halo-seq

<br>

> ⑥ multi-NTT seq (nanobody tethered transposition followed by sequencing)

> ⑦ [Epigenomics Sequencing](https://jb243.github.io/pages/1431)(epigenomics sequencing)

> ⑧ [3D Sequencing](https://www.science.org/doi/10.1126/sciadv.abb3446?url_ver=Z39.88-2003&rfr_id=ori:rid:crossref.org&rfr_dat=cr_pub%20%200pubmed)

> ⑨ Temporal Sequencing

>> ○ [Record-seq](https://www.nature.com/articles/s41596-019-0253-4)

>> ○ [Live-seq](https://www.nature.com/articles/s41586-022-05046-9)

>> ○ [TMI](https://www.biorxiv.org/content/10.1101/2022.08.22.504781v1.full)

>> ○ [molecular recording](https://amjad.mcdb.ucla.edu/research/)

> ⑩ Spatiotemporal Omics

>> ○ [ORBIT](https://www.nature.com/articles/s41586-019-1397-7.epdf?shared_access_token=3i3U430q2qLQV0x7sK6YBtRgN0jAjWel9jnR3ZoTv0P1z_iF2KzWeM8MdvOqRyOhVwD-UmF4piQyH4k0q2ZGmjqKtV7ZuIF9EtLgYzNrbaH5yunNuXyfh1mVEVB14ssbcwk1gMVYOIe9LIottGEL-A%3D%3D) (single-molecule DNA origami rotation measurement)

>> ○ 4D spatiotemporal MRI or hyperpolarized MR

>> ○ _in vivo_ 4D omics with transparent mice

⒄ NGS (next-generation sequencing) Summary

> ① Cost of genome analysis

>> ○ 2001: Human Genome Project benchmark $100 million / person

>> ○ 2007: 100 billion won / 4 years

>> ○ 2008: 454 Life Sciences standard $1,000,000 / person. 1.5 billion won / 4.5 months

>> ○ 2009: Helicos BioSciences standard $48,000 / person

>> ○ Predicted to be sufficient with one million won by 2014 (Nature 456, 23-25, 2008)

> ② Scale of genome analysis

<br>

![image](https://github.com/user-attachments/assets/287653e0-f5b1-4d4b-b8c1-8488f3b73e0b)

 **Figure 18.** Trend of genome analysis scale 

<br>

> ③ Relationship between depth and coverage

>> ○ sequencing depth (read depth): Indicates how many times a specific nucleotide appears on average

<br>

![image](https://github.com/user-attachments/assets/e50295ed-070f-40cc-8ca8-2c5f97a9ce93)

**Figure 19.** Definition of depth

<br>

>>> ○ "10x" means it was read 10 times

>>> ○ Can be defined for each nucleotide

>> ○ coverage (c)

>>> ○ c: = LN / G

>>> ○ L: read length

>>> ○ N: number of reads

>>> ○ G: haploid genome length

>> ○ [Comparison of depth and coverage](https://www.ecseq.com/support/ngs/how-to-calculate-the-coverage-for-a-sequencing-experiment#:~:text=The%20term%20%E2%80%9Ccoverage%E2%80%9D%20in%20NGS%20always%20describes%20a%20relation%20between%20sequence%20reads%20and%20a%20reference%20\(e.g.%20a%20whole%20genome%20or%20al%20locus\)%2C%20unlike%20sequencing%20depth%20which%20describes%20a%20total%20read%20number%20\(Fig.%201\).)

>>> ○ Sequencing depth represents **total read number**

>>> ○ Coverage represents the relationship between sequence reads and **reference** (e.g. whole genome, al locus)

>>> ○ Otherwise, depth and coverage are very similar concepts

> ④ Relationship between bulk and read

>> ○ bulk: total RNA production

>> ○ In case of equal depth, as bulk increases, [RNA read count is inversely proportional](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2010-11-3-r25#:~:text=Estimated%20normalization%20factors,factor%20of%202.), causing irrationality

>>> ○ Example: In spatial transcriptomics, bulk is typically large and depth is low, resulting in low RNA read count

>> ○ Normalization: Various methods have been introduced to resolve this irrationality

> ⑤ Relationship between read count and number of reads

>> ○ If read length is less than 250 bp, it is impossible to detect sequence error

>> ○ Relationship between read length and number of reads per run: there is a trade-off

<br>

![image](https://github.com/user-attachments/assets/7e9f7fb1-44c7-4694-aa6f-ab016f4265ff)

**Figure 20.** Relationship between read length and number of reads per run 

<br>

> ⑥ Relationship between transcriptome read count and gene expression

>> ○ read count: Actual number of transcripts

>> ○ gene expression: Value corrected from read count through [normalization process](https://jb243.github.io/pages/2050#footnote_link_67_50)

<br>

---

_Entered: 2015.07.02 23:31_

_Updated: 2022.03.13 13:11_
