## **Chapter 9. DNA Technology**

Higher category **:** 【Biology】 [Biology Index](https://jb243.github.io/pages/1457) 

---

**1.** [DNA preparation](#1-dna-preparation)

**2.** [Gene library](#2-gene-library--dna-storage)

**3.** [PCR](#3-pcr)

**4.** [DNA fingerprinting](#4-dna-fingerprinting)

**5.** [Hybridization](#5-hybridization)

**6.** [Gene deletion](#6-gene-deletion)

**7.** [Nuclear substitution](#7-nuclear-substitution)

**8.** [DNA-protein interaction studies](#8-dna-protein-interaction-studies)

**9.** [Protein-protein interaction studies](#9-protein-protein-interaction-studies)

**10**. [Gene therapy](#10-gene-therapy)

---

<br>

## **1\. DNA recombination** **:** DNA synthesis

⑴ 1<sup>st</sup>. Target gene generation from mRNA (for eukaryotic DNA)

> ① Reverse Transcription of mRNA to Complementary DNA to mRNA (cDNA)

> ② RT-PCR **:** DNA primer added to mRNA solution and one-time reverse transcriptase treatment

>> ○ oligo dT primer **:** Complete cDNA can be synthesized using poly A as a template for eukaryotic-derived mRNA

>> ○ random primer (usually hexomer) **:** Can form a variety of cDNA pools  

> ③ RT-PCR **:** Amplify the amount of cDNA like a normal PCR

⑵ 2<sup>nd</sup>. Recombinant plasmid

> ① Restriction enzyme **:** A kind of endonuclease, a scissors that cuts DNA

>> ○ Purpose to attack foreign DNA in bacteria

>> ○ DNA that can be attacked is methylated. 4-⑽-③Protection

>> ○ Sticky end and blunt end

>>> ○ Sticky ends **:** Terminals that can be cut back to restriction enzymes to form complementary hydrogen bonds

>>> ○ Smooth end **:** Terminal that cannot be rebound after being cut by restriction enzyme

>>> ○ Restriction enzymes that form sticky ends are used for DNA recombination

>> ○ Palindrome awareness **:** If you read 5 'to 3', the sense and anti-sense strands are identical.

>>> ○ TATCGTACGAAC    →    TATC                +            GTACGAAC

>>> ○ ATAGCATGCTTG    →    ATAGCATG                    CTTG

>>> ○ The above example is an example of sticky ends

>> ○ Restriction enzyme example

>>> ○ AgeⅠ **:** 5'-A ▼ CCGGT-3 '

>>> ○ BamHⅠ **:** 5'-G ▼ GATCC-3 '

>>> ○ BglⅡ **:** 5'-A ▼ GATCT-3 '

>>> ○ EcoRⅠ **:** 5'-G ▼ AATTC-3 '

>>> ○ HindⅢ **:** 5'-A ▼ AGCTT-3 '

>>> ○ HpaⅡ **:** 5'-CC ▼ GG-3 'cleavage, 5'-CmC ▼ GG-3' cleavage, but mC is methylated cytosine

>>> ○ MspⅠ **:** 5'-CC ▼ GG-3 'is cleaved, 5'-CmC ▼ GG-3' is also cleaved, but mC is methylated cytosine

>>> ○ MSTⅡ **:** 5'-CCTN ▼ AGG-3 ', where N is the base sequence of n palindromic structures

>>> ○ NcoⅠ **:** 5'-C ▼ CATGG-3 '  

>>> ○ PstⅠ **:** 5'-CTGCA ▼ G-3 '

>>> ○ SalⅠ **:** 5'-G ▼ TCGAC-3 '

>>> ○ Sau3AⅠ **:** 5'- ▼ GATC-3 '

>>> ○ SmaⅠ **:** 5'-GGG ▼ CCC-3 ', blunt-ended formation

>>> ○ Recombinant sequence by BamHI is also cleaved by Sau3AI, vice versa only in certain cases

>>> ○ The fragment by BglII and the fragment by BamHⅠ bind to each other and are not cut again by restriction enzymes.

>> ○ If linear DNA is cut once with restriction enzymes, it becomes two pieces, whereas circular DNA is cut once with restriction enzymes.

>> ○ Restriction mapping (later updated)

> ② 2<sup>nd</sup>\-1<sup>st</sup>. Cut the target DNA and the plasmid DNA with the same restriction enzyme to form the same sticky ends

>> ○ 2<sup>nd</sup>\-1<sup>st</sup>\-1<sup>st</sup>. Alkiline phospatase (eg, to prevent self-ligation when only one restriction enzyme is processed) **:** Cal intestinal phosphatase (CIP) treatment

>> ○ 2<sup>nd</sup>\-1<sup>st</sup>\-2<sup>nd</sup>. Orientation to Plasmid DNA Binding to Target DNA by Treatment with Two or More Restriction Enzymes

> ③ 2<sup>nd</sup>\-2<sup>nd</sup>. Insert the truncated gene and plasmid into one test tube → Complementary cohesive ends bind to each other

> ④ 2<sup>nd</sup>\-3<sup>rd</sup>. Linkage with DNA ligase to generate recombinant plasmid

⑶ 3<sup>rd</sup>. Transformation **:** Transformation of recombinant genes into host cells (bacteria)

> ① Transformation process

>> ○ 3<sup>rd</sup>\-1<sup>st</sup>. CaCl2 **:** Ca2+ binds to phospholipids and stabilizes cell membranes

>> ○ 3<sup>rd</sup>\-2<sup>nd</sup>. Thermal Shock 42 ℃ **:** Temporary perforation in the cell membrane inserts the recombinant plasmid into the host

> ② Host **:** Escherichia coli, yeast, insect cells, etc.

>> ○ To obtain the product of a recombinant gene, it must be put into a host cell with metabolic function

>> ○ **Condition 1.** First generation period should be short

>> ○ **Condition 2.** Easy to cultivate and grow well on cheap media

>> ○ **Condition 3.** Not pathogenic

>> ○ **Condition 4.** Antibiotic intolerance, screening marker purpose (See. ↓)

> ③ Vector **:** Plasmid or bacteriophage DNA that carries and replicates recombinant DNA

> ④ Cloning vector **:** Vectors aimed at cloning genes

>> ○ Four conditions **:** Origin of origin (_ori_), promoter, restriction enzyme recognition sequence, selectable marker (antibiotic resistance gene)

>> ○ Additional condition **:** Small size, copy number ↑

>> ○ Cloning site

> ⑤ Expression vector **:** Vectors for expressing foreign genes in other hosts with different expression systems

>> ○ Prokaryotic host condition **:** promoter, SD sequence, 70S ribosomal binding site, Transcription terminator, cloning gene cDNA

>>> ○ Tertiary structure (example **:** Disulfide bond)

>> ○ Eukaryotic Host Condition **:** promoter, poly A sequence

>> ○ Example **:** When promoter is added after the promoter of _lac_ operon

> ⑥ **Vector example 1.** pBR322 plasmid

>> ○ 4361 bp, ori present

>> ○ Restriction enzyme recognition sequences such as EcoRI, BamHI and HindIII

>> ○ amp<sup>r</sup> **:** Ampicillin resistance gene in plasmid pBR322, Pst I recognition

>> ○ tet<sup>r</sup> **:** Recognition of the tetracycline resistance gene, HindIII, BamHI, SalI in plasmid pBR322

>> ○ Recombinant strains **:** Selecting tet non-resistant amp resistant strains

> ⑦ **Vector example 2.** pUC19 plasmid

>> ○ 2686 bp, ori present

>> ○ Has restriction enzyme recognition sequence such as ApaLI

>> ○ lac Z **:** X-gal as a substrate to produce a blue product

>> ○ amp<sup>r</sup>

> ⑧ **Vector example 3.** Ti plasmid

>> ○ Agrobacterium tumefaciens (Rhizobium radiobacter) **:** Pathogens of plant myocarcinoma (benign tumor)

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/5a14fa7f-f238-4aa6-b74e-a4a7df656e59)

**Figure 1.** The mycorrhizal fungus attached to carrot cells.

<br>

>> ○ Ti plasmid **:** Myocarcinoma plasmid, including T-DNA

>> ○ T-DNA **:** 20 kb of DNA encodes opines and auxin and cytokinin (myocarcinoma formation) synthase

>> ○ Opine **:** Nutrient Sources of Myocardium Carcinoma

>> ○ Lapse **:** Myocardium carcinoma plant infection → vir (virulance DNA, 25 bp) cuts both ends of T-DNA and inserts randomly into plant nucleochromosomes

>> ○ DNA recombination **:** Inserting target DNA into the T-DNA region, wounding plants and infecting recombinant myocarcinoma

> ⑨ Vector example 4. Gene library vector (See. 2-⑴) **:** BAC Library, YAC Library

> ⑩ Miscellaneous Vector Examples

>> ○ Bacterial plasmid **:** F plasmid, R plasmid (antibiotic resistance gene insertion), approximately 1-10 kb

>> ○ Virus vector

>>> ○ Genetic information is inserted into bacterial DNA through bacterial infection, proliferation, lysis, release and cleavage

>>> ○ λ phage is mainly used, about 25 kb

>>> ○ Sometimes use a retrovirus

>> ○ Cosmid **:** Introduction of phage cos end, about 50 kb by in vitro packaging method

>> ○ Beta-lactamidase gene creates beta-lactam ring to confer penicillin resistance

⑷ 4<sup>th</sup>. Screening **:** Screening for Cells or Populations with Recombinant Genes

> ① Antibiotic resistance gene (selection following normal insertion of recombinant plasmid)

>> Example **:** Ampicillin resistance gene, tetracycline resistance gene

>> ○ 1<sup>st</sup>. replica formation **:** After incubating the entire colony, the entire culture dish is buried in the filter paper

>> ○ 2<sup>nd</sup>. Add antibiotics to some of the colonies on the filter paper

>> ○ 3<sup>rd</sup>. Dead colonies due to antibiotics did not transform

>> ○ 4<sup>th</sup>. Colony selection corresponding to colonies that did not die due to antibiotics on filter paper

> ② Antibiotic resistance genes (screened according to recombination success)

>> ○ 1<sup>st</sup>. replica formation **:** After incubating the entire colony, the entire culture dish is buried in the filter paper

>> ○ 2<sup>nd</sup>. Add antibiotics to some of the colonies on the filter paper

>> ○ 3<sup>rd</sup>. Colonies killed by antibiotics indicate that the restriction enzyme cut the antibiotic resistance gene

>> ○ 4<sup>th</sup>. Colony selection corresponding to dead colonies due to antibiotic on filter paper

> ③ X-gal (screening according to recombination success)  

>> ○ Cloning gene is inserted into lac Z gene

>> ○ 1<sup>st</sup>. X-gal is originally white, but becomes blue due to lac Z.

>> ○ 2<sup>nd</sup>. X-gal white in some colonies containing lac Z-containing plasmids indicates transfection

> ④ Southern blotting (See. ↓↓)

⑸ 5<sup>th</sup>. Cloning amplification **:** Large quantities of product are obtained by mass fermentation and purification

> ① Why antibiotic-resistant plasmids and antibiotics for non-selective purposes in large scale culture

>> ○ 1<sup>st</sup>. Plasmids in the strain randomly enter daughter cells whenever the strain divides

>> ○ 2<sup>nd</sup>. As division progresses, some cells have no plasmids

>> ○ 3<sup>rd</sup>. Cells with plasmids use a lot of energy to make plasmids, so growth rate is different from cells without plasmids

>> ○ 4<sup>th</sup>. Ultimately more cells without plasmids

⑹ Success story

> ① Hormones and bioactive substances **:** Growth hormone, Insulin, secretin, etc.

> ② Therapeutic and diagnostic reagents **:** Hepatitis B vaccine, diagnosis of HIV infection, interferon, neoendorphin, analgesic, blood coagulation factor

> ③ Agriculture **:** Golden rice, herbicide-tolerant crops, pest-tolerant crops

> ④ Ti plasmid transgenic neofunctional crop **:** Herbicide tolerance, anthrax crops (seed potatoes, etc.)

> ⑤ Recombinant Strain Development Strategy

>> ○ feedback / antibiotics resistant mutant

>> ○ Auxotrophic mutant

>> ○ Overexpression mutant  

<br>

<br>

## **2\. Gene library :** DNA storage

⑴ Vector for genetic library

> ① Bacterial Artificial Chromosome

>> ○ Advantages **:** No intersection

>> ○ Disadvantages **:** Match the expression system of prokaryotes

>> ○ Used in [human genome projects](https://jb243.github.io/pages/75#1-genome-project).

>> ○ It is still used today

> ② YAC (Yeast Artificial Chromosome) (pYAC3)

>> ○ Artificial chromosome into yeast (~ 1 Mb), ARS retention, DNA insertion between the tautomer and telomere

>> ○ Advantages **:** Eukaryotes are used to make human proteins easy

>> ○ Disadvantages **:** Cross from the first Meiosis electricity

> ③ Virus genome library

>> ○ Disadvantages **:** Small capacity, difficult to control breeding

⑵ Required DNA Elements

> ① Centromere: A GC-rich sequence that attaches to the spindle fibers during cell division, ensuring proper distribution to each daughter cell.

> ② Telomere: A sequence of base pairs located at the end of a chromosome, serving to counteract the shortening of chromosomes during cell division.

> ③ Origin of replication: The site where DNA replication begins.

> ④ Selection marker: A specific DNA sequence used to confirm the proper insertion of artificial chromosomes into cells.

> ⑤ Other: Promoters, gene expression regulatory mechanisms, and other related elements.

⑶ Types

> ① gDNA Library

>> ○ 1<sup>st</sup>. Precipitate histones by treating with phenol.

>> ○ 2<sup>nd</sup>. Nucleic acids exist in the upper layer of the genomic DNA (gDNA) solution.

>> ○ 3<sup>rd</sup>. Separate the upper layer → Tri chloroacetic acid → Nucleic acid aggregates.

>> ○ 4<sup>th</sup>. Clone all gDNA after restriction enzyme cleavage into a vector (BAC).

>> ○ 5<sup>th</sup>. Selectively amplify Escherichia coli containing the target DNA.

>> ○ 6<sup>th</sup>. Used in intron research.

>> ○ Triton X-100 is amphipathic and does not denature the DNA structure.

>> ○ SDS-PAGE denatures the DNA structure.

> ② cDNA Library

>> ○ 1<sup>st</sup>. Affinity chromatography: Attach oligo-dT (poly T) to beads.

>> ○ 2<sup>nd</sup>. Reverse transcribe separated mRNA using reverse transcriptase.

>> ○ 3<sup>rd</sup>. Treat with low-concentration RNAase acting as primers.

>> ○ 4<sup>th</sup>. Treat with DNA polymerase I.

>> ○ 5<sup>th</sup>. Treat with ligase.

>> ○ 6<sup>th</sup>. Treat with restriction enzyme.

>> ○ 7<sup>th</sup>. Insert into a vector.

>> ○ 8<sup>th</sup>. Clone and amplify.

>> ○ 9<sup>th</sup>. Used in the study of gene expression.

> ③ Library Screening

>> ○ Move library clones to a filter, and activate the probe.

>> ○ Locate clones with the same sequence as the probe when screened against the library.

<br>

<br>

## **3\. PCR** (polymerase chain reaction) **:** DNA amplification

⑴ Defition **:** DNA amplification technology applying the principle of DNA replication

> ① Kerry Mullis, 1993 Nobel Prize in Chemistry

⑵ Sample **:** DNA sample, DNA primer (2 types), dNTP, Taq polymerase, Buffer (pH stabilization), MgCl2 (DNA stabilization)

> ① NTP RNA Polymeric Materials

> ② Taq polymerase

>> ○ Taq polymerase is obtained from the archaeological bacterium Thermus aquaticus that lives in hot spring water.

>> ○ Taq polymerase 55 ℃ optimum polymerization, 92 ~ 95 ℃ heat denaturation

>> ○ Taq polymerase additionally adds A after polymerization to the 3 'end.

>>> ○ No template required, so can be applied to recombination of blunt-ended DNA

> ③ DNA has anion repulsion of phosphate skeleton, so unstable structure → Na + or Mg2 + can stabilize DNA

⑶ Step

> ① 1<sup>st</sup>. DNA denaturing **:** Heat to separate two strands of DNA, 94 ° C, 30-60 seconds

> ② 2<sup>nd</sup>. Annealing **:** As the mixture cools, the primers bind to the DNA template, 50-55 ° C, for 1 minute

> ③ 3<sup>rd</sup>. DNA extension **:** Polymerase starts synthesis at primer, 72 ° C, 1 minute per 1000 bp

> ④ 4<sup>th</sup>. ① To③ Repeating the process would theoretically duplicate an exponentially DNA sample of 2

> ⑤ 5<sup>th</sup>. DNA ligase (ligase) action (40 ° C) and stabilization (4 ° C)

> ⑥ 6<sup>th</sup>. Switch to E. coli cloning amplification after a few cycles **:** In reality, the amount of artificial nucleotides is insufficient, producing much less DNA

⑷ T<sub>m</sub> vs annealing temperature

> ① Too high annealing results in poor hybridization, too low an increase in mis-hybridization  

> ② T<sub>m</sub> **:** 50% retains double strands, 50% proportionate to the released temperature, typically (A + T) × 2 + (G + C) × 4

> ③ Usually the annealing temperature is set as low as 5-10 ℃ at Tm

> ④ The presence or absence of metal ions affects the optimum annealing temperature.

⑸ Limit **:** Need to know the nucleotide sequence at both ends of template DNA

> ① Home **:** If there is a known sequence in the middle and both ends of the template DNa are blunt ends, the limitations can be overcome.

> ② **Solution 1.** inverse-PCR

>> ○ 1<sup>st</sup>. Template DNA Generation through Cyclic DNA Ligase

>> ○ 2<sup>nd</sup>. Restriction enzymes cut inside known sequences, resulting in linear DNA

>> ○ 3<sup>rd</sup>. PCR is possible because we know sequence of both ends  

> ③ **Solution 2.** anchored-PCR

>> ○ 1<sup>st</sup>. Amplify half by connecting new primer with template DNA and DNA ligase

>> ○ 2<sup>nd</sup>. The other half goes through the same process.

> ④ **Solution 3.** oligonucleotide-directed mutagenesis

>> ○ **Case 1.** How to replicate template DNA with primers that only match a few bases

>> ○ **Case 2.** Method of replicating template DNA with primer added with base sequence to form hairpin structure in the middle

>> ○ **Case 3.** Method of replicating template DNA with primers without base sequence to make template DNA form hairpin structure

⑹ Kinds

> ① RT-PCR **:** Involves the creation of cDNA from RNA using reverse transcription enzymes

> ② Real Time PCR (QPCR, Quantitative PCR) **:** Check the amount of amplified DNA in real time

> ③ multiplex PCR **:** Amplification of multiple DNA samples in one PCR device

<br>

<br>

## **4\. DNA fingerprinting** **:** DNA discrimination, no two genetically identical twins are genetically identical

⑴ Objective **:** Paternity confirmation, prediction of genetic diseases, forensic perpetrator identification.

⑵ Electrophoresis **:** Material separation and quantity estimation by size

> ① Stationary phase, mobile phase

>> ○ Compression gel(stacking gel) 

>>> ○ Higher gel density results in slower transport of DNA samples or proteins

>>> ○ Reason for high gel density **:** To keep the moving speed constant regardless of DNA sample or protein size

>> ○ Running gel 

>>> ○ Low gel density allows rapid movement of DNA samples or proteins

>>> ○ Gel density is 6-15%

> ② Nucleic acid electrophoresis **:** Agarose gel

> ③ Protein electrophoresis **:** Polyacrylamide Gel + Bisacrylamide Gel, SDS-PAGE, TEMED

>> ○ SDS-PAGE **:** Used for more efficient separation, consisting of SDS (negative charge surfactant) and β-ME (β-mercaptoethanol)

>> ○ SDS **:** Elimination of ionic bonds, hydrophobic interactions, etc., to ensure that portions of the protein have a constant negative charge density

>> ○ β-ME **:** Disulfide bond removal

>> ○ When SDS-PAGE is attached to each amino acid, all proteins become secondary structure of the same charge density due to the repulsive force between negative charges.

>> ○ Ammonium persulfate (APS) **:** Role of Enzymes to Form Crosslinks of Polyacrylamide Gels and Bisacrylamide Gels

>> ○ TEMED (N, N, N ', N'-tetramethylethylenediamine) **:** Role of stabilizing APS

>> ○ native-PAGE **:** SDS-PAGE to be a modified gel and to be an unmodified gel

> ④ Typically the top is the anode, the bottom is the cathode

> ⑤ moving distance = Δt × (a × log M + b), a < 0, M : molecular weight

>> ○ Molecular weight is linearly proportional to the size of special materials, which reduces the travel distance due to high resistance in the mobile phase.

>> ○ Travel distance is proportional to time (Δt)

>> ○ DNA ladder (size marker) **:** It is possible to infer the size of an unknown substance by presenting the movement distance according to the size.

>> ○ supercoil DNA has little resistance in electrophoresis

> ⑥ **Type 1.** General electrophoresis

>> ○ More low molecular weight nucleic acids and proteins

>> ○ In protein electrophoresis, proteins move from the (-) pole to the (+) pole by SDS-PAGE.

> ⑦ **Type 2.** Isoelectric electrophoresis **:** Electrophoresis using isoelectric points, which are unique values ??of proteins

>> ○ Electrophoresis with a pH gradient moves proteins to the point where the total charge is zero (isoelectric point)

>> ○ After electrophoresis, staining with protein dyes such as coomassie blue or silver nitrate

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/b02150a3-25da-4633-915e-4c89b02fd8cd)

**Figure. 2.** Isoelectric electrophoresis

<br>

> ⑧ Type 3. 2D electrophoresis

>> ○ O'Farrel law is common

>> ○ x axis **:** Isolation according to charge by isoelectric point electrophoresis in the presence of elements or electrophoresis under undenatured conditions

>>> ○ Element **:** Eliminates all R-R interactions except disulfide bonds

>> ○ y axis **:** Separation Depending on Molecular Weight by Electrophoresis in the Presence of SDS

>> ○ One can separate more than 1,000 kinds of proteins

>> ○ **Application 1.** Separate the x-axis and y-axis in the same way, but add a specific substance during the y-axis separation

>> ○ **Application 2.** tRNA, for small RNA molecules **:** 1D 10%, 2D 20% polyacrylamide gel concentration → Effective separation of various RNAs

⑶ **Fingerprint method 1.** RFLP (restriction fragment length polymorphism) **:** Paternity, genetic analysis

> ① Process **:** Target genes

>> ○ 1<sup>st</sup>. DNA isolation from tissue

>> ○ 2<sup>nd</sup>. PCR **:** Amplify the amount of DNA

>> ○ 3<sup>rd</sup>. Cutting DNA into Fragments Using Restriction Enzymes

>> ○ 4<sup>th</sup>. Generating DNA fragments of different sizes with the same restriction enzyme recognition site

>> ○ 5<sup>th</sup>. Electrophoresis **:** Allows you to see the pieces separated by size differences

> ② Cause **:** Single nucleotide polymorphism (SNP)

>> ○ Defition **:** Difference between people due to one base being different by point mutation

>> ○ One SNP per 1000 bp per person

>> ○ SNPs can occur in both genes and introns

>> ○ RFP may occur when SNP occurs at restriction enzyme recognition site

>> ○ Difference between Mutation and Polymorphism: If it occurs in less than 1% of the entire population, it is considered a mutation; otherwise, it is classified as polymorphism.

>> ○ Number of SNPs **:** As of 2017, there are approximately 10,000 single nucleotide polymorphisms (SNPs).

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/da6a1a42-d2e2-48ec-be9c-a0d593272a49)

**Figure. 3.** Number of SNPs

<br>

> ③ Limit **:** RFLP does not vary greatly from person to person since SNPs cannot occur at the restriction sites

>> ○ It is hard to utilize for identification of criminal

⑷ **Fingerprinting 2.** Repeat repeat **:** Targeting criminals, genetic fingerprints stronger than RFLP, and repetitive sequences

> ① VNTR (variable number tandem repeat) **:** 15-100 repetitions

>> ○ Principle **:** Different sizes of repeat sequences are created between individuals

>> ○ Cause **:** Uneven Crossover of Homologous Chromosome

>> ○ 1<sup>st</sup>. DNA fragments separated and amplified along the length by electrophoresis

>> ○ 2<sup>nd</sup>. Separated DNA fragments were transferred to filter paper

>> ○ 3<sup>rd</sup>. Transfer to filter paper and treat chemical (basic) to break hydrogen bonds → one strand of DNA

>> ○ 4<sup>th</sup>. Compare the location of the VNTR fragments with a VNTR probe labeled with radioactive material

>> ○ 5<sup>th</sup>. Band appears when exposing filter paper to X-ray film

> ② STR (Short Tendom Repeat) **:** 2-4 repetitions

>> ○ Not much different than VNTR

<br>

<br>

## **5\. Hybridization** **:** Check DNA presence

⑴ Southern blotting **:** Nucleic Acid Probe Hybridization to DNA

> ① Summary

>> ○ Some sort of RFLP

>> ○ Edward M 1975. Developed by Southern

> ② 1<sup>st</sup>. Restriction enzyme treatment → PCR amplification → electrophoresis

>> ○ Electrophoresis **:** Use agarose gel

>> ○ Master plate can be used instead of electrophoresis **:** List the colonies into which the recombinant plasmid was introduced in a solid medium

> ③ 2<sup>nd</sup>. Gel 0.Shake in 5 M HCl solution

>> ○ Faster DNA transfer from agarose gel to membrane by HCl treatment

> ④ 3<sup>rd</sup>. Gel 1.5 M NaCl, 0.Shaking in 5 M NaOH solution (DNA denaturation)

>> ○ Separate DNA into single strands by placing NaOH solution-sponge-gel

>> ○ Sponge absorbs NaOH solution and provides it to the gel

>> ○ Additional nucleic acid denaturant is added to prevent partial double bond formation of ssDNA

>> ○ Nucleic acid denaturants **:** Form ions, such as metal ions, formamide and formaldehyde, weaken DNA binding ability

> ⑤ 4<sup>th</sup>. Gel 1.5 M NaCl, 1 mM EDTA, 0.5 M Tris-HCl (pH 7.2) shaking in solution

>> ○ Trsi-HCl **:** Buffer

> ⑥ 5<sup>th</sup>. Capillary phenomenon + blotting

>> ○ 5<sup>th</sup>-1<sup>st</sup>. Place a positively charged nylon filter and a paper towel on the gel

>>> ○ Nitrocellulose paper can be used instead of nylon filter

>> ○ 5<sup>th</sup>-2<sup>nd</sup>. NaOH solution transferred to gel is absorbed by carrying single-stranded DNA and moving to the towel bundle

>> ○ 5<sup>th</sup>-3<sup>rd</sup>. Negatively charged single-stranded DNA sticks to nylon membrane

> ⑦ 6<sup>th</sup>. Autoclave **:** Nylon membrane combined with DNA is treated under high pressure and high temperature to fix the bond.

>> ○ UV light can cause DNA to bind to the membrane

> ⑧ 7<sup>th</sup>. Pre-Blocking **:** Improved sharpness by adding blocking agent

>> ○ 7<sup>th</sup>-1<sup>st</sup>. Identify DNA that is considered insignificant through repeated or allele-specific electrophoresis

>> ○ 7<sup>th</sup>-2<sup>nd</sup>. Add salmon sperm DNA fragments to the DNA

>> ○ 7<sup>th</sup>-3<sup>rd</sup>. Prevents probe DNA from binding to specific DNAs during 8<sup>th</sup> hybridization (enhances sharpness)

> ⑨ 8<sup>th</sup>. Hybridization

>> ○ 8<sup>th</sup>-1<sup>st</sup>. DNA-bound nylon membranes in seal-a-meal bags

>> ○ 8<sup>th</sup>-2<sup>nd</sup>. Add radiolabeled DNA probe to seal-a-meal bag at 62 ° C

>>> ○ Labeled probes have complementary sequences of genes of interest

>>> ○ Probe DNA **:** 100-500 bp, single stranded DNA

>>> ○ Probe DNA typically labels P32, a radioisotope

>>> ○ Probe DNA can be labeled with dyes such as EtBr and SYBR Green

>> ○ 8<sup>th</sup>-3<sup>rd</sup>. Probes bind to complementary DNA

>>> ○ Other DNA that is considered insignificant through repeated experiments is covered with salmon sperm DNA by prehybridization

> ⑩ 9<sup>th</sup>. Eliminate unhybridized radioactive probes in seal-a-meal bags

> ⑪ 10<sup>th</sup>. Autoradiogram **:** Irradiation on nylon membrane

> ⑫ 11<sup>th</sup>. Identifies the location (or colony location) on the radioactive electrophoresis

⑵ Northern blotting **:** Nucleic Acid Probe Hybridization for RNA

> ① Very similar to Southern blotting but with the following differences

> ② **Difference 1.** Extraction of samples

>> ○ Southern blotting uses DNA extraction

>> ○ Northern blotting effectively isolates mRNA from extracts by affinity chromatography using oligo dT.

> ③ **Difference 2.** Use of restriction enzyme

>> ○ Southern blotting is very long and must be used with restriction enzymes

>> ○ Northern blotting uses RNA, not DNA, so restriction enzymes cannot be used

> ④ **Difference 3.** Type of solution in capillary action

>> ○ Southern blotting uses basic solutions to denature single-stranded DNA

>> ○ Use of Basic Solution in Northern Blots Decomposes RNA

>> ○ Northern blotting uses a salt solution to stabilize RNA

> ④ **Difference 4.** Type of probe

>> ○ Southern blotting uses a gDNA probe

>> ○ Northern blotting uses cDNA probe

⑶ Western blotting **:** Antibody Probe Hybridization for Proteins

> ① Numerous differences with nucleic acid blotting

> ② **Difference 1.** Type of gel

>> ○ Nucleic Acid Blotting Using Agarose Gel

>> ○ Western blotting uses polyacrylamide gel, SDS-PAGE, etc.

> ③ **Difference 2.** electrophoresis voltage condition

>> ○ In nucleic acid blotting, nucleic acid transfers well to electrophoresis, so a low voltage must be applied.

>> ○ In Western blotting, proteins are not well transported by electrophoresis and high voltage must be applied.

> ④ **Difference 3.** Blotting Process

>> ○ Nucleic Acid Blotting Uses Capillary Phenomenon During Blotting

>> ○ Western blotting uses an electric field during blotting

> ⑤ **Difference 4.** Nucleic Acid Pretreatment

>> ○ Nucleic Acid Blotting Using Salmon Sperm DNA Fragments

>> ○ Western blotting uses casein, skim milk, bovine serum albumin (BSA), etc.

>> ○ BSA can also be used for nucleic acid blotting

> ⑥ **Difference 5.** Type of probe

>> ○ Nucleic Acid Blotting Using DNA Probes

>> ○ Western blotting uses an antibody probe

>>> ○ Primary antibody **:** Target Proteins and Specific Antibodies

>>> ○ For specificity of primary antibodies, enzymes from other animals should be used

>>> ○ Secondary antibody **:** Combined with primary and specific antibodies, enzymes that degrade chromogenic substrates

>>> ○ For specificity of secondary antibodies, enzymes from target and primary antibody derived animals and other animals should be used

> ⑦ **Difference 6.** Imaging Method

>> ○ Nucleic Acid Blotting Using EtBr + Self-radiation

>> ○ Western blotting uses Coomas blue staining

⑷ DNA chip (microarray)

> ① Simultaneously check the expression patterns of multiple genes

>> ○ One compartment contains one cDNA probe.

> ② Confirmation of tissue-specific gene expression through cDNA DNA chip

>> ○ 1<sup>st</sup>. abrication of cDNA Chips by Dropping Human Gene cDNA Libraries on Glass Slides

>> ○ 2<sup>nd</sup>. Treated 1% BSA Solution on cDNA Chip

>>> ○ BSA already binds to DNA, allowing only complementary DNA to bind

>> ○ 3<sup>rd</sup>. Normal tissue, abnormal tissue (e.g., **:** MRNA X and Y are each prepared from cancer)

>> ○ 4<sup>th</sup>. Add oligo-dT to X and Y, respectively

>>> ○ oligo-dT complementarily binds to poly A sequences in X and Y

>> ○ 5<sup>th</sup>. RT-PCR **:** X adds dNTP and cy3 (green fluorescent substance) -dTTP, and Y adds dNTP and cy5 (red fluorescent substance) -dTTP, respectively, and performs reverse transcription reaction to generate fluorescently labeled cDNA.

>> ○ 6<sup>th</sup>. 0 each for X and Y.1 N NaOH and reacted for 10 minutes at 70 ℃ 0.Neutralize with 1 N HCl

>>> ○ Template RNA digestion

>> ○ 7<sup>th</sup>. The cDNA synthesized in X and Y was purified and mixed in equal amounts to hybridize with the cDNA chip.

>> ○ 8<sup>th</sup>. Washing cDNA Chips with Buffer

>> ○ 9<sup>th</sup>. scanning (scanning) **:** Measure and calibrate the fluorescence intensity of cy3, cy5

>>> ○ Black **:** X and Y do not express both

>>> ○ Green **:** X only expression

>>> ○ Red **:** Y only expression

>>> ○ Yellow **:** Expression of both X and Y

⑸ Fluorescence in situ hybridization (FISH)

> ① Summary

>> ○ Experimental Techniques to Hybridize Specific DNA Sequences with Chromosome Phases

>> ○ Used to locate RNA as well as the location of specific sequences

>> ○ Hybridization with fluorescent probes complementary to specific sequences, followed by fluorescence microscopy

>> ○ Diagnose chromosomal abnormalities quickly but cannot diagnose DNA abnormalities

> ② Process

>> ○ 1<sup>st</sup>. Prepare epithelial tissue sections on slides

>> ○ 2<sup>nd</sup>. Treated with RNase and incubated for 1 hour at 37 ℃

>>> ○ Probes that bind to DNA may bind to RNA, so there is a process to remove RNA

>> ○ 3<sup>rd</sup>. Change the buffer to acidic acid, treat pepsin and react for 10 minutes at 37 ℃

>>> ○ Pepsin acts in acid, changing pH composition to acid

>>> ○ Pepson removes proteins in cells, facilitating further penetration of probes

>> ○ 4<sup>th</sup>. Wash

>> ○ 5<sup>th</sup>. Prepare a probe that can bind to isotope, but biotin is attached to dTTP of this probe

>> ○ 6<sup>th</sup>. Hybridize 4<sup>th</sup> Sample and 5<sup>th</sup> Probe

>> ○ 7<sup>th</sup>. Wash

>> ○ 8<sup>th</sup>. Fully avidin processed and washed

>>> ○ Fluorescent material should be directly attached

>> ○ 9<sup>th</sup>. Dye with DAPI **:** DAPI Nuclear Staining

>>> ○ DAPI (4'-6-diamidino-2-phenylindole)

>>> ○ Fluorescent dye that binds strongly to the AT-rich region of DNA

>>> ○ DAPI is cell membrane permeable which can be used for staining both living and fixed cells

>>> ○ UV irradiation on DNA bound to DAPI produces blue fluorescence

>> ○ 10<sup>th</sup>. Observation of the isotope region will show a fluorescent color of avidin, DNA will show a blue fluorescent color

⑹ Dideoxy chain termination **:** DNA sequence determination

<br>

<br>

## **6\. Gene deletion :** DNA function check

⑴ Knock-out mouse **:** Study the function of specific genes

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/b88b7a38-3aaa-4fc5-bd40-c0b28abcc223)

**Figure. 4.** Knockout process

<br>

> ① 1<sup>st</sup>. Knockout Induction of Gene X in Embryonic Stem (ES) Cells

>> ○ 1<sup>st</sup>-1<sup>st</sup>. Construction of plasmid vector **:** Contains the thymidine kinase (TK) gene, which is remote from the inactivated gene X by inserting the neor (neomycin resistant gene) gene

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/4ae2fe47-8217-4d3d-9bcc-db7f1f9f0945)

**Figure. 5.** The structure of plasmid vector 

<br>

>> ○ 1<sup>st</sup>-2<sup>nd</sup>. Insert targeting vector inside ES cell

>> ○ 1<sup>st</sup>-3<sup>rd</sup>. Some ES cells take care of replacing existing gene X with inactivated gene X

>>> ○ Be sure to include the same gene next to gene X so that genetic recombination can occur

>> ○ 1<sup>st</sup>-4<sup>th</sup>. Selection **:** Cells are cultured in medium containing G418 (geneticin, neomycin derivative), gancyclovir

>>> ○ 1<sup>st</sup>-4<sup>th</sup>-1<sup>st</sup>. Untransformed ES Cells **:** Killed by G418

>>> ○ 1<sup>st</sup>-4<sup>th</sup>-2<sup>nd</sup>. ES cell with site other than gene X replaced **:** TK gene is killed by gancyclovir (TK gene breaks down gancyclovir and causes toxicity)

>>> ○ 1<sup>st</sup>-4<sup>th</sup>-3<sup>rd</sup>. Surviving only ES cells replaced with gene X only

> ② 2<sup>nd</sup>. Knocked Out ES Cell Selection

> ③ 3<sup>rd</sup>. Injecting knocked out ES cells into the embryo

> ④ 4<sup>th</sup>. Germ cells of chimeric mice are produced by germ cells from normal germ cells and knocked out ES cells.

> ⑤ 5<sup>th</sup>. F1 mice have (+ / +) and (+/-) objects

> ⑥ 6<sup>th</sup>. Genetic testing for self-breeding by selecting (+/-) individuals from F1 mice

>> ○ There is a limit to screening through mouse hair color

> ⑦ 7<sup>th</sup>. 25% of F2 mice are knockout mice and should be screened by genetic testing

⑵ Cre-lox

> ① DNA recombination by Cre occurs only between identical lox sequences

> ② Cre deletes DNA when the lox sequence pairs are in the same direction, and DNA inversion occurs when the lox sequences are in the opposite direction

⑶ SiRNA **:** RNA interference can be used to suppress all gene expression

<br>

<br>

## **7\. Nuclear substitution**

⑴ GMO (Genetically Modified Organism): Genetically altered substances, food, organisms.

⑵ Methods:

> ① Nuclear Transplantation

>> ○ Utilizes techniques such as microinjection.

>> ○ Applicable to gene transformation and animal cloning.

> ② Trait Transformation without Using a Vector

>> ○ Can induce trait transformation using gene guns, among other methods.

>> ○ **Example 1 :** Formation of transgenic plant cells by "shooting" particles containing recombinant genes into plant cells.

>> ○ **Example 2 :** Transgenic plant cells → callus → organism formation (in nutrient medium).

>> ○ Plant cells exhibit totipotency.

>> ○ **Example 3 :** Maintenance and propagation of purebred, proliferation of useful plants.

>> ○ **Example 4 :** Tissue culture with meristematic regions, induction of differentiation.

⑶ Examples:

> ① **Example 1 :** Transgenic Animals - Mass production of beneficial gene products in animals (e.g., farming).

> ② **Example 2 :** Edible Vaccines.

> ③ **Example 3 :** Genetically Modified Food (GM Food)

>> ○ Past **:** Gene-modified crops obtained by increasing the frequency of specific alleles through selective breeding (artificial selection).

>> ○ Present **:** Genetic recombination technology → Increased shelf life, production rate (resistance to pests, weeds, diseases, drought, and cold).

>> ○ Example **:** Golden Rice - Genetically modified to produce beta-carotene (increases nutritional value of rice).

⑷ [GMO Debate and Principles of Safety Assessment](https://jb243.github.io/pages/1432)

<br>

<br>

## **8\. DNA-protein interaction studies**

⑴ Gene footprinting assay, also known as DNA footprinting technology:

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/cf5c3183-a4da-43b8-aedf-0f09163d6116)

**Figure. 6.** Application of gene footprinting

<br>

> ① Genes binding with proteins appear to be missing on the gel electrophoresis.

> ② It allows the identification of transcription factor binding sites.

⑵ electrophoretic mobility shift assay (EMSA) **:** Nucleic acid-protein binding investigation

> ① Also called a gel shift assay

> ② 1<sup>st</sup>. Probe production **:** Labeling DNA with Radioactive Isotopes

> ③ 2<sup>nd</sup>. Radiophoresis after electrophoresis of protein and DNA hybrids

> ④ 3<sup>rd</sup>. Result analysis

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/ec8f01e3-fb82-403e-8fa1-29b97860d6c2)

**Figure. 7.** Example EMSA Results</center>

<br>

>> ○ Premise **:** DNA is moved from top to bottom because the top is the cathode and the bottom is the anode.

>> ○ interpretation of a **:** Proteins A and B bind to the probe, and A and B bind with the probe attached..)

>> ○ interpretation of b **:** C protein and D protein bind to probe, C and D do not bind to each other

>> ○ interpretation of c **:** E protein binds to probe, F protein does not bind to probe, but binds to E protein

> ⑤ Generally, protein means transcription factor and it can be applied to measure transcription activity.

⑶ ChIP (Chromatin Immunoprecipitation)

⑷ South-Western Blotting

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/8d1e2640-403f-4d9e-a687-5518501840dd)

**Figure. 8.** South-Western Blotting

<br>

> ① 1<sup>st</sup> step: Proteins are initially subjected to Western blotting.

> ② 2<sup>nd</sup> step: Subsequently, DNA labeled with fluorescence is subjected to Southern blotting.

⑸ Yeast One-Hybrid Assay

⑹ Phage Display Assay

> ① 1<sup>st</sup> step: Introduce mutations or alter nucleotide sequences to create a diverse phage library.

> ② 2<sup>nd</sup> step: Screen for phages with desired activity and later confirm the nucleotide sequence.

⑺ Filter Binding Assay: Only transcription factors bind to the filter.

⑻ DNA Affinity Chromatography

<br>

<br>

## **9\. Protein-protein interaction studies**

⑴ Dual Hybrid System: A method for identifying proteins that interact with protein X.

> ① 1<sup>st</sup>. Gene of transcription factor is divided into two parts: DNA binding domain and transcription activation domain.
   
> ② 2<sup>nd</sup>. Bait: Protein X + DNA binding domain.
   
> ③ 3<sup>rd</sup>. Prey: Protein of interest for determining binding + transcription activation domain.
   
> ④ 4<sup>th</sup>. When the two hybrid proteins interact and bind, the prey induces the expression of a reporter gene.

>> ○ Example of a reporter gene: GFP protein.

⑵ Yeast Two-Hybrid: Confirming protein binding relationships using protein-protein interactions.

⑶ Protein separation and purification using GST-tagged fusion proteins.

⑷ Phage Display Method: Used to obtain monoclonal antibodies.

> ① Aim: Obtaining a single clone antibody that binds to the target protein.
   
> ② Method to obtain a single clone antibody in a test tube.
   
> ③ Typical process for obtaining a single clone antibody: Fusion of B lymphocytes and myeloma cells (hybridoma), becoming stem cells and proliferating to produce a large amount of antibodies.

<br>

<br>

## **10\. Gene therapy**

⑴ CRISPR-Cas9 Gene Editing Technology (also called simply CRISPR technology)

> ① Overview

>> ○ Gene Scissors: A mechanism of gene cleavage in bacteria (e.g., E. coli) to eliminate viral DNA.

>> ○ CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats)

>>> ○ Derived from DNA fragments of bacteriophages that infected unicellular organisms.

>>> ○ Found in 50% of bacterial genomes and 90% of archaeal genomes.

>>> ○ CRISPR arrays consist of repetitive DNA sequences and spacers located between each repetition.

>>> ○ Bacteria or archaea record fragments of external DNA, such as viruses, as spacers between repeats.

>>> ○ Subsequently, the CRISPR-Cas9 system detects and cleaves viral DNA that matches the spacer.

>> ○ Cas9 (CRISPR-associated protein 9)

>>> ○ Acquired from organisms like Streptococcus pyogenes. It has two nuclease domains, HNH and RuVC9.

>>> ○ dCas9 (nuclease-deficient Cas9, dead Cas9): Unable to cut DNA but capable of binding to DNA through sgRNA.

> ② **Step 1.** CRISPR Processing **:** The following describes the CRISPR processing of the subtype I-C/Dvulg Cas5d system in Bacillus halodurans.

>> ○ 1<sup>st</sup>. Cas5d recognizes hairpin structures and 3' single-stranded sequences in the CRISPR repeat region, cleaving pre-crRNA into unit-length fragments.

>> ○ 2<sup>nd</sup>. Pre-crRNA processing: Further processing of pre-crRNA into smaller crRNAs.

>> ○ 3<sup>rd</sup>. Cas5d forms a complex with crRNA, Csd1, and Csd2 proteins.

>> ○ 4<sup>th</sup>. The crRNA portion in this complex detects and removes viral DNA.

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/08c3dc77-1c4c-403d-9c04-38bda4b3eccd)

**Figure. 9.** CRISPR Processing

<br>

> ③ **Step 2.** Reaction to External Nucleic Acids

>> ○ 1<sup>st</sup>. Cas9 creates a bubble in the double-stranded DNA, and the complementary sgRNA (small guide RNA) binds to the target RNA next to the PAM site within that bubble.

>>> ○ The bubble is referred to as DNA double-strand breaks (DSBs).

>>> ○ sgRNA **:** Fragments of viral DNA that remain in the genomic DNA of bacteria (e.g., E. coli) surviving viral infection. Also known as gRNA (guide RNA).

>>> ○ PAM (Protospacer Adjacent Motif) site: A 5'-NGG-3' nucleotide sequence that distinguishes self from non-self.

>> ○ 2<sup>nd</sup>. The sgRNA-Cas9 complex cleaves both the template and non-template strands of the DNA at the same location.

>> ○ 3<sup>rd</sup>. Cas9 and sgRNA separate.

>> ○ 4<sup>th</sup>. The DNA that formed the bubble reanneals through hydrogen bonding.

>> ○ 5<sup>th</sup>. The cleaved template and non-template DNA undergo DNA repair mechanisms.

>>> ○ 5<sup>th</sup> - 1<sup>st</sup>. Viruses without DNA repair mechanisms are eliminated by the CRISPR/Cas9 mechanism.

>>> ○ 5<sup>th</sup> - 2<sup>nd</sup>. Non-homologous End Joining (NHEJ) **:** Allows for the joining of completely different chromosomes, following the Holliday model.

>>> ○ 5<sup>th</sup> - 3<sup>rd</sup>. Homology Directed Repair (HDR) **:** Shows effects like the substitution of specific base pairs.

>> ○ 6<sup>th</sup>. Research is ongoing to replace unnecessary portions of Variable Number Tandem Repeat (VNTR) genes in genetic recombination.

>>> ○ Utilizes homology-directed repair mechanisms.

>>> ○ Can be applied not only to DNA but also to RNA, epigenomes, and other single nucleotides for editing.

>>> ○ Types of gene editing: OE (overexpression), KD (knock-down).

> ④ Off-Target Issues

>> ○ Definition **:** The problem of editing regions other than the target gene when applying the CRISPR/Cas9 system to gene therapy.

>> ○ Streptococcus pyogenes Cas9 (SpCas9) nuclease is commonly used due to its efficiency but has a high off-target ratio.

>> ○ [Solutions](https://en.wikipedia.org/wiki/Off-target_genome_editing)

>>> ○ Nuclease mutation

>>> ○ PAM sequence modification

>>> ○ gRNA truncation

<br>

![image](https://github.com/JB243/jb243.github.io/assets/55747737/97cdf026-4d08-4075-91ee-10aa29c65939)

**Figure. 11.** CRISPR/Cas9 Technology Diagram

<br>

> ⑤ Applications

>> ○ For signal transduction research purposes: Multiplexing guide RNAs enable perturbation screening.

>> ○ For imaging research purposes.

>> ○ For drug research purposes: Studying changes in drug intake when specific genes are suppressed.

>> ○ Gene therapy: Editing genes responsible for rare diseases.

>> ○ Temporal sequencing (e.g., [Record-seq](https://www.nature.com/articles/s41596-019-0253-4)).

⑵ siRNA, miRNA Therapeutics **:** Current market situation is not favorable

⑶ Nucleic Acid Delivery Systems

> ① Overview

>> ○ Adopted commercially with great success by Moderna and Pfizer during the COVID-19 pandemic.

>> ○ Since mRNA is unstable in the body, a delivery vehicle is necessary to ensure stability until it reaches the target tissue or cell.

> ② **Type 1.** mRNA Drug Delivery Systems

>> ○ Solid Lipid Nanoparticle (SLN): The most popular mRNA delivery vehicle adopted by Moderna and Pfizer.

>>> ○ A single SLN of 80-100 nm can encapsulate around 100 mRNA molecules.

>>> ○ Examples **:** ALC-0315 (Pfizer/BioNTech), SM-102 (Moderna), ALC-0159 (Pfizer/BioNTech), PEG-DMG (Moderna).

>> ○ Cationic liposome

>> ○ polymer and polymer/lipid hybrid particle

>> ○ micelle

>> ○ emulsion

> ③ **Type 2.** DNA Drug Delivery Systems

⑹ Mega Nuclease

⑺ TALEN (Transcription Activator-Like Effector Nuclease)

⑻ ZFN (Zinc Finger Nuclease)

<br>

---

*Input **:** 2015.7.03 21:57*

*Modify **:** 2019.1.25 00:18*
