## **Chapter 9. DNA Technology**

Higher category: 【Biology】 [Biology Index](https://jb243.github.io/pages/1457) 

---

**1.** [DNA recombination](#1-dna-recombination)

**2.** [DNA library](#2-dna-library)

**3.** [PCR](#3-pcr)

**4.** [DNA fingerprinting](#4-dna-fingerprinting)

**5.** [Hybridization](#5-hybridization)

**6.** [Gene deletion](#6-gene-deletion)

**7.** [Nuclear substitution](#7-nuclear-substitution)

**8.** [DNA-protein interaction studies](#8-dna-protein-interaction-studies)

**9.** [Protein-protein interaction studies](#9-protein-protein-interaction-studies)

**10**. [Gene editing](#10-gene-editing)

---

**a.** [Nucleic Acid (DNA, RNA) Extraction and Purification Experiment](https://jb243.github.io/pages/1482)

**b.** [Western Blotting Experiment](https://jb243.github.io/pages/1419) 

**c.** [The Romanov Dynasty and Forensic Science](https://jb243.github.io/pages/73)

---

<br>

## **1\. DNA recombination**

⑴ 1<sup>st</sup>. Target gene generation from mRNA (for eukaryotic DNA)

> ① mRNA is reverse transcribed to synthesize cDNA (complementary DNA to mRNA).

> ② RT-PCR: Add DNA primers to the mRNA solution and treat with reverse transcriptase (RT) once.

>> ○ Oligo dT primer: Enables the synthesis of complete cDNA by using the poly A tail, which is present in eukaryotic mRNA, as a template.

>> ○ Random primer (usually hexomer): Can form a variety of cDNA pools.

> ③ RT-PCR: Amplify the amount of cDNA like a normal PCR.

⑵ 2<sup>nd</sup>. Recombinant plasmid

> ① Restriction enzyme: A kind of endonuclease. A scissors that cuts DNA.

>> ○ Restriction enzymes are produced in bacteria to target and degrade foreign DNA.

>> ○ Their own DNA, which could be targeted, is protected by methylation.

>> ○ Blunt ends: Ends that cannot rejoin after being cut by a restriction enzyme.

>> ○ Sticky ends: Ends that can re-form complementary hydrogen bonds after being cut by a restriction enzyme.

>>> ○ Restriction enzymes that form sticky ends are used for DNA recombination.

>> ○ Palindrome recognition: When reading from 5' to 3', the sequences on the sense strand and the antisense strand are identical.

<br>

<img width="377" alt="스크린샷 2025-02-01 오후 3 07 58" src="https://github.com/user-attachments/assets/fffd84b9-bd25-4db2-ba02-c56a108ad517" />

<br>

>>> ○ The above example is a sticky end.

>> ○ Examples of restriction enzymes

>>> ○ AgeI: 5'-A▼CCGGT-3'

>>> ○ AluI: 5'-AG▼CT-3'

>>> ○ BamHI: 5'-G▼GATCC-3'

>>> ○ BglII: 5'-A▼GATCT-3'

>>> ○ DpnI: 5'-G metA ▼TC-3'

>>> ○ DpnII: 5'-▼GATC-3'

>>> ○ EcoRI: 5'-G▼AATTC-3'

>>> ○ FspEI: 5'-C<sup>m</sup>C(N)<sub>12</sub>▼-3' 

>>> ○ HindIII: 5'-A▼AGCTT-3'

>>> ○ HpaII: Cuts at 5'-C▼CGG-3', but not at 5'-C<sup>m</sup>C▼GG-3' where <sup>m</sup>C represents methylated cytosine.

>>> ○ LpnPI: 5'-C<sup>m</sup>CDG(N)<sub>10</sub>▼-3'

>>> ○ McrBC: 5'-Pu<sup>m</sup>C(N<sub>40-3000</sub>)Pu<sup>m</sup>C-3' 

>>> ○ MspI: 5'-C▼CGG-3', 5'-C▼<sup>m</sup>CGG-3' where <sup>m</sup>C represents methylated cytosine.

>>> ○ MspJI: 5'-<sup>m</sup>CNNR(n)<sub>9</sub>▼-3' 

>>> ○ MSTII: 5'-CCTN▼AGG-3', where is a nucleotide sequence with **n** palindromic structures.

>>> ○ NcoI: 5'-C▼CATGG-3'

>>> ○ PstI: 5'-CTGCA▼G-3'

>>> ○ SalI: 5'-G▼TCGAC-3'

>>> ○ Sau3AI: 5'-▼GATC-3'

>>> ○ SmaI: 5'-GGG▼CCC-3', forming blunt ends. 

>>> ○ Recombinant sequence by BamHI is also cleaved by Sau3AI. The opposite holds true only in specific cases.

>>> ○ The fragment by BglII and the fragment by BamHⅠ bind to each other and are not cut again by restriction enzymes.

>> ○ When linear DNA is cut once by a restriction enzyme, it becomes two fragments, whereas circular DNA, when cut once by a restriction enzyme, becomes a single fragment.

>> ○ Restriction mapping 

> ② 2<sup>nd</sup>\-1<sup>st</sup>. Cut the target DNA and the plasmid DNA with the same restriction enzyme to form the same sticky ends.

>> ○ 2<sup>nd</sup>\-1<sup>st</sup>\-1<sup>st</sup>. To prevent self-ligation when treating with only one restriction enzyme, treat with alkaline phosphatase (e.g., Calf Intestinal Phosphatase (CIP)).

>> ○ 2<sup>nd</sup>\-1<sup>st</sup>\-2<sup>nd</sup>. Provide directionality to plasmid DNA binding to target DNA by treating with two or more types of restriction enzymes.

> ③ 2<sup>nd</sup>\-2<sup>nd</sup>. Insert the truncated gene and plasmid into one test tube → Complementary sticky ends bind to each other.

> ④ 2<sup>nd</sup>\-3<sup>rd</sup>. Create a recombinant plasmid by ligating with DNA ligase.

⑶ 3<sup>rd</sup>. Transformation: Insert the recombinant gene into the host cell (bacteria) to induce transformation.

> ① Transformation process

>> ○ 3<sup>rd</sup>\-1<sup>st</sup>. CaCl<sub>2</sub>: Ca<sup>2+</sup> binds to phospholipids and stabilizes cell membranes.

>> ○ 3<sup>rd</sup>\-2<sup>nd</sup>. Thermal Shock 42 ℃: A temporary pore is formed in the cell membrane, allowing the recombinant plasmid to be inserted into the host.

> ② Host: Escherichia coli, yeast, insect cells, etc.

>> ○ To obtain the product of a recombinant gene, it must be put into a host cell with metabolic function.

>> ○ **Condition 1.** First generation period should be short.

>> ○ **Condition 2.** Easy to cultivate and grow well on cheap media.

>> ○ **Condition 3.** Not pathogenic.

>> ○ **Condition 4.** Antibiotic intolerance for screening process.

> ③ Vector: Plasmid or bacteriophage DNA that carries and replicates recombinant DNA.

> ④ Cloning vector: A vector used for gene replication.

>> ○ Four conditions: Origin of replication (ori), promoter, restriction enzyme recognition sequence, selection marker (antibiotic resistance gene)

>> ○ Additional condition: Small size, higy copy number

>> ○ Cloning site

> ⑤ Expression vector: Vectors for expressing foreign genes in other hosts with different expression systems.

>> ○ Prokaryotic host condition: Promoter, SD sequence, 70S ribosomal binding site, transcription terminator. Note that cloning gene is **cDNA**.

>>> ○ Proteins that process tertiary structures (e.g., disulfide bonds) are also required.

>> ○ Eukaryotic host condition: Promoter, poly A sequence

>> ○ Example: When a promoter is attached after the activator gene of the lac operon.

> ⑥ Vector examples

>> ○ **Example 1.** pBR322 plasmid

>>> ○ 4361 bp. ori is present.

>>> ○ It has recognition sequences for restriction enzymes such as EcoRI, BamHI, and HindIII.

>>> ○ amp<sup>r</sup>: The ampicillin resistance gene from plasmid pBR322, recognized by PstI.

>>> ○ tet<sup>r</sup>: The tetracycline resistance gene from plasmid pBR322, recognized by HindIII, BamHI, and SalI.

>>> ○ Recombinant strain: Select a tetracycline-sensitive, ampicillin-resistant strain.

>> ○ **Example 2.** pUC19 plasmid

>>> ○ 2686 bp. ori is present.

>>> ○ Has restriction enzyme recognition sequences such as ApaLI.

>>> ○ lacZ: Produces a blue-colored product using X-gal as a substrate.

>>> ○ amp<sup>r</sup>

>> ○ **Example 3.** Ti plasmid

>>> ○ Agrobacterium tumefaciens (Rhizobium radiobacter): Pathogens of plant crown gall (benign tumor).

<br>

![image](https://github.com/user-attachments/assets/71652fcd-316a-420b-a863-1aa923d91889)

**Figure 1.** Rhizobium radiobacter attached to carrot cells

<br>

>>> ○ Ti plasmid: Rhizobium radiobacter plasmid, including T-DNA.

>>> ○ T-DNA: 20 kb of DNA encodes opines and auxin·cytokinin synthase (forming crown gall).

>>> ○ Opine: Nutrient sources of Rhizobium radiobacter.

>>> ○ Process: Agrobacterium tumefaciens infects the plant → vir (virulence DNA, 25 bp) cleaves both ends of T-DNA and randomly inserts it into the plant nuclear chromosome.

>>> ○ DNA recombination: Inserting target DNA into the T-DNA region after injuring the plant and infecting it with recombinant Rhizobium radiobacter.

>>> ○ The recombinant Agrobacterium tumefaciens does not synthesize auxin or cytokinin, preventing the formation of crown gall tumors.

>> ○ **Example 4.** pBHA

>> ○ **Example 5.** pBIC-A

>> ○ **Example 6.** Gene library vector: BAC Library, YAC Library

>> ○ **Example 7.** Bacterial plasmid: F plasmid, R plasmid (antibiotic resistance gene inserted). Approximately 1-10 kb.

>> ○ **Example 8.** Virus vector

>>> ○ Genetic information is inserted into bacterial DNA through bacterial infection, proliferation, lysis, release and cleavage.

>>> ○ λ phage (about 25 kb) is mainly used.

>>> ○ Sometimes retrovirus is used.

>> ○ **Example 9.** Cosmid: Introduces the phage cos end using the in vitro packaging method, approximately 50 kb in size.

>> ○ **Example 10.** pOXA-48 plasmid

>>> ○ Synthesizes carbapenemase to give resistance to carbapenem, a beta-lactam antibiotic

>>> ○ Like this, Beta-lactamidase gene creates beta-lactam ring to confer penicillin resistance.

⑷ 4<sup>th</sup>. Screening: Screening for cells or populations with recombinant genes

> ① Antibiotic resistance gene (selection based on proper insertion of recombinant plasmid)

>> ○ Example: Ampicillin resistance gene, tetracycline resistance gene

>> ○ 1<sup>st</sup>. Replica formation: After incubating the entire colony, imprint the entire culture plate onto filter paper.

>> ○ 2<sup>nd</sup>. Add antibiotics to a portion of the colonies imprinted on the filter paper.

>> ○ 3<sup>rd</sup>. Colonies that die due to the antibiotic indicate that they were not transformed.

>> ○ 4<sup>th</sup>. Select colonies corresponding to those that survived the antibiotic on the filter paper.

> ② Antibiotic resistance genes (selection according to recombination success)

>> ○ 1<sup>st</sup>. Replica formation: After incubating the entire colony, imprint the entire culture plate onto filter paper.

>> ○ 2<sup>nd</sup>. Add antibiotics to a portion of the colonies imprinted on the filter paper.

>> ○ 3<sup>rd</sup>. Colonies that died due to the antibiotic indicate that the restriction enzyme cut the antibiotic resistance gene.

>> ○ 4<sup>th</sup>. Select colonies corresponding to those that died due to the antibiotic on the filter paper.

> ③ X-gal (selection according to recombination success)  

>> ○ Cloning gene is inserted into lac Z gene.

>> ○ 1<sup>st</sup>. X-gal is originally white, but becomes blue due to lac Z.

>> ○ 2<sup>nd</sup>. If a colony treated with lacZ-bearing plasmid appears white on X-gal, it indicates successful transformation.

> ④ Reporter Gene  

>> ○ Use genes that are not expressed in the host cell.  

>> ○ Examples: eGFP, tdTomato.

> ⑤ Southern blotting

⑸ 5<sup>th</sup>. Cloning amplification: Large quantities of product are obtained by mass fermentation and purification.

> ① Reason for using antibiotics in large-scale culture with antibiotic-resistant plasmids (non-selective purpose)  

>> ○ 1<sup>st</sup>. Plasmids within the bacterial strain are randomly distributed to daughter cells during cell division.  

>> ○ 2<sup>nd</sup>. As division continues, some cells end up without any plasmids.  

>> ○ 3<sup>rd</sup>. Cells containing plasmids consume more energy to replicate them, leading to a difference in growth rate compared to plasmid-free cells.  

>> ○ 4<sup>th</sup>. Ultimately, plasmid-free cells become the majority."

> ② Inclusion Body  

>> ○ Since the gene introduced during transformation is not originally present in the host, the produced protein may not fold properly and aggregates.

⑹ Examples

> ① Hormones and bioactive substances: Growth hormone, insulin, secretin, etc.

> ② Therapeutic and diagnostic reagents: Hepatitis B vaccine, diagnosis of HIV infection, interferon, neoendorphin, analgesic, blood coagulation factor, etc.

> ③ Agriculture: Golden rice, herbicide-tolerant crops, pest-tolerant crops

> ④ Ti plasmid transgenic neofunctional crop: Herbicide tolerance, anti-anthrax crops (seed potatoes, etc.)

> ⑤ Recombinant strain development strategy

>> ○ Feedback / antibiotics resistant mutant

>> ○ Auxotrophic mutant

>> ○ Overexpression mutant  

<br>

<br>

## 2. DNA library

⑴ Vector for DNA library

> ① Bacterial Artificial Chromosome (BAC)

>> ○ Advantage: No cross occurs.

>> ○ Disadvantage: Must be adapted to the prokaryotic expression system.

>> ○ Used in [human genome projects](https://jb243.github.io/pages/75#1-genome-project).

>> ○ It is still used today.

> ② Yeast Artificial Chromosome (YAC; pYAC3)

>> ○ Artificial chromosome into yeast (~1 Mb), contains ARS, and inserts DNA between the centromere and telomere.

>> ○ Advantage: Suitable for producing human proteins as it utilizes a eukaryotic system.

>> ○ Disadvantage: Crossover occurs during prophase I of meiosis.

> ③ Virus genome library

>> ○ Disadvantage: Small capacity. Difficult to control breeding.

⑵ Required DNA elements

> ① Centromere: A GC-rich sequence that attaches to the spindle fibers during cell division, ensuring proper distribution of chromosomes to each daughter cell.

> ② Telomere: A nucleotide sequence located at the end of a chromosome that prevents chromosome shortening due to cell division.

> ③ Origin of replication: The site where DNA replication begins.

> ④ Selection marker: A specific DNA sequence used to confirm the proper insertion of artificial chromosomes into cells.

> ⑤ Others: Promoter, gene expression regulatory mechanisms.

⑶ Types

> ① gDNA Library

>> ○ 1<sup>st</sup>. Precipitate histones by treating with phenol.

>> ○ 2<sup>nd</sup>. Nucleic acids are present in the genomic DNA supernatant.

>> ○ 3<sup>rd</sup>. Separate the supernatant → Trichloroacetic acid → Nucleic acid precipitate.

>> ○ 4<sup>th</sup>. After digesting gDNA with a restriction enzyme, clone all fragments and insert them into a BAC vector.

>> ○ 5<sup>th</sup>. Selectively amplify Escherichia coli containing the target DNA.

>> ○ 6<sup>th</sup>. Used in intron research.

>> ○ Triton X-100 is amphipathic and does not denature the DNA structure.

>> ○ SDS-PAGE denatures the DNA structure.

> ② cDNA Library

>> ○ 1<sup>st</sup>. [Affinity chromatography](https://jb243.github.io/pages/1339#6-chromatography-solution): Attach oligo-dT (poly T) to beads.

>> ○ 2<sup>nd</sup>. Reverse transcribe the separated mRNA using reverse transcriptase.

>> ○ 3<sup>rd</sup>. Treat with a low concentration of RNase: Functions as a primer.

>> ○ 4<sup>th</sup>. Treat with DNA polymerase I.

>> ○ 5<sup>th</sup>. Treat with ligase.

>> ○ 6<sup>th</sup>. Treat with restriction enzyme.

>> ○ 7<sup>th</sup>. Insert into a vector.

>> ○ 8<sup>th</sup>. Cloning amplification.

>> ○ 9<sup>th</sup>. Used in the study of gene expression.

> ③ Library Screening

>> ○ Transfer library clones onto a filter and activate with a probe to identify the locations of clones containing the same sequence as the probe.

<br>

<br>

## **3\. PCR** 

⑴ Overview

> ① Definition: A DNA amplification technique based on the principles of DNA replication.  

> ② Developed by Kerry Mullis in 1983; he was awarded the Nobel Prize in Chemistry in 1993.

⑵ Sample: DNA sample, DNA primer (2 types), dNTP, Taq polymerase, Buffer (pH stabilization), MgCl<sub>2</sub> (DNA stabilization)

> ① DNA Primer  

>> ○ **Condition 1:** GC content ＞ 50%  

>>> ○ Example: 10 out of 18 bases (= 18/2 + 1) are G or C  

>>> ○ Example: 11 out of 20 bases (= 20/2 + 1) are G or C  

>> ○ **Condition 2:** The 5' end should end with AT, and the 3' end should end with GC  

>>> ○ Reason: Prevents hairpin formation due to self-ligation  
  
>>> ○ Example: 5'-ATGCCTATGCG-3'  
  
>>> ○ Example: 5'-ATGCAGGCTAAT-3' can cause self-ligation  

>> ○ **Condition 3:** The 3' end should have 2 to 3 overlapping Gs or Cs  

>>> ○ Reason: A=T bonds are double bonds, whereas G≡C bonds are triple bonds, making GC-rich sequences more stable  

>>> ○ Example: 5'-ATGCCTATGCG-3'

> ② dNTP  

>> ○ Note that NTPs are the building blocks for RNA polymerization.  

>> ○ Reason why an *E. coli* colony can be used instead of a DNA template in colony PCR: Due to heat-induced denaturation.  

>> ○ In other words, the *E. coli* colony is broken down under denaturation, providing the necessary materials for DNA polymerization.

> ③ Taq polymerase

>> ○ Taq polymerase is derived from the archaea *Thermus aquaticus*, which inhabits high-temperature hot springs.  

>>> ○ It was first discovered in the 1960s in Yellowstone National Park.  

>> ○ Taq polymerase has an optimal polymerization temperature of 55°C and denatures at 92–95°C.  

>> ○ After completing polymerization to the 3' end, Taq polymerase adds an extra A nucleotide.  

>>> ○ Since it does not require a template, it can be applied to recombinant DNA with blunt-ended sequences.  

>> ○ Taq polymerase lacks 3' → 5' proofreading ability, resulting in an error rate of 1/10<sup>4</sup>.  

>>> ○ In contrast, typical DNA polymerases have a replication error rate of 1/10<sup>7</sup>.

> ④ MgCl<sub>2</sub>

>> ○ DNA has a negatively charged phosphate backbone, leading to structural instability → Na<sup>+</sup> or Mg<sup>2+</sup> can stabilize DNA.  

> ⑤ IPP (Inorganic Pyrophosphatase)  

>> ○ Enhances the binding of NTPs.

⑶ Process

> ① 1<sup>st</sup>. The heating step to separate the two DNA strands. 94 ℃. 30-60 seconds.

> ② 2<sup>nd</sup>. Annealing: As the mixture cools, the primers bind to the DNA template. 50-55 ℃. 1 minute.

> ③ 3<sup>rd</sup>. DNA elongation: Polymerase starts synthesis at primer. 72 ℃. 1 minute per 1000 bp.

> ④ 4<sup>th</sup>. Repeating steps ① to ③ theoretically results in the exponential amplification of the DNA sample by a factor of 2.

> ⑤ 5<sup>th</sup>. DNA ligase action (40 ℃) and stabilization (4 ℃).

> ⑥ 6<sup>th</sup>. Switch to E. coli cloning amplification after a few cycles: In reality, the amount of artificial nucleotides is insufficient, producing much less DNA.

⑷ T<sub>m</sub> vs annealing temperature

> ① DNA Melting Curve: A graphical representation of the degree of DNA denaturation as a function of temperature.  

>> ○ It shows the denaturation level of linear double-stranded DNA at different temperatures using absorbance at 260 nm (A260).  

>> ○ A260 value comparison: Double-stranded DNA ＜ Single-stranded DNA ＜ Fragmented nucleic acids  

>> ○ Reason: Less base stacking leads to lower shielding and higher absorbance.  

> ② T<sub>m</sub> (Melting Temperature): The temperature at which 50% of the DNA remains double-stranded, while the other 50% has denatured into single strands.

>> ○ Generally proportional to (A + T) × 2 + (G + C) × 4. 

>> ○ T<sub>m</sub> value is independent of repetitive sequences.  

>> ○ The T<sub>m</sub> of dsRNA is about 15 ℃ higher than that of dsDNA with the same sequence due to the -OH group in RNA stabilizing dsRNA.  

> ③ Annealing  

>> ○ The annealing temperature is typically set 5–10°C below T<sub>m</sub>.  

>> ○ If the annealing temperature is too high, hybridization is inefficient; if too low, mis-hybridization increases.  

>> ○ The presence of metal ions affects the optimal annealing temperature.

⑸ C<sub>t</sub> (Threshold Cycle)  

> ① Definition: The cycle at which the signal exceeds the threshold value.  

> ② C<sub>t</sub> and copy number have an inverse correlation.

⑹ Limitation: The nucleotide sequences at both ends of the template DNA must be known. 

> ① Assumption: If there is a known sequence in the middle and the template DNA has blunt ends, this limitation can be overcome.

> ② **Solution 1.** Inverse-PCR

>> ○ 1<sup>st</sup>. Form circular DNA from the template DNA using DNA ligase.

>> ○ 2<sup>nd</sup>. Generate linear DNA by cutting within the known sequence using a restriction enzyme.

>> ○ 3<sup>rd</sup>. PCR is possible because we know sequence of both ends.

> ③ **Solution 2.** Anchored-PCR

>> ○ 1<sup>st</sup>. Attach a new primer to the template DNA using DNA ligase and amplify one half.

>> ○ 2<sup>nd</sup>. The other half goes through the same process.

> ④ **Solution 3.** Oligonucleotide-directed mutagenesis

>> ○ **Case 1:** Replicating the template DNA using a primer with only a few mismatched bases.  

>> ○ **Case 2:** Replicating the template DNA using a primer with an added nucleotide sequence that induces a hairpin structure in the middle.  

>> ○ **Case 3:** Replicating the template DNA using a primer with a removed nucleotide sequence to allow the template DNA to form a hairpin structure.

⑺ Types 

> ① RT-PCR: Includes the process of synthesizing cDNA from RNA using reverse transcriptase.  

> ② Real-time PCR (qPCR, quantitative PCR): Allows real-time monitoring of amplified DNA quantity.  

>> ○ Initially, the given nucleotide fragment does not fluoresce because the distance between the fluorescent reporter and quencher is too short.  

>> ○ During polymerization in PCR, the fluorescent reporter and quencher separate, allowing fluorescence to appear.  

>> ○ TaqMan qPCR: A TaqMan probe (containing a fluorophore and quencher) binds to the target DNA and is cleaved during DNA synthesis, generating fluorescence.  

> ③ Multiplex PCR: Amplifies multiple types of DNA samples within a single PCR reaction.  

> ④ Allele-Specific PCR (AS-PCR): A method for detecting allele-specific variations using PCR.

<br>

<br>

## **4\. DNA fingerprinting**

⑴ Overview

> ① Objective: Paternity confirmation, prediction of genetic diseases, forensic perpetrator identification.

> ② Using the fact that no two individuals, except for identical twins, are genetically identical.

⑵ Electrophoresis: Allows for the separation of substances by size and the estimation of their quantity.

> ① Stationary phase, mobile phase

>> ○ Stationary phase (compression gel, stacking gel) 

>>> ○ Higher gel density results in slower transport of DNA samples or proteins.

>>> ○ Reason for high gel density: To keep the moving speed constant regardless of DNA sample or protein size.

>> ○ Mobile phase (running gel)

>>> ○ Low gel density allows rapid movement of DNA samples or proteins.

>>> ○ Gel density is 6-15%.

> ② Nucleic acid electrophoresis: Agarose gel

> ③ Protein electrophoresis: Polyacrylamide gel + Bisacrylamide gel, SDS-PAGE, TEMED

>> ○ **Material 1.** Acrylamide: Forms polymers in the gel.  

>> ○ **Material 2.** RIPA Buffer: Isotonic solution. Used as a buffer for cell lysis.  

>> ○ **Material 3.** Loading Buffer or Laemmli Sample Buffer: A buffer that breaks disulfide bonds.  

>>> ○ Does not contain methanol but includes SDS-PAGE.

>>> ○ SDS-PAGE: Used for more efficient separation. Applied in denaturing gels.  

>>> ○ SDS-PAGE consists of SDS (an anionic surfactant) and β-ME (β-mercaptoethanol).

>>> ○ SDS: Removes ionic bonds and hydrophobic interactions. Equalizes the charge density per unit mass of proteins.  

>>> ○ β-ME: Eliminates disulfide bonds.  

>>> ○ When SDS-PAGE binds to each amino acid, the repulsion between negative charges results in all proteins having the same charge density, forming a secondary structure.

>>> ○ DTT: An additive that breaks disulfide bonds between cysteine residues.

>> ○ **Material 4.** Transfer Buffer: Used for transferring from gel to membrane.  

>>> ○ Contains methanol but does not include SDS-PAGE.

>>> ○ Native-PAGE: Used in non-denaturing gels.  

>>> ○ Methanol: Functions to cool the heat generated during the transfer process.  

>> ○ **Material 5.** Ammonium Persulfate (APS): Acts as an enzyme that forms cross-links between polyacrylamide and bis-acrylamide gels.  

>> ○ **Material 6.** TEMED (N,N,N',N'-tetramethylethylenediamine): Stabilizes APS and helps the gel solidify quickly.

> ④ Typically the top is the anode, the bottom is the cathode.

> ⑤ Moving distance = Δt × (a × log M + b), a < 0, M: molecular weight

>> ○ When treated with special substances such as SDS, molecular weight is linearly proportional to size, resulting in greater resistance in the mobile phase and a shorter migration distance.

>> ○ Molecular weight has the greatest influence on migration distance.

>> ○ Migration distance is proportional to time (Δt).

>> ○ DNA ladder (size marker): Provides migration distances based on size, allowing estimation of the size of unknown substances.

>> ○ Supercoiled DNA experiences less resistance in electrophoresis.

> ⑥ **Type 1.** General electrophoresis

>> ○ Nucleic acids and proteins with smaller molecular weights migrate further.

>> ○ **In protein electrophoresis, SDS-PAGE causes proteins to migrate from the (-) electrode to the (+) electrode.

> ⑦ **Type 2.** Isoelectric electrophoresis: Electrophoresis using isoelectric points, which are unique values of proteins.

>> ○ Electrophoresis with a pH gradient moves proteins to the point where the total charge is zero (isoelectric point).

>> ○ After electrophoresis, proteins are stained using protein dyes such as Coomassie Blue or silver nitrate.

<br>

![image](https://github.com/user-attachments/assets/03c4d15f-e322-45c0-8102-301c8d22117e)

**Figure 2.** Isoelectric electrophoresis

<br>

> ⑧ **Type 3.** 2D electrophoresis

>> ○ O'Farrel law is common.

>> ○ X-axis: Separation based on charge through isoelectric electrophoresis in the presence of urea or electrophoresis under non-denaturing conditions.  

>>> ○ In other words, separation occurs according to the isoelectric point.

>>> ○ Urea: Removes all R-R interactions except disulfide bonds.

>> ○ Y-axis: Separation based on molecular weight through electrophoresis in the presence of SDS.  

>> ○ More than 1,000 types of proteins can be separated.

>> ○ **Application 1.** Separate both the X-axis and Y-axis using the same method, but add a specific substance during Y-axis separation → Substances deviating from the diagonal indicate interaction with the added substance.  

>> ○ **Application 2.** For tRNA and small RNA molecules: Use 10% polyacrylamide gel for 1D and 20% for 2D → Effectively separates various types of RNA.

⑶ **Fingerprinting Method 1.** RFLP (restriction fragment length polymorphism): Paternity testing, genetic disease analysis

> ① Process: Target genes.

>> ○ 1<sup>st</sup>. DNA isolation from tissue.

>> ○ 2<sup>nd</sup>. PCR: Amplify the amount of DNA.

>> ○ 3<sup>rd</sup>. Cutting DNA into fragments using restriction enzymes.

>> ○ 4<sup>th</sup>. Generate DNA fragments of different sizes with the same restriction enzyme recognition sites.

>> ○ 5<sup>th</sup>. Electrophoresis: Separates fragments based on size differences for visualization.

> ② Cause: Single nucleotide polymorphism (SNP)

>> ○ Defition: Difference between people due to one base being different by point mutation.

>> ○ In humans, approximately one SNP appears per 1,000 base pairs (bp).

>> ○ SNPs can occur in both exons and introns.

>> ○ If an SNP occurs in a restriction enzyme recognition site, RFLP may occur.

>> ○ Difference between mutation and polymorphism: If it occurs in less than 1% of the entire population, it is considered a mutation; otherwise, it is classified as polymorphism.

>> ○ Number of SNPs: As of 2017, there are approximately 10,000 SNPs.

<br>

![image](https://github.com/user-attachments/assets/a1c28633-e487-4453-9e16-a35a229392a6)

**Figure 3.** Number of SNPs

<br>

> ③ Limitation: Since SNPs do not necessarily occur at restriction enzyme recognition sites, RFLP variations among individuals are not significant.  

>> ○ Difficult to use for criminal identification and similar applications.

⑷ **Fingerprinting Method 2.** Tandem repeat: Used for suspect identification. A more powerful genetic fingerprinting method than RFLP, targeting repeat sequences.

> ① VNTR (variable number tandem repeat): 15-100 repetitions

>> ○ Principle: The number of repeats in repeat sequences varies among individuals, resulting in fragments of different sizes.

>> ○ Cause: Uneven crossover of homologous chromosome

>> ○ 1<sup>st</sup>. DNA fragments are separated and amplified based on length using electrophoresis.

>> ○ 2<sup>nd</sup>. Separated DNA fragments are transferred to filter paper.

>> ○ 3<sup>rd</sup>. After transferring to filter paper, a chemical treatment (alkaline) is applied to break hydrogen bonds → Results in single-stranded DNA.

>> ○ 4<sup>th</sup>. Compare the location of the VNTR fragments with a VNTR probe labeled with radioactive material.

>> ○ 5<sup>th</sup>. Band appears when exposing filter paper to X-ray film.

> ② STR (Short Tandem Repeat): Repeats of 2-4 times.  

>> ○ Not significantly different from VNTR.

⑸ **Fingerprinting Method 3.** Other Fingerprinting Methods

> ① CNV (Copy Number Variation)

> ② LOH (Loss of Heterozygosity)

> ③ Genomic Rearrangement

> ④ Rare Variant

<br>

<br>

## **5\. Hybridization**

⑴ Southern blotting: Nucleic acid probe hybridization to DNA.

> ① Overview

>> ○ A kind of RFLP.

>> ○ Developed by Edward M. Southern in 1975.

> ② 1<sup>st</sup>. Restriction enzyme treatment → PCR amplification → Electrophoresis

>> ○ Electrophoresis: Agarose gel is used.

>> ○ A master plate can be used instead of electrophoresis: Colonies with recombinant plasmids are arranged on a solid medium.

> ③ 2<sup>nd</sup>. The gel is shaken in a 0.5 M HCl solution.  

>> ○ HCl treatment accelerates the transfer of DNA from the agarose gel to the membrane.

> ④ 3<sup>rd</sup>. The gel is shaken in a **1.5 M NaCl, 0.5 M NaOH solution** (DNA denaturation).  

>> ○ The NaOH solution, sponge, and gel are stacked in order to separate DNA into single strands.

>> ○ The sponge absorbs the NaOH solution and supplies it to the gel.

>> ○ To prevent partial reformation of double-stranded DNA, an additional nucleic acid denaturant is added.

>> ○ Nucleic acid denaturants: Metal ions, formamide, formaldehyde, etc. The form series weakens DNA binding strength.

> ⑤ 4<sup>th</sup>. The gel is shaken in a 1.5 M NaCl, 1 mM EDTA, 0.5 M Tris-HCl (pH 7.2) solution.

>> ○ Tris-HCl: Buffer solution.

> ⑥ 5<sup>th</sup>. Capillary phenomenon + blotting

>> ○ 5<sup>th</sup>-1<sup>st</sup>. Place a positively charged nylon filter and a paper towel on the gel.

>>> ○ Nitrocellulose paper can be used instead of nylon filter.

>> ○ 5<sup>th</sup>-2<sup>nd</sup>. The NaOH solution that moves into the gel carries single-stranded DNA and travels to the paper towel stack, where it is absorbed.

>> ○ 5<sup>th</sup>-3<sup>rd</sup>. Negatively charged single-stranded DNA sticks to nylon membrane.

> ⑦ 6<sup>th</sup>. Autoclave: Nylon membrane combined with DNA is treated under high pressure and high temperature to fix the bond.

>> ○ DNA can also bind to the membrane when exposed to ultraviolet (UV) light.

>> ○ In general, sterilization at 121°C for 15 minutes kills most microorganisms.

> ⑧ 7<sup>th</sup>. Blocking: A blocking agent is added to enhance clarity.

>> ○ 7<sup>th</sup>-1<sup>st</sup>. Identify DNA that is considered insignificant through repeated experiments or electrophoresis for each allele.

>> ○ 7<sup>th</sup>-2<sup>nd</sup>. Add salmon sperm DNA fragments to the DNA.

>> ○ Purpose: To prevent probe DNA from non-specifically binding to the membrane during the 8th hybridization process (enhancing clarity).

>> ○ Note: The membrane strongly interacts with DNA or proteins, requiring a blocking process.

> ⑨ 8<sup>th</sup>. Hybridization

>> ○ 8<sup>th</sup>-1<sup>st</sup>. The nylon membrane with bound DNA is placed in a Seal-a-Meal bag.

>> ○ 8<sup>th</sup>-2<sup>nd</sup>. At 62 °C, a **radioactively labeled DNA probe** is added to the **Seal-a-Meal bag**.

>>> ○ The labeled probe has a complementary sequence to the gene of interest.

>>> ○ Probe DNA: 100–500 bp, single-stranded DNA.  

>>> ○ Probe DNA is typically labeled with the radioactive isotope P<sup>32</sup>.

>>> ○ Probe DNA can also be labeled with dyes such as EtBr or SYBR Green.

>> ○ 8<sup>th</sup>-3<sup>rd</sup>. Probes bind to complementary DNA.

>>> ○ Through repeated experiments, other DNA considered insignificant is blocked and covered with salmon sperm DNA.

> ⑩ 9<sup>th</sup>. Eliminate unhybridized radioactive probes in seal-a-meal bags.

> ⑪ 10<sup>th</sup>. Autoradiogram: Irradiation on nylon membrane.

> ⑫ 11<sup>th</sup>. Identify the radioactive positions on the electrophoresis gel (or colony positions).

> ⑬ Housekeeping Gene  

>> ○ A gene that is present in all cells and continuously expressed.  

>> ○ Used as a control in Southern blotting.

>> ○ Examples: β-actin (ACTB), tubulin, GAPDH, B2M, RPL11, 18S rRNA (most reliable).  

> ⑭ Stringency

>> ○ Conditions that lower stringency: Create an environment where probe binding is easier.  

>> ○ Low temperature or high salt concentration facilitates nucleic acid hybridization, leading to lower stringency.

⑵ Northern blotting: Nucleic acid probe hybridization for RNA.

> ① Very similar to Southern blotting but with the following differences.

> ② **Difference 1.** Extraction of samples

>> ○ Southern blotting uses DNA extraction.

>> ○ Northern blotting effectively isolates mRNA from extracts by [affinity chromatography](https://jb243.github.io/pages/1339#6-chromatography-solution) using oligo dT.

> ③ **Difference 2.** Use of restriction enzyme

>> ○ DNA dealt by Southern blotting is very long and must be used with restriction enzymes.

>> ○ Northern blotting uses RNA, not DNA, so restriction enzymes cannot be used.

> ④ **Difference 3.** Type of solution in capillary action

>> ○ Southern blotting uses basic solutions to denature single-stranded DNA.

>> ○ In Northern blotting, RNA is degraded when exposed to an alkaline solution.

>> ○ Northern blotting uses a salt solution to stabilize RNA.

> ④ **Difference 4.** Types of probe

>> ○ Southern blotting uses gDNA probe.

>> ○ Northern blotting uses cDNA probe.

⑶ Western blotting: Antibody probe hybridization for proteins.

> ① Numerous differences with nucleic acid blotting.

> ② **Difference 1.** Type of gel

>> ○ Nucleic acid blotting uses agarose gel.

>> ○ Western blotting uses polyacrylamide gel, SDS-PAGE, etc.

> ③ **Difference 2.** Electrophoresis voltage condition

>> ○ In nucleic acid blotting, nucleic acid transfers well to electrophoresis, so a low voltage must be applied.

>> ○ In Western blotting, proteins are not well transported by electrophoresis, so high voltage must be applied.

> ④ **Difference 3.** Blotting Process

>> ○ Nucleic acid blotting uses capillary phenomenon during blotting.

>> ○ Western blotting uses an electric field during blotting.

> ⑤ **Difference 4.** Nucleic acid pretreatment

>> ○ Nucleic acid blotting uses salmon sperm DNA fragments.

>> ○ Western blotting uses casein, skim milk, bovine serum albumin (BSA), etc.

>> ○ Tween: Surfactants such as Tween-20 are used.  

>>> ○ A brand name developed by the American company Uniqema.

>>> ○ A detergent composed of a hydrophilic group (polyethylene glycol) and a hydrophobic hydrocarbon group.  

>>> ○ Uses: Helps mix water and oil, disrupts cell membranes.  

>>> ○ The numbers 20, 40, 60, 80 indicate increasing carbon chain length, making the detergent more hydrophobic as the value increases.

>> ○ BSA can also be used for nucleic acid blotting.

> ⑥ **Difference 5.** Types of probe

>> ○ Nucleic acid blotting uses DNA probes.

>> ○ Western blotting uses antibody probe.

>>> ○ Primary antibody: An antibody that specifically binds to the target protein.

>>> ○ To ensure the specificity of the primary antibody, an enzyme extracted from a different species than the target animal must be used.

>>> ○ Secondary antibody: An antibody that specifically binds to the primary antibody. It is typically conjugated with an enzyme that breaks down a chromogenic substrate.

>>> ○ To ensure the specificity of the secondary antibody, an enzyme extracted from a species different from both the target animal and the primary antibody host must be used.

>>> ○ In Western blotting, the primary and secondary antibodies originate from different species.

> ⑦ **Difference 6.** Imaging Method

>> ○ Nucleic acid blotting uses EtBr + self-radiation.

>> ○ Western blotting uses the Coomassie Blue staining method.

⑷ DNA chip (microarray)

> ① Characteristics

>> ○ Simultaneously detects the expression patterns of multiple genes.

>> ○ Each spot contains a single cDNA probe.

>> ○ Allows identification of tissue-specific gene expression using cDNA DNA chips.

>> ○ Microarray data consists of continuous raw data, whereas RNA-seq raw data is count-based.

>> ○ RNA-seq is more robust in detecting genes with extremely low or high expression levels compared to microarray data.

>> ○ Becoming an obsolete technology due to advancements in NGS (Next-Generation Sequencing).

> ② Process

>> ○ 1<sup>st</sup>. A cDNA library of human genes is spotted onto a glass slide to create a cDNA chip.

>> ○ 2<sup>nd</sup>. The cDNA chip is treated with a 1% BSA solution.

>>> ○ BSA prevents nonspecific binding between the chip and DNA, allowing only complementary DNA to bind.

>> ○ 3<sup>rd</sup>. mRNA X (from normal tissue) and mRNA Y (from abnormal tissue, e.g., cancer) are extracted separately.  

>> ○ 4<sup>th</sup. Oligo-dT is added to X and Y, binding complementarily to poly-A sequences in mRNA.  

>> ○ 5<sup>th</sup>. RT-PCR (Reverse Transcription PCR)

>>> ○ X: dNTP + Cy3 (green fluorescent dye)-dTTP.  

>>> ○ Y: dNTP + Cy5 (red fluorescent dye)-dTTP.  
   
>>> ○ Reverse transcription reaction is performed to generate fluorescently labeled cDNA.

>> ○ 6<sup>th</sup>. 0.1 N NaOH is added to X and Y, incubated at 70°C for 10 minutes, then neutralized with 0.1 N HCl.
   
>>> ○ Template RNA is degraded due to basic condition.

>> ○ 7<sup>th</sup>. Synthesized cDNA from X and Y is purified and mixed in equal amounts, then hybridized onto the cDNA chip.

>> ○ 8<sup>th</sup>. The cDNA chip is washed with **buffer solution** to remove unbound sequences.  

>> ○ 9<sup>th</sup>. Fluorescence intensity of Cy3 and Cy5 is measured and corrected.

>>> ○ Black: Neither X nor Y is expressed.  

>>> ○ Green: Only X is expressed.  

>>> ○ Red: Only Y is expressed.  

>>> ○ Yellow: Both X and Y are expressed.

⑸ ISH (in situ hybridization)

> ① Overview

>> ○ An experimental technique that hybridizes a specific DNA sequence with a chromosome.

>> ○ Used not only to determine the location of specific nucleotide sequences but also to identify the location of RNA.

>> ○ Rapidly diagnoses chromosomal abnormalities but cannot diagnose DNA mutations.

> ② **Type 1.** Fluorescence in situ hybridization (FISH)

>> ○ Hybridize a specific nucleotide sequence with a fluorescent probe complementary to the nucleotide sequence, then observe using a fluorescence microscope.

>> ○ 1<sup>st</sup>. Prepare epithelial tissue sections on slides.

>> ○ 2<sup>nd</sup>. Treat with RNase, incubate at 37°C for 1 hour, then wash.

>>> ○ Probes that bind to DNA may bind to RNA, so there is a process to remove RNA.

>> ○ 3<sup>rd</sup>. Remove the culture medium, add a 2% formaldehyde solution, and incubate for 15 minutes.  

>>> ○ This step fixes the cells. 

>> ○ 4<sup>th</sup>. Remove the formaldehyde solution, add a 0.2% Triton X-100 solution, and incubate for 5 minutes.  

>>> ○ Triton X-100 is a surfactant.

>> ○ 5<sup>th</sup>. Remove the Triton X-100 solution, add 2 M HCl, then treat with pepsin and incubate at 37°C for 10 minutes.

>>> ○ Pepsin functions in acidic conditions, so the pH is adjusted accordingly.

>>> ○ Pepsin degrades intracellular proteins, facilitating probe penetration in subsequent steps.

>> ○ 6<sup>th</sup>. Wash the sample.

>> ○ 7<sup>th</sup>. Prepare a probe that binds to the centromere.  

>>> ○ This probe contains biotin-labeled dTTP.

>> ○ 8<sup>th</sup>. Hybridize the sample from **step 4** with the probe from **step 5**.  

>> ○ 9<sup>th</sup>. Wash the sample.

>> ○ 10<sup>th</sup>. After sufficiently treating with fluorescently labeled avidin, wash the sample.

>> ○ 11<sup>th</sup>. Stain with DAPI.

>>> ○ DAPI (4'-6-diamidino-2-phenylindole) is a nuclear stain.

>>> ○ DAPI strongly binds to AT-rich regions of DNA minor groove.  

>>> ○ It is also used as an apoptosis marker.  

>>> ○ Since DAPI is cell-membrane-permeable, it can be used to stain both live and fixed cells.  

>>> ○ When UV light is applied to DNA bound with DAPI, it emits blue fluorescence.

>> ○ 12<sup>th</sup>. The centromere region will display avidin fluorescence, while DNA will appear as blue fluorescence.  

>> ○ **Application 1.** RNA ISH (In Situ Hybridization)

>>> ○ rRNA molecules are observed in situ rather than being isolated from individual cells.

>>> ○ A labeled riboprobe (complementary RNA sequence) hybridizes with the transcript.

>>> ○ Visualization is achieved through a probe that binds to the complementary transcript.

>> ○ **Application 2.** [WM ISH](https://pubmed.ncbi.nlm.nih.gov/2476281/) (Whole-Mount ISH) (1989)

>>> ○ Requires thousands of animals to create an atlas, making it less commonly used after 2010.

> ③ **Type 2.** [Radioactive ISH](https://pubmed.ncbi.nlm.nih.gov/5261036/): The first ISH method, developed in 1969.

⑹ Sequencing technology: Determines DNA nucleotide sequences.  

⑺ Gene-trap screen

<br>

<br>

## **6\. Gene deletion**

⑴ Knock-out mouse: Study the function of specific genes.

<br>

![image](https://github.com/user-attachments/assets/fe778a2e-9394-413a-9822-fd20ed66a8ff)

**Figure 4.** Knockout process

<br>

> ① 1<sup>st</sup>. Knockout induction of gene X in embryonic stem (ES) cells

>> ○ 1<sup>st</sup>-1<sup>st</sup>. Construction of a plasmid vector (targeting vector): Includes gene X which is inactivated by the insertion of neo<sup>r</sup> (neomycin-resistant gene), along with a TK (thymidine kinase) gene located at a distance from the gene X.

<br>

![image](https://github.com/user-attachments/assets/10ee9371-2062-4fb8-becb-cd6950dcd524)

**Figure 5.** The structure of plasmid vector 

<br>

>> ○ 1<sup>st</sup>-2<sup>nd</sup>. Insert targeting vector inside ES cell.

>> ○ 1<sup>st</sup>-3<sup>rd</sup>. Some ES cells autonomously replace the existing gene X with the inactivated gene X.

>>> ○ To enable genetic recombination, the same gene must be included next to gene X.

>> ○ 1<sup>st</sup>-4<sup>th</sup>. Selection: Cells are cultured in medium containing G418 (geneticin, neomycin derivative) and gancyclovir.

>>> ○ 1<sup>st</sup>-4<sup>th</sup>-1<sup>st</sup>. Untransformed ES Cells: Killed by G418.

>>> ○ 1<sup>st</sup>-4<sup>th</sup>-2<sup>nd</sup>. ES cells in which regions other than gene X have been replaced: These cells contain the TK gene, making them susceptible to ganciclovir-induced cell death (TK gene metabolizes ganciclovir into a toxic compound).

>>> ○ 1<sup>st</sup>-4<sup>th</sup>-3<sup>rd</sup>. Only ES cells in which only gene X has been replaced survive.

> ② 2<sup>nd</sup>. Selection of knockout ES cells.

> ③ 3<sup>rd</sup>. Inject knockout ES cells into the embryo.

> ④ 4<sup>th</sup>. The germ cells of the **chimera mouse** consist of both **normal germ cells** and **germ cells derived from knockout ES cells**.

> ⑤ 5<sup>th</sup>. F1 mice include both **(+/+) individuals** and **(+/-) individuals**.

>> ○ Reason: Because the chimera mouse can be considered a heterozygous mutant.

> ⑥ 6<sup>th</sup>. Select (+/-) individuals among the F1 mice through genetic testing and perform self-crossing.

>> ○ There are limitations in selection based on mouse fur color.

> ⑦ 7<sup>th</sup>. Among the **F2 mice, 25% are knockout mice** (**homozygous mutants**) and must be **identified through genetic testing**.

⑵ Cre-Lox

> ① The Cre gene encodes recombinase.

> ② Cre-mediated DNA recombination occurs only between identical lox sequences.

> ③ If the paired lox sequences are in the same orientation, DNA deletion occurs by Cre; if they are in opposite orientations, DNA inversion occurs by Cre.

⑶ miRNA, siRNA: Gene expression can be suppressed for all genes using RNA interference (RNAi).

<br>

<br>

## **7\. Nuclear substitution**

⑴ GMO (Genetically Modified Organism): Genetically altered substances, food, organisms.

⑵ Methods

> ① Nuclear Transplantation

>> ○ Utilizes techniques such as microinjection.

>> ○ Applicable to trait transformation and animal cloning.

> ② Trait transformation without using a vector

>> ○ Can induce trait transformation using gene guns, etc.

>> ○ **Example 1.** Formation of recombinant plant cells: Particles coated with recombinant genes are "shot" into plant cells.

>> ○ **Example 2.** Recombinant plant cells → Callus → Whole plant formation (in nutrient medium).

>>> ○ Plant cells exhibit totipotency.

>> ○ **Example 3.** Maintenance and propagation of purebred 

>> ○ **Example 4.** Proliferation of useful plants

>> ○ **Example 5.** Tissue culture with meristematic regions

>> ○ **Example 6.** Induction of differentiation

⑶ Examples

> ① **Example 1.** Recombinant animals: Mass production of beneficial gene products in animals.

> ② **Example 2.** Edible vaccines

> ③ **Example 3.** Genetically Modified Food (GM Food)

>> ○ Past: Gene-modified crops obtained by increasing the frequency of specific alleles through selective breeding (artificial selection).

>> ○ Present: Genetic recombination technology → Increased shelf life, production rate (resistance to pests, weeds, diseases, drought, and cold).

>> ○ **Example 3-1.** Golden rice: Genetically modified to produce beta-carotene (increases nutritional value of rice).

⑷ [GMO Debate and Principles of Safety Assessment](https://jb243.github.io/pages/1432)

<br>

<br>

## **8\. DNA-protein interaction studies**

⑴ Gene footprinting assay (DNA footprinting technology)

<br>

![image](https://github.com/user-attachments/assets/dd53e5d6-9450-4011-b5bb-2bdf739843e9)

**Figure 6.** Application of gene footprinting

<br>

> ① Genes binding with proteins appear to be missing on the gel electrophoresis.

> ② It allows the identification of transcription factor binding sites.

⑵ Electrophoretic mobility shift assay (EMSA; gel shift assay, GMSA) 

> ① Confirms the binding between the transcription factor and the promoter.

> ② 1<sup>st</sup>. Probe production: Labels DNA with radioactive isotopes.

> ③ 2<sup>nd</sup>. Electrophorese the protein-DNA hybrid, followed by radiation sensitivity analysis.

> ④ 3<sup>rd</sup>. Result analysis

<br>

![image](https://github.com/user-attachments/assets/d0d0ee8b-bd47-4e31-b95d-729cf146ee56)

**Figure 7.** Example of EMSA results

<br>

>> ○ Premise: Since the top is the cathode and the bottom is the anode, DNA will move from top to bottom. (-) refers to the case where only the probe is moved.

>> ○ Interpretation of a: Protein A and Protein B bind to the probe, and A and B bind together with the probe (they could also bind exclusively with each other...).

>> ○ Interpretation of b: Protein C and Protein D bind to the probe, and C and D do not bind to each other.

>> ○ Interpretation of c: Protein E binds to the probe, while protein F does not bind to the probe but binds to protein E.

> ⑤ Generally, protein means transcription factor and it can be applied to measure transcription activity.

⑶ [ChIP](https://jb243.github.io/pages/1431#2-type-2-identifying-transcription-regulation) (Chromatin Immunoprecipitation)

⑷ South-Western Blotting

<br>

![image](https://github.com/user-attachments/assets/ff2da2b2-d9be-436b-9de1-f9d921dea442)

**Figure 8.** South-Western Blotting

<br>

> ① 1<sup>st</sup> step: Proteins are initially subjected to Western blotting.

> ② 2<sup>nd</sup> step: Subsequently, DNA labeled with fluorescence is subjected to Southern blotting.

⑸ Yeast One-Hybrid Assay

⑹ Filter Binding Assay: Only transcription factors bind to the filter.

⑺ DNA Affinity Chromatography

<br>

<br>

## **9\. Protein-protein interaction studies**

⑴ Dual Hybrid System: A method for identifying proteins that interact with protein X.

> ① 1<sup>st</sup>. Transcription factor gene is divided into two parts: DNA binding domain and transcription activation domain.
   
> ② 2<sup>nd</sup>. Bait: Protein X + DNA binding domain.
   
> ③ 3<sup>rd</sup>. Prey: Protein of interest for determining binding + transcription activation domain.
   
> ④ 4<sup>th</sup>. When the two hybrid proteins interact and bind, the prey induces the expression of a reporter gene.

>> ○ Example of a reporter gene: GFP protein.

⑵ Yeast Two-Hybrid: Protein-protein interactions are used to identify protein binding relationships.

⑶ Protein separation and purification using GST-tagged fusion proteins.

⑷ Phage Display Method: Used to obtain monoclonal antibodies.

> ① Objective: To obtain a monoclonal antibody that binds to a specific protein.  

> ② Method for obtaining monoclonal antibodies in vitro.

> ③ Conventional monoclonal antibodies: B lymphocytes are fused with myeloma cells (hybridoma), becoming stem cells that proliferate and produce a large amount of antibodies.

> ④ 1<sup>st</sup> step. By introducing mutagens and altering the base sequence, a vast phage library is created.

> ⑤ 2<sup>nd</sup> step. Screen for phages with desired activity and later confirm the nucleotide sequence.

<br>

<br>

## **10\. Gene editing**

⑴ CRISPR-Cas9 Gene Editing Technology 

> ① Overview

>> ○ Gene Scissors: A mechanism of gene cleavage in bacteria (e.g., E. coli) to eliminate viral DNA.

>> ○ CRISPR (Clustered Regularly Interspaced Short Palindromic Repeats)

>>> ○ Derived from DNA fragments of bacteriophages that infected prokaryotic cells.

>>> ○ Found in 50% of bacterial genomes and 90% of archaeal genomes.

>>> ○ CRISPR arrays consist of repetitive DNA sequences and spacers located between each repetition.

>>> ○ Bacteria or archaea record spacers, which are fragments of external DNA such as viruses, and add repeat sequences.

>>> ○ Subsequently, the CRISPR-Cas9 system detects and cleaves viral DNA that matches the spacer.

>> ○ Cas9 (CRISPR-associated protein 9)

>>> ○ Acquired from organisms like Streptococcus pyogenes. It has two nuclease domains, HNH and RuVC9.

>>> ○ dCas9 (nuclease-deficient Cas9, dead Cas9): Unable to cut DNA but capable of binding to DNA through sgRNA.

> ② **Step 1.** CRISPR Processing: The following describes the CRISPR processing of the subtype I-C/Dvulg Cas5d system in Bacillus halodurans.

>> ○ 1<sup>st</sup>. Cas5d recognizes hairpin structures and 3' single-stranded sequences in the CRISPR repeat region, cleaving pre-crRNA into unit-length fragments.

>> ○ 2<sup>nd</sup>. Pre-crRNA processing: Further processing of pre-crRNA into smaller crRNAs.

>> ○ 3<sup>rd</sup>. Cas5d forms a complex with crRNA, Csd1, and Csd2 proteins.

>> ○ 4<sup>th</sup>. The crRNA portion in this complex detects and removes viral DNA.

<br>

![image](https://github.com/user-attachments/assets/8f4db44d-8dfc-4653-a26a-da08aed8d527)

**Figure 9.** CRISPR Processing

<br>

> ③ **Step 2.** Reaction to External Nucleic Acids

>> ○ 1<sup>st</sup>. Cas9 creates a bubble in the double-stranded DNA, and the complementary sgRNA (small guide RNA) binds to the target RNA next to the PAM site within that bubble.

>>> ○ The bubble is referred to as DNA double-strand breaks (DSBs).

>>> ○ sgRNA: Fragments of viral DNA that remain in the genomic DNA of bacteria (e.g., E. coli) surviving viral infection. Also known as Spacer or gRNA (guide RNA).

>>> ○ PAM (Protospacer Adjacent Motif) site: A 5'-NGG-3' nucleotide sequence that distinguishes self from non-self.

<br>

![image](https://github.com/user-attachments/assets/55ada28d-d1b5-427d-bf83-7010adf983b6)

**Figure 10.** Location of sgRNA in dsDNA

<br>

>> ○ 2<sup>nd</sup>. The sgRNA-Cas9 complex cleaves the template DNA, non-template DNA, and sgRNA at the same location.

>> ○ 3<sup>rd</sup>. Cas9 and sgRNA separate.

>> ○ 4<sup>th</sup>. The DNA that formed the bubble reanneals through hydrogen bonding.

>> ○ 5<sup>th</sup>. The cleaved template and non-template DNA undergo DNA repair mechanisms.

>>> ○ 5<sup>th</sup>-1<sup>st</sup>. Viruses without DNA repair mechanisms are eliminated by the CRISPR/Cas9 mechanism.

>>> ○ 5<sup>th</sup>-2<sup>nd</sup>. Non-homologous End Joining (NHEJ): Allows for the joining of completely different chromosomes, following the [Holliday model](https://jb243.github.io/pages/1460).

>>> ○ 5<sup>th</sup>-3<sup>rd</sup>. Homology Directed Repair (HDR): Effects such as the substitution of specific bases or the addition of repeat sequences occur.

>> ○ 6<sup>th</sup>. In genetic recombination, research is conducted to replace the unnecessary parts of VNTR genes with useful DNA.

>>> ○ Utilizes homology-directed repair mechanisms.

>>> ○ Can be applied not only to DNA but also to RNA, epigenomes, and other single nucleotides for editing.

>>> ○ Types of gene editing: OE (overexpression), KD (knock-down).

> ④ Off-Target Issues

>> ○ Definition: The problem of editing regions other than the target gene when applying the CRISPR/Cas9 system to gene therapy.

>> ○ Streptococcus pyogenes Cas9 (SpCas9) nuclease is commonly used due to its efficiency but has a high off-target ratio.

>> ○ [Solutions](https://en.wikipedia.org/wiki/Off-target_genome_editing)

>>> ○ Nuclease mutation

>>> ○ PAM sequence modification

>>> ○ gRNA truncation

<br>

![image](https://github.com/user-attachments/assets/fa345b98-f650-4656-88d2-aa6273ba7077)

**Figure 11.** CRISPR/Cas9 Technology Diagram

<br>

> ⑤ Applications

>> ○ For signal transduction research purposes: Multiplexing guide RNAs enable perturbation screening.

>> ○ For imaging research purposes.

>> ○ For drug research purposes: Studying changes in drug uptake when specific genes are suppressed.

>> ○ Gene therapy: Editing genes responsible for rare diseases.

>> ○ Temporal sequencing (e.g., [Record-seq](https://www.nature.com/articles/s41596-019-0253-4)).

⑵ Lysogenic viruses: The viruses primarily used in gene therapy are adenoviruses and retroviruses.

⑶ siRNA, miRNA Therapeutics: Current market situation is not favorable

⑷ Nucleic Acid Delivery Systems

> ① Overview

>> ○ Adopted commercially with great success by Moderna and Pfizer during the COVID-19 pandemic.

>> ○ Since mRNA is unstable in the body, a delivery vehicle is necessary to ensure stability until it reaches the target tissue or cell.

> ② **Type 1.** mRNA Drug Delivery Systems

>> ○ Solid Lipid Nanoparticle (SLN): The most popular mRNA delivery vehicle adopted by Moderna and Pfizer.

>>> ○ A single SLN of 80-100 nm can encapsulate around 100 mRNA molecules.

>>> ○ Examples: ALC-0315 (Pfizer/BioNTech), SM-102 (Moderna), ALC-0159 (Pfizer/BioNTech), PEG-DMG (Moderna).

>> ○ Cationic liposome

>> ○ Polymer and polymer/lipid hybrid particle

>> ○ Micelle

>> ○ Emulsion

> ③ **Type 2.** DNA Drug Delivery Systems

> ④ **Type 3.** AAV(adeno-associated virus)

>> ○ Utilized as a formulation for treating Alzheimer's disease.

<br>

![image](https://github.com/user-attachments/assets/fe9c4a0f-1312-4f77-993d-a89a9ffe84af)

**Figure 12.** Comparison between AAV and LNP

<br>

⑹ Mega Nuclease

⑺ TALEN (Transcription Activator-Like Effector Nuclease)

⑻ ZFN (Zinc Finger Nuclease)

<br>

---

*Input: 2015.7.03 21:57*

*Modify: 2019.1.25 00:18*
