## **Ligand-Receptor Interaction Database**

Recommended Post: 【Bioinformatics】 [Bioinformatics Analysis Index](https://jb243.github.io/pages/836)

---

**1.** [Algorithm and Database](#1-algorithm-and-database)

**2.** [Ligand-Receptor Library](#2-ligand-receptor-library)

---

**a.** [Gene Library](https://jb243.github.io/pages/2212)

**b.** [Gene Score Library](https://jb243.github.io/pages/2215)

**c.** [Cell Type Marker Genes](https://jb243.github.io/pages/1846)

---

<br>

## **1. Algorithm and Database**

⑴ LR interaction score 

<br>

<img width="251" alt="스크린샷 2024-12-08 오후 10 50 57" src="https://github.com/user-attachments/assets/bd9cf94d-6dca-4b02-b729-ab471a3e0c9e">

<br>

⑵ bulk-RNA-seq-based

> ○ [BulkSignalR](https://github.com/jcolinge/BulkSignalR)

> ○ squidy

> ○ IPA (upstream regulator analysis of ingenuity pathway analysis)

> ○ Omnipath: For code that outputs all ligand-receptor pairs containing a specific gene, refer [here](https://jb243.github.io/pages/1892#10-bioinformatics).

> ○ BioCypher

⑶ scRNA-seq-based

> ○ [CellTalkDB](https://github.com/ZJUFanLab/CellTalkDB) ([Human Database File](https://blog.kakaocdn.net/dn/czeZZx/btsjBsLASyx/No0QweqLHKVRlu0CgKKe2K/CellTalkDB_human.csv?attach=1&knm=tfile.csv), [Mouse Database File](https://blog.kakaocdn.net/dn/chmMJS/btsjD8lDfej/yeVXUQmw72FiJktVjRZgK1/CellTalkDB_mouse.csv?attach=1&knm=tfile.csv))

> ○ [CellPhoneDB](https://jb243.github.io/pages/2088) ([tutorial](https://jb243.github.io/pages/2088))

> ○ [CellChat](https://www.nature.com/articles/s41467-021-21246-9) ([Database File](https://blog.kakaocdn.net/dn/rWlGz/btsNcXaZMTZ/a244WkSdoNJKK5kXAwZDL0/all_interacting_LR.csv?attach=1&knm=tfile.csv); 359 unique interaction, 13284 possible combination)

> ○ ICELLNET

> ○ [NicheNet](https://jb243.github.io/pages/168) 

> ○ SoptSC

> ○ CytoTalk

> ○ scTensor

> ○ CCCExplorer

> ○ Connectome

> ○ Ramilowski 

> ○ [FlowSig](https://pubmed.ncbi.nlm.nih.gov/39187683/)

> ○ scSeqComm: LR pairs from Reactome + TTRUST + RegNetwork

> ○ NATMI

> ○ SingleCellSignalR

⑷ ST-based

> ○ Giotto

> ○ spata2

> ○ CellPhoneDB v3

> ○ stLearn

> ○ SVCA

> ○ MISTy

> ○ [NCEM](https://www.nature.com/articles/s41587-022-01467-z)

> ○ [COMMOT](https://www.nature.com/articles/s41592-022-01728-4): Uses [optimal transport](https://jb243.github.io/pages/2386).

> ○ [SCOTIA](https://pubmed.ncbi.nlm.nih.gov/37425692/): In addition to the [optimal transport theorem](https://jb243.github.io/pages/2386), it also considers physical distance for greater accuracy

> ○ [STopover](https://github.com/bsungwoo/STopover): Employs diffusion and Jaccard index 

> ○ [cytosignal](https://github.com/welch-lab/cytosignal) 

> ○ [SpatialDM](https://github.com/StatBiomed/SpatialDM)

> ○ SpaTalk

> ○ stMLnet

> ○ HoloNet

> ○ DeepLinc

> ○ [scMultiSim](https://doi.org/10.1038/s41592-025-02651-0)

> ○ [GCNG](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-020-02214-w)

> ○ [conST](https://www.biorxiv.org/content/10.1101/2022.01.14.476408.full)

> ○ [Tangram](https://www.nature.com/articles/s41592-021-01264-7)

> ○ [DIALOGUE](https://www.nature.com/articles/s41587-022-01288-0)

> ○ CellTrek

> ○ [LIANA+](https://www.nature.com/articles/s41556-024-01469-w): An all-in-one framework incorporating CellphoneDB, CellChat, Connectome, scSeqComm, NATMI, Moran, PyDESeq2, Tensor-cell2cell (higher-order factorizations), scHOT, SpatialDM, etc.

>> ○ Spacia

⑸ [Protein-protein interaction](https://jb243.github.io/pages/2085#2-protein-protein-interaction-ppi-molecular-docking)(PPI, molecular docking)

> ○ AlphaFold2 multimer, [AFM-LIS](https://github.com/flyark/AFM-LIS), [AlphaFold3](https://alphafoldserver.com/)

> ○ [boltz-2](https://github.com/jwohlwend/boltz) 

> ○ DeepDTA

> ○ DeepDTAF

> ○ DeepFusionDTA

> ○ GraphDTA

> ○ [CAPLA](https://academic.oup.com/bioinformatics/article/39/2/btad049/6998204)

> ○ GNINA

> ○ SMINA

> ○ GLIDE

> ○ EquiBind

> ○ TANKBind

> ○ DIFFDOCK

> ○ STRINGdb

<br>

<br>

## **2. Ligand-Receptor Library**

> ○ Common target receptors and target ligands

<br>

| Organ | Cell type | Receptor | Ligand | Common targeting applications | Ref. |
| --- | --- | --- | --- | --- | --- |
| Liver | Hepatocytes, cancer cells | Asialoglyco-protein | Galactose | Cancer, Wilson's disease, hemochromatosis | . |
| . | . | . | Galactosamine | . | . |
| . | . | . | Lactose | . | . |
| Liver | Endothelial cells and Kupffer cells | Mannose receptor (C-type lectin receptor) | Mannose | Bacterial infections, inflammation, cardiovascular disease, cancer, cerebral ischemia/stroke, genetic metabolic diseases, Leishmaniasis | . |
| Liver | Hepatocytes | Galactosyl receptor | Galactose | Blood filtration | . |
| Liver | Kupffer cells | Fucose receptor, Lectin B | Fucose | Tissue binding, bio film formation | . |
| Liver, spleen, brain, bone marrow, and lungs | Macrophages | Mannose receptor (C-type lectin receptor) | Mannose | Bacterial infections, inflammation, cardiovascular disease, cancer, cerebral ischemia/stroke, genetic metabolic diseases, Leishmaniasis | . |
| Spleen, lymph nodes, blood, skin | Dendritic cells | Mannose receptor (C-type lectin receptor) | Mannose | Adaptive immune respose/vaccination | . |
| Tumor sites | Malignant cells | Lectin-like receptors (e.g., galectins) | Galactose, lactose, mannose, fucose, Sialic Lewis' glycosides | Anti-proliferation | . |
| Oral cavity | Buccal cells | Lectin receptors | Lactose and mannose | Bioadhesion | . |
| Eye | Corneal/conjunctival cells | Lectin receptors | Glucose and galactose, chitosan | Conjunctivitis | . |
| Colon | Epithelial cells | Galectin | Galactose | Cancer, bioadhesion | . |
| Lung | Pulmonary cells | Lectin receptors | Galactose | Cystic fibrosis, lung cancer, and pulmonary tuberculosis | . |
| Brain | Neurons | Anionic membranes and lectins | Trimethylated chitosan | Crossing the blood brain barrier, memory impairment | . |

**Table. 1.** Common target receptors and target ligands

<br>

> ○ ALOX5 ↔︎ ALOX5AP: proinflammatory interaction

> ○ ANGPTL4 ↔︎ SDC1: adhesive. It may inhibit Wnt signaling

> ○ BST2 ↔️ LILRA4: anti-inflammatory interaction

> ○ C3 ↔️ C3AR1: proinflammatory interaction

> ○ CCL5 ↔️ CCR1: proinflammatory interaction

> ○ CD28 ↔️ CD80: co-stimulatory

> ○ CD47 ↔️ SIRPA: innate immunity suppressor 

> ○ COL1A1 ↔︎ SDC1: adhesive

> ○ CTLA4 ↔️ CD80: co-inhibitory

> ○ CTLA4 ↔️ CD86: co-inhibitory

> ○ CX3CL1 ↔️ CX3CR1: anti-inflammatory interaction

> ○ CXCR3 ↔️ CXCL9: T cell homing interaction

> ○ CXCR3 ↔️ CXCL10: T cell homing interaction

> ○ CXCR3 ↔️ CXCL11: T cell homing interaction

> ○ DNAM1 ↔️ CD155, CD112: cell mediated cytotoxicity

> ○ DSC2 ↔︎ DSG1: adhesive 

> ○ FGF9 ↔︎ FGFR3: observed in astrocytes

> ○ FN1 ↔︎ SDC1: adhesive

> ○ GAS6 ↔︎ TYRO3: innate immunity suppressor 

> ○ HAVCR2 (TIM3) ↔️ LGALS9 (Galectin9): co-inhibitory

> ○ HLA-F ↔️ LILRB1: anti-inflammatory interaction

> ○ ICOS ↔️ ICOSLG: co-stimulatory

> ○ LAIR1 ↔️ LILRB4: suppressive interactions with myeloid cells

> ○ LAMB3 ↔︎ ITGA6_ITGB1: adhesive

> ○ LAMB3 ↔︎ ITGA6_ITGB4: adhesive

> ○ LFA1 ↔️ ICAM1: T cell recruitment

> ○ MIF ↔︎ CD74_CD44: inflammatory

> ○ NECTIN1 ↔︎ FGFR3: observed in astrocytes

> ○ PDCD1 (PD1) ↔️ CD274 (PDL1): co-inhibitory

> ○ PDCD1 (PD1) ↔️ PDCD1LG2 (PDL2): co-inhibitory

> ○ POMC ↔️ MC1R: anti-inflammatory interaction

> ○ PROS1 ↔️ AXL: anti-inflammatory interaction

> ○ PSAP ↔️ GPR37L1: observed in oligodendrocytes

> ○ TGFBR3 ↔️ TGFB1: suppressive interactions with myeloid cells

> ○ TIM-3 (T-cell Ig and mucin domain 3) receptor ↔️ galectin-9 ligand

> ○ TNFRSF9 ↔️ TNFSF9: co-stimulatory

> ○ TNFSF10 ↔️ TNFRSF10A: pro-apoptotic interaction

> ○ VEGF ↔️ NRP1

> ○ VEGF ↔️ NRP2

<br>

---

_Input**:** 2023.06.12 18:19_

